<HTML>

<HEAD>
  <TITLE>
  EMBOSS: cusp
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="index.html" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

cusp
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>




<H2>
    Function
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->
Create a codon usage table


<H2>
    Description
</H2>


Reads one or more coding sequences (CDS sequence only) and calculates
a codon frequency table.

<p>
The output file can be used as a codon usage table in other applications.

<H2>
    Usage
</H2>


<b>Here is a sample session with cusp</b>
<p>
This example uses only one input sequence. The normal use would be to use a set of coding sequences as the input.
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>cusp -sbeg 135 -send 1292</b>
Create a codon usage table
Input sequence(s): <b>tembl:paamir</b>
Output file [paamir.cusp]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>


<!--
Here is a sample session with cusp, using just one sequence.  Normal
use would be to extract a set of coding sequences and to use these as
input.

<pre>

% cusp -sbeg 135 -send 1292
Create a codon usage table
Input sequence: embl:paamir
Output file [paamir.cusp]: 

</pre>
-->


<H2>
    Command line arguments
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Mandatory qualifiers:
  [-sequence]          seqall     Sequence database USA
  [-outfile]           outfile    Output file name

   Optional qualifiers: (none)
   Advanced qualifiers:
   -cfile              codon      Codon usage table name

   General qualifiers:
  -help                boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose


</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#f5f5ff">
<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Mandatory qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>Sequence database USA</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;sequence&gt;</i>.cusp</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Optional qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Advanced qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-cfile</td>
<td>Codon usage table name</td>
<td>Codon usage file in EMBOSS data path</td>
<td>Ehum.cut</td>
</tr>

</table>



<H2>
    Input file format
</H2>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:paamir' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:paamir</h3>
<table width="90%"><tr><td bgcolor="#FFDDFF">
<pre>
&gt;PAAMIR X13776.1 Pseudomonas aeruginosa amiC and amiR gene for aliphatic amidase regulation
ggtaccgctggccgagcatctgctcgatcaccaccagccgggcgacgggaactgcacgat
ctacctggcgagcctggagcacgagcgggttcgcttcgtacggcgctgagcgacagtcac
aggagaggaaacggatgggatcgcaccaggagcggccgctgatcggcctgctgttctccg
aaaccggcgtcaccgccgatatcgagcgctcgcacgcgtatggcgcattgctcgcggtcg
agcaactgaaccgcgagggcggcgtcggcggtcgcccgatcgaaacgctgtcccaggacc
ccggcggcgacccggaccgctatcggctgtgcgccgaggacttcattcgcaaccgggggg
tacggttcctcgtgggctgctacatgtcgcacacgcgcaaggcggtgatgccggtggtcg
agcgcgccgacgcgctgctctgctacccgaccccctacgagggcttcgagtattcgccga
acatcgtctacggcggtccggcgccgaaccagaacagtgcgccgctggcggcgtacctga
ttcgccactacggcgagcgggtggtgttcatcggctcggactacatctatccgcgggaaa
gcaaccatgtgatgcgccacctgtatcgccagcacggcggcacggtgctcgaggaaatct
acattccgctgtatccctccgacgacgacttgcagcgcgccgtcgagcgcatctaccagg
cgcgcgccgacgtggtcttctccaccgtggtgggcaccggcaccgccgagctgtatcgcg
ccatcgcccgtcgctacggcgacggcaggcggccgccgatcgccagcctgaccaccagcg
aggcggaggtggcgaagatggagagtgacgtggcagaggggcaggtggtggtcgcgcctt
acttctccagcatcgatacgcccgccagccgggccttcgtccaggcctgccatggtttct
tcccggagaacgcgaccatcaccgcctgggccgaggcggcctactggcagaccttgttgc
tcggccgcgccgcgcaggccgcaggcaactggcgggtggaagacgtgcagcggcacctgt
acgacatcgacatcgacgcgccacaggggccggtccgggtggagcgccagaacaaccaca
gccgcctgtcttcgcgcatcgcggaaatcgatgcgcgcggcgtgttccaggtccgctggc
agtcgcccgaaccgattcgccccgacccttatgtcgtcgtgcataacctcgacgactggt
ccgccagcatgggcgggggaccgctcccatgagcgccaactcgctgctcggcagcctgcg
cgagttgcaggtgctggtcctcaacccgccgggggaggtcagcgacgccctggtcttgca
gctgatccgcatcggttgttcggtgcgccagtgctggccgccgccggaagccttcgacgt
gccggtggacgtggtcttcaccagcattttccagaatggccaccacgacgagatcgctgc
gctgctcgccgccgggactccgcgcactaccctggtggcgctggtggagtacgaaagccc
cgcggtgctctcgcagatcatcgagctggagtgccacggcgtgatcacccagccgctcga
tgcccaccgggtgctgcctgtgctggtatcggcgcggcgcatcagcgaggaaatggcgaa
gctgaagcagaagaccgagcagctccaggaccgcatcgccggccaggcccggatcaacca
ggccaaggtgttgctgatgcagcgccatggctgggacgagcgcgaggcgcaccagcacct
gtcgcgggaagcgatgaagcggcgcgagccgatcctgaagatcgctcaggagttgctggg
aaacgagccgtccgcctgagcgatccgggccgaccagaacaataacaagaggggtatcgt
catcatgctgggactggttctgctgtacgttggcgcggtgctgtttctcaatgccgtctg
gttgctgggcaagatcagcggtcgggaggtggcggtgatcaacttcctggtcggcgtgct
gagcgcctgcgtcgcgttctacctgatcttttccgcagcagccgggcagggctcgctgaa
ggccggagcgctgaccctgctattcgcttttacctatctgtgggtggccgccaaccagtt
cctcgag
</pre>
</td></tr></table><p>


<H2>
    Output file format
</H2>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: paamir.cusp</h3>
<table width="90%"><tr><td bgcolor="#DDFFDD">
<pre>
# CUSP codon usage file
# Codon	Amino acid	Fract   /1000	Number
GCA	A		0.077	7.772	3
GCC	A		0.462	46.632	18
GCG	A		0.462	46.632	18
GCT	A		0.000	0.000	0
TGC	C		1.000	10.363	4
TGT	C		0.000	0.000	0
GAC	D		0.864	49.223	19
GAT	D		0.136	7.772	3
GAA	E		0.269	18.135	7
GAG	E		0.731	49.223	19
TTC	F		1.000	28.497	11
TTT	F		0.000	0.000	0
GGA	G		0.062	5.181	2
GGC	G		0.719	59.585	23
GGG	G		0.125	10.363	4
GGT	G		0.094	7.772	3
CAC	H		0.727	20.725	8
CAT	H		0.273	7.772	3
ATA	I		0.000	0.000	0
ATC	I		0.800	41.451	16
ATT	I		0.200	10.363	4
AAA	K		0.000	0.000	0
AAG	K		1.000	5.181	2
CTA	L		0.000	0.000	0
CTC	L		0.269	18.135	7
CTG	L		0.577	38.860	15
CTT	L		0.000	0.000	0
TTA	L		0.000	0.000	0
TTG	L		0.154	10.363	4
ATG	M		1.000	15.544	6
AAC	N		1.000	28.497	11
AAT	N		0.000	0.000	0
CCA	P		0.074	5.181	2
CCC	P		0.222	15.544	6
CCG	P		0.630	44.041	17
CCT	P		0.074	5.181	2
CAA	Q		0.062	2.591	1
CAG	Q		0.938	38.860	15
AGA	R		0.000	0.000	0
AGG	R		0.029	2.591	1
CGA	R		0.000	0.000	0
CGC	R		0.629	56.995	22
CGG	R		0.314	28.497	11
CGT	R		0.029	2.591	1
AGC	S		0.304	18.135	7
AGT	S		0.087	5.181	2
TCA	S		0.000	0.000	0
TCC	S		0.261	15.544	6
TCG	S		0.304	18.135	7
TCT	S		0.043	2.591	1
ACA	T		0.000	0.000	0
ACC	T		0.733	28.497	11
ACG	T		0.267	10.363	4
ACT	T		0.000	0.000	0
GTA	V		0.030	2.591	1
GTC	V		0.394	33.679	13
GTG	V		0.576	49.223	19
GTT	V		0.000	0.000	0
TGG	W		1.000	12.953	5
TAC	Y		0.619	33.679	13
TAT	Y		0.381	20.725	8
TAA	*		0.000	0.000	0
TAG	*		0.000	0.000	0
TGA	*		1.000	2.591	1
</pre>
</td></tr></table><p>


<p>

<!--
<pre>

# CUSP codon usage file
# Codon Amino acid      Fract   /1000   Number
GCA     A               0.077   7.772   3
GCC     A               0.462   46.632  18
GCG     A               0.462   46.632  18
GCT     A               0.000   0.000   0

.........

</pre>
-->

The example usage read in a
single CDS from Pseudomonas aeruginosa which has a very high GC
content ands a strong coding bias, as shown by the codons for Alanine
where those ending with G or C are used almost exclusively.

<p>

The 'Fract' column gives the faction of all amino acids coded for by
this codon triplet. 

<p>

The /1000 column represents the number of codons, given the
input sequence(s), there are per 1000 bases. This will be an
extrapolation if the sequence is shorter than 1000 bases.

<p>

If multiple sequences are input then the statistics are given for all of
the sequences together, not individually. 

<H2>
    Data files
</H2>


cusp reads a codon usage file, but only as a template and does not
use any of the data so any file will give the same results.

<H2>
    Notes
</H2>

None.

<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

Always exits with status 0.

<H2>
    Known bugs
</H2>

None.

<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="cai.html">cai</a></td><td>CAI codon adaptation index</td></tr>
<tr><td><a href="chips.html">chips</a></td><td>Codon usage statistics</td></tr>
<tr><td><a href="codcmp.html">codcmp</a></td><td>Codon usage table comparison</td></tr>
<tr><td><a href="syco.html">syco</a></td><td>Synonymous codon usage Gribskov statistic plot</td></tr>
</table>

<!--
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find.
   -->


<H2>
    Author(s)
</H2>


This application was written by Alan Bleasby <a
href="mailto:ableasby@hgmp.mrc.ac.uk">(ableasby@hgmp.mrc.ac.uk)</a>

<H2>
    History
</H2>

Spring 2000 - written

<H2>
    Target users
</H2>


This program is intended to be used by everyone and everything,
from naive users to embedded scripts. 

<H2>
    Comments
</H2>


<pre>

</pre>


</BODY>
</HTML>

