<HTML>

<HEAD>
  <TITLE>
  EMBOSS
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
diffseq
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
    Function
</H2>
Find differences between nearly identical sequences
<H2>
    Description
</H2>

<b>diffseq</b> takes two overlapping, nearly identical sequences and
reports the differences between them, together with any features that
overlap with these regions. GFF files of the differences in each
sequence are also produced.

<p>

<b>diffseq</b> finds the region of overlap of the input sequences and
then reports differences within this region, like a local alignment.

<p>

The start and end positions of the overlap are reported.

<p>

<b>diffseq</b> should be of value when looking for SNPs, differences
between strains of an organism and anything else that requires the
differences between sequences to be highlighted.

<p>  

The sequences can be very long. The program does a match of all
sequence words of size 10 (by default). It then reduces this to the
minimum set of overlapping matches by sorting the matches in order of
size (largest size first) and then for each such match it removes any
smaller matches that overlap. The result is a set of the longest
ungapped alignments between the two sequences that do not overlap with
each other. The mismatched regions between these matches are reported.

<p>   

It should be possible to find differences between sequences that are Mega-bases
long.


<H2>
    Usage
</H2>
<b>Here is a sample session with diffseq</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>diffseq tembl:ba000025 tembl:af129756 </b>
Find differences between nearly identical sequences
Word size [10]: <b></b>
Output report [ba000025.diffseq]: <b></b>
Features output [BA000025.diffgff]: <b></b>
Second features output [AF129756.diffgff]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>

<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-asequence]         sequence   Sequence filename and optional format, or
                                  reference (input USA)
  [-bsequence]         sequence   Sequence filename and optional format, or
                                  reference (input USA)
   -wordsize           integer    [10] The similar regions between the two
                                  sequences are found by creating a hash table
                                  of 'wordsize'd subsequences. 10 is a
                                  reasonable default. Making this value larger
                                  (20?) may speed up the program slightly,
                                  but will mean that any two differences
                                  within 'wordsize' of each other will be
                                  grouped as a single region of difference.
                                  This value may be made smaller (4?) to
                                  improve the resolution of nearby
                                  differences, but the program will go much
                                  slower. (Integer 2 or more)
  [-outfile]           report     [*.diffseq] Output report file name
  [-aoutfeat]          featout    [$(asequence.name).diffgff] File for output
                                  of first sequence's features
  [-boutfeat]          featout    [$(bsequence.name).diffgff] File for output
                                  of second sequence's features

   Additional (Optional) qualifiers:
   -globaldifferences  boolean    [N] Normally this program will find regions
                                  of identity that are the length of the
                                  specified word-size or greater and will then
                                  report the regions of difference between
                                  these matching regions. This works well and
                                  is what most people want if they are working
                                  with long overlapping nucleic acid
                                  sequences. You are usually not interested in
                                  the non-overlapping ends of these
                                  sequences. If you have protein sequences or
                                  short RNA sequences however, you will be
                                  interested in differences at the very ends .
                                  It this option is set to be true then the
                                  differences at the ends will also be
                                  reported.

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-asequence" associated qualifiers
   -sbegin1            integer    Start of the sequence to be used
   -send1              integer    End of the sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-bsequence" associated qualifiers
   -sbegin2            integer    Start of the sequence to be used
   -send2              integer    End of the sequence to be used
   -sreverse2          boolean    Reverse (if DNA)
   -sask2              boolean    Ask for begin/end/reverse
   -snucleotide2       boolean    Sequence is nucleotide
   -sprotein2          boolean    Sequence is protein
   -slower2            boolean    Make lower case
   -supper2            boolean    Make upper case
   -sformat2           string     Input sequence format
   -sdbname2           string     Database name
   -sid2               string     Entryname
   -ufo2               string     UFO features
   -fformat2           string     Features format
   -fopenfile2         string     Features file name

   "-outfile" associated qualifiers
   -rformat3           string     Report format
   -rname3             string     Base file name
   -rextension3        string     File name extension
   -rdirectory3        string     Output directory
   -raccshow3          boolean    Show accession number in the report
   -rdesshow3          boolean    Show description in the report
   -rscoreshow3        boolean    Show the score in the report
   -rusashow3          boolean    Show the full USA in the report
   -rmaxall3           integer    Maximum total hits to report
   -rmaxseq3           integer    Maximum hits to report for one sequence

   "-aoutfeat" associated qualifiers
   -offormat4          string     Output feature format
   -ofopenfile4        string     Features file name
   -ofextension4       string     File name extension
   -ofdirectory4       string     Output directory
   -ofname4            string     Base file name
   -ofsingle4          boolean    Separate file for each entry

   "-boutfeat" associated qualifiers
   -offormat5          string     Output feature format
   -ofopenfile5        string     Features file name
   -ofextension5       string     File name extension
   -ofdirectory5       string     Output directory
   -ofname5            string     Base file name
   -ofsingle5          boolean    Separate file for each entry

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write standard output
   -filter             boolean    Read standard input, write standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-asequence]<br>(Parameter 1)</td>
<td>Sequence filename and optional format, or reference (input USA)</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-bsequence]<br>(Parameter 2)</td>
<td>Sequence filename and optional format, or reference (input USA)</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-wordsize</td>
<td>The similar regions between the two sequences are found by creating a hash table of 'wordsize'd subsequences. 10 is a reasonable default. Making this value larger (20?) may speed up the program slightly, but will mean that any two differences within 'wordsize' of each other will be grouped as a single region of difference. This value may be made smaller (4?) to improve the resolution of nearby differences, but the program will go much slower.</td>
<td>Integer 2 or more</td>
<td>10</td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 3)</td>
<td>Output report file name</td>
<td>Report output file</td>
<td><i>&lt;*&gt;</i>.diffseq</td>
</tr>

<tr>
<td>[-aoutfeat]<br>(Parameter 4)</td>
<td>File for output of first sequence's features</td>
<td>Writeable feature table</td>
<td>$(asequence.name).diffgff</td>
</tr>

<tr>
<td>[-boutfeat]<br>(Parameter 5)</td>
<td>File for output of second sequence's features</td>
<td>Writeable feature table</td>
<td>$(bsequence.name).diffgff</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-globaldifferences</td>
<td>Normally this program will find regions of identity that are the length of the specified word-size or greater and will then report the regions of difference between these matching regions. This works well and is what most people want if they are working with long overlapping nucleic acid sequences. You are usually not interested in the non-overlapping ends of these sequences. If you have protein sequences or short RNA sequences however, you will be interested in differences at the very ends . It this option is set to be true then the differences at the ends will also be reported.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>

<H2>
    Input file format
</H2>
This program reads in two nucleic acid sequence USAs or two protein
sequence USAs. 

<p>

<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:ba000025' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:ba000025</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   BA000025; SV 2; linear; genomic DNA; STD; HUM; 2229817 BP.
XX
AC   BA000025; AP000502-AP000521;
XX
DT   09-DEC-2004 (Rel. 82, Created)
DT   14-NOV-2006 (Rel. 89, Last updated, Version 4)
XX
DE   Homo sapiens genomic DNA, chromosome 6p21.3, HLA Class I region.
XX
KW   .
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-2229817
RA   Hirakawa M., Yamaguchi H., Imai K., Shimada J.;
RT   ;
RL   Submitted (21-AUG-2001) to the EMBL/GenBank/DDBJ databases.
RL   Mika Hirakawa, Japan Science and Technology Corporation (JST), Advanced
RL   Databases Department; 5-3, Yonbancho, Chiyoda-ku, Tokyo 102-0081, Japan
RL   (E-mail:mika@tokyo.jst.go.jp, URL:http://www-alis.tokyo.jst.go.jp/,
RL   Tel:81-3-5214-8491, Fax:81-3-5214-8470)
XX
RN   [2]
RA   Shiina S., Tamiya G., Oka A., Inoko H.;
RT   "Homo sapiens 2,229,817bp genomic DNA of 6p21.3 HLA class I region";
RL   Unpublished.
XX
DR   EPD; EP11158; HS_TNF.
DR   EPD; EP11159; HS_LTA.
DR   EPD; EP73522; HS_HLA-B.
DR   EPD; EP73908; HS_GTF2H4.
DR   EPD; EP73940; HS_NEU1.
DR   EPD; EP74013; HS_VARS2.
DR   EPD; EP74203; HS_MRPS18B.
DR   EPD; EP74346; HS_HLA-E.
DR   EPD; EP74389; HS_BAT1.
DR   EPD; EP74485; HS_IER3.
DR   GDB; 11515913.
DR   GOA; P59942.
DR   IMGT/HLA; HLA02629; J*01010102.
DR   RFAM; RF00017.
DR   RFAM; RF00019.
DR   RFAM; RF00026.
DR   RFAM; RF00100.
DR   RFAM; RF00137.
DR   RFAM; RF00276.


<font color=red>  [Part of this file has been deleted for brevity]</font>

     ttggccccac cccagcatgt ctccaggttc ctctcagccc tggttccttt tggccctgca   2226900
     gtcacaatgg gcaacactgt gacgcaccct gtcctgtgtc acagtgtcat acactcaggc   2226960
     tcacattgcc cctaggccac ttgccagcca agggacatgg ccacattttg tgtcttctgc   2227020
     acctcagcct tgctttcaag tgcaggtgat gatggcaccc acgcagaaca aatgttattt   2227080
     gctatcttcg tcgagtttag tcatccaatt ttccaaccct cactgggcaa ggaagagtgt   2227140
     ggtttccacc aagaaggcag gatgtcagca gtcacagggg caaccaacag ggaaagccgc   2227200
     cggaaaatag accccacagg aagcacaggt gtccagtgga gatgggaacc ctgcagattt   2227260
     gaccgtcttt aagcagatta gagagattac cgttactaac aacttagcca taaaagttta   2227320
     ttagctattt tcaaaaagca taaaattatg taatataatt ttttttaaat ttccatcaat   2227380
     acaaaactaa tctgggcact gcaacttccg gtgggcaact gggataggcg gcatcatcag   2227440
     gaaggcgagc cctgccgtgc cccatgtgcc agtgccccag atggcggcag cctccccaga   2227500
     agcaccttgt atctcccctg cacagggcca gggtcccagc ttcccataca ccttctcctg   2227560
     ctttttcttt tctgtccttt cctttttcaa taaaccacct gcaaaaaggg aaaaccattc   2227620
     tgaggacaag aaacatgtca atgggaaata cacagttgcc agagggtaaa aggccctgtt   2227680
     cattctcatt gaaaagctca ggtatttctg ttaaagtctc tccttttact ttaggatgct   2227740
     gactcctgcg tccatctcaa cctgggcatc gtgccaccac cttcaagaag agaaaaacta   2227800
     agtagtgctt tgcaaagggg cagcagcatt tctcatttct gaccatgtca ggcacatggc   2227860
     catgcagatg agcaggtggg ggacacaggt gagtctccag acctgctctc ctcccacagt   2227920
     acattcttga gtctttttaa acagttgtga aaatgccaca gatgcaagca cctgtgggcc   2227980
     actcccatgg ggaccgttgc acaaggcagt gccactcatt ctcagaacct cctaccatgg   2228040
     gctatgctta gtgacccgag gccaagccaa ggaagacgcc agccacaggg tgccatcctc   2228100
     aggggcatgc tgccagcagg ggcaaagtta tccctagcaa caagatacag aaagaaagaa   2228160
     aaaaggaagg aaatgtagcc aatgggccgg ttcaggttct tgactttgcc acacaaaaga   2228220
     atttgagagc aagtccaaag taaaagtcag caagagaatt tattgcaaag tgaaagtaca   2228280
     ctctgacagc tgatcagagc agctgctcaa aagagagaca gtaccctccc ctcacgggag   2228340
     tcttacatga ttattcatga ataggtggga aggggtattg ttttaagcat gttctgtggt   2228400
     ctcttgaacg tgcatgcact gtggttgtac atatcagcac acacatctta cgtctcatta   2228460
     gcatcttaac ttccctctca gagttgtgtt tgctactatt gtaatgagca taggtcagcc   2228520
     caaggacact attcatgggt ttctgggctt cctcagatgt ggggatgcct cccttggctc   2228580
     ttctacctct ttgctgcagg atgttctaac cacaagccca ggatatggtt tgcgcactgt   2228640
     cgaacagctt gttctctcca tcaacctgac aagtctcttg tttcctttca agggaggctg   2228700
     tgaacaccct atctcactga cctcagaagg acagtacagc agtagccacc atgaccaaaa   2228760
     agatgattcc agaagtgcag gacaactccc tacccagagg ctgtggctgt gcagtaacac   2228820
     accaagaggg gagtccagct ggctctcagg gtgctcacta ccctcatctg ggggcctgga   2228880
     ggacgtcaat tcctgagaac gccacgttct agtgagtaga atgaactgag agatacacag   2228940
     caaagctcca catacttttc cttttctttg tgcccgcagt gttcttcatc agtgtgctct   2229000
     cgcttttcag ctactactgt tggctggctg gaaaaaatag aacaatagta aaaattagag   2229060
     accagtcttt ggtgatgaag agaaatattg gctacttcca gtattttcta gctttggtta   2229120
     tggttgcagt tttccagctc accttgtggg gatgaattca gaaaaaagtt acaaattgaa   2229180
     atgaacatgc cagaagtatt ggctcaaatc aacgttgtcc tattaagcca cttagtgaat   2229240
     caaaagaccg cttgttggac tgttaatctc ggtggccaga gaaaggagct gaagaaggtg   2229300
     ttgccagatc aggaacaaat aattacagcg gcaatagaaa atggaagacc acttgttcat   2229360
     aaccatttga ataagggcaa ggtgtatgga aacacattat gaactgatat tttcagtttt   2229420
     gtttgcaaga aaatgattaa taaggtgaaa taattgaagt atcacggaag atacattaaa   2229480
     aaaaaaaaaa gcctttgtac agtttgctgg agccacagat gtcctactcc agagcagaac   2229540
     aatgcctgaa tcttcagggt ccatttctgc cgcattcact agcaaccaca aatgtgactt   2229600
     aattttactt tggaaataat gcttacccat tgtgagatgc tgtaatatga accatcatta   2229660
     catgttaaca tggcacatgg aattttgagt gtctaagtta catttttaga gttgtttctt   2229720
     agtagccatg tgagtttcca ctccaaaaac acaagctaaa aacttgtttt gagtgaagga   2229780
     catctagggc aaatggtggc tgaaagtgaa tgagatc                            2229817
//
</pre>
</td></tr></table><p>
<p><h3>Database entry: tembl:af129756</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   AF129756; SV 1; linear; genomic DNA; STD; HUM; 184666 BP.
XX
AC   AF129756;
XX
DT   12-MAR-1999 (Rel. 59, Created)
DT   14-NOV-2006 (Rel. 89, Last updated, Version 5)
XX
DE   Homo sapiens MSH55 gene, partial cds; and CLIC1, DDAH, G6b, G6c, G5b, G6d,
DE   G6e, G6f, BAT5, G5b, CSK2B, BAT4, G4, Apo M, BAT3, BAT2, AIF-1, 1C7, LST-1,
DE   LTB, TNF, and LTA genes, complete cds.
XX
KW   .
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-184666
RX   DOI; 10.1101/gr.1736803.
RX   PUBMED; 14656967.
RA   Xie T., Rowen L., Aguado B., Ahearn M.E., Madan A., Qin S., Campbell R.D.,
RA   Hood L.;
RT   "Analysis of the gene-dense major histocompatibility complex class III
RT   region and its comparison to mouse";
RL   Genome Res. 13(12):2621-2636(2003).
XX
RN   [2]
RP   1-184666
RA   Rowen L., Madan A., Qin S., Shaffer T., James R., Ratcliffe A., Abbasi N.,
RA   Dickhoff R., Loretz C., Madan A., Dors M., Young J., Lasky S., Hood L.;
RT   "Sequence of the human major histocompatibility complex class III region";
RL   Unpublished.
XX
RN   [3]
RP   1-184666
RA   Rowen L.;
RT   ;
RL   Submitted (22-FEB-1999) to the EMBL/GenBank/DDBJ databases.
RL   Department of Molecular Biotechnology, Box 357730 University of Washington,
RL   Seattle, WA 98195, USA
XX
RN   [4]
RP   1-184666
RA   Rowen L.;
RT   ;
RL   Submitted (28-OCT-1999) to the EMBL/GenBank/DDBJ databases.
RL   Multimegabase Sequencing Center, University of Washington, PO Box 357730,
RL   Seattle, WA 98195, USA


<font color=red>  [Part of this file has been deleted for brevity]</font>

     aaaccagttt accaccactc ctaacactaa acttaaatct gactctaaat gtaagtccaa    181740
     tctgagccac aagcctaaag ttgaacttta tcctgcttta tgaattattc atccattcct    181800
     ccatttagtg agtatctgcg tgcctaacac atgctgggca ttgtcctaag gcaggaggga    181860
     catggaggca aagggatcag agaaggtacc agcacctgtg gagcttgtat tccagtgagg    181920
     ccagacggaa aagaaagaaa ctgaagaaga aattggtact atgagaaaat aagacaggct    181980
     gatgttgtaa gagtggcagg gagctacttt taaatacagt agtcagcaaa atcctctttg    182040
     agtgtttggg tggcactgga gctgagaccc aaatgacaaa aaatagtgac caggtaaaag    182100
     tttgggagca aagcatttca ggtaaaggga gcagctactg caaaggctgg aaggcggaac    182160
     caagctgggg gtgttgacga caaacagaag gccagtgtgg ctggagcaga gagagagact    182220
     gggaggcggg tgggagatga ggtcagagag gagggcaggg gccaggtcat gcagggccat    182280
     gcaagaaggg taaagcctct agatttcatc cagccacagg aagcctttaa aggtcgtcag    182340
     agtgtgtggt gcgtgcgtgt gtgtgtgtgt gtgtgtgtgt gttgcagggg agagaggggg    182400
     agggagagag agagagagag agagaagagg gaggtgagca gaggtgattg gatttttttt    182460
     tcttttgaca tggtgtcttg ctctgtggcc taggctggag tgcagtggca ccatcatagc    182520
     ccactgcaac ctcaaaacca tgggctcaag tcatccttcc acctcagctt cccaagtatc    182580
     taggactaca ggtgtgtgcc actgtgcctg gctaatttta aaaaatattt taaaattttt    182640
     gttgagacag ggtctatgct gctcaggctg gtctcgaact cctggtttca agtgatctgc    182700
     ccatcttggc ctcccaaagt ttttttttgt tagtttgaga ggcggtttcg ctcgttgccc    182760
     aggctggagt gcaatgactg atctcatctc actgcaacct ctgcctcctg ggttcaagcg    182820
     attctcctgc ttcagcctcc caagtagctg ggattacagg tgcatgccac cattcccggc    182880
     taattttttg tatttagtag agatggggtt tcaccatgtt agtcaggctg atctcaaact    182940
     cctgacctca ggtgatccgc ctgcctcagc ctcccaaagt tttgggatta caggtgtgag    183000
     ccaccatgct gggccagcct cccaaagttt tgggattaca ggcatgagtc accacactgg    183060
     ccctggattt tttttctttc ttttttttgg agacggagtc tcactctgtt gcccaggctg    183120
     gagtgcaatg gcgtaatctc agctcactgc aacctctgct gcccgggttc aaacgattct    183180
     cctgtcttag cctcctgagt agctgggatt ataggtgcat gccaccatgc ctggctaatt    183240
     tttgtacttt tagtagagaa agtacaccat cttggccagg ctggtctcga actcctgacc    183300
     tcaggtgatc cacttgcgtc ggcctcccaa agtgctggga ttacaggcgt gagacaccgc    183360
     acccagcctt tttttttttt tttcttttaa gacagaatcg ctctgtcacc caggctggag    183420
     tgcagtggca caatctcggc tcactgcaac ctctgcctcc caggtttaag caatccacct    183480
     atgtcagtct cccaagtagc tgggattata ggtgcatgtc accatgcctg gctaattttt    183540
     gtacttttag tatagaaagt acaccatgtt ggccaggctg gtcttgaact cctgacctca    183600
     agtgatccgc ctgcctcagc ctcccgaagt gctggaatta cagacatgtg ccactgcacc    183660
     cggcctggtt ttttttttct aagagatgga gtctcacttt tctgcccagg ttggagtgca    183720
     atggcaccat catagctcac tgcagccttc aactcttggc ctcaggcaat ccttgcacct    183780
     tagcctcgca gtgttgggat tacaggcatg agccactgag ccttgcctgg actttttttt    183840
     ttttttgaga tggcgtctcg ctctgttgcc caggttggag tgctacggca tgatcttggc    183900
     tcactgcaac ttccacctcc caggttcaag cgattctctt gcctcggccc cccgagtagc    183960
     tgggattaca ggcatgcgcc accgtgcctg gctaattttg gtatttttag tagagatagg    184020
     gtttcatcat gttgggcagg ctggtcttga actcctgacc tcgtgatcca cccacctcgg    184080
     cctcccaaag tgctgggatt ataggcatag ccaacgcgcc cagcctggac ttgtttttaa    184140
     aagatcactg tggctcctgt gtttaggctg gctggtagga gacaggtggc agtggcattg    184200
     atggtgaaga gaaaatagtg gcagccatgg agatggagag aagtagacaa gtttgggata    184260
     tattatacat tccaggggta gaaacaacag gactagatga tggattgatg ggtgggagat    184320
     gtagatactg ggagagaagc aggattctga tggatggaaa aactaaaaaa ttctattttg    184380
     ggtgtggtaa gtctaagtct attagacatg caagtagaga tgtcactggg cagatacaca    184440
     tctggatttc aggggcaagg tccaagctag agaaagaaac ctgggcatgg tcagcatgag    184500
     gatggtgttt aaagccatgg aacttatctt gtgcatccct ataagacccc tttgaggcac    184560
     ttgtttcccc tcacaatgga tgcagtgcat cttccattct gaattccaga ggcaacaacc    184620
     tcctgctcct agaagctaaa ctctccagac ttagtcttct gaattc                   184666
//
</pre>
</td></tr></table><p>


<H2>
    Output file format
</H2>

<p>

The output is a standard EMBOSS report file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <b>-rformat xxx</b>, where 'xxx' is replaced by
the name of the required format.  The available format names are: embl,
genbank, gff, pir, swiss, trace, listfile, dbmotif, diffseq, excel,
feattable, motif, regions, seqtable, simple, srs, table, tagseq

<p>

See:
<A href="http://emboss.sf.net/docs/themes/ReportFormats.html">
http://emboss.sf.net/docs/themes/ReportFormats.html</A>
for further information on report formats.

<p>
<p>

By default <b>diffseq</b> writes a 'diffseq' report file.

<p>

<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: ba000025.diffseq</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: diffseq
# Rundate: Sat 15 Jul 2006 12:00:00
# Commandline: diffseq
#    [-asequence] tembl:ba000025
#    [-bsequence] tembl:af129756
# Report_format: diffseq
# Report_file: ba000025.diffseq
# Additional_files: 2
# 1: BA000025.diffgff (Feature file for first sequence)
# 2: AF129756.diffgff (Feature file for second sequence)
########################################

#=======================================
#
# Sequence: BA000025     from: 1   to: 2229817
# HitCount: 236
#
# Compare: AF129756     from: 1   to: 184666
# 
# BA000025 overlap starts at 193957
# AF129756 overlap starts at 1
# 
# (BA000025) start end length sequence
# (AF129756) start end length sequence
# 
#
#=======================================


BA000025 194109-194109 Length: 1
Sequence: t
Sequence: c
AF129756 153-153 Length: 1

BA000025 197911-197914 Length: 4
Sequence: tcac
Sequence: 
Feature: repeat_region 3674-3957 rpt_family='AluSq'
AF129756 3954 Length: 0

BA000025 198249-198254 Length: 6
Sequence: tctcaa
Sequence: cctc
Feature: repeat_region 4013-4310 rpt_family='AluY'
AF129756 4289-4292 Length: 4

BA000025 198912-198914 Length: 3
Sequence: ttt
Sequence: 


<font color=red>  [Part of this file has been deleted for brevity]</font>


BA000025 377595-377595 Length: 1
Sequence: t
Sequence: c
Feature: misc_feature 130479-184666 note='Overlap with GenBank entry Y14768.'
Feature: misc_feature 180792-184666 note='Overlap with cosmid M9A (AC004181).'
Feature: repeat_region 183369-183665 rpt_family='AluSx'
Feature: variation 183597-183597 replace='t' note='201G24: c; M9A and Y14768: t'
AF129756 183597-183597 Length: 1

BA000025 377789-377790 Length: 2
Sequence: aa
Sequence: 
Feature: misc_feature 130479-184666 note='Overlap with GenBank entry Y14768.'
Feature: misc_feature 180792-184666 note='Overlap with cosmid M9A (AC004181).'
Feature: repeat_region 183669-183822 rpt_family='FRAM'
AF129756 183790 Length: 0

BA000025 377931-377931 Length: 1
Sequence: t
Sequence: c
Feature: misc_feature 130479-184666 note='Overlap with GenBank entry Y14768.'
Feature: misc_feature 180792-184666 note='Overlap with cosmid M9A (AC004181).'
Feature: repeat_region 183833-184125 rpt_family='AluSg'
Feature: variation 183931-183931 replace='t' note='210G24: c; M9A and Y14768: t'
AF129756 183931-183931 Length: 1

BA000025 378423-378423 Length: 1
Sequence: c
Sequence: t
Feature: misc_feature 130479-184666 note='Overlap with GenBank entry Y14768.'
Feature: misc_feature 180792-184666 note='Overlap with cosmid M9A (AC004181).'
Feature: repeat_region 184316-184524 rpt_family='LINE2'
Feature: variation 184423-184423 replace='c' note='201G24: t; M9A and Y14768: c'
AF129756 184423-184423 Length: 1

#---------------------------------------
#
# Overlap_end: 378666 in BA000025
# Overlap_end: 184666 in AF129756
# 
# SNP_count: 166
# Transitions: 113
# Transversions: 53
#
#---------------------------------------

#---------------------------------------
# Total_sequences: 1
# Total_hitcount: 236
#---------------------------------------
</pre>
</td></tr></table><p>
<p><h3>File: AF129756.diffgff</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
##gff-version 2.0
##date 2006-07-15
##Type DNA AF129756
AF129756	diffseq	conflict	153	153	1.000	+	.	Sequence "AF129756.1" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	4289	4292	1.000	+	.	Sequence "AF129756.2" ; note "BA000025" ; replace "tctcaa"
AF129756	diffseq	conflict	6882	6882	1.000	+	.	Sequence "AF129756.3" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	7830	7830	1.000	+	.	Sequence "AF129756.4" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	8500	8500	1.000	+	.	Sequence "AF129756.5" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	10945	10962	1.000	+	.	Sequence "AF129756.6" ; note "Insertion of 18 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	10999	11001	1.000	+	.	Sequence "AF129756.7" ; note "BA000025" ; replace "aaa"
AF129756	diffseq	conflict	12915	12915	1.000	+	.	Sequence "AF129756.8" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	15139	15139	1.000	+	.	Sequence "AF129756.9" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	17192	17192	1.000	+	.	Sequence "AF129756.10" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	19761	19761	1.000	+	.	Sequence "AF129756.11" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	20291	20291	1.000	+	.	Sequence "AF129756.12" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	20462	20462	1.000	+	.	Sequence "AF129756.13" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	25686	25686	1.000	+	.	Sequence "AF129756.14" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	26192	26192	1.000	+	.	Sequence "AF129756.15" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	27227	27227	1.000	+	.	Sequence "AF129756.16" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	27837	27837	1.000	+	.	Sequence "AF129756.17" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	29328	29328	1.000	+	.	Sequence "AF129756.18" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	29458	29458	1.000	+	.	Sequence "AF129756.19" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	29629	29629	1.000	+	.	Sequence "AF129756.20" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	29646	29646	1.000	+	.	Sequence "AF129756.21" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	30838	30838	1.000	+	.	Sequence "AF129756.22" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	31349	31349	1.000	+	.	Sequence "AF129756.23" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	31901	31901	1.000	+	.	Sequence "AF129756.24" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	36682	36682	1.000	+	.	Sequence "AF129756.25" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	38225	38226	1.000	+	.	Sequence "AF129756.26" ; note "Insertion of 2 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	38379	38379	1.000	+	.	Sequence "AF129756.27" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	38537	38537	1.000	+	.	Sequence "AF129756.28" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	39114	39114	1.000	+	.	Sequence "AF129756.29" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	39816	39816	1.000	+	.	Sequence "AF129756.30" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	40807	40807	1.000	+	.	Sequence "AF129756.31" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	40977	40977	1.000	+	.	Sequence "AF129756.32" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	41204	41204	1.000	+	.	Sequence "AF129756.33" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	42548	42548	1.000	+	.	Sequence "AF129756.34" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	45315	45315	1.000	+	.	Sequence "AF129756.35" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	48382	48382	1.000	+	.	Sequence "AF129756.36" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	50635	50635	1.000	+	.	Sequence "AF129756.37" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	50809	50809	1.000	+	.	Sequence "AF129756.38" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	51286	51286	1.000	+	.	Sequence "AF129756.39" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	51645	51645	1.000	+	.	Sequence "AF129756.40" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	52388	52388	1.000	+	.	Sequence "AF129756.41" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	52646	52646	1.000	+	.	Sequence "AF129756.42" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	53596	53596	1.000	+	.	Sequence "AF129756.43" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	53621	53621	1.000	+	.	Sequence "AF129756.44" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	54883	54883	1.000	+	.	Sequence "AF129756.45" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	55377	55377	1.000	+	.	Sequence "AF129756.46" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	55571	55571	1.000	+	.	Sequence "AF129756.47" ; note "SNP in BA000025" ; replace "c"


<font color=red>  [Part of this file has been deleted for brevity]</font>

AF129756	diffseq	conflict	135941	135946	1.000	+	.	Sequence "AF129756.156" ; note "BA000025" ; replace "gagtt"
AF129756	diffseq	conflict	136800	136802	1.000	+	.	Sequence "AF129756.157" ; note "Insertion of 3 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	137143	137143	1.000	+	.	Sequence "AF129756.158" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	137303	137303	1.000	+	.	Sequence "AF129756.159" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	137335	137335	1.000	+	.	Sequence "AF129756.160" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	137597	137597	1.000	+	.	Sequence "AF129756.161" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	137886	137886	1.000	+	.	Sequence "AF129756.162" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	138985	138993	1.000	+	.	Sequence "AF129756.163" ; note "Insertion of 9 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	139070	139070	1.000	+	.	Sequence "AF129756.164" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	140737	140744	1.000	+	.	Sequence "AF129756.165" ; note "BA000025" ; replace "tttttt"
AF129756	diffseq	conflict	140794	140794	1.000	+	.	Sequence "AF129756.166" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	140979	140979	1.000	+	.	Sequence "AF129756.167" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	144318	144318	1.000	+	.	Sequence "AF129756.168" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	145509	145509	1.000	+	.	Sequence "AF129756.169" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	147960	147960	1.000	+	.	Sequence "AF129756.170" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	147977	147977	1.000	+	.	Sequence "AF129756.171" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	148139	148139	1.000	+	.	Sequence "AF129756.172" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	148568	148568	1.000	+	.	Sequence "AF129756.173" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	149373	149373	1.000	+	.	Sequence "AF129756.174" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	151782	151783	1.000	+	.	Sequence "AF129756.175" ; note "Insertion of 2 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	154604	154604	1.000	+	.	Sequence "AF129756.176" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	155080	155080	1.000	+	.	Sequence "AF129756.177" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	155714	155714	1.000	+	.	Sequence "AF129756.178" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	156664	156664	1.000	+	.	Sequence "AF129756.179" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	156689	156690	1.000	+	.	Sequence "AF129756.180" ; note "Insertion of 2 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	156933	156933	1.000	+	.	Sequence "AF129756.181" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	161113	161113	1.000	+	.	Sequence "AF129756.182" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	161187	161187	1.000	+	.	Sequence "AF129756.183" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	161642	161646	1.000	+	.	Sequence "AF129756.184" ; note "BA000025" ; replace "cctct"
AF129756	diffseq	conflict	162430	162430	1.000	+	.	Sequence "AF129756.185" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	162692	162692	1.000	+	.	Sequence "AF129756.186" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	166509	166509	1.000	+	.	Sequence "AF129756.187" ; note "BA000025" ; replace "ga"
AF129756	diffseq	conflict	166756	166756	1.000	+	.	Sequence "AF129756.188" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	167262	167262	1.000	+	.	Sequence "AF129756.189" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	170308	170308	1.000	+	.	Sequence "AF129756.190" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	173995	173995	1.000	+	.	Sequence "AF129756.191" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	175340	175340	1.000	+	.	Sequence "AF129756.192" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	177038	177038	1.000	+	.	Sequence "AF129756.193" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	177393	177393	1.000	+	.	Sequence "AF129756.194" ; note "SNP in BA000025" ; replace "g"
AF129756	diffseq	conflict	177509	177509	1.000	+	.	Sequence "AF129756.195" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	177681	177681	1.000	+	.	Sequence "AF129756.196" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	177751	177751	1.000	+	.	Sequence "AF129756.197" ; note "SNP in BA000025" ; replace "c"
AF129756	diffseq	conflict	178054	178054	1.000	+	.	Sequence "AF129756.198" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	179324	179324	1.000	+	.	Sequence "AF129756.199" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	181023	181023	1.000	+	.	Sequence "AF129756.200" ; note "SNP in BA000025" ; replace "a"
AF129756	diffseq	conflict	181595	181595	1.000	+	.	Sequence "AF129756.201" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	182424	182424	1.000	+	.	Sequence "AF129756.202" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	182732	182732	1.000	+	.	Sequence "AF129756.203" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	183597	183597	1.000	+	.	Sequence "AF129756.204" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	183931	183931	1.000	+	.	Sequence "AF129756.205" ; note "SNP in BA000025" ; replace "t"
AF129756	diffseq	conflict	184423	184423	1.000	+	.	Sequence "AF129756.206" ; note "SNP in BA000025" ; replace "c"
</pre>
</td></tr></table><p>
<p><h3>File: BA000025.diffgff</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
##gff-version 2.0
##date 2006-07-15
##Type DNA BA000025
BA000025	diffseq	conflict	194109	194109	1.000	+	.	Sequence "BA000025.1" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	197911	197914	1.000	+	.	Sequence "BA000025.2" ; note "Insertion of 4 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	198249	198254	1.000	+	.	Sequence "BA000025.3" ; note "AF129756" ; replace "cctc"
BA000025	diffseq	conflict	198912	198914	1.000	+	.	Sequence "BA000025.4" ; note "Insertion of 3 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	200847	200847	1.000	+	.	Sequence "BA000025.5" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	201795	201795	1.000	+	.	Sequence "BA000025.6" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	202273	202273	1.000	+	.	Sequence "BA000025.7" ; note "Insertion of 1 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	202466	202466	1.000	+	.	Sequence "BA000025.8" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	202655	202658	1.000	+	.	Sequence "BA000025.9" ; note "Insertion of 4 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	204951	204953	1.000	+	.	Sequence "BA000025.10" ; note "AF129756" ; replace "tat"
BA000025	diffseq	conflict	206600	206600	1.000	+	.	Sequence "BA000025.11" ; note "Insertion of 1 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	206868	206868	1.000	+	.	Sequence "BA000025.12" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	208218	208221	1.000	+	.	Sequence "BA000025.13" ; note "Insertion of 4 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	209096	209096	1.000	+	.	Sequence "BA000025.14" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	211149	211149	1.000	+	.	Sequence "BA000025.15" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	213718	213718	1.000	+	.	Sequence "BA000025.16" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	214248	214248	1.000	+	.	Sequence "BA000025.17" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	214419	214419	1.000	+	.	Sequence "BA000025.18" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	219643	219643	1.000	+	.	Sequence "BA000025.19" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	220149	220149	1.000	+	.	Sequence "BA000025.20" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	221316	221319	1.000	+	.	Sequence "BA000025.21" ; note "Insertion of 4 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	221797	221797	1.000	+	.	Sequence "BA000025.22" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	223288	223288	1.000	+	.	Sequence "BA000025.23" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	223418	223418	1.000	+	.	Sequence "BA000025.24" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	223589	223589	1.000	+	.	Sequence "BA000025.25" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	223606	223606	1.000	+	.	Sequence "BA000025.26" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	224798	224798	1.000	+	.	Sequence "BA000025.27" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	225309	225309	1.000	+	.	Sequence "BA000025.28" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	225861	225861	1.000	+	.	Sequence "BA000025.29" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	228039	228040	1.000	+	.	Sequence "BA000025.30" ; note "Insertion of 2 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	230644	230644	1.000	+	.	Sequence "BA000025.31" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	232339	232339	1.000	+	.	Sequence "BA000025.32" ; note "SNP in AF129756" ; replace "g"
BA000025	diffseq	conflict	232497	232497	1.000	+	.	Sequence "BA000025.33" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	233074	233074	1.000	+	.	Sequence "BA000025.34" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	233776	233776	1.000	+	.	Sequence "BA000025.35" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	234767	234767	1.000	+	.	Sequence "BA000025.36" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	235163	235163	1.000	+	.	Sequence "BA000025.37" ; note "SNP in AF129756" ; replace "g"
BA000025	diffseq	conflict	236507	236507	1.000	+	.	Sequence "BA000025.38" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	237760	237762	1.000	+	.	Sequence "BA000025.39" ; note "Insertion of 3 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	238680	238683	1.000	+	.	Sequence "BA000025.40" ; note "Insertion of 4 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	242347	242347	1.000	+	.	Sequence "BA000025.41" ; note "SNP in AF129756" ; replace "g"
BA000025	diffseq	conflict	242637	242638	1.000	+	.	Sequence "BA000025.42" ; note "Insertion of 2 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	244602	244602	1.000	+	.	Sequence "BA000025.43" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	244776	244776	1.000	+	.	Sequence "BA000025.44" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	245253	245253	1.000	+	.	Sequence "BA000025.45" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	246354	246354	1.000	+	.	Sequence "BA000025.46" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	246612	246612	1.000	+	.	Sequence "BA000025.47" ; note "SNP in AF129756" ; replace "g"


<font color=red>  [Part of this file has been deleted for brevity]</font>

BA000025	diffseq	conflict	331285	331285	1.000	+	.	Sequence "BA000025.158" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	331547	331547	1.000	+	.	Sequence "BA000025.159" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	331836	331836	1.000	+	.	Sequence "BA000025.160" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	333011	333011	1.000	+	.	Sequence "BA000025.161" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	334349	334356	1.000	+	.	Sequence "BA000025.162" ; note "Insertion of 8 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	334686	334691	1.000	+	.	Sequence "BA000025.163" ; note "AF129756" ; replace "atttttta"
BA000025	diffseq	conflict	334741	334741	1.000	+	.	Sequence "BA000025.164" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	334926	334926	1.000	+	.	Sequence "BA000025.165" ; note "SNP in AF129756" ; replace "g"
BA000025	diffseq	conflict	337483	337497	1.000	+	.	Sequence "BA000025.166" ; note "Insertion of 15 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	338280	338280	1.000	+	.	Sequence "BA000025.167" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	339471	339471	1.000	+	.	Sequence "BA000025.168" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	339761	339762	1.000	+	.	Sequence "BA000025.169" ; note "Insertion of 2 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	341940	341940	1.000	+	.	Sequence "BA000025.170" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	342102	342102	1.000	+	.	Sequence "BA000025.171" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	342531	342531	1.000	+	.	Sequence "BA000025.172" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	343336	343336	1.000	+	.	Sequence "BA000025.173" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	343800	343801	1.000	+	.	Sequence "BA000025.174" ; note "Insertion of 2 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	348567	348567	1.000	+	.	Sequence "BA000025.175" ; note "SNP in AF129756" ; replace "g"
BA000025	diffseq	conflict	349043	349043	1.000	+	.	Sequence "BA000025.176" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	350892	350892	1.000	+	.	Sequence "BA000025.177" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	352621	352640	1.000	+	.	Sequence "BA000025.178" ; note "Insertion of 20 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	353771	353771	1.000	+	.	Sequence "BA000025.179" ; note "Insertion of 1 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	355093	355093	1.000	+	.	Sequence "BA000025.180" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	355167	355167	1.000	+	.	Sequence "BA000025.181" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	355622	355626	1.000	+	.	Sequence "BA000025.182" ; note "AF129756" ; replace "tctcc"
BA000025	diffseq	conflict	356410	356410	1.000	+	.	Sequence "BA000025.183" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	356672	356672	1.000	+	.	Sequence "BA000025.184" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	360489	360490	1.000	+	.	Sequence "BA000025.185" ; note "AF129756" ; replace "t"
BA000025	diffseq	conflict	360737	360737	1.000	+	.	Sequence "BA000025.186" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	361243	361243	1.000	+	.	Sequence "BA000025.187" ; note "SNP in AF129756" ; replace "g"
BA000025	diffseq	conflict	364289	364289	1.000	+	.	Sequence "BA000025.188" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	367976	367976	1.000	+	.	Sequence "BA000025.189" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	369321	369321	1.000	+	.	Sequence "BA000025.190" ; note "SNP in AF129756" ; replace "g"
BA000025	diffseq	conflict	371019	371019	1.000	+	.	Sequence "BA000025.191" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	371374	371374	1.000	+	.	Sequence "BA000025.192" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	371490	371490	1.000	+	.	Sequence "BA000025.193" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	371662	371662	1.000	+	.	Sequence "BA000025.194" ; note "SNP in AF129756" ; replace "g"
BA000025	diffseq	conflict	371732	371732	1.000	+	.	Sequence "BA000025.195" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	372035	372035	1.000	+	.	Sequence "BA000025.196" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	372964	372964	1.000	+	.	Sequence "BA000025.197" ; note "Insertion of 1 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	373306	373306	1.000	+	.	Sequence "BA000025.198" ; note "SNP in AF129756" ; replace "t"
BA000025	diffseq	conflict	374230	374231	1.000	+	.	Sequence "BA000025.199" ; note "Insertion of 2 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	375007	375007	1.000	+	.	Sequence "BA000025.200" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	375579	375579	1.000	+	.	Sequence "BA000025.201" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	376341	376352	1.000	+	.	Sequence "BA000025.202" ; note "Insertion of 12 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	376727	376727	1.000	+	.	Sequence "BA000025.203" ; note "SNP in AF129756" ; replace "a"
BA000025	diffseq	conflict	377379	377381	1.000	+	.	Sequence "BA000025.204" ; note "Insertion of 3 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	377595	377595	1.000	+	.	Sequence "BA000025.205" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	377789	377790	1.000	+	.	Sequence "BA000025.206" ; note "Insertion of 2 bases in BA000025" ; replace ""
BA000025	diffseq	conflict	377931	377931	1.000	+	.	Sequence "BA000025.207" ; note "SNP in AF129756" ; replace "c"
BA000025	diffseq	conflict	378423	378423	1.000	+	.	Sequence "BA000025.208" ; note "SNP in AF129756" ; replace "t"
</pre>
</td></tr></table><p>

<p>

The first line is the title giving the names of the sequences used.
<p>
The next two non-blank lines state the positions in each sequence
where the detected overlap between them starts.
<p>
There then follows a set of reports of the mismatches between the sequences.
<br>
Each report consists of 4 or more lines.
<br>
<ul>

<li>The first line has the name of the first sequence followed by the
start and end positions of the mismatched region in that sequence,
followed by the length of the mismatched region. If the mismatched
region is of zero length in this sequence, then only the position of
the last matching base before the mismatch is given.</li>

<li>If a feature of the first sequence overlaps with this mismatch
region, then one or more lines starting with 'Feature:' comes next
with the type, position and tag field of the feature.</li>

<li>Next is a line starting "Sequence:" giving the sequence of the
mismatch in the first sequence.</li> </ul>

<p>

This is followed by the equivalent information for the second
sequence, but in the reverse order, namely 'Sequence:' line,
'Feature:' lines and line giving the position of the mismatch in the
second sequence.
<p>

At the end of the report are two non-blank lines giving the positions in each
sequence where the detected
overlap between them ends.
<p>

The last three lines of the report gives the counts of SNPs (defined
as a change of one nucleotide to one other nucleotide, no deletions or
insertions are counted, no multi-base changes are counted).
<p>

If the input sequences are nucleic acid, The counts of transitions
(Pyrimide to Pyrimidine or Purine to Purine) and transversions
(Pyrimidine to Purine) are also given. 

<p>

It should be noted that not all features are reported.

<p>

The 'source' feature found in all EMBL/Genbank feature table entries is
not reported as this covers all of the sequence and so overlaps with any
difference found in that sequence and so is uninformative and
irritating.  It has therefore been removed from the output report. 

<p>

The translation information of CDS features is often extremely long and
does not add useful information to the report.  It has therefore been
removed from the output report. 

<H2>
    Data files
</H2>

None

<H2>
    Notes
</H2>

It should be noted that not all features are reported.

<p>

The 'source' feature found in all EMBL/Genbank feature table entries is
not reported as this covers all of the sequence and so overlaps with any
difference found in that sequence and so is uninformative and
irritating.  It has therefore been removed from the output report. 

<p>

The translation information of CDS features is often extremely long and
does not add useful information to the report.  It has therefore been
removed from the output report. 

<p>

If you run out of memory, use a larger word size.

<p>

Using a larger word size increases the length between mismatches that
will be reported as one event.  Thus a word size of 50 will report two
single-base differences that are with 50 bases of each other as one mismatch. 

   
<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
</table>

<H2>
    Author(s)
</H2>

Gary Williams (gwilliam&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK


<H2>
    History
</H2>

Written 15th Aug 2000 - Gary Williams.
<p>
18th Aug 2000 - Added writing out GFF files of the mismatched regions

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.

<H2>
    Comments
</H2>
None


</BODY>
</HTML>

