<HTML>

<HEAD>
  <TITLE>
  EMBOSS
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="index.html" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

diffseq
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>





<!--END OF HEADER-->





<H2>
    Function
</H2>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->
Find differences (SNPs) between nearly identical sequences


<H2>
    Description
</H2>

<!-- 
	This is the main part of the document.

        This is what the user looks at to find out if this program will
        do what he wants. 

        Long description.

        Why it is useful.

        When it should/shouldn't be used.

        Algorithms

        Expanded description.

        eg: Reverse engineers Wheedle's 'talpsgob' program. This does a 
        splurge of the 5' end of the sequence followed by a durm.
        The net effect is a 5' end splurt.
        etc. for as long as is required (several pages is not unreasonable)
  -->

   <b>diffseq</b> takes two overlapping, nearly identical sequences and reports the
   differences between them, together with any features that overlap with these
   regions. GFF files of the differences in each sequence are also produced.
<p>
   <b>diffseq</b> should be of value when looking for SNPs, differences between
   strains of an organism and anything else that requires the differences between
   sequences to be highlighted.
<p>  
   The sequences can be very long. The program does a match of all
   sequence words of size 10 (by default). It then reduces this to the
   minimum set of overlapping matches by sorting the matches in order of
   size (largest size first) and then for each such match it removes any
   smaller matches that overlap. The result is a set of the longest
   ungapped alignments between the two sequences that do not overlap with
   each other. The mismatched regions between these matches are reported.
<p>   
   It should be possible to find differences between sequences that are Mega bytes
   long.
<p>
     



<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->


<b>Here is a sample session with diffseq</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>diffseq tembl:ap000504 tembl:af129756</b>
Find differences (SNPs) between nearly identical sequences
Word size [10]: <b></b>
Output report [ap000504.diffseq]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>


<!--
Here is a sample session with diffseq:

<pre>

% diffseq tembl:ap000504 tembl:af129756
Find differences (SNPs) between nearly identical sequences
Word size [10]: 
Output file [ap000504.diffseq]:

</pre>
-->

<H2>
    Command line arguments
</H2>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Mandatory qualifiers:
  [-asequence]         sequence   Sequence USA
  [-bsequence]         sequence   Sequence USA
   -wordsize           integer    The similar regions between the two
                                  sequences are found by creating a hash table
                                  of 'wordsize'd subsequences. 10 is a
                                  reasonable default. Making this value larger
                                  (20?) may speed up the program slightly,
                                  but will mean that any two differences
                                  within 'wordsize' of each other will be
                                  grouped as a single region of difference.
                                  This value may be made smaller (4?) to
                                  improve the resolution of nearby
                                  differences, but the program will go much
                                  slower.
   -outfile            report     Output report file

   Optional qualifiers:
   -afeatout           featout    File for output of first sequence's normal
                                  tab delimited gff's
   -bfeatout           featout    File for output of second sequence's normal
                                  tab delimited gff's
   -columns            boolean    The default format for the output report
                                  file is to have several lines per difference
                                  giving the sequence positions, sequences
                                  and features.
                                  If this option is set true then the output
                                  report file's format is changed to a set of
                                  columns and no feature information is given.

   Advanced qualifiers: (none)
   General qualifiers:
  -help                boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose


</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#f5f5ff">
<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Mandatory qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-asequence]<br>(Parameter 1)</td>
<td>Sequence USA</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-bsequence]<br>(Parameter 2)</td>
<td>Sequence USA</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-wordsize</td>
<td>The similar regions between the two sequences are found by creating a hash table of 'wordsize'd subsequences. 10 is a reasonable default. Making this value larger (20?) may speed up the program slightly, but will mean that any two differences within 'wordsize' of each other will be grouped as a single region of difference. This value may be made smaller (4?) to improve the resolution of nearby differences, but the program will go much slower.</td>
<td>Integer 2 or more</td>
<td>10</td>
</tr>

<tr>
<td>-outfile</td>
<td>Output report file</td>
<td>Report output file</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Optional qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-afeatout</td>
<td>File for output of first sequence's normal tab delimited gff's</td>
<td>Writeable feature table</td>
<td>$(asequence.name).diffgff</td>
</tr>

<tr>
<td>-bfeatout</td>
<td>File for output of second sequence's normal tab delimited gff's</td>
<td>Writeable feature table</td>
<td>$(bsequence.name).diffgff</td>
</tr>

<tr>
<td>-columns</td>
<td>The default format for the output report file is to have several lines per difference giving the sequence positions, sequences and features.
If this option is set true then the output report file's format is changed to a set of columns and no feature information is given.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Advanced qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>



<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        Except for sequences which can be described by 'sequence USA',
        this should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

This program reads in two nucleic acid sequence USAs or two protein
sequence USAs. 

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:ap000504' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:ap000504</h3>
<table width="90%"><tr><td bgcolor="#FFDDFF">
<pre>
&gt;AP000504 AP000504.1 Homo sapiens genomic DNA, chromosome 6p21.3, HLA Class I region, section 3/20.
gaccaatctcactgtgaggaggcagtcaaagggaataatggaagagaggaagaggatttt
ctcagtggcagtcatggcgtctgggatgaaggagtagtttccagaaaggaggcgttgttt
gcttatctccagacctatttgagggaggcaagcaaagggaacggtcttgtagctcaattt
tttcaccccattttaagaatgagacaatagaagcaagagagattatttgacttgcccaag
ctcacacaggcagttaatggaaagctagagcaagaaccaaattttcagactcttagtcta
attctctttttattctacatataatataaagatacttgtctgaaagcacagcctgagaaa
gataaatggctgaggaaagtagacatctgtctggaattgaggattttggtcaaaataatg
gtattaatagaactagtaacactaatgccttaatatctaattaggatagtacactcctgt
tcttattgtaaacctaggaaagttatagaagtgccttatggatcataataagggtcactg
aggcagtgccttttggtttggtgataaaaggctttaacttaatggggagaattccaacaa
taaaaccctgtccaaaaagtgtcaccactcctcaggggaggccctcatccctagacatga
cttaagcagaggcttcccaataagctgcaggttattaaagggtagggagcaggagagatc
ttggggggacaggtcatagggcatgaggagcacaaaggtttaggatgacataaggcagag
gggagatctgtgatgatgaaggtagagttgggggaaagaatgggacaccggaacagggag
ttaggcaaagcaaaaggaaggagataccaaaatccacacttggcaaaaatatgatttcag
gtcttttaggctctctgtgctcctgggaggctgtgggggaggaaagaaaaggctatcatt
ctttacatctcagtccttctacctctgtctgacactccctctcacccaattctagccccc
tggaatattccatatattagtccttccccattttccctctatcctttaccaagtccttac
caagctttcccagaaatcgagtcatattctcatcctgtttggcactcgtaacaacagact
ggggattgatctcatccagaacttggaaggagaacagagatcaaatgagttaaaggatct
ttgtctttgactaagagaaaacccatagccctcctcttcctacccctctccttctcaaaa
acatttcctccctaggagtagggagtgctctgcacagtgggaacacaggtagaagttgag
atttagaaaagtagttaagagtggtgggatggtgagagggaagtgggatgttctggatgt
tgtcactaggctgtaaacccctggagaacagacatgactgatttgcccagggctgaatct
gaagcacctgaaacattgtaaatacgtcatatatatttgtggccaggcacagtggctcat
gcctataatccctgccctttgggaggccaaggcaggcagatcactggaggccaggagctc
aagacaagcctagccaacgtggtgaaaccctgcctctactaaaaatataaaaattagcca
ggcgtgatggcagattcttgtaatcccagctactcgggagactgaggcaggagaattgct
tgaatccgggagacggaggttgcagtgagccaagatggcaccactacacttccagcctga
gtgacggagcaagacactgtctcaaaaaagaacaaccaaacaaaccaaaaaacagcctca
caaatatttgttaaataatgaaatgaattcataaaaacaaaagagggagcctctgtgaag
caactgtaaaatatattgagtcagtgctatagtttggatgtgatttgtccctgccaaata
tcgtgttgaaatttaatccccagtgtgatagtgttgtgaggtagggcctagcaggaggtg
tgtgggtgatgggagtggatcgctcatgaacagattaatgcccttcctggagtgtgttgg


<font color=red>  [Part of this file has been deleted for brevity]</font>

aaccccatctctactgaaaatataaacattagctggatgtagtggtgcacacctgtaatc
ccagctactcaggaggctgaggcaggagaatcgctagaactcgggaggagggggttgcag
tgagccgagattgcactactgcactctagcctgggtgacagagcgagactgtctcaaaaa
acaaaacaaaacaaaaaaacacacattgcaacaaaacaatttctctctaaacctgtaagt
gattttgtcctcccttacagagaaggtgataatctttgctgtaagcactgtcctcgtatc
gtaccccttgtgcccctgaatgaatttagaaaatgtaaagtacaggagatcagtatatga
tgacttactgattcatagtagtgttttaataggatgttccttatgtgaataagatataat
ttatttgcaaagatttggtctacatgtaaacttccaaggatataactgaaagttttggag
gacatggtattctcagtaggcattattgcttttattagtgagatggactccagcttgata
ttttctgcctttttgtgtttggctggttgtgcgcagcacgagggccgggaggaggatcag
cggttgatcaacttggtaggggagagcctgcgactgctgggcaacacctttgttgcactg
tctgacctgcgctgcaatctggcctgcacgcccccacgacacctgcatgtggtccggcct
atgtctcactacaccacccccatggtgctccagcaggcagccattcccatacaggtgggt
tagggggagtctggcctgagggagagtgaggggtgttgatagagtgacccagggtagcta
ctgggcctgaaggaggttaggaaaggaggagactggaaacatggtgatgaaggctggaga
tactttagaggtttatcatgaggttttcttggttaggctcttgtatttttctcacatctg
cctgtccatctgtctttttcagatcaatgtgggaaccactgtgaccatgacaggaaatgg
gactcggccccccccaactcccaatgcagaggcacctccccctggtcctgggcaggcctc
atccgtggctccgtcttctaccaatgtcgagtcctcagctgagggggctcccccgccagg
tccagctcccccgccagccaccagccacccgagggtcatccggatttcccaccagagtgt
ggaacccgtggtcatgatgcacatgaacattcaaggtgagaatagttgctggcgagaaga
gcaggatcagcatgatgagggaggttcatgctgaggtgtgagggaacagggtggggaagg
gagaggcacatgctggtggtggtagcctggggaccagagcagaagcttaagtagacagat
gtggggggtgtgggggttggtttgtctttggaggtgtgtttgtgtggtgaagggagtacc
tctccctgtttagatggagggaaaggcaggctttctgattgggggattatgggcctgaag
tatgcctgatctcagaaggatatagttaggccttggccctacctacctcagggccactgt
ctctgtctccctgcccagattctggcacacagcctggtggtgttccgagtgctcccactg
gccccctgggaccccctggtcatggccaaaccctgggtaagagtgagggcatcagggcag
gctgagctctgggtagagaaagggaagggctgagtgggtgggttgaaggggtccaggttc
aaggttacatcagacccgccccccaggctccaccctcatccagctgccctccctgccccc
tgagttcatgcacgccgtcgcccaccagatcactcatcaggccatggtggcagctgttgc
ctccgcggccgcaggtaatgacctggaaggggaggcttgggaggtagggcacagtccatg
gtggcagctggctggcaagggcctggccctcagccctcttcggtctgtctcttctgccac
ccacaggacagcaggtgccaggcttcccaacagctccaacccgggtggtgattgcccggc
ccactcctccacaggctcggccttcccatcctggagggcccccagtctctgggacactgg
tgagcaagggtcggggagttctagtgcgtaacagtctagg
</pre>
</td></tr></table><p>
<p><h3>Database entry: tembl:af129756</h3>
<table width="90%"><tr><td bgcolor="#FFDDFF">
<pre>
&gt;AF129756 AF129756.1 Homo sapiens MSH55 gene, partial cds; and CLIC1, DDAH, G6b, G6c, G5b, G6d, G6e, G6f, BAT5, G5b, CSK2B, BAT4, G4, Apo M, BAT3, BAT2, AIF-1, 1C7, LST-1, LTB, TNF, and LTA genes, complete cds.
gaattctctccctcccatctgtggctgagattaaagatctgcacctgaagcactgaagaa
tgtgtgggtaattaaattaccctgccgattcctggagatgctgattacctggagatgacc
tcagagattatcctaaattaactcctacaagacacacattgcagcgggaggtgaggtagg
ggaaggattgtgcacctgaggctagcaaaggtttccactctgtttagagatgatgtcacc
agtgcgtttacatttgcgttgtgttcacacattgagtgctactatgtacaagaccatgtg
tcagacactgaggtaacaggtatagctactagatagagttcatagctatggaggcaagca
gattcactgactgctaattctaacatgatgtgacagtgcaactagaaaaacataaacaag
cactatgtgagcacaaagaaggtgcacatcaactccttacaggtacctgtaaaagccaaa
gggtaacagttggattgcaccttgaagaggatgcacttttttttttttttaagacagaat
ctcactctgttgcccaggctggagtgcagtggggcaatctgggctcacttcaacctctac
ctcccgagttcaagcaattctcctgcctcagcctcctgagtagctggtactagaggcatg
cgccatgatgctgggctaatttttgtattttcggtagacgtgaagtttcaccaagttggc
caggctggtcttgaactcctgacctcaaatcatccacccacctcagcctcccaaaatgct
gagactacaggcgtgagccaccgcgcctgacctggatgtaagattttgataggtacagaa
caaggaaaagactttccaggccgggcacagtggcttatgcctgtaatcccagcactttgg
gaggccgaggtgggcagatcacgaggtcagcagttcaagaccagcgtggccaacatggtg
aaatcccatctctactaaaaatacaaaaattagccaggcatggtggtgggtccctgtaat
cccagctactccggaggctgaggcaggagaattgcttgaacctgggaggtggaggttgca
gtgagcaaagaccgcgccactgcactccagcctgggtgacagagggagactccgtctcaa
aaaaaaaaaaaaaagactttccagaaggagcagcataaacacaggcatgacatgtttcca
taatggcaagtggccctaaatgactagaatataaggtagatccagtaggaaaggacttag
aaggggctttggaaggtgagtctggaaattaaaactggggtaaacgtgatggaccctgaa
catcattatactgcttaagatgctaatcttaatcctgaaggtaatgggaaaacctcctaa
ggtttatgttattttctttctacttaggctatttaaaaagtggagtgacggccaggcgca
gtgactcatgcctgtaatcccagcactttgggaggccgaggtgggcggatcaccaggagt
tcgagaccagcctgaccaacatggtgaaaccccgcctctactgaaaatacaaaaattagc
caggtgtggtggtgggcgcctgtaatcccagctacttgggaggctgaggcagaagaattg
cttgaacccgggaagtggaggttgcagtgagcagagatcgtgccattgtactccagcctg
ggcaacaagagcgaaactcagtcacaaaaaaaaaaaaaaaaaaaaaggagtgacatgctt
agatctctgttttggaatgacaggttttttgtttctagcatcaatccaaggttcatggct
tgagaaggtgtactgccagcaatgccattaaccagcaaagggaatgcaggaagaggaaca
gatctggtgggcatcagtttggatgctctgagtttgagctgcctgtgaaaactgcaggtg
gtgatatgcaattaacattcacatacggagttcaaaactagagacacaaatttgagagtc
atcacagaaatgtgaagtgtgttttctataactaaagataaccatgctaacatagccatg


<font color=red>  [Part of this file has been deleted for brevity]</font>

ccactgcaacctcaaaaccatgggctcaagtcatccttccacctcagcttcccaagtatc
taggactacaggtgtgtgccactgtgcctggctaattttaaaaaatattttaaaattttt
gttgagacagggtctatgctgctcaggctggtctcgaactcctggtttcaagtgatctgc
ccatcttggcctcccaaagtttttttttgttagtttgagaggcggtttcgctcgttgccc
aggctggagtgcaatgactgatctcatctcactgcaacctctgcctcctgggttcaagcg
attctcctgcttcagcctcccaagtagctgggattacaggtgcatgccaccattcccggc
taattttttgtatttagtagagatggggtttcaccatgttagtcaggctgatctcaaact
cctgacctcaggtgatccgcctgcctcagcctcccaaagttttgggattacaggtgtgag
ccaccatgctgggccagcctcccaaagttttgggattacaggcatgagtcaccacactgg
ccctggattttttttctttcttttttttggagacggagtctcactctgttgcccaggctg
gagtgcaatggcgtaatctcagctcactgcaacctctgctgcccgggttcaaacgattct
cctgtcttagcctcctgagtagctgggattataggtgcatgccaccatgcctggctaatt
tttgtacttttagtagagaaagtacaccatcttggccaggctggtctcgaactcctgacc
tcaggtgatccacttgcgtcggcctcccaaagtgctgggattacaggcgtgagacaccgc
acccagcctttttttttttttttcttttaagacagaatcgctctgtcacccaggctggag
tgcagtggcacaatctcggctcactgcaacctctgcctcccaggtttaagcaatccacct
atgtcagtctcccaagtagctgggattataggtgcatgtcaccatgcctggctaattttt
gtacttttagtatagaaagtacaccatgttggccaggctggtcttgaactcctgacctca
agtgatccgcctgcctcagcctcccgaagtgctggaattacagacatgtgccactgcacc
cggcctggttttttttttctaagagatggagtctcacttttctgcccaggttggagtgca
atggcaccatcatagctcactgcagccttcaactcttggcctcaggcaatccttgcacct
tagcctcgcagtgttgggattacaggcatgagccactgagccttgcctggactttttttt
ttttttgagatggcgtctcgctctgttgcccaggttggagtgctacggcatgatcttggc
tcactgcaacttccacctcccaggttcaagcgattctcttgcctcggccccccgagtagc
tgggattacaggcatgcgccaccgtgcctggctaattttggtatttttagtagagatagg
gtttcatcatgttgggcaggctggtcttgaactcctgacctcgtgatccacccacctcgg
cctcccaaagtgctgggattataggcatagccaacgcgcccagcctggacttgtttttaa
aagatcactgtggctcctgtgtttaggctggctggtaggagacaggtggcagtggcattg
atggtgaagagaaaatagtggcagccatggagatggagagaagtagacaagtttgggata
tattatacattccaggggtagaaacaacaggactagatgatggattgatgggtgggagat
gtagatactgggagagaagcaggattctgatggatggaaaaactaaaaaattctattttg
ggtgtggtaagtctaagtctattagacatgcaagtagagatgtcactgggcagatacaca
tctggatttcaggggcaaggtccaagctagagaaagaaacctgggcatggtcagcatgag
gatggtgtttaaagccatggaacttatcttgtgcatccctataagacccctttgaggcac
ttgtttcccctcacaatggatgcagtgcatcttccattctgaattccagaggcaacaacc
tcctgctcctagaagctaaactctccagacttagtcttctgaattc
</pre>
</td></tr></table><p>



<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats 
        Except for sequences which can be described by 'sequence USA',
        this should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 
   -->

A report of the differences between the two sequences is produced,
together with any features that overlap with these differing regions. 
<p>

<p>

The output is a standard EMBOSS report file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <b>-rformat xxx</b>, where 'xxx' is replaced by
the name of the required format.  The available format names are: embl,
genbank, gff, pir, swiss, trace, listfile, dbmotif, diffseq, excel,
feattable, motif, regions, seqtable, simple, srs, table, tagseq

<p>

See:
<A href="http://www.uk.embnet.org/Software/EMBOSS/Themes/ReportFormats.html">
http://www.uk.embnet.org/Software/EMBOSS/Themes/ReportFormats.html</A>
for further information on report formats.

<p>


<p>

By default <b>diffseq</b> writes a 'diffseq' report file.

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: AF129756.diffgff</h3>
<table width="90%"><tr><td bgcolor="#DDFFDD">
<pre>
##gff-version 2.0
##date 2002-08-23
##Type DNA AF129756
AF129756	diffseq	conflict	6882	6882	1.000	+	.	Sequence "AF129756.1" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	7830	7830	1.000	+	.	Sequence "AF129756.2" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	8500	8500	1.000	+	.	Sequence "AF129756.3" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	10945	10962	1.000	+	.	Sequence "AF129756.4" ; note "Insertion of 18 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	10999	11001	1.000	+	.	Sequence "AF129756.5" ; note "AP000504" ; replace "aaa"
AF129756	diffseq	conflict	12915	12915	1.000	+	.	Sequence "AF129756.6" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	15139	15139	1.000	+	.	Sequence "AF129756.7" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	17192	17192	1.000	+	.	Sequence "AF129756.8" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	19761	19761	1.000	+	.	Sequence "AF129756.9" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	20291	20291	1.000	+	.	Sequence "AF129756.10" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	20462	20462	1.000	+	.	Sequence "AF129756.11" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	25686	25686	1.000	+	.	Sequence "AF129756.12" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	26192	26192	1.000	+	.	Sequence "AF129756.13" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	27227	27227	1.000	+	.	Sequence "AF129756.14" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	27837	27837	1.000	+	.	Sequence "AF129756.15" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	29328	29328	1.000	+	.	Sequence "AF129756.16" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	29458	29458	1.000	+	.	Sequence "AF129756.17" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	29629	29629	1.000	+	.	Sequence "AF129756.18" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	29646	29646	1.000	+	.	Sequence "AF129756.19" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	30838	30838	1.000	+	.	Sequence "AF129756.20" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	31349	31349	1.000	+	.	Sequence "AF129756.21" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	31901	31901	1.000	+	.	Sequence "AF129756.22" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	36682	36682	1.000	+	.	Sequence "AF129756.23" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	38225	38226	1.000	+	.	Sequence "AF129756.24" ; note "Insertion of 2 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	38379	38379	1.000	+	.	Sequence "AF129756.25" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	38537	38537	1.000	+	.	Sequence "AF129756.26" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	39114	39114	1.000	+	.	Sequence "AF129756.27" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	39816	39816	1.000	+	.	Sequence "AF129756.28" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	40807	40807	1.000	+	.	Sequence "AF129756.29" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	40977	40977	1.000	+	.	Sequence "AF129756.30" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	41204	41204	1.000	+	.	Sequence "AF129756.31" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	42548	42548	1.000	+	.	Sequence "AF129756.32" ; note "SNP in AP000504" ; replace "g"


<font color=red>  [Part of this file has been deleted for brevity]</font>

AF129756	diffseq	conflict	80042	80042	1.000	+	.	Sequence "AF129756.70" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	80115	80115	1.000	+	.	Sequence "AF129756.71" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	81882	81882	1.000	+	.	Sequence "AF129756.72" ; note "AP000504" ; replace "tttggaat"
AF129756	diffseq	conflict	82132	82132	1.000	+	.	Sequence "AF129756.73" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	83649	83649	1.000	+	.	Sequence "AF129756.74" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	84290	84290	1.000	+	.	Sequence "AF129756.75" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	86465	86465	1.000	+	.	Sequence "AF129756.76" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	86842	86844	1.000	+	.	Sequence "AF129756.77" ; note "Insertion of 3 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	87014	87014	1.000	+	.	Sequence "AF129756.78" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	87102	87102	1.000	+	.	Sequence "AF129756.79" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	87605	87605	1.000	+	.	Sequence "AF129756.80" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	87893	87893	1.000	+	.	Sequence "AF129756.81" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	88359	88359	1.000	+	.	Sequence "AF129756.82" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	88635	88635	1.000	+	.	Sequence "AF129756.83" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	88750	88750	1.000	+	.	Sequence "AF129756.84" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	88822	88826	1.000	+	.	Sequence "AF129756.85" ; note "Insertion of 5 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	89118	89118	1.000	+	.	Sequence "AF129756.86" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	89738	89738	1.000	+	.	Sequence "AF129756.87" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	91271	91271	1.000	+	.	Sequence "AF129756.88" ; note "Insertion of 1 bases in AF129756" ; replace ""
AF129756	diffseq	conflict	92311	92311	1.000	+	.	Sequence "AF129756.89" ; note "SNP in AP000504" ; replace "g"
AF129756	diffseq	conflict	92345	92345	1.000	+	.	Sequence "AF129756.90" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	93979	93979	1.000	+	.	Sequence "AF129756.91" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	94959	94959	1.000	+	.	Sequence "AF129756.92" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	95246	95246	1.000	+	.	Sequence "AF129756.93" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	95809	95810	1.000	+	.	Sequence "AF129756.94" ; note "AP000504" ; replace "aat"
AF129756	diffseq	conflict	96756	96756	1.000	+	.	Sequence "AF129756.95" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	97713	97713	1.000	+	.	Sequence "AF129756.96" ; note "AP000504" ; replace "tgtgtgtgtgtgtgtgt"
AF129756	diffseq	conflict	97827	97827	1.000	+	.	Sequence "AF129756.97" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	98195	98195	1.000	+	.	Sequence "AF129756.98" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	99280	99280	1.000	+	.	Sequence "AF129756.99" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	99726	99726	1.000	+	.	Sequence "AF129756.100" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	99890	99890	1.000	+	.	Sequence "AF129756.101" ; note "SNP in AP000504" ; replace "t"
AF129756	diffseq	conflict	101481	101481	1.000	+	.	Sequence "AF129756.102" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	102680	102680	1.000	+	.	Sequence "AF129756.103" ; note "SNP in AP000504" ; replace "c"
AF129756	diffseq	conflict	103744	103744	1.000	+	.	Sequence "AF129756.104" ; note "SNP in AP000504" ; replace "a"
AF129756	diffseq	conflict	103855	103855	1.000	+	.	Sequence "AF129756.105" ; note "SNP in AP000504" ; replace "c"
</pre>
</td></tr></table><p>
<p><h3>File: AP000504.diffgff</h3>
<table width="90%"><tr><td bgcolor="#DDFFDD">
<pre>
##gff-version 2.0
##date 2002-08-23
##Type DNA AP000504
AP000504	diffseq	conflict	847	847	1.000	+	.	Sequence "AP000504.1" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	1795	1795	1.000	+	.	Sequence "AP000504.2" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	2273	2273	1.000	+	.	Sequence "AP000504.3" ; note "Insertion of 1 bases in AP000504" ; replace ""
AP000504	diffseq	conflict	2466	2466	1.000	+	.	Sequence "AP000504.4" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	2655	2658	1.000	+	.	Sequence "AP000504.5" ; note "Insertion of 4 bases in AP000504" ; replace ""
AP000504	diffseq	conflict	4951	4953	1.000	+	.	Sequence "AP000504.6" ; note "AF129756" ; replace "tat"
AP000504	diffseq	conflict	6600	6600	1.000	+	.	Sequence "AP000504.7" ; note "Insertion of 1 bases in AP000504" ; replace ""
AP000504	diffseq	conflict	6868	6868	1.000	+	.	Sequence "AP000504.8" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	8218	8221	1.000	+	.	Sequence "AP000504.9" ; note "Insertion of 4 bases in AP000504" ; replace ""
AP000504	diffseq	conflict	9096	9096	1.000	+	.	Sequence "AP000504.10" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	11149	11149	1.000	+	.	Sequence "AP000504.11" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	13718	13718	1.000	+	.	Sequence "AP000504.12" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	14248	14248	1.000	+	.	Sequence "AP000504.13" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	14419	14419	1.000	+	.	Sequence "AP000504.14" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	19643	19643	1.000	+	.	Sequence "AP000504.15" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	20149	20149	1.000	+	.	Sequence "AP000504.16" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	21316	21319	1.000	+	.	Sequence "AP000504.17" ; note "Insertion of 4 bases in AP000504" ; replace ""
AP000504	diffseq	conflict	21797	21797	1.000	+	.	Sequence "AP000504.18" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	23288	23288	1.000	+	.	Sequence "AP000504.19" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	23418	23418	1.000	+	.	Sequence "AP000504.20" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	23589	23589	1.000	+	.	Sequence "AP000504.21" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	23606	23606	1.000	+	.	Sequence "AP000504.22" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	24798	24798	1.000	+	.	Sequence "AP000504.23" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	25309	25309	1.000	+	.	Sequence "AP000504.24" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	25861	25861	1.000	+	.	Sequence "AP000504.25" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	28039	28040	1.000	+	.	Sequence "AP000504.26" ; note "Insertion of 2 bases in AP000504" ; replace ""
AP000504	diffseq	conflict	30644	30644	1.000	+	.	Sequence "AP000504.27" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	32339	32339	1.000	+	.	Sequence "AP000504.28" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	32497	32497	1.000	+	.	Sequence "AP000504.29" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	33074	33074	1.000	+	.	Sequence "AP000504.30" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	33776	33776	1.000	+	.	Sequence "AP000504.31" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	34767	34767	1.000	+	.	Sequence "AP000504.32" ; note "SNP in AF129756" ; replace "c"


<font color=red>  [Part of this file has been deleted for brevity]</font>

AP000504	diffseq	conflict	71553	71553	1.000	+	.	Sequence "AP000504.70" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	72048	72048	1.000	+	.	Sequence "AP000504.71" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	73627	73627	1.000	+	.	Sequence "AP000504.72" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	73998	73998	1.000	+	.	Sequence "AP000504.73" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	74071	74071	1.000	+	.	Sequence "AP000504.74" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	75838	75845	1.000	+	.	Sequence "AP000504.75" ; note "AF129756" ; replace "g"
AP000504	diffseq	conflict	76095	76095	1.000	+	.	Sequence "AP000504.76" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	77612	77612	1.000	+	.	Sequence "AP000504.77" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	78253	78253	1.000	+	.	Sequence "AP000504.78" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	80428	80428	1.000	+	.	Sequence "AP000504.79" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	80974	80974	1.000	+	.	Sequence "AP000504.80" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	81564	81564	1.000	+	.	Sequence "AP000504.81" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	81852	81852	1.000	+	.	Sequence "AP000504.82" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	82318	82318	1.000	+	.	Sequence "AP000504.83" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	82594	82594	1.000	+	.	Sequence "AP000504.84" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	82709	82709	1.000	+	.	Sequence "AP000504.85" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	83072	83072	1.000	+	.	Sequence "AP000504.86" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	83692	83692	1.000	+	.	Sequence "AP000504.87" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	86264	86264	1.000	+	.	Sequence "AP000504.88" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	86298	86298	1.000	+	.	Sequence "AP000504.89" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	87932	87932	1.000	+	.	Sequence "AP000504.90" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	88912	88912	1.000	+	.	Sequence "AP000504.91" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	89199	89199	1.000	+	.	Sequence "AP000504.92" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	89762	89764	1.000	+	.	Sequence "AP000504.93" ; note "AF129756" ; replace "ca"
AP000504	diffseq	conflict	90710	90710	1.000	+	.	Sequence "AP000504.94" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	91667	91683	1.000	+	.	Sequence "AP000504.95" ; note "AF129756" ; replace "g"
AP000504	diffseq	conflict	91797	91797	1.000	+	.	Sequence "AP000504.96" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	92165	92165	1.000	+	.	Sequence "AP000504.97" ; note "SNP in AF129756" ; replace "a"
AP000504	diffseq	conflict	93250	93250	1.000	+	.	Sequence "AP000504.98" ; note "SNP in AF129756" ; replace "c"
AP000504	diffseq	conflict	93696	93696	1.000	+	.	Sequence "AP000504.99" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	93860	93860	1.000	+	.	Sequence "AP000504.100" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	95451	95451	1.000	+	.	Sequence "AP000504.101" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	96650	96650	1.000	+	.	Sequence "AP000504.102" ; note "SNP in AF129756" ; replace "t"
AP000504	diffseq	conflict	97273	97274	1.000	+	.	Sequence "AP000504.103" ; note "Insertion of 2 bases in AP000504" ; replace ""
AP000504	diffseq	conflict	97716	97716	1.000	+	.	Sequence "AP000504.104" ; note "SNP in AF129756" ; replace "g"
AP000504	diffseq	conflict	97827	97827	1.000	+	.	Sequence "AP000504.105" ; note "SNP in AF129756" ; replace "t"
</pre>
</td></tr></table><p>
<p><h3>File: ap000504.diffseq</h3>
<table width="90%"><tr><td bgcolor="#DDFFDD">
<pre>
########################################
# Program: diffseq
# Rundate: Fri Aug 23 15:46:50 2002
# Report_format: diffseq
# Report_file: ap000504.diffseq
# Additional_files: 2
# 1: AP000504.diffgff (Feature file for first sequence)
# 2: AF129756.diffgff (Feature file for second sequence)
########################################

#=======================================
#
# Sequence: AP000504     from: 1   to: 100000
# HitCount: 119
#
# Compare: AF129756     from: 1   to: 184666
# 
# AP000504 overlap starts at 1
# AF129756 overlap starts at 6036
# 
# (AP000504) start end length sequence
# (AF129756) start end length sequence
# 
#
#=======================================


AP000504 847-847 Length: 1
Sequence: a
Sequence: t
AF129756 6882-6882 Length: 1

AP000504 1795-1795 Length: 1
Sequence: g
Sequence: a


<font color=red>  [Part of this file has been deleted for brevity]</font>

Sequence: c
Sequence: t
AF129756 101481-101481 Length: 1

AP000504 96650-96650 Length: 1
Sequence: c
Sequence: t
AF129756 102680-102680 Length: 1

AP000504 97273-97274 Length: 2
Sequence: aa
Sequence: 
Feature: repeat_region 103299-103402 rpt_family='AluSq'
AF129756 103302 Length: 0

AP000504 97716-97716 Length: 1
Sequence: a
Sequence: g
AF129756 103744-103744 Length: 1

AP000504 97827-97827 Length: 1
Sequence: c
Sequence: t
Feature: repeat_region 103784-104083 rpt_family='AluSx'
AF129756 103855-103855 Length: 1

#---------------------------------------
#
# Overlap_end: 100000 in AP000504
# Overlap_end: 106028 in AF129756
# 
# SNP_count: 86
# Transitions: 58
# Transversions: 28
#
#---------------------------------------
</pre>
</td></tr></table><p>


<p>

The first line is the title giving the names of the sequences used.
<p>
The next two non-blank lines state the positions in each sequence where the detected
overlap between them starts.
<p>
There then follows a set of reports of the mismatches between the sequences.
<br>
Each report consists of 4 or more lines.
<br>
<ul>
<li>The first line has the name of the first sequence followed by the start and end
positions of the mismatched region in that sequence, followed by the length of
the mismatched region. If the mismatched region is of zero length in this sequence,
then only the position of the last matching base before the mismatch is given.</li>

<li>If a feature of the first sequence overlaps with this mismatch region, then one or more
lines starting with 'Feature:' comes next with the type, position and tag field
of the feature.</li>

<li>Next is a line starting "Sequence:" giving the sequence of the mismatch in the
first sequence.</li>
</ul>
<p>

This is followed by the equivalent information for the second sequence,
but in the reverse order, namely 'Sequence:' line, 'Feature:' lines and line
giving the position of the mismatch in the second sequence.
<p>

At the end of the report are two non-blank lines giving the positions in each
sequence where the detected
overlap between them ends.
<p>

The last three lines of the report gives the counts of SNPs (defined as a change of one nucleotide to
one other nucleotide, no deletions or insertions are counted, no multi-base changes are counted).
<p>

If the input sequences are nucleic acid, The counts of transitions
(Pyrimide to Pyrimidine or Purine to Purine) and transversions
(Pyrimidine to Purine) are also given. 

<p>

It should be noted that not all features are reported.

<p>

The 'source' feature found in all EMBL/Genbank feature table entries is
not reported as this covers all of the sequence and so overlaps with any
difference found in that sequence and so is uninformative and
irritating.  It has therefore been removed from the output report. 

<p>

The translation information of CDS features is often extremely long and
does not add useful information to the report.  It has therefore been
removed from the output report. 

<p>

If no regions of alignment are found, the following output is given:

<p>

<pre>
<hr>
########################################
# Program: diffseq
# Rundate: Mon Feb 11 13:21:20 2002
# Report_file: ap000504.diffseq
# Additional_files: 2
# 1: AP000504.diffgff (Feature file for first sequence)
# 2: fred.diffgff (Feature file for second sequence)
########################################

#=======================================
#
# Sequence: AP000504     from: 1   to: 100000
# HitCount: 0
#=======================================


#---------------------------------------
#
# No regions of alignment found.
#
#
#---------------------------------------              
<hr>
</pre>
<p>

If the <b>-rformat table</b> qualifier is given then the output is given
in a columnar format. 

<p>

The columns are separated by one or more spaces or TAB characters in the
order:

<p>

<ul>
<li>sequence 1 name
<li>sequence 1 start
<li>sequence 1 end
<li>score (always 0.000)
<li>sequence 2 start
<li>sequence 2 end
<li>sequence 2 name
<li>sequence difference length
<li>sequence 
<li>first feature
<li>second feature
</ul>

<p>


If no regions of alignment are found, the following output is given:

<p>

<pre>
<hr>
########################################
# Program: diffseq
# Rundate: Mon Feb 11 13:34:34 2002
# Report_file: ap000504.diffseq
# Additional_files: 2
# 1: AP000504.diffgff (Feature file for first sequence)
# 2: fred.diffgff (Feature file for second sequence)
########################################

#=======================================
#
# Sequence: AP000504     from: 1   to: 100000
# HitCount: 0
#=======================================

USA               Start     End   Score start  end    length name   sequence first_feature second_feature

#---------------------------------------
#
# No regions of alignment found.
#
#
#---------------------------------------    
<hr>
</pre>
<p>



<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.

        eg:
        data file format:

	blank lines and comment lines are ignored.
        comment lines start with '#'
        There are three columns separated by one or more space or TAB
	characters for X and Y values of durm, followed by energy
        all coefficients are given as floating point values
        An example is:

        # Durm coefficient enery file
        # X     Y       Energy
        0.1     0.2     0.331
        0.2     0.4     0.332
        0.3     0.6     0.333
        0.3     0.8     0.334



If you wish to include the standard description of what data files
are and how to use embossdata to inspect and retrieve them, use:

<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>


   -->



<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

It should be noted that not all features are reported.

<p>

The 'source' feature found in all EMBL/Genbank feature table entries is
not reported as this covers all of the sequence and so overlaps with any
difference found in that sequence and so is uninformative and
irritating.  It has therefore been removed from the output report. 

<p>

The translation information of CDS features is often extremely long and
does not add useful information to the report.  It has therefore been
removed from the output report. 

<p>

If you run out of memory, use a larger word size.

<p>

Using a larger word size increases the length between mismatches that
will be reported as one event.  Thus a word size of 50 will report two
SNP that are with 50 bases of each other as one mismatch. 

   
<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.

<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
        eg:
        It doesn't check to see if the sequence is a protein sequence or not.
   -->

None.

<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program.
        eg:

        "Threshold value greater than 1." - means that the threshold has
        been set to be greater than 1 but a burgle exception has
        occurred making this invalid - you should reduce the wobble
        parameter and try again. 

   -->
None.

<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
        eg:
        "0"
   -->
It always exits with status 0.

<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.

<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
</table>

<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->
<p>
   A graphical dotplot of the matches used in this program can be displayed
   using the program <b>dotpath</b>.



<H2>
    Author(s)
</H2>
<!--
        Who has worked on the program in the past.
e.g.

This application was written by Alan Bleasby <a
href="mailto:ableasby@hgmp.mrc.ac.uk">(ableasby@hgmp.mrc.ac.uk)</a>
   -->

This application was written by Gary Williams <a
href="mailto:gwilliam@hgmp.mrc.ac.uk">(gwilliam@hgmp.mrc.ac.uk)</a>

<H2>
    History
</H2>

<!--
        What changes have been made.
   -->

Written 15th Aug 2000 - Gary Williams.
<br>
18th Aug 2000 - Added writing out GFF files of the mismatched regions

<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:

   -->
This program is intended to be used by everyone and everything, from naive users to embedded scripts.



<H2>
    Comments
</H2>

<!--
	Any comments, requests etc. that people have about the program
   -->



</BODY>
</HTML>

