<!--START OF HEADER - DON'T ALTER -->

<HTML>
<HEAD>
  <TITLE>
  EMBOSS: domainer
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

domainer
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>



<!--END OF HEADER-->






<H2>
    Function
</H2>
[an error occurred while processing this directive]

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<!-- 
	This is the main part of the document.
        This is what the user looks at to find out if this program will do what he wants. 
        Long description.
        Why it is useful.
        When it should/shouldn't be used.
  -->

This program is part of a suite of EMBOSS applications that directly or
indirectly make use of the protein structure databases pdb and scop.
This program is part of an experimental analysis pipeline described in an
accompanying document.  We provide the software in the hope that it will
be useful.  The applications were designed for specific research purposes
and may not be useful or reliable in contexts other than the described
pipeline.  The development of the suite was coordinated by Jon Ison to
whom enquiries and bug reports should be sent (email jison@hgmp.mrc.ac.uk).

<p>

There is a need for a convenient source of protein coordinate data which
includes protein strucutral domain definitions and is therefore appropriate
for domain-centred approaches.  domainer reads protein coordinate files and
writes files of coordinate data for individual scop domains in embl-like
and pdb formats. 


<p>

domainer reads a scop classification file and a directory of protein
coordinate files and writes, for each domain in the scop classification
file, domain coordinate files in embl-like and pdb formats.  Each domain
coordinate file contains coordinates for a single scop domain.  The paths
and extensions for the protein (input) and domain (output) coordinate
files are specified by the user.  The scop domain identifier codes are used
to name the output files.  A log file for each of the embl-like and pdb
format builds is written.  











<H2>
    Algorithm
</H2>

<!-- 
        Algorithms
	  Pseudocode/Vague description/Handwaving/web links to useful pages
  -->

All domainer does is read a clean coordinate file for the full length
protein and, using the domain definition given in a scop classification
file, write a clean coordinate file just containing the coordinates for
the domain.  The 'algorithm' is trivial; just reading stuff to and from
data structures. 

<p>

The scop classification file gives the pdb residue numbers of the first
and last residue of the segment(s) of the chain comprising the domain. 
Coordinate and sequence data corresponding to these region(s) only are
written to the output file. 










<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->





<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
[an error occurred while processing this directive]
</pre>
</td></tr></table>
<P>
[an error occurred while processing this directive]

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>domainer</b> reads any normal sequence USAs.

<p>



<p>

The format of the protein coordinate file is described in pdbparse



<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
<p>

The output is a standard EMBOSS report file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <b>-rformat xxx</b>, where 'xxx' is replaced by
the name of the required format.  The available format names are: embl,
genbank, gff, pir, swiss, trace, listfile, dbmotif, diffseq, excel,
feattable, motif, regions, seqtable, simple, srs, table, tagseq

<p>

See:
<A href="http://www.uk.embnet.org/Software/EMBOSS/Themes/ReportFormats.html">
http://www.uk.embnet.org/Software/EMBOSS/Themes/ReportFormats.html</A>
for further information on report formats.

<p>

   -->



<p>



<p>

The embl-like format used for domain coordinate files is exactly the same
as that used for protein coordinate files and is described in pdbparse.

<p>

An excerpt of the pdb format used for domain coordinate files is shown
below.  pdb format is explained on the pdb website:

<p>

(1) http://www.rcsb.org/pdb/docs/format/pdbguide2.2/guide2.2_frame.html
domainer writes the following records in pdb format (Figure 1):

<p>

(1)  HEADER - bibliographic information.  The text 'CLEANED-UP PDB FILE FOR
SCOP DOMAIN XXXXXXX' is always given (where XXXXXXX is a 7-character domain
identifier code).

<p>

(2)  TITLE - bibliographic information. The text ' THIS FILE IS MISSING
MOST RECORDS FROM THE ORIGINAL PDB FILE' is always given.

<p>

(3)  COMPND - compound information.  The COMPND records from the original
pdb file are given.

<p>

(4)  SOURCE - protein source information.  The SOURCE records from the
original pdb file are given.

<p>

(5)  REMARK - remark records. Remark records are used for spacing. One
REMARK line containing the protein resolution is always given.

<p>

(6)  SEQRES - protein sequence.

<p>

(7)  ATOM - atomic coordinates.

<p>

(8)  TER - indicates the end of a chain.**

<p>


Excerpt of a domain coordinate file in pdb format

<p>

<pre>
HEADER     CLEANED-UP PDB FILE FOR SCOP DOMAIN D1CS4A_
TITLE      THIS FILE IS MISSING MOST RECORDS FROM THE ORIGINAL PDB FILE
COMPND     MOL_ID: 1; MOLECULE: TYPE V ADENYLATE CYCLASE;
SOURCE     MOL_ID: 1; ORGANISM_SCIENTIFIC: CANIS FAMILIARIS;
REMARK
REMARK     RESOLUTION. 2.50  ANGSTROMS.
REMARK
SEQRES   1 A   52  ALA ASP ILE GLU GLY PHE THR SER LEU ALA SER GLN CYS
SEQRES   2 A   52  THR ALA GLN GLU LEU VAL MET THR LEU ASN GLU LEU PHE
SEQRES   3 A   52  ALA ARG PHE ASP LYS LEU ALA ALA GLU ASN HIS CYS LEU
SEQRES   4 A   52  ARG ILE LYS ILE LEU GLY ASP CYS TYR TYR CYS VAL SER
ATOM      1  N   ASP A   2      51.711 -11.782  62.798  1.00 51.17           N
ATOM      2  CA  ASP A   2      52.810 -11.644  61.848  1.00 54.45           C
ATOM      3  C   ASP A   2      54.137 -11.314  62.530  1.00 55.11           C
ATOM      4  O   ASP A   2      54.175 -10.524  63.469  1.00 55.34           O
ATOM      5  CB  ASP A   2      52.437 -10.555  60.831  1.00 57.50           C
ATOM      6  CG  ASP A   2      53.460 -10.391  59.729  1.00 61.38           C
ATOM      7  OD1 ASP A   2      54.316  -9.485  59.841  1.00 65.55           O
ATOM      8  OD2 ASP A   2      53.390 -11.146  58.736  1.00 63.68           O
ATOM      9  N   ILE A   3      55.216 -11.941  62.066  1.00 57.14           N
ATOM     10  CA  ILE A   3      56.546 -11.705  62.624  1.00 59.46           C
ATOM     11  C   ILE A   3      57.020 -10.305  62.230  1.00 60.12           C
ATOM     12  O   ILE A   3      56.963  -9.927  61.060  1.00 59.12           O
ATOM     13  CB  ILE A   3      57.583 -12.722  62.094  1.00 60.84           C
ATOM     14  CG1 ILE A   3      57.184 -14.163  62.447  1.00 63.12           C
ATOM     15  CG2 ILE A   3      58.975 -12.384  62.632  1.00 61.24           C
ATOM     16  CD1 ILE A   3      57.408 -14.554  63.895  1.00 63.92           C
ATOM     17  N   GLU A   4      57.492  -9.548  63.212  1.00 60.23           N
ATOM     18  CA  GLU A   4      57.975  -8.198  62.971  1.00 62.14           C
ATOM     19  C   GLU A   4      59.401  -8.277  62.424  1.00 60.59           C
ATOM     20  O   GLU A   4      60.244  -8.972  62.987  1.00 61.84           O
ATOM     21  CB  GLU A   4      57.917  -7.386  64.272  1.00 65.47           C
ATOM     22  CG  GLU A   4      58.037  -5.874  64.091  1.00 70.25           C
ATOM     23  CD  GLU A   4      57.588  -5.089  65.324  1.00 72.94           C
ATOM     24  OE1 GLU A   4      58.262  -5.175  66.377  1.00 70.76           O
ATOM     25  OE2 GLU A   4      56.555  -4.380  65.232  1.00 74.36           O
ATOM     26  N   GLY A   5      59.642  -7.608  61.298  1.00 57.58           N
ATOM     27  CA  GLY A   5      60.956  -7.615  60.681  1.00 56.14           C
ATOM     28  C   GLY A   5      61.452  -8.993  60.265  1.00 58.03           C
ATOM     29  O   GLY A   5      62.620  -9.322  60.480  1.00 57.47           O
ATOM     30  N   PHE A   6      60.576  -9.789  59.649  1.00 58.00           N
**
< data ommitted for clarity >
**
ATOM    354  O   CYS A  47      50.803 -16.789  58.301  1.00 50.50           O
ATOM    355  CB  CYS A  47      51.409 -13.972  57.565  1.00 51.82           C
ATOM    356  SG  CYS A  47      49.727 -13.277  57.723  1.00 54.53           S
TER     357      CYS A  47
END
</pre>        



<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.

	If you wish to include the standard description of what data
	files are and how to use embossdata to inspect and retrieve
	them, use:
<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>

   -->

None.













<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

Where coordinates for multiple models are given in a protein coordinate
file, the domain coordinate file will contain coordinates for the first
model.

<p>

In the rare cases where a domain is comprised of segments from more than
one chain, the data in the domain coordinate file will be presented as
belonging to a single chain with a chain identifier of '.'.  A single
sequence will be given

<p>

The start and end positions of domains in scop coincide in several cases
to residues which either lack a CA atom or for which coordinates for a
single atom only are given in the pdb file.  To ensure these domains are
processed during a domainer run, it is important to use clean protein
coordinate files which have NOT been masked to remove such residues.
Otherwise errors of the following type may occur in the log file:

<p>

<pre>
//
D1QFUL1
ERROR Domain end not found in ajXyzCpdbWriteDomain
**
//
D1QFUL1
ERROR Domain start not found in ajXyzCpdbWriteDomain
</pre>







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

domainer generates a log file an excerpt of which is shown below. If
there is a problem in processing a domain, three lines containing the
record '//', the domain identifier code and an error message respectively
are written. The text 'WARN  filename not found' is given in cases where a
clean coordinate file could not be found. 'ERROR filename file read error'
or 'ERROR filename file write error' will be reported when an error was
encountered during a file read or write respectively.  The 'ERROR Domain
start found by wildcard match only' is reported when wildcard matching
was needed to find the start (or end) of a domain in a pdb file (see below).
Various other error messages may also be given (in case of difficulty
email Jon Ison, jison@hgmp.mrc.ac.uk).

<p>

Excerpt of log file

<p>

<pre>
//
DS002__
WARN  Could not open for reading cpdb file s002.pxyz
//
DS003__
WARN  Could not open for reading cpdb file s003.pxyz
</pre>

<p>


Messages of the type:

<p>

<pre>
//
D0LPC_1
WARN  0lpc.pxyz not found
</pre>

<p>

can appear in the log file if

<p>

(i)  The domain is of a pdb file which is in holding, but has not made it
into the main pdb release yet.

<p>

(ii) The domain is of a pdb file that is now obsolete - having been
replaced by a more recent entry.

<p>

There may be other cases too.

<p>

Messages of the type:

<p>

<pre>
//
D1QFUL1
ERROR Domain start not found in ajXyzCpdbWriteDomain
//
D1QFUL1
ERROR Domain end not found in ajXyzCpdbWriteDomain
indicate that residues for the start or end of a domain are missing from
the protein coordinate file (see 'Notes' above).
</pre>







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
[an error occurred while processing this directive]
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->

<p>

A 'scop classification file' contains classification and other data for
domains from the scop database.  The file is in embl-like format and is
generated by scopparse.  Domain sequence information can be added to the
file by using scopseqs.

<p>

A 'protein coordinate file' contains protein coordinate and other data
extracted from a single pdb file.  The files, generated by pdbparse, are in
embl-like format and contain 'cleaned-up' data that is self-consistent and
error-corrected.

<p>

pdbplus, scopreso, scopseqs, scopalign and siggen use domain coordinate
files as input.  








<H2>
    Author(s)
</H2>
<!--
        Who has worked on the program in the past.
e.g. one of:
Alan Bleasby (ableasby&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK


Bernd Jagla (bernd&nbsp;&copy;&nbsp;golgi.ski.mskcc.org)
<br>
Cellular Biochemistry and Biophysics Program, Rockefeller
Research Laboratories, Memorial Sloan-Kettering Cancer Center, 1275 York
Avenue, Box 251,New York, NY 10021.

David Martin (dmartin&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>

Gos Micklem (gos&nbsp;&copy;&nbsp;ebi.ac.uk)
<br>
Informatics Division, European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

Gary Williams (gwilliam&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK


Ian Longden (il&nbsp;&copy;&nbsp;sanger.ac.uk)
<br>
Sanger Institute, Wellcome Trust Genome Campus, Hinxton,
Cambridge, CB10 1SA, UK.                      

Mark Faller (current e-mail address unknown)
<br>
while he was with:
<br>
HGMP-RC, Genome Campus, Hinxton, Cambridge CB10 1SB, UK

Michael K. Schuster and Martin Grabner (martin.grabner&nbsp;&copy;&nbsp;univie.ac.at)
<br>
from the Austrian National EMBnet node.


Michael Schmitz (mschmitz&nbsp;&copy;&nbsp;lbl.gov)
<br>
Lawrence Berkeley Labs, USA

Nicolas Tourasse (nicolas.tourasse&nbsp;&copy;&nbsp;biotek.uio.no)
<br>
Biotechnology Center of Oslo

Peter Rice (pmr&nbsp;&copy;&nbsp;ebi.ac.uk)
<br>
Informatics Division, European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

Richard Durbin (rd&nbsp;&copy;&nbsp;sanger.ac.uk)
<br>
Sanger Institute, Wellcome Trust Genome Campus, Hinxton,
Cambridge, CB10 1SA, UK.                      

Richard Bruskiewich (current e-mail address unknown)
<br>
while he was at:
<br>
Sanger Institute, Wellcome Trust Genome Campus, Hinxton,
Cambridge, CB10 1SA, UK.                      

Rodrigo Lopez (rls&nbsp;&copy;&nbsp;ebi.ac.uk)
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

Sinead O'Leary (current e-mail address unknown)
<br>
while she was at:
<br>
HGMP-RC, Genome Campus, Hinxton, Cambridge CB10 1SB, UK


Tim Carver (tcarver&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK

Thomas Laurent (thomas.laurent&nbsp;&copy;&nbsp;uk.lionbioscience.com)
<br>
Lion Bioscience Ltd,
     Compass House,
     80-82 Newmarket Road,
     Cambridge,
     CB5 8DZ,
     UK

Val Curwen (vac&nbsp;&copy;&nbsp;sanger.ac.uk)
<br>
Sanger Institute, Wellcome Trust Genome Campus, Hinxton,
Cambridge, CB10 1SA, UK. 


   -->


Jon Ison (jison&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK













<H2>
    History
</H2>

<!--
        What changes have been made.
   -->

Written (2003) - Jon Ison

<!--
<p>
[an error occurred while processing this directive]
-->












<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.












</BODY>
</HTML>

