<html>
<head>
<title>EMBASSY domainatrix: domainnr</title>
<base href="http://emboss.sourceforge.net/embassy/domainatrix/">
</head>
<body>
<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBOSS: DOMAINNR documentation. </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">
<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="../index.html" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="../../emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

<H2> DOMAINNR documentation </H2>
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>








<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<br><H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>

<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<br><br><br><H2> 1.0   SUMMARY  </H2>
 Removes redundant domains from a DCF file (domain classification file). 
 The file must contain domain sequence information, which can be added by 
 using DOMAINSEQS.





<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<br><br><br><H2> 2.0   INPUTS & OUTPUTS          </H2>

  DOMAINNR reads a DCF file (domain classification file) containing domain sequence 
  information and writes a DCF file in which the redundant domains are removed from each node 
  (e.g. family, superfamily etc).
 Optionally, the redundant domains are written to a second DCF output file.
  The node of operation and input and output files are specified by the user.  A log file is also 
  written.




<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.

     Cannot use the test data files because they might be empty or need 
     hand-editing 

     Use "<b>DOMAINNR</b> reads any normal sequence USAs." if
     appropriate.
-->
<a name="3.0"></a>
<br><br><br><H2> 3.0   INPUT FILE FORMAT         </H2>
The format of the DCF file is described in the <a href="scopparse.html">SCOPPARSE documentation.</a>





<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<br><br><br><H2> 4.0   OUTPUT FILE FORMAT         </H2>
The format of the DCF file is described in the <a href="scopparse.html">SCOPPARSE documentation.</a>




<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml" 
-->

<a name="5.0"></a>
<br><br><br><H2> 5.0   DATA FILES                </H2>

DOMAINNR requires a residue substitution matrix.



<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<br><br><br><H2> 6.0   USAGE                     </H2>
<H3> 6.1   COMMAND LINE ARGUMENTS </H3>
 
 
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td>[-dcfin]<br>(Parameter 1)</td>
<td>This option specifies name of DCF file (domain classification file) (input). A 'domain classification file' contains classification and other data for domains from SCOP or CATH, in DCF format (EMBL-like). The files are generated by using SCOPPARSE and CATHPARSE. Domain sequence information can be added to the file by using DOMAINSEQS.</td>
<td>Input file</td>
<td><b>Required</b></td>
</tr>
 
<tr>
<td>-retain</td>
<td>This option specifies whether to write redundant domains to a separate file. If this option is selected, redundant domains are written to a separate output file.</td>
<td>Toggle value Yes/No</td>
<td>N</td>
</tr>
 
<tr>
<td>-node</td>
<td>This option specifies the node for redundancy removal. Redundancy can be removed at any specified node in the SCOP or CATH hierarchies. For example by selecting 'Class' entries belonging to the same Class will be non-redundant.</td>
<td><table><tr><td>1</td> <td><i>(Class (SCOP))</i></td></tr><tr><td>2</td> <td><i>(Fold (SCOP))</i></td></tr><tr><td>3</td> <td><i>(Superfamily (SCOP))</i></td></tr><tr><td>4</td> <td><i>(Family (SCOP))</i></td></tr><tr><td>5</td> <td><i>(Class (CATH))</i></td></tr><tr><td>6</td> <td><i>(Architecture (CATH))</i></td></tr><tr><td>7</td> <td><i>(Topology (CATH))</i></td></tr><tr><td>8</td> <td><i>(Homologous Superfamily (CATH))</i></td></tr><tr><td>9</td> <td><i>(Family (CATH))</i></td></tr></table></td>
<td>1</td>
</tr>
 
<tr>
<td>-mode</td>
<td>This option specifies whether to remove redundancy at a single threshold % sequence similarity or remove redundancy outside a range of acceptable threshold % similarity. All permutations of pair-wise sequence alignments are calculated for each domain family in turn using the EMBOSS implementation of the Needleman and Wunsch global alignment algorithm. Redundant sequences are removed in one of two modes as follows: (i) If a pair of proteins achieve greater than a threshold percentage sequence similarity (specified by the user) the shortest sequence is discarded. (ii) If a pair of proteins have a percentage sequence similarity that lies outside an acceptable range (specified by the user) the shortest sequence is discarded.</td>
<td><table><tr><td>1</td> <td><i>(Remove redundancy at a single threshold % sequence similarity)</i></td></tr><tr><td>2</td> <td><i>(Remove redundancy outside a range of acceptable threshold % similarity)</i></td></tr></table></td>
<td>1</td>
</tr>
 
<tr>
<td>-thresh</td>
<td>This option specifies the % sequence identity redundancy threshold, which determines the redundancy calculation. If a pair of proteins achieve greater than this threshold the shortest sequence is discarded.</td>
<td>Any numeric value</td>
<td>95.0</td>
</tr>
 
<tr>
<td>-threshlow</td>
<td>This option specifies the % sequence identity redundancy threshold, which determines the redundancy calculation. If a pair of proteins have a percentage sequence similarity that lies outside an acceptable range the shortest sequence is discarded.</td>
<td>Any numeric value</td>
<td>30.0</td>
</tr>
 
<tr>
<td>-threshup</td>
<td>This option specifies the % sequence identity redundancy threshold, which determines the redundancy calculation. If a pair of proteins have a percentage sequence similarity that lies outside an acceptable range the shortest sequence is discarded.</td>
<td>Any numeric value</td>
<td>90.0</td>
</tr>
 
<tr>
<td>[-dcfout]<br>(Parameter 2)</td>
<td>This option specifies the name of non-redundant DCF file (domain classification file) (output). A 'domain classification file' contains classification and other data for domains from SCOP or CATH, in DCF format (EMBL-like). The files are generated by using SCOPPARSE and CATHPARSE. Domain sequence information can be added to the file by using DOMAINSEQS.</td>
<td>Output file</td>
<td>test.scop</td>
</tr>
 
<tr>
<td>-redout</td>
<td>This option specifies the name of DCF file (domain classification file) for redundant sequences (output). A 'domain classification file' contains classification and other data for domains from SCOP or CATH, in DCF format (EMBL-like). The files are generated by using SCOPPARSE and CATHPARSE. Domain sequence information can be added to the file by using DOMAINSEQS.</td>
<td>Output file</td>
<td>test.scop</td>
</tr>
 
<tr>
<td>-errf</td>
<td>This option specifies the name of log file for the build. The log file contains messages about any errors arising while domainnr ran.</td>
<td>Output file</td>
<td>domainnr.log</td>
</tr>
 
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td>-datafile</td>
<td>This option specifies the residue substitution matrix. This is used for sequence comparison.</td>
<td>Comparison matrix file in EMBOSS data path</td>
<td>EBLOSUM62</td>
</tr>
 
<tr>
<td>-gapopen</td>
<td>This option specifies the gap insertion penalty. This is the score taken away when a gap is created. The best value depends on the choice of comparison matrix. The default value assumes you are using the EBLOSUM62 matrix for protein sequences, and the EDNAFULL matrix for nucleotide sequences.</td>
<td>Floating point number from 1.0 to 100.0</td>
<td>10.0 for any sequence</td>
</tr>
 
<tr>
<td>-gapextend</td>
<td>This option specifies the gap extension penalty. This is added to the standard gap penalty for each base or residue in the gap. This is how long gaps are penalized. Usually you will expect a few long gaps rather than many short gaps, so the gap extension penalty should be lower than the gap penalty. An exception is where one or both sequences are single reads with possible sequencing errors in which case you would expect many single base gaps. You can get this result by setting the gap open penalty to zero (or very low) and using the gap extension penalty to control gap scoring.</td>
<td>Floating point number from 0.0 to 10.0</td>
<td>0.5 for any sequence</td>
</tr>
 
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td colspan=4>(none)</td>
</tr>
 
</table>


<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of DOMAINNR is shown below.
<table><td bgcolor="#FFCCFF">
<pre>
Unix % domainnr
Removes redundant domains from a DCF file (domain classification file). The
file must contain domain sequence information, which can be added by using
DOMAINSEQS.
Name of DCF file (domain classification file for input (DCF format): test_data/all_s.scop
Write redundant sequences to separate file. [N]:
Node at which to remove redundancy
         1 : Class (SCOP)
         2 : Fold (SCOP)
         3 : Superfamily (SCOP)
         4 : Family (SCOP)
         5 : Class (CATH)
         6 : Architecture (CATH)
         7 : Topology (CATH)
         8 : Homologous Superfamily (CATH)
         9 : Family (CATH)
Select number [1]: 4
Redundancy removal options
         1 : Remove redundancy at a single threshold % sequence similarity
         2 : Remove redundancy outside a range of acceptable threshold % similarity
Select number [1]: 1
The % sequence identity redundancy threshold [95.0]:
Name of non-redundant DCF file (domain classification file) for output (DCF format) [test.scop]: test_data/domainnr/all_nr.scop
Name of log file for the build [domainnr.log]: test_data/domainnr/domainnr.log
// Adenylyl and guanylyl cyclase catalytic domain
D1CS4A_
// DNA double-strand break repair nuclease
D1II7A_
Unix %
</pre>
</table>

<br><br>
The DCF file /test_data/all_s.scop with families containing
redundant domains was read and a DCF file called 
/test_data/domainnr/all_nr.scop was written in which redundant domains were removed
from each family.  A redundant domain was one sharing at least 95% sequence
similarity to another domain.  Redundant domains were not retained. The log file 
/test_data/domainnr/domainnr.log was written.



<br> 
<br>The following command line would achieve the same result.
<br>
<table><td bgcolor="#FFCCFF">
<pre> 
domainnr test_data/all_s.scop test_data/domainnr/all_nr.scop -mode 1 -thresh 95 -errf test_data/domainnr/domainnr.log -retain N -node 4
</pre> 
</table>

<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<br><br><br><H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
None.


<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<br><br><br><H2> 8.0   NOTES                     </H2>

If for example the user selected the node to be "Family (SCOP)" then DOMAINNR 
removes redundancy at the level of the SCOP family, i.e. entries belonging 
to the same family will be non-redundant.  


<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<tr>
<td><b> Domain classification file (for SCOP)</b></td>
<td> DCF format (EMBL-like format for domain classification data). </td>
<td> Classification and other data for domains from SCOP.  </td>
<td> <a href="scopparse.html">SCOPPARSE</a> </td>
<td> Domain sequence information can be added to the file by using DOMAINSEQS.  </td>
</tr>
<tr>
<td><b> Domain classification file (for CATH)</b></td>
<td> DCF format (EMBL-like format for domain classification data). </td>
<td> Classification and other data for domains from CATH.  </td>
<td> <a href="cathparse.html">CATHPARSE</a> </td>
<td> Domain sequence information can be added to the file by using DOMAINSEQS.  </td>
</tr>
</table>





<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="9.0"></a>
<br><br><br><H2> 9.0   DESCRIPTION               </H2>

The inclusion of very similar sequences in certain analyses will introduce
undesirable bias.  For example, a family may possess 100 sequences in the 
sequence database, but 90 of these might be essentially the same sequence,
e.g. very close relatives or mutations of a single sequence.  Although 100
sequences are known, the family only contains 11 sequences that are 
essentially unique.  For many methods it is desirable to use sets of 
sequences that are truly representative of the larger family.  DOMAINNR reads
a DCF file (domain classification file) and writes a DCF file with 
redundant domains removed from each node in the domain classification 
hierarchy, e.g. family, superfamily or class.  





<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<br><br><br><H2> 10.0  ALGORITHM                 </H2>

All permutations of pair-wise sequence alignments are calculated for each 
node (family etc) in turn using the EMBOSS implementation of the Needleman and 
Wunsch global alignment algorithm.  Redundant sequences are removed in one
of two modes as follows: (i) If a pair of proteins achieve greater 
than a threshold percentage sequence similarity (specified by the user) 
the shortest sequence is discarded. (ii) If a pair of proteins have a 
percentage sequence similarity that lies outside an acceptable range 
(specified by the user) the shortest sequence is discarded.
The user must specify gap insertion 
and extension penalties and a residue substitution matrix for use in the 
alignments.  % sequence similarity is calculated by using the EMBOSS 
function embAlignCalcSimilarity.




<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
     (Take this from "Sister applications" in the old documentation)
-->
<a name="11.0"></a>
<br><br><br><H2> 11.0  RELATED APPLICATIONS      </H2>

<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>APPLICATION</b></td>
<td><b>DESCRIPTION</b></td>
</tr>
<tr>
<td><b><a href="domainer.html">DOMAINER</a> </b></td>
<td> Reads CCF files (clean coordinate files) for proteins and writes CCF files for domains in a DCF file (domain classification file).</td>
</tr>
<tr>
<td><b><a href="funky.html">FUNKY</a> </b></td>
<td> Reads CCF files (clean coordinate files) and writes CON files (contact files) of residue-ligand contact data for domains in a DCF file (domain classification file). </td>
</tr>
<tr>
<td><b><a href="domainreso.html">DOMAINRESO</a> </b></td>
<td> Removes low resolution domains from a DCF file (domain classification file). </td>
</tr>
<tr>
<td><b><a href="domainseqs.html">DOMAINSEQS</a> </b></td>
<td> Adds sequence records to a DCF file (domain classification file). </td>
</tr>
<tr>
<td><b><a href="domainrep.html">DOMAINREP</a> </b></td>
<td> Reorder DCF file (domain classification file) so that the representative structure of each
      user-specified node is given first. </td>
</tr>
<tr>
<td><b><a href="domainalign.html">DOMAINALIGN</a> </b></td>
<td> Generates structure-based sequence alignments for nodes in a DCF
      file (domain classification file). </td>
</tr>
<tr>
<td><b><a href="seqsearch.html">SEQSEARCH</a> </b></td>
<td> Generate DHF files (domain hits files) of database hits (sequences) from a DAF file (domain alignment file) or other sequences by using PSI-BLAST.</td>
</tr>
<tr>
<td><b><a href="seqwords.html">SEQWORDS</a> </b></td>
<td>Generates DHF files (domain hits files) of database hits (sequences) for nodes in a DCF file (domain classification file) by keyword search of UniProt.</td>
</tr>
<tr>
<td><b><a href="seqsort.html">SEQSORT</a> </b></td>
<td>  Reads DHF files (domain hits files) of database hits (sequences) and removes hits of ambiguous classification. </td>
</tr>
</table>





<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="12.0"></a>
<br><br><br><H2> 12.0  DIAGNOSTIC ERROR MESSAGES </H2>
DOMAINNR generates a log file an excerpt of which is shown in Figure 1. The 
first two lines give the level in the SCOP or CATH hierarchy at which redundancy 
was removed (e.g. 'Families') and the value
of the redundancy threshold.  The file then contains a section for each node, e.g.
each family. Each section contains a line with the record '//' immediately 
followed by the name of the node (family in this case), and two lines containing 'Retained' 
and 'Rejected' respectively. Domain identifier codes of domains that appear in output 
file are listed under 'Retained', while redundant domains are listed under 'Rejected'.
Tthese will be saved in a second output file if the user has specified redundant domains to be retained.
The text  'ERROR filename file read error' will be given when an error was encountered during a file read.

<br><br><b>Figure 1  Summary of ROCPLOT output </b>
<table><td bgcolor="#FFCCFF">
<pre>
Excerpt from DOMAINNR log file
Families are non-redundant
95% redundancy threshold
// Homeodomain
Retained
D2HDDA_
D1AKHA_
D1MNMC_
Rejected
D2HDDB_
D1ENH__
D3HDDA_
WARN  d3hdda_.pxyz not found
// Di-haem cytohrome c peroxidase
WARN  ds005__.pxyz not found
WARN  Empty family
// Nuclear receptor coactivator Src-1
Retained
D2PRGC_
Rejected
</pre>
</table>




<!-- AUTHORS -->
<a name="13.0"><br><br><br><H2> 13.0  AUTHORS                   </H2></a>
 Ranjeeva Ranasinghe  <a href="mailto:rranasin@hgmp.mrc.ac.uk">(rranasin@hgmp.mrc.ac.uk)</a>
<br><br>
Jon Ison <a href="mailto:jison@rfcgr.mrc.ac.uk">(jison@rfcgr.mrc.ac.uk)</a>
<br>MRC Rosalind Franklin Centre for Genomics Research, Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK






<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  REFERENCES                </H2>

Please cite the authors and EMBOSS.
<br><br>
<i>Rice P, Longden I and Bleasby A (2000) "EMBOSS - The European
Molecular Biology Open Software Suite"  Trends in Genetics,
15:276-278.

<p>
See also <a href="http://emboss.sourceforge.net/">http://emboss.sourceforge.net/</a></i>



<H3>14.1 Other useful references  </H3>

<br>

</BODY>
</HTML>

</body>
</html>
