<HTML>

<HEAD>
  <TITLE>
  EMBOSS
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="index.html" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

dotpath
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>





<!--END OF HEADER-->





<H2>
    Function
</H2>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->
Displays a non-overlapping wordmatch dotplot of two sequences


<H2>
    Description
</H2>

<!-- 
	This is the main part of the document.

        This is what the user looks at to find out if this program will
        do what he wants. 

        Long description.

        Why it is useful.

        When it should/shouldn't be used.

        Algorithms

        Expanded description.

        eg: Reverse engineers Wheedle's 'talpsgob' program. This does a 
        splurge of the 5' end of the sequence followed by a durm.
        The net effect is a 5' end splurt.
        etc. for as long as is required (several pages is not unreasonable)
  -->

A dotplot is a graphical representation of the regions of similarity
between two sequences. 
<P>

The two sequences are placed on the axes of a rectangular image and
wherever there is a similarity between the sequences a dot is placed on
the image. 
<P>

Where the two sequences have substantial regions of similarity, many
dots align to form diagonal lines.  It is therefore possible to see at a
glance where there are local regions of similarity. 
<P>

<b>dotpath</b> is very similar to the program <b>dottup</b> which looks
for places where words (tuples) of a specified length have an exact
match in both sequences and draws a diagonal line over the position of
these words.
<P>


Using a longer word size thus displays less random noise, runs
extremely quickly, but is less sensitive. 
<P>

<b>dotpath</b> finds all matches of size <b>-wordsize</b> or greater
between two sequences.  It then reduces the matches found to the minimal
set of long matches that do not overlap.  This is a way of finding the
(nearly) optimal path aligning two sequences.  It is not the true
optimal path as produced by the algorithms used in <b>water</b> or
<b>needle</b>, but for very closely related sequences it will produce
the same result and will work well with very long sequences. 
<P>

If you wish to compare the path found by <b>dotpath</b> to the set of
all matches found then the qualifier <b>-overlaps</b> will show all
matches in red except for the matches in the minimal path which are
shown in black, as normal. 

<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with dotpath</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>dotpath tembl:AF129756 tembl:AP000504 -word 20 -graph cps -overlaps</b>
Displays a non-overlapping wordmatch dotplot of two sequences

Created dotpath.ps

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>


<!--
Here is a sample session with dotpath:

<pre>

% dotpath embl:AF129756 embl:AP000504 -word 20
Displays a non-overlapping wordmatch dotplot of two sequences
Graph type [x11]: 

</pre>
-->

<H2>
    Command line arguments
</H2>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Mandatory qualifiers (* if not always prompted):
  [-sequencea]         sequence   Sequence USA
  [-sequenceb]         sequence   Sequence USA
   -wordsize           integer    Word size
*  -graph              graph      Graph type
*  -outfile            outfile    Output file name

   Optional qualifiers:
   -overlaps           boolean    Displays the overlapping matches (in red) as
                                  well as the minimal set of non-overlapping
                                  matches
   -[no]boxit          boolean    Draw a box around dotplot

   Advanced qualifiers:
   -data               boolean    Output the match data to a file instead of
                                  plotting it

   General qualifiers:
  -help                boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose


</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#f5f5ff">
<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Mandatory qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequencea]<br>(Parameter 1)</td>
<td>Sequence USA</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-sequenceb]<br>(Parameter 2)</td>
<td>Sequence USA</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-wordsize</td>
<td>Word size</td>
<td>Integer 2 or more</td>
<td>4</td>
</tr>

<tr>
<td>-graph</td>
<td>Graph type</td>
<td>EMBOSS has a list of known devices, including postscript, ps, hpgl, hp7470, hp7580, meta, colourps, cps, xwindows, x11, tektronics, tekt, tek4107t, tek, none, null, text, data, xterm, png</td>
<td><i>EMBOSS_GRAPHICS</i> value, or x11</td>
</tr>

<tr>
<td>-outfile</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;sequence&gt;</i>.dotpath</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Optional qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-overlaps</td>
<td>Displays the overlapping matches (in red) as well as the minimal set of non-overlapping matches</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-[no]boxit</td>
<td>Draw a box around dotplot</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Advanced qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-data</td>
<td>Output the match data to a file instead of plotting it</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

</table>



<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        Except for sequences which can be described by 'sequence USA',
        this should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:AF129756' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:AF129756</h3>
<table width="90%"><tr><td bgcolor="#FFDDFF">
<pre>
&gt;AF129756 AF129756.1 Homo sapiens MSH55 gene, partial cds; and CLIC1, DDAH, G6b, G6c, G5b, G6d, G6e, G6f, BAT5, G5b, CSK2B, BAT4, G4, Apo M, BAT3, BAT2, AIF-1, 1C7, LST-1, LTB, TNF, and LTA genes, complete cds.
gaattctctccctcccatctgtggctgagattaaagatctgcacctgaagcactgaagaa
tgtgtgggtaattaaattaccctgccgattcctggagatgctgattacctggagatgacc
tcagagattatcctaaattaactcctacaagacacacattgcagcgggaggtgaggtagg
ggaaggattgtgcacctgaggctagcaaaggtttccactctgtttagagatgatgtcacc
agtgcgtttacatttgcgttgtgttcacacattgagtgctactatgtacaagaccatgtg
tcagacactgaggtaacaggtatagctactagatagagttcatagctatggaggcaagca
gattcactgactgctaattctaacatgatgtgacagtgcaactagaaaaacataaacaag
cactatgtgagcacaaagaaggtgcacatcaactccttacaggtacctgtaaaagccaaa
gggtaacagttggattgcaccttgaagaggatgcacttttttttttttttaagacagaat
ctcactctgttgcccaggctggagtgcagtggggcaatctgggctcacttcaacctctac
ctcccgagttcaagcaattctcctgcctcagcctcctgagtagctggtactagaggcatg
cgccatgatgctgggctaatttttgtattttcggtagacgtgaagtttcaccaagttggc
caggctggtcttgaactcctgacctcaaatcatccacccacctcagcctcccaaaatgct
gagactacaggcgtgagccaccgcgcctgacctggatgtaagattttgataggtacagaa
caaggaaaagactttccaggccgggcacagtggcttatgcctgtaatcccagcactttgg
gaggccgaggtgggcagatcacgaggtcagcagttcaagaccagcgtggccaacatggtg
aaatcccatctctactaaaaatacaaaaattagccaggcatggtggtgggtccctgtaat
cccagctactccggaggctgaggcaggagaattgcttgaacctgggaggtggaggttgca
gtgagcaaagaccgcgccactgcactccagcctgggtgacagagggagactccgtctcaa
aaaaaaaaaaaaaagactttccagaaggagcagcataaacacaggcatgacatgtttcca
taatggcaagtggccctaaatgactagaatataaggtagatccagtaggaaaggacttag
aaggggctttggaaggtgagtctggaaattaaaactggggtaaacgtgatggaccctgaa
catcattatactgcttaagatgctaatcttaatcctgaaggtaatgggaaaacctcctaa
ggtttatgttattttctttctacttaggctatttaaaaagtggagtgacggccaggcgca
gtgactcatgcctgtaatcccagcactttgggaggccgaggtgggcggatcaccaggagt
tcgagaccagcctgaccaacatggtgaaaccccgcctctactgaaaatacaaaaattagc
caggtgtggtggtgggcgcctgtaatcccagctacttgggaggctgaggcagaagaattg
cttgaacccgggaagtggaggttgcagtgagcagagatcgtgccattgtactccagcctg
ggcaacaagagcgaaactcagtcacaaaaaaaaaaaaaaaaaaaaaggagtgacatgctt
agatctctgttttggaatgacaggttttttgtttctagcatcaatccaaggttcatggct
tgagaaggtgtactgccagcaatgccattaaccagcaaagggaatgcaggaagaggaaca
gatctggtgggcatcagtttggatgctctgagtttgagctgcctgtgaaaactgcaggtg
gtgatatgcaattaacattcacatacggagttcaaaactagagacacaaatttgagagtc
atcacagaaatgtgaagtgtgttttctataactaaagataaccatgctaacatagccatg


<font color=red>  [Part of this file has been deleted for brevity]</font>

ccactgcaacctcaaaaccatgggctcaagtcatccttccacctcagcttcccaagtatc
taggactacaggtgtgtgccactgtgcctggctaattttaaaaaatattttaaaattttt
gttgagacagggtctatgctgctcaggctggtctcgaactcctggtttcaagtgatctgc
ccatcttggcctcccaaagtttttttttgttagtttgagaggcggtttcgctcgttgccc
aggctggagtgcaatgactgatctcatctcactgcaacctctgcctcctgggttcaagcg
attctcctgcttcagcctcccaagtagctgggattacaggtgcatgccaccattcccggc
taattttttgtatttagtagagatggggtttcaccatgttagtcaggctgatctcaaact
cctgacctcaggtgatccgcctgcctcagcctcccaaagttttgggattacaggtgtgag
ccaccatgctgggccagcctcccaaagttttgggattacaggcatgagtcaccacactgg
ccctggattttttttctttcttttttttggagacggagtctcactctgttgcccaggctg
gagtgcaatggcgtaatctcagctcactgcaacctctgctgcccgggttcaaacgattct
cctgtcttagcctcctgagtagctgggattataggtgcatgccaccatgcctggctaatt
tttgtacttttagtagagaaagtacaccatcttggccaggctggtctcgaactcctgacc
tcaggtgatccacttgcgtcggcctcccaaagtgctgggattacaggcgtgagacaccgc
acccagcctttttttttttttttcttttaagacagaatcgctctgtcacccaggctggag
tgcagtggcacaatctcggctcactgcaacctctgcctcccaggtttaagcaatccacct
atgtcagtctcccaagtagctgggattataggtgcatgtcaccatgcctggctaattttt
gtacttttagtatagaaagtacaccatgttggccaggctggtcttgaactcctgacctca
agtgatccgcctgcctcagcctcccgaagtgctggaattacagacatgtgccactgcacc
cggcctggttttttttttctaagagatggagtctcacttttctgcccaggttggagtgca
atggcaccatcatagctcactgcagccttcaactcttggcctcaggcaatccttgcacct
tagcctcgcagtgttgggattacaggcatgagccactgagccttgcctggactttttttt
ttttttgagatggcgtctcgctctgttgcccaggttggagtgctacggcatgatcttggc
tcactgcaacttccacctcccaggttcaagcgattctcttgcctcggccccccgagtagc
tgggattacaggcatgcgccaccgtgcctggctaattttggtatttttagtagagatagg
gtttcatcatgttgggcaggctggtcttgaactcctgacctcgtgatccacccacctcgg
cctcccaaagtgctgggattataggcatagccaacgcgcccagcctggacttgtttttaa
aagatcactgtggctcctgtgtttaggctggctggtaggagacaggtggcagtggcattg
atggtgaagagaaaatagtggcagccatggagatggagagaagtagacaagtttgggata
tattatacattccaggggtagaaacaacaggactagatgatggattgatgggtgggagat
gtagatactgggagagaagcaggattctgatggatggaaaaactaaaaaattctattttg
ggtgtggtaagtctaagtctattagacatgcaagtagagatgtcactgggcagatacaca
tctggatttcaggggcaaggtccaagctagagaaagaaacctgggcatggtcagcatgag
gatggtgtttaaagccatggaacttatcttgtgcatccctataagacccctttgaggcac
ttgtttcccctcacaatggatgcagtgcatcttccattctgaattccagaggcaacaacc
tcctgctcctagaagctaaactctccagacttagtcttctgaattc
</pre>
</td></tr></table><p>
<p><h3>Database entry: tembl:AP000504</h3>
<table width="90%"><tr><td bgcolor="#FFDDFF">
<pre>
&gt;AP000504 AP000504.1 Homo sapiens genomic DNA, chromosome 6p21.3, HLA Class I region, section 3/20.
gaccaatctcactgtgaggaggcagtcaaagggaataatggaagagaggaagaggatttt
ctcagtggcagtcatggcgtctgggatgaaggagtagtttccagaaaggaggcgttgttt
gcttatctccagacctatttgagggaggcaagcaaagggaacggtcttgtagctcaattt
tttcaccccattttaagaatgagacaatagaagcaagagagattatttgacttgcccaag
ctcacacaggcagttaatggaaagctagagcaagaaccaaattttcagactcttagtcta
attctctttttattctacatataatataaagatacttgtctgaaagcacagcctgagaaa
gataaatggctgaggaaagtagacatctgtctggaattgaggattttggtcaaaataatg
gtattaatagaactagtaacactaatgccttaatatctaattaggatagtacactcctgt
tcttattgtaaacctaggaaagttatagaagtgccttatggatcataataagggtcactg
aggcagtgccttttggtttggtgataaaaggctttaacttaatggggagaattccaacaa
taaaaccctgtccaaaaagtgtcaccactcctcaggggaggccctcatccctagacatga
cttaagcagaggcttcccaataagctgcaggttattaaagggtagggagcaggagagatc
ttggggggacaggtcatagggcatgaggagcacaaaggtttaggatgacataaggcagag
gggagatctgtgatgatgaaggtagagttgggggaaagaatgggacaccggaacagggag
ttaggcaaagcaaaaggaaggagataccaaaatccacacttggcaaaaatatgatttcag
gtcttttaggctctctgtgctcctgggaggctgtgggggaggaaagaaaaggctatcatt
ctttacatctcagtccttctacctctgtctgacactccctctcacccaattctagccccc
tggaatattccatatattagtccttccccattttccctctatcctttaccaagtccttac
caagctttcccagaaatcgagtcatattctcatcctgtttggcactcgtaacaacagact
ggggattgatctcatccagaacttggaaggagaacagagatcaaatgagttaaaggatct
ttgtctttgactaagagaaaacccatagccctcctcttcctacccctctccttctcaaaa
acatttcctccctaggagtagggagtgctctgcacagtgggaacacaggtagaagttgag
atttagaaaagtagttaagagtggtgggatggtgagagggaagtgggatgttctggatgt
tgtcactaggctgtaaacccctggagaacagacatgactgatttgcccagggctgaatct
gaagcacctgaaacattgtaaatacgtcatatatatttgtggccaggcacagtggctcat
gcctataatccctgccctttgggaggccaaggcaggcagatcactggaggccaggagctc
aagacaagcctagccaacgtggtgaaaccctgcctctactaaaaatataaaaattagcca
ggcgtgatggcagattcttgtaatcccagctactcgggagactgaggcaggagaattgct
tgaatccgggagacggaggttgcagtgagccaagatggcaccactacacttccagcctga
gtgacggagcaagacactgtctcaaaaaagaacaaccaaacaaaccaaaaaacagcctca
caaatatttgttaaataatgaaatgaattcataaaaacaaaagagggagcctctgtgaag
caactgtaaaatatattgagtcagtgctatagtttggatgtgatttgtccctgccaaata
tcgtgttgaaatttaatccccagtgtgatagtgttgtgaggtagggcctagcaggaggtg
tgtgggtgatgggagtggatcgctcatgaacagattaatgcccttcctggagtgtgttgg


<font color=red>  [Part of this file has been deleted for brevity]</font>

aaccccatctctactgaaaatataaacattagctggatgtagtggtgcacacctgtaatc
ccagctactcaggaggctgaggcaggagaatcgctagaactcgggaggagggggttgcag
tgagccgagattgcactactgcactctagcctgggtgacagagcgagactgtctcaaaaa
acaaaacaaaacaaaaaaacacacattgcaacaaaacaatttctctctaaacctgtaagt
gattttgtcctcccttacagagaaggtgataatctttgctgtaagcactgtcctcgtatc
gtaccccttgtgcccctgaatgaatttagaaaatgtaaagtacaggagatcagtatatga
tgacttactgattcatagtagtgttttaataggatgttccttatgtgaataagatataat
ttatttgcaaagatttggtctacatgtaaacttccaaggatataactgaaagttttggag
gacatggtattctcagtaggcattattgcttttattagtgagatggactccagcttgata
ttttctgcctttttgtgtttggctggttgtgcgcagcacgagggccgggaggaggatcag
cggttgatcaacttggtaggggagagcctgcgactgctgggcaacacctttgttgcactg
tctgacctgcgctgcaatctggcctgcacgcccccacgacacctgcatgtggtccggcct
atgtctcactacaccacccccatggtgctccagcaggcagccattcccatacaggtgggt
tagggggagtctggcctgagggagagtgaggggtgttgatagagtgacccagggtagcta
ctgggcctgaaggaggttaggaaaggaggagactggaaacatggtgatgaaggctggaga
tactttagaggtttatcatgaggttttcttggttaggctcttgtatttttctcacatctg
cctgtccatctgtctttttcagatcaatgtgggaaccactgtgaccatgacaggaaatgg
gactcggccccccccaactcccaatgcagaggcacctccccctggtcctgggcaggcctc
atccgtggctccgtcttctaccaatgtcgagtcctcagctgagggggctcccccgccagg
tccagctcccccgccagccaccagccacccgagggtcatccggatttcccaccagagtgt
ggaacccgtggtcatgatgcacatgaacattcaaggtgagaatagttgctggcgagaaga
gcaggatcagcatgatgagggaggttcatgctgaggtgtgagggaacagggtggggaagg
gagaggcacatgctggtggtggtagcctggggaccagagcagaagcttaagtagacagat
gtggggggtgtgggggttggtttgtctttggaggtgtgtttgtgtggtgaagggagtacc
tctccctgtttagatggagggaaaggcaggctttctgattgggggattatgggcctgaag
tatgcctgatctcagaaggatatagttaggccttggccctacctacctcagggccactgt
ctctgtctccctgcccagattctggcacacagcctggtggtgttccgagtgctcccactg
gccccctgggaccccctggtcatggccaaaccctgggtaagagtgagggcatcagggcag
gctgagctctgggtagagaaagggaagggctgagtgggtgggttgaaggggtccaggttc
aaggttacatcagacccgccccccaggctccaccctcatccagctgccctccctgccccc
tgagttcatgcacgccgtcgcccaccagatcactcatcaggccatggtggcagctgttgc
ctccgcggccgcaggtaatgacctggaaggggaggcttgggaggtagggcacagtccatg
gtggcagctggctggcaagggcctggccctcagccctcttcggtctgtctcttctgccac
ccacaggacagcaggtgccaggcttcccaacagctccaacccgggtggtgattgcccggc
ccactcctccacaggctcggccttcccatcctggagggcccccagtctctgggacactgg
tgagcaagggtcggggagttctagtgcgtaacagtctagg
</pre>
</td></tr></table><p>


<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats 
        Except for sequences which can be described by 'sequence USA',
        this should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 
   -->

In normal operation, a dotplot image is displayed.
<p>

With the <b>-data</b> qualifier a file of the positions of the
matches in the minimal non-overlapping set of matches is output.

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>Graphics File: dotpath.ps</h3>
<p><img src="dotpath.1.dotpath.gif" alt="[dotpath results]">




<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.

        eg:
        data file format:

	blank lines and comment lines are ignored.
        comment lines start with '#'
        There are three columns separated by one or more space or TAB
	characters for X and Y values of durm, followed by energy
        all coefficients are given as floating point values
        An example is:

        # Durm coefficient enery file
        # X     Y       Energy
        0.1     0.2     0.331
        0.2     0.4     0.332
        0.3     0.6     0.333
        0.3     0.8     0.334

   -->



<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->


<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->


<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
        eg:
        It doesn't check to see if the sequence is a protein sequence or not.
   -->

If you give a small word size with a very large sequence you will run
out of memory.  If this happens, try again with a larger word size. 

<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program.
        eg:

        "Threshold value greater than 1." - means that the threshold has
        been set to be greater than 1 but a burgle exception has
        occurred making this invalid - you should reduce the wobble
        parameter and try again. 

   -->


<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
        eg:
        "It always exits with status 0"
   -->

<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->



<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="dotmatcher.html">dotmatcher</a></td><td>Displays a thresholded dotplot of two sequences</td></tr>
<tr><td><a href="dottup.html">dottup</a></td><td>Displays a wordmatch dotplot of two sequences</td></tr>
<tr><td><a href="polydot.html">polydot</a></td><td>Displays all-against-all dotplots of a set of sequences</td></tr>
</table>

<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->

<P>
This program is closely based on <b>dottup</b> with the addition of
by default displaying only the minimal set of non-overlapping matches. 
<P>

This program uses the same algorithm as <b>diffseq</b> for finding a
minimal set of very good matches between two sequences.  <b>diffseq</b>
may be more convenient if you are looking at the differences between two
nearly identical sequences. 


<H2>
    Author(s)
</H2>
<!--
        Who has worked on the program in the past.
e.g.

   -->
This application was written by Gary Williams <a
href="mailto:gwilliam@hgmp.mrc.ac.uk">(gwilliam@hgmp.mrc.ac.uk)</a>



<H2>
    History
</H2>

<!--
        What changes have been made.
   -->
Written 14 Aug 2000.

<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:

   -->
This program is intended to be used by everyone and everything, from naive users to embedded scripts.



<H2>
    Comments
</H2>

<!--
	Any comments, requests etc. that people have about the program
   -->



</BODY>
</HTML>

