<HTML>

<HEAD>
  <TITLE>
  EMBOSS: isochore
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">




<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="index.html" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

isochore
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>





<H2>
    Function
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->
Plots isochores in large DNA sequences


<H2>
    Description
</H2>


The nuclear genomes of vertebrates are mosaics of isochores, very long
stretches (&gt;300kb) of DNA that are homogeneous in base composition and
are compositionally correlated with the coding sequences that they
embed.  Isochores can be partitioned in a small number of families that
cover a range of GC levels (GC is the molar ratio of guanine+cytosine in
DNA), which is narrow in cold-blooded vertebrates, but broad in
warm-blooded vertebrates. 

<p>
This application plots GC content over a sequence.  It is intended for
large sequences such as complete chromosomes or large genomic contigs,
although interesting results can also be obtained from shorter
sequences. 

<H2>
    Usage
</H2>


<b>Here is a sample session with isochore</b>
<p>
This reads yeast chromosome 3 from a remote web server so the sequence reading may take a little time.
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>isochore tgb:scchriii -graph cps</b>
Plots isochores in large DNA sequences
Output file [scchriii.iso]:
Created isochore.ps

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Mandatory qualifiers:
  [-sequence]          sequence   Sequence USA
  [-out]               outfile    Output file name
   -graph              xygraph    Graph type

   Optional qualifiers:
   -window             integer    Window size
   -shift              integer    Shift increment

   Advanced qualifiers: (none)
   General qualifiers:
  -help                boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose


</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#f5f5ff">
<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Mandatory qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>Sequence USA</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-out]<br>(Parameter 2)</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;sequence&gt;</i>.isochore</td>
</tr>

<tr>
<td>-graph</td>
<td>Graph type</td>
<td>EMBOSS has a list of known devices, including postscript, ps, hpgl, hp7470, hp7580, meta, colourps, cps, xwindows, x11, tektronics, tekt, tek4107t, tek, none, null, text, data, xterm, png</td>
<td><i>EMBOSS_GRAPHICS</i> value, or x11</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Optional qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-window</td>
<td>Window size</td>
<td>Integer 1 or more</td>
<td>1000</td>
</tr>

<tr>
<td>-shift</td>
<td>Shift increment</td>
<td>Integer 1 or more</td>
<td>100</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Advanced qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>



<H2>
    Input file format
</H2>

<b>isochore</b> reads a normal nucleic acid USA.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>Database entry: tgb:scchriii</h3>
<table width="90%"><tr><td bgcolor="#FFDDFF">
<pre>
&gt;SCCHRIII X59720.2 S.cerevisiae chromosome III complete DNA sequence
cccacacaccacacccacaccacacccacacaccacacacaccacacccacacacccaca
ccacaccacacccacaccacacccacacacccacacccacacaccacacccacacacacc
acacccacacacacccacacccacacaccacacccacacacacaccacacccacacacac
cacaccacacccacaccacacccacacccacacaccacacccacacccacaccccacacc
cacacaccacacccacacacaccacacccacacacacccacaccacacccacacaccaca
cccacacacccacacccacacacaccacacccacaccacacccacacccacacacccaca
ccctaacactaccctaacactaccctattctaaccctgattttacctgtctcccaactta
ctctccattaccctacctctccactcgttaccctgtctcattcaaccgtaccactcccaa
ccaccatccatctctctacttactaccaccaacccaccgtccaccataaccgttaccctc
caactacccgtatccaactccactaccgcttaccctaccatcgaccatgtcctactcact
gtactgttgttcacccaccatattgaaacgtctacaaatgatcgtaaataatacacatat
acttatcctaccactctaatcccactaccacatgccatactcaccttcacttgtattctg
atcggtcatacgcacacggatgctacagtatataccatctcaaacttaccctactttcat
attccactccatcacccatctctcaccatcagtaccaaatgcactcgcatcattatgcac
ggcacttgcctcagcggtctataccctgtgccatttacgcataacgcccatcattatcca
cattttaatatctatatctcattcggcgacaccaaatattgtataactgcccttaataca
tacgttataccacttttacaccatatactaaccactcaatttatacacacttatgtcaat
ataaccacaaaatcaccactaaaatcacctaaacataaaaatattctactcttcaacaat
aatacataaacacactcaattgcgtatctataccaccatgacgtcattaacgtaaaagtt
ccttaatattgcaaattgcttgaacggatgccatttcatgttgaatgtggtaacccaata
gcatgatatgagtaatgctttagtattgtttcagagttgtttcagtaatgttttagacaa
ggagaacatatagtagcaaacctctaatccggtagtacttaagaaactacagtttctatg
tacgaaagcagtaactatgtaattattacatttacatgacatataggaaggtccaataaa
cttactacattatgacctataagctagatcgtaattcattacgtcaacaggttatgagcc
ctcagagcaatgcttctgagaacataatcaatctatctagccccaacaattataaacagt
ggctgtacggtatcgagaccgctgctgaatatgctaacgaatatatgaacgaattcgttc
ataccggagatatccaatcaatgaaaagggattacaatctcagcgcgaatgatgaaagct
ttgtcaaaaccgtatttaacagtttcctggtaaagctctacaagaaaactatcgtgggtg
aagctgcatgtgaaatgaactggatatgtgatgattcgcttggaagggtctctgcttatg
atattttctcgcacttcgaagaaaactataatgaagtcactattggatccaggcttactc
ttatagaggacctaccaaatatatcctccaagcctgtagatgaaattgcttcctttttga
aaactctattcacgatgcttgaagacaatagcgaagaacaggacaaaaaaaaaagacgcg
acaccaatatcgcgttgttattaatgaccttcttacccgagttaaaagaatcattccacg
agaaattcggtgactctaaggctcttcagctatcacaagtcattagattctgtaaattaa


<font color=red>  [Part of this file has been deleted for brevity]</font>

atcttatggtaaaactccctttagtatctaccaaggtaaatggaacgtgttgaacagaga
ttttgagcgtgatattattccaatggctaggcatttcggtatggccctcgccccatggga
tgtcatgggaggtggaagatttcagagtaaaaaagcaatggaggaacggaggaagaatgg
agagggtattcgttctttcgttggcgcctccgaacaaacagatgcagaaatcaagattag
tgaagcattggccaagattgctgaggaacatggcactgagtctgttactgctattgctat
tgcctatgttcgctctaaggcgaaaaatttttttccgtcggttgaaggaggaaaaattga
ggatctcaaagagaacattaaggctctcagtatcgatctaacgccagacaatataaaata
cttagaaagtatagttccttttgacatcggatttcctaataattttatcgtgttaaattc
cttgactcaaaaatatggtacgaataatgtttagataatttttcagtaatcaactacgca
agtaaagcagtaaatacgttactgctggtattaatgtcatgtattgaggcaatgatgcta
tgctcttaacgacatgtagcttacaaaacgttcctatggttttactgcatggtatttaca
aattagatagcagttccatccgcccttgacatattattattcagacaaggtgtatatgag
cataaatatgtatatatgcacatgaggtttgtattaaccttgcaactgtaccaaaataca
atcttctttgctgctatttaacggcgtatgtgcagttcataaatgcgtgttctgtatagt
acatccgcgcacattcttcctagcggaagacatattaacgtagcccgtacgccctggtta
agacttggtatagttccaataattggaatattactgcatagtggtccaatagcaggattt
agcataaacatgataatatttagagatgctttcatccctctgacgaaggtggtagagaac
aagaaaatgaataacattatatcatattcctatatatatatatatatatatatatatatc
atataacggtggaaaatcccggcgatattacagataaacattacacccgcatgaatgtga
gccactactatattataacttagtgaataaagagtgttacaatagtgaggtgcatattat
tctaatgtaaggtctgtataagtacgaaatattctgaagtggcatccagtcaagcaaatg
gtaatattaaggaacttttaagttaatgacgtcatggtagtgctcgtacttcaagtcaaa
gtgtttatgtattatggttgaagaatagaatatttttatgtttaggtgattttagtggtg
atttttctgtaatattgacataagtgtatataaattaagtggttagtatacggtgaaaaa
gaggtataacgtatgtattaagggaatttatacgatatttgggcccgccgaatgagatat
agatattaaaatgtggataatcgtgggctttatgggtaaatggcacagggtatagaccgc
tgaggcaagtgccgtgcataatgatgtgggtgcatttggtactgatttagtgagaatggg
ccatggattggagtgtgagagtagggtaacttgagagtggtatatactgtagcatccgtg
tgcgtatgccccatcaatataagtgaaggtgagtatggcatgtggtggtggtatagagtg
gtagggtaagtatgtatgtattatttacgatcatttgttaacgtttcaatatggtgggtg
aacaacagtacagtgagtaggacatggtggatggtagggtaatagtagggtaagtggtgg
tggagttggatatgggtaattggagggtaacggttatgatgggcggtggatggtggtagt
aagtagagagatggatggtggttgggagtggtatggttgagtggggcagggtaacgagtg
gggaggtagggtaatgtgagggtaggtttggagacaggtaaaatcagggttagaataggg
tagtgttagggtagtgtgtgggtgtgggtgtgtgggtgtggtgtgtgggtgtggtgtgtg
tgggtgtggtgtgtgggtgtgggtgtgtgggtgtggtgggtgtggtgtgtgtg
</pre>
</td></tr></table><p>


<H2>
    Output file format
</H2>



<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>Graphics File: isochore.ps</h3>
<p><img src="isochore.1.isochore.gif" alt="[isochore results]">
<p><h3>File: scchriii.iso</h3>
<table width="90%"><tr><td bgcolor="#DDFFDD">
<pre>
Position	Percent G+C 1 .. 316613
500	0.506
600	0.469
700	0.446
800	0.417
900	0.390
1000	0.379
1100	0.367
1200	0.370
1300	0.367
1400	0.355
1500	0.362
1600	0.375
1700	0.375
1800	0.380
1900	0.398
2000	0.404
2100	0.400
2200	0.404
2300	0.408
2400	0.414
2500	0.417
2600	0.421
2700	0.422
2800	0.419
2900	0.417
3000	0.426
3100	0.429
3200	0.437
3300	0.432
3400	0.422
3500	0.416
3600	0.412
3700	0.401
3800	0.391


<font color=red>  [Part of this file has been deleted for brevity]</font>

312600	0.337
312700	0.341
312800	0.348
312900	0.351
313000	0.352
313100	0.360
313200	0.381
313300	0.375
313400	0.379
313500	0.394
313600	0.394
313700	0.389
313800	0.390
313900	0.392
314000	0.399
314100	0.408
314200	0.403
314300	0.419
314400	0.415
314500	0.397
314600	0.396
314700	0.391
314800	0.386
314900	0.390
315000	0.384
315100	0.363
315200	0.349
315300	0.341
315400	0.330
315500	0.338
315600	0.351
315700	0.365
315800	0.367
315900	0.372
316000	0.389
316100	0.421
</pre>
</td></tr></table><p>


<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>

None.


<H2>
    References
</H2>


<ol>

<li>
Bernardi G 
Isochores and the evolutionary genomics of vertebrates. 
Gene 2000 Jan 4;241(1):3-17

<li>
Pesole G, Bernardi G, Saccone C 
Isochore specificity of AUG initiator context of human genes. 
FEBS Lett 1999 Dec 24;464(1-2):60-2
<li>

Bernardi G 
The human genome: organization and evolutionary history.
Annu Rev Genet 1995;29:445-76

</ol>

<H2>
    Warnings
</H2>


None.

<H2>
    Diagnostic Error Messages
</H2>

None.


<H2>
    Exit status
</H2>

It always exits with a status of 0.


<H2>
    Known bugs
</H2>

None.


<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="banana.html">banana</a></td><td>Bending and curvature plot in B-DNA</td></tr>
<tr><td><a href="btwisted.html">btwisted</a></td><td>Calculates the twisting in a B-DNA sequence</td></tr>
<tr><td><a href="chaos.html">chaos</a></td><td>Create a chaos game representation plot for a sequence</td></tr>
<tr><td><a href="compseq.html">compseq</a></td><td>Counts the composition of dimer/trimer/etc words in a sequence</td></tr>
<tr><td><a href="dan.html">dan</a></td><td>Calculates DNA RNA/DNA melting temperature</td></tr>
<tr><td><a href="freak.html">freak</a></td><td>Residue/base frequency table or plot</td></tr>
<tr><td><a href="wordcount.html">wordcount</a></td><td>Counts words of a specified size in a DNA sequence</td></tr>
</table>

<!--
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find.
   -->


<H2>
    Author(s)
</H2>


This application was written by Peter Rice <a
href="mailto:pmr@sanger.ac.uk">(pmr@sanger.ac.uk)</a> Informatics
Division, The Sanger Centre, Wellcome Trust Genome Campus, Hinxton,
Cambridge, CB10 1SA, UK.

<H2>
    History
</H2>


<H2>
    Target users
</H2>


This program is intended to be used by everyone and everything,
from naive users to embedded scripts. 

<H2>
    Comments
</H2>


<pre>

</pre>


</BODY>
</HTML>

