<HTML>

<HEAD>
  <TITLE>
  EMBOSS
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
megamerger
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
    Function
</H2>
Merge two large overlapping nucleic acid sequences

<H2>
    Description
</H2>

<b>megamerger</b> takes two overlapping sequences and merges them into one sequence. 
It could thus be regarded as the opposite of what splitter does.
<P>

The sequences can be very long.
The program does a match of all sequence words of size 20 (by default). It then reduces
this to the minimum set of overlapping matches by sorting the matches in order of size (largest size first)
and then for each such match it removes any smaller matches that overlap. The result is a set of the longest
ungapped alignments between the two sequences that do not overlap with each other. If the two sequences are
identical in their region of overlap then there will be one region of match and no mismatches.
<P>

It should be possible to merge sequences that are Mega bytes long.
Compare this with the program <b>merger</b> which does a more accurate
alignment of more divergent sequences using the Needle and Wunsch algorithm but which uses much more memory.
<P>

The sequences should ideally be identical in their region of overlap.
If there are any mismatches between the two sequences then <b>megamerger</b> will still attempt to
create a merged sequence, but you should check that this is what you required.
<P>

A report of the actions of <b>megamerger</b> is written out.
Any actions that require a choice between using regions of the two sequences
where they have a mismatch is marked with the word <b>WARNING!</b>.
The sequence in these regions is written out in uppercase.
All other regions of the output sequence are written in lowercase.
<p>

Where there is a mismatch then the sequence that is chosen to supply the region of the mismatch in the
final merged sequence is that sequence whose mismatch region is furthest from the start of end of the sequence.



<p>

<H2>
    Usage
</H2>

<b>Here is a sample session with megamerger</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>megamerger tembl:ba000025 tembl:af129756 </b>
Merge two large overlapping nucleic acid sequences
Word size [20]: <b></b>
output sequence [ba000025.merged]: <b></b>
Output file [ba000025.megamerger]: <b>report</b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>


<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-asequence]         sequence   Nucleotide sequence filename and optional
                                  format, or reference (input USA)
  [-bsequence]         sequence   Nucleotide sequence filename and optional
                                  format, or reference (input USA)
   -wordsize           integer    [20] Word size (Integer 2 or more)
  [-outseq]            seqout     [<sequence>.<format>] Sequence filename and
                                  optional format (output USA)
  [-outfile]           outfile    [*.megamerger] Output file name

   Additional (Optional) qualifiers:
   -prefer             boolean    [N] When a mismatch between the two sequence
                                  is discovered, one or other of the two
                                  sequences must be used to create the merged
                                  sequence over this mismatch region. The
                                  default action is to create the merged
                                  sequence using the sequence where the
                                  mismatch is closest to that sequence's
                                  centre. If this option is used, then the
                                  first sequence (seqa) will always be used in
                                  preference to the other sequence when there
                                  is a mismatch.

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-asequence" associated qualifiers
   -sbegin1            integer    Start of the sequence to be used
   -send1              integer    End of the sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-bsequence" associated qualifiers
   -sbegin2            integer    Start of the sequence to be used
   -send2              integer    End of the sequence to be used
   -sreverse2          boolean    Reverse (if DNA)
   -sask2              boolean    Ask for begin/end/reverse
   -snucleotide2       boolean    Sequence is nucleotide
   -sprotein2          boolean    Sequence is protein
   -slower2            boolean    Make lower case
   -supper2            boolean    Make upper case
   -sformat2           string     Input sequence format
   -sdbname2           string     Database name
   -sid2               string     Entryname
   -ufo2               string     UFO features
   -fformat2           string     Features format
   -fopenfile2         string     Features file name

   "-outseq" associated qualifiers
   -osformat3          string     Output seq format
   -osextension3       string     File name extension
   -osname3            string     Base file name
   -osdirectory3       string     Output directory
   -osdbname3          string     Database name to add
   -ossingle3          boolean    Separate file for each entry
   -oufo3              string     UFO features
   -offormat3          string     Features format
   -ofname3            string     Features file name
   -ofdirectory3       string     Output directory

   "-outfile" associated qualifiers
   -odirectory4        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write standard output
   -filter             boolean    Read standard input, write standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-asequence]<br>(Parameter 1)</td>
<td>Nucleotide sequence filename and optional format, or reference (input USA)</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-bsequence]<br>(Parameter 2)</td>
<td>Nucleotide sequence filename and optional format, or reference (input USA)</td>
<td>Readable sequence</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-wordsize</td>
<td>Word size</td>
<td>Integer 2 or more</td>
<td>20</td>
</tr>

<tr>
<td>[-outseq]<br>(Parameter 3)</td>
<td>Sequence filename and optional format (output USA)</td>
<td>Writeable sequence</td>
<td><i>&lt;*&gt;</i>.<i>format</i></td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 4)</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.megamerger</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-prefer</td>
<td>When a mismatch between the two sequence is discovered, one or other of the two sequences must be used to create the merged sequence over this mismatch region. The default action is to create the merged sequence using the sequence where the mismatch is closest to that sequence's centre. If this option is used, then the first sequence (seqa) will always be used in preference to the other sequence when there is a mismatch.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>


<H2>
    Input file format
</H2>

<b>megamerger</b> reads any two Sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:ba000025' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:ba000025</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   BA000025; SV 2; linear; genomic DNA; STD; HUM; 2229817 BP.
XX
AC   BA000025; AP000502-AP000521;
XX
DT   09-DEC-2004 (Rel. 82, Created)
DT   14-NOV-2006 (Rel. 89, Last updated, Version 4)
XX
DE   Homo sapiens genomic DNA, chromosome 6p21.3, HLA Class I region.
XX
KW   .
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-2229817
RA   Hirakawa M., Yamaguchi H., Imai K., Shimada J.;
RT   ;
RL   Submitted (21-AUG-2001) to the EMBL/GenBank/DDBJ databases.
RL   Mika Hirakawa, Japan Science and Technology Corporation (JST), Advanced
RL   Databases Department; 5-3, Yonbancho, Chiyoda-ku, Tokyo 102-0081, Japan
RL   (E-mail:mika@tokyo.jst.go.jp, URL:http://www-alis.tokyo.jst.go.jp/,
RL   Tel:81-3-5214-8491, Fax:81-3-5214-8470)
XX
RN   [2]
RA   Shiina S., Tamiya G., Oka A., Inoko H.;
RT   "Homo sapiens 2,229,817bp genomic DNA of 6p21.3 HLA class I region";
RL   Unpublished.
XX
DR   EPD; EP11158; HS_TNF.
DR   EPD; EP11159; HS_LTA.
DR   EPD; EP73522; HS_HLA-B.
DR   EPD; EP73908; HS_GTF2H4.
DR   EPD; EP73940; HS_NEU1.
DR   EPD; EP74013; HS_VARS2.
DR   EPD; EP74203; HS_MRPS18B.
DR   EPD; EP74346; HS_HLA-E.
DR   EPD; EP74389; HS_BAT1.
DR   EPD; EP74485; HS_IER3.
DR   GDB; 11515913.
DR   GOA; P59942.
DR   IMGT/HLA; HLA02629; J*01010102.
DR   RFAM; RF00017.
DR   RFAM; RF00019.
DR   RFAM; RF00026.
DR   RFAM; RF00100.
DR   RFAM; RF00137.
DR   RFAM; RF00276.


<font color=red>  [Part of this file has been deleted for brevity]</font>

     ttggccccac cccagcatgt ctccaggttc ctctcagccc tggttccttt tggccctgca   2226900
     gtcacaatgg gcaacactgt gacgcaccct gtcctgtgtc acagtgtcat acactcaggc   2226960
     tcacattgcc cctaggccac ttgccagcca agggacatgg ccacattttg tgtcttctgc   2227020
     acctcagcct tgctttcaag tgcaggtgat gatggcaccc acgcagaaca aatgttattt   2227080
     gctatcttcg tcgagtttag tcatccaatt ttccaaccct cactgggcaa ggaagagtgt   2227140
     ggtttccacc aagaaggcag gatgtcagca gtcacagggg caaccaacag ggaaagccgc   2227200
     cggaaaatag accccacagg aagcacaggt gtccagtgga gatgggaacc ctgcagattt   2227260
     gaccgtcttt aagcagatta gagagattac cgttactaac aacttagcca taaaagttta   2227320
     ttagctattt tcaaaaagca taaaattatg taatataatt ttttttaaat ttccatcaat   2227380
     acaaaactaa tctgggcact gcaacttccg gtgggcaact gggataggcg gcatcatcag   2227440
     gaaggcgagc cctgccgtgc cccatgtgcc agtgccccag atggcggcag cctccccaga   2227500
     agcaccttgt atctcccctg cacagggcca gggtcccagc ttcccataca ccttctcctg   2227560
     ctttttcttt tctgtccttt cctttttcaa taaaccacct gcaaaaaggg aaaaccattc   2227620
     tgaggacaag aaacatgtca atgggaaata cacagttgcc agagggtaaa aggccctgtt   2227680
     cattctcatt gaaaagctca ggtatttctg ttaaagtctc tccttttact ttaggatgct   2227740
     gactcctgcg tccatctcaa cctgggcatc gtgccaccac cttcaagaag agaaaaacta   2227800
     agtagtgctt tgcaaagggg cagcagcatt tctcatttct gaccatgtca ggcacatggc   2227860
     catgcagatg agcaggtggg ggacacaggt gagtctccag acctgctctc ctcccacagt   2227920
     acattcttga gtctttttaa acagttgtga aaatgccaca gatgcaagca cctgtgggcc   2227980
     actcccatgg ggaccgttgc acaaggcagt gccactcatt ctcagaacct cctaccatgg   2228040
     gctatgctta gtgacccgag gccaagccaa ggaagacgcc agccacaggg tgccatcctc   2228100
     aggggcatgc tgccagcagg ggcaaagtta tccctagcaa caagatacag aaagaaagaa   2228160
     aaaaggaagg aaatgtagcc aatgggccgg ttcaggttct tgactttgcc acacaaaaga   2228220
     atttgagagc aagtccaaag taaaagtcag caagagaatt tattgcaaag tgaaagtaca   2228280
     ctctgacagc tgatcagagc agctgctcaa aagagagaca gtaccctccc ctcacgggag   2228340
     tcttacatga ttattcatga ataggtggga aggggtattg ttttaagcat gttctgtggt   2228400
     ctcttgaacg tgcatgcact gtggttgtac atatcagcac acacatctta cgtctcatta   2228460
     gcatcttaac ttccctctca gagttgtgtt tgctactatt gtaatgagca taggtcagcc   2228520
     caaggacact attcatgggt ttctgggctt cctcagatgt ggggatgcct cccttggctc   2228580
     ttctacctct ttgctgcagg atgttctaac cacaagccca ggatatggtt tgcgcactgt   2228640
     cgaacagctt gttctctcca tcaacctgac aagtctcttg tttcctttca agggaggctg   2228700
     tgaacaccct atctcactga cctcagaagg acagtacagc agtagccacc atgaccaaaa   2228760
     agatgattcc agaagtgcag gacaactccc tacccagagg ctgtggctgt gcagtaacac   2228820
     accaagaggg gagtccagct ggctctcagg gtgctcacta ccctcatctg ggggcctgga   2228880
     ggacgtcaat tcctgagaac gccacgttct agtgagtaga atgaactgag agatacacag   2228940
     caaagctcca catacttttc cttttctttg tgcccgcagt gttcttcatc agtgtgctct   2229000
     cgcttttcag ctactactgt tggctggctg gaaaaaatag aacaatagta aaaattagag   2229060
     accagtcttt ggtgatgaag agaaatattg gctacttcca gtattttcta gctttggtta   2229120
     tggttgcagt tttccagctc accttgtggg gatgaattca gaaaaaagtt acaaattgaa   2229180
     atgaacatgc cagaagtatt ggctcaaatc aacgttgtcc tattaagcca cttagtgaat   2229240
     caaaagaccg cttgttggac tgttaatctc ggtggccaga gaaaggagct gaagaaggtg   2229300
     ttgccagatc aggaacaaat aattacagcg gcaatagaaa atggaagacc acttgttcat   2229360
     aaccatttga ataagggcaa ggtgtatgga aacacattat gaactgatat tttcagtttt   2229420
     gtttgcaaga aaatgattaa taaggtgaaa taattgaagt atcacggaag atacattaaa   2229480
     aaaaaaaaaa gcctttgtac agtttgctgg agccacagat gtcctactcc agagcagaac   2229540
     aatgcctgaa tcttcagggt ccatttctgc cgcattcact agcaaccaca aatgtgactt   2229600
     aattttactt tggaaataat gcttacccat tgtgagatgc tgtaatatga accatcatta   2229660
     catgttaaca tggcacatgg aattttgagt gtctaagtta catttttaga gttgtttctt   2229720
     agtagccatg tgagtttcca ctccaaaaac acaagctaaa aacttgtttt gagtgaagga   2229780
     catctagggc aaatggtggc tgaaagtgaa tgagatc                            2229817
//
</pre>
</td></tr></table><p>
<p><h3>Database entry: tembl:af129756</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   AF129756; SV 1; linear; genomic DNA; STD; HUM; 184666 BP.
XX
AC   AF129756;
XX
DT   12-MAR-1999 (Rel. 59, Created)
DT   14-NOV-2006 (Rel. 89, Last updated, Version 5)
XX
DE   Homo sapiens MSH55 gene, partial cds; and CLIC1, DDAH, G6b, G6c, G5b, G6d,
DE   G6e, G6f, BAT5, G5b, CSK2B, BAT4, G4, Apo M, BAT3, BAT2, AIF-1, 1C7, LST-1,
DE   LTB, TNF, and LTA genes, complete cds.
XX
KW   .
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Euarchontoglires; Primates; Haplorrhini; Catarrhini; Hominidae;
OC   Homo.
XX
RN   [1]
RP   1-184666
RX   DOI; 10.1101/gr.1736803.
RX   PUBMED; 14656967.
RA   Xie T., Rowen L., Aguado B., Ahearn M.E., Madan A., Qin S., Campbell R.D.,
RA   Hood L.;
RT   "Analysis of the gene-dense major histocompatibility complex class III
RT   region and its comparison to mouse";
RL   Genome Res. 13(12):2621-2636(2003).
XX
RN   [2]
RP   1-184666
RA   Rowen L., Madan A., Qin S., Shaffer T., James R., Ratcliffe A., Abbasi N.,
RA   Dickhoff R., Loretz C., Madan A., Dors M., Young J., Lasky S., Hood L.;
RT   "Sequence of the human major histocompatibility complex class III region";
RL   Unpublished.
XX
RN   [3]
RP   1-184666
RA   Rowen L.;
RT   ;
RL   Submitted (22-FEB-1999) to the EMBL/GenBank/DDBJ databases.
RL   Department of Molecular Biotechnology, Box 357730 University of Washington,
RL   Seattle, WA 98195, USA
XX
RN   [4]
RP   1-184666
RA   Rowen L.;
RT   ;
RL   Submitted (28-OCT-1999) to the EMBL/GenBank/DDBJ databases.
RL   Multimegabase Sequencing Center, University of Washington, PO Box 357730,
RL   Seattle, WA 98195, USA


<font color=red>  [Part of this file has been deleted for brevity]</font>

     aaaccagttt accaccactc ctaacactaa acttaaatct gactctaaat gtaagtccaa    181740
     tctgagccac aagcctaaag ttgaacttta tcctgcttta tgaattattc atccattcct    181800
     ccatttagtg agtatctgcg tgcctaacac atgctgggca ttgtcctaag gcaggaggga    181860
     catggaggca aagggatcag agaaggtacc agcacctgtg gagcttgtat tccagtgagg    181920
     ccagacggaa aagaaagaaa ctgaagaaga aattggtact atgagaaaat aagacaggct    181980
     gatgttgtaa gagtggcagg gagctacttt taaatacagt agtcagcaaa atcctctttg    182040
     agtgtttggg tggcactgga gctgagaccc aaatgacaaa aaatagtgac caggtaaaag    182100
     tttgggagca aagcatttca ggtaaaggga gcagctactg caaaggctgg aaggcggaac    182160
     caagctgggg gtgttgacga caaacagaag gccagtgtgg ctggagcaga gagagagact    182220
     gggaggcggg tgggagatga ggtcagagag gagggcaggg gccaggtcat gcagggccat    182280
     gcaagaaggg taaagcctct agatttcatc cagccacagg aagcctttaa aggtcgtcag    182340
     agtgtgtggt gcgtgcgtgt gtgtgtgtgt gtgtgtgtgt gttgcagggg agagaggggg    182400
     agggagagag agagagagag agagaagagg gaggtgagca gaggtgattg gatttttttt    182460
     tcttttgaca tggtgtcttg ctctgtggcc taggctggag tgcagtggca ccatcatagc    182520
     ccactgcaac ctcaaaacca tgggctcaag tcatccttcc acctcagctt cccaagtatc    182580
     taggactaca ggtgtgtgcc actgtgcctg gctaatttta aaaaatattt taaaattttt    182640
     gttgagacag ggtctatgct gctcaggctg gtctcgaact cctggtttca agtgatctgc    182700
     ccatcttggc ctcccaaagt ttttttttgt tagtttgaga ggcggtttcg ctcgttgccc    182760
     aggctggagt gcaatgactg atctcatctc actgcaacct ctgcctcctg ggttcaagcg    182820
     attctcctgc ttcagcctcc caagtagctg ggattacagg tgcatgccac cattcccggc    182880
     taattttttg tatttagtag agatggggtt tcaccatgtt agtcaggctg atctcaaact    182940
     cctgacctca ggtgatccgc ctgcctcagc ctcccaaagt tttgggatta caggtgtgag    183000
     ccaccatgct gggccagcct cccaaagttt tgggattaca ggcatgagtc accacactgg    183060
     ccctggattt tttttctttc ttttttttgg agacggagtc tcactctgtt gcccaggctg    183120
     gagtgcaatg gcgtaatctc agctcactgc aacctctgct gcccgggttc aaacgattct    183180
     cctgtcttag cctcctgagt agctgggatt ataggtgcat gccaccatgc ctggctaatt    183240
     tttgtacttt tagtagagaa agtacaccat cttggccagg ctggtctcga actcctgacc    183300
     tcaggtgatc cacttgcgtc ggcctcccaa agtgctggga ttacaggcgt gagacaccgc    183360
     acccagcctt tttttttttt tttcttttaa gacagaatcg ctctgtcacc caggctggag    183420
     tgcagtggca caatctcggc tcactgcaac ctctgcctcc caggtttaag caatccacct    183480
     atgtcagtct cccaagtagc tgggattata ggtgcatgtc accatgcctg gctaattttt    183540
     gtacttttag tatagaaagt acaccatgtt ggccaggctg gtcttgaact cctgacctca    183600
     agtgatccgc ctgcctcagc ctcccgaagt gctggaatta cagacatgtg ccactgcacc    183660
     cggcctggtt ttttttttct aagagatgga gtctcacttt tctgcccagg ttggagtgca    183720
     atggcaccat catagctcac tgcagccttc aactcttggc ctcaggcaat ccttgcacct    183780
     tagcctcgca gtgttgggat tacaggcatg agccactgag ccttgcctgg actttttttt    183840
     ttttttgaga tggcgtctcg ctctgttgcc caggttggag tgctacggca tgatcttggc    183900
     tcactgcaac ttccacctcc caggttcaag cgattctctt gcctcggccc cccgagtagc    183960
     tgggattaca ggcatgcgcc accgtgcctg gctaattttg gtatttttag tagagatagg    184020
     gtttcatcat gttgggcagg ctggtcttga actcctgacc tcgtgatcca cccacctcgg    184080
     cctcccaaag tgctgggatt ataggcatag ccaacgcgcc cagcctggac ttgtttttaa    184140
     aagatcactg tggctcctgt gtttaggctg gctggtagga gacaggtggc agtggcattg    184200
     atggtgaaga gaaaatagtg gcagccatgg agatggagag aagtagacaa gtttgggata    184260
     tattatacat tccaggggta gaaacaacag gactagatga tggattgatg ggtgggagat    184320
     gtagatactg ggagagaagc aggattctga tggatggaaa aactaaaaaa ttctattttg    184380
     ggtgtggtaa gtctaagtct attagacatg caagtagaga tgtcactggg cagatacaca    184440
     tctggatttc aggggcaagg tccaagctag agaaagaaac ctgggcatgg tcagcatgag    184500
     gatggtgttt aaagccatgg aacttatctt gtgcatccct ataagacccc tttgaggcac    184560
     ttgtttcccc tcacaatgga tgcagtgcat cttccattct gaattccaga ggcaacaacc    184620
     tcctgctcct agaagctaaa ctctccagac ttagtcttct gaattc                   184666
//
</pre>
</td></tr></table><p>

<H2>
    Output file format
</H2>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: report</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
# Report of megamerger of: BA000025 and AF129756

BA000025 overlap starts at 193957
AF129756 overlap starts at 1

Using BA000025 1-193956 as the initial sequence

Matching region BA000025 193957-194108 : AF129756 1-152
Length of match: 152

WARNING!
Mismatch region found:
Mismatch BA000025 194109-194109
Mismatch AF129756 153-153
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 194110-197910 : AF129756 154-3954
Length of match: 3801

WARNING!
Mismatch region found:
Mismatch BA000025 197911-197914
Mismatch AF129756 3954
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 197915-198248 : AF129756 3955-4288
Length of match: 334

WARNING!
Mismatch region found:
Mismatch BA000025 198249-198254
Mismatch AF129756 4289-4292
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 198255-198911 : AF129756 4293-4949
Length of match: 657

WARNING!
Mismatch region found:
Mismatch BA000025 198912-198914
Mismatch AF129756 4949
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 198915-200846 : AF129756 4950-6881
Length of match: 1932

WARNING!
Mismatch region found:
Mismatch BA000025 200847-200847
Mismatch AF129756 6882-6882


<font color=red>  [Part of this file has been deleted for brevity]</font>


WARNING!
Mismatch region found:
Mismatch BA000025 377379-377381
Mismatch AF129756 183383
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 377382-377594 : AF129756 183384-183596
Length of match: 213

WARNING!
Mismatch region found:
Mismatch BA000025 377595-377595
Mismatch AF129756 183597-183597
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 377596-377788 : AF129756 183598-183790
Length of match: 193

WARNING!
Mismatch region found:
Mismatch BA000025 377789-377790
Mismatch AF129756 183790
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 377791-377930 : AF129756 183791-183930
Length of match: 140

WARNING!
Mismatch region found:
Mismatch BA000025 377931-377931
Mismatch AF129756 183931-183931
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 377932-378422 : AF129756 183932-184422
Length of match: 491

WARNING!
Mismatch region found:
Mismatch BA000025 378423-378423
Mismatch AF129756 184423-184423
Mismatch is closer to the ends of AF129756, so use BA000025 in the merged sequence

Matching region BA000025 378424-378666 : AF129756 184424-184666
Length of match: 243

BA000025 overlap ends at 378666
AF129756 overlap ends at 184666

Using BA000025 378667-2229817 as the final sequence

</pre>
</td></tr></table><p>
<p><h3>File: ba000025.merged</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&gt;BA000025 BA000025.2 Homo sapiens genomic DNA, chromosome 6p21.3, HLA Class I region.
gatctccagagcactcttccctgcagggcaccctcccatcccagactccaggcacctggc
atgggtggacatctttactttctgggccagcttcagcagagctatgtcatcaccatagaa
ctccaggattccctggttctttttggcaaagacatcaaaccctggggagatcaccgcctt
ctcaataaggaattctttgccccactgggatttggggtctcctggaaatgatacactaga
ttaggctagaccagggctcctgcaggggccagaggctgggtgaggtggtaggatctgtgg
cttcaggatcaggaggctggtgcatcccctgccttacccacattgaccctccacagggag
tggtcgttgccatcgcggaagcaatgagctgctgtcaggacccattggtcggagatgagg
gccccccggcaggtctcttggctcttgggctgcaggggaacaggtgattttcagagattg
cagtatgtctggcccatggccgcttttacctctggaatccaagccctgcccctccttcct
ggtaccttaatagtgacatgccagggtgtcctctcctggtcagaggcgtttgctgacatg
ttccccaccccgcagatggtgtctgtgagcttggagacatctgtgggtgtgaggatcaga
tggggaaggaggcaagtgaggggcactgtgtccaggttcccaacacgggcctctggcggg
ctcctcaccatcctccccacaccaaggagggcaaagctcactcacccagcatatgttcaa
agacctggtgcagagcctttgtgtcctgcagaatgaaggcatgcctctcaccatccttct
tggaccctagctcattcagttctctccagtccacatccagcttgcccaccccgatggcat
agatgtctggtggggaagagggaaatcaccagactcctgtggctttggggctaccccatg
agacaggaggctgtcatctgaaactcactgtgtccaatcaagacctacatgagctggacc
cctgcgtcctccccactgctacctgtctgccttcatttcctgccactccctgcccttcac
tctcctgcagcacacagcctctttgaagttcctcaaatccataggcatggtcacacctca
ggccctttgcccagctgtgcctctgcctagttcactcctcccccccagacttccacatgg
ctcactttcgtacctttttaagtcttggctcaaatgtcaccttctcagtgaggccttccc
tggtcttcctgtctaaaactgcaatgccccagacaaactttcatccccactttgggaggc
aaggtgggaggatcccttgaagccagaagtttgagaccagcctgggcaacatggcaacac
cccttagcttgtgtcacctaccacctgctgggttctatggttttcttatcctgtttattc
cctgtaatggtggaattgtgtcccccagaaagatgtgttcgagtcctaatccccagtatc
tgtgactttatttggaaaaagggtctttgcagatgtaatcaagttaagattaagtcatac
tagattagggtgagctctaatccaatgactgaggtccttataagaagaggtaagccagag
ccaggcgtggtggctcacacctgtaatcaccaggaggcggtggttgtggtgagccaagat
cgcgccattgcactccagcctgggcaacaagagcaaaaccccgtctcaaaaaaaaaaaaa
gaagaggtgagccgggcacggtggctcacacctgtaatcccagcactctgggaggctgag
gcgggcagatcacgaggtcaggaattcaagaccagcctgaccaacatggtgaaaccctgt
ctctactaaaaatacaaaaattagccagacatgctggcacacacctgtaatcccagctac
tcaggaggctgaggcaggagaatcgcttgaaccgggaggcggatgttgcagtgagccgag
attgcaccactgcactccagcctgggcaacagagcaagactccatctcaaaaaaaaaaaa
aaaaaaaaaaagtgaactggctgggcatggtggtgactcatgcctgtaatcccggcagtt
tttttgaggcgaaggcaggcagatcgccttgaggccaggagtttaagaccagcctagcca
acatggcgagaccatgtctctactaaaaatacaaaaatttgccgggcatggtggcacatg
cctgtaatcccagcttcttgggagactgaggcacgagaatcacctgaacccaggaggcag
aggttacagtgagccgggatcccgccactgcactgcagcctgggcttctgggtgacagag
cgagactctgtctcaaacaaatgaacagaaaaagaagaaaggaatttggacacaaagaca
caggtagtgggtctcctatctatataagagaacagcatgtaatgacacagaggcacacac
agaaaagaaggcgagttgaagacagaggcagagaatgggtttatgctgccgcaagccaag
gttggagctgccggcagccggaaaaggcaggaaagaattcttcccaagagccttctgagg
aagcacggccctgccaacaccttgatttcagacttctaacctccagaactgtaagaaaaa
gaaattctgtgttctaagccacccaggtttgtggtagtttggtaagtacttttaaatgac
tgaatgaatagaaagaactcagaacacaacatggaaactaaacctcagatctggtcttcc
tctgtaaaaggtagcatctgggagaagggcctaaagccacgttttcccactggaggccct
ggacccacacaacaggccgcgcctgtcctccgactgtggtgccagtcagaactgccctca
gacagaccacagagtctactcctctcccagcctttgcaccccttgtggcccatttttgtt


<font color=red>  [Part of this file has been deleted for brevity]</font>

cctcggtctgtctccaccaggccctgtgagggtgggtggaggctctctccaagccctcgt
ttggccccaccccagcatgtctccaggttcctctcagccctggttccttttggccctgca
gtcacaatgggcaacactgtgacgcaccctgtcctgtgtcacagtgtcatacactcaggc
tcacattgcccctaggccacttgccagccaagggacatggccacattttgtgtcttctgc
acctcagccttgctttcaagtgcaggtgatgatggcacccacgcagaacaaatgttattt
gctatcttcgtcgagtttagtcatccaattttccaaccctcactgggcaaggaagagtgt
ggtttccaccaagaaggcaggatgtcagcagtcacaggggcaaccaacagggaaagccgc
cggaaaatagaccccacaggaagcacaggtgtccagtggagatgggaaccctgcagattt
gaccgtctttaagcagattagagagattaccgttactaacaacttagccataaaagttta
ttagctattttcaaaaagcataaaattatgtaatataattttttttaaatttccatcaat
acaaaactaatctgggcactgcaacttccggtgggcaactgggataggcggcatcatcag
gaaggcgagccctgccgtgccccatgtgccagtgccccagatggcggcagcctccccaga
agcaccttgtatctcccctgcacagggccagggtcccagcttcccatacaccttctcctg
ctttttcttttctgtcctttcctttttcaataaaccacctgcaaaaagggaaaaccattc
tgaggacaagaaacatgtcaatgggaaatacacagttgccagagggtaaaaggccctgtt
cattctcattgaaaagctcaggtatttctgttaaagtctctccttttactttaggatgct
gactcctgcgtccatctcaacctgggcatcgtgccaccaccttcaagaagagaaaaacta
agtagtgctttgcaaaggggcagcagcatttctcatttctgaccatgtcaggcacatggc
catgcagatgagcaggtgggggacacaggtgagtctccagacctgctctcctcccacagt
acattcttgagtctttttaaacagttgtgaaaatgccacagatgcaagcacctgtgggcc
actcccatggggaccgttgcacaaggcagtgccactcattctcagaacctcctaccatgg
gctatgcttagtgacccgaggccaagccaaggaagacgccagccacagggtgccatcctc
aggggcatgctgccagcaggggcaaagttatccctagcaacaagatacagaaagaaagaa
aaaaggaaggaaatgtagccaatgggccggttcaggttcttgactttgccacacaaaaga
atttgagagcaagtccaaagtaaaagtcagcaagagaatttattgcaaagtgaaagtaca
ctctgacagctgatcagagcagctgctcaaaagagagacagtaccctcccctcacgggag
tcttacatgattattcatgaataggtgggaaggggtattgttttaagcatgttctgtggt
ctcttgaacgtgcatgcactgtggttgtacatatcagcacacacatcttacgtctcatta
gcatcttaacttccctctcagagttgtgtttgctactattgtaatgagcataggtcagcc
caaggacactattcatgggtttctgggcttcctcagatgtggggatgcctcccttggctc
ttctacctctttgctgcaggatgttctaaccacaagcccaggatatggtttgcgcactgt
cgaacagcttgttctctccatcaacctgacaagtctcttgtttcctttcaagggaggctg
tgaacaccctatctcactgacctcagaaggacagtacagcagtagccaccatgaccaaaa
agatgattccagaagtgcaggacaactccctacccagaggctgtggctgtgcagtaacac
accaagaggggagtccagctggctctcagggtgctcactaccctcatctgggggcctgga
ggacgtcaattcctgagaacgccacgttctagtgagtagaatgaactgagagatacacag
caaagctccacatacttttccttttctttgtgcccgcagtgttcttcatcagtgtgctct
cgcttttcagctactactgttggctggctggaaaaaatagaacaatagtaaaaattagag
accagtctttggtgatgaagagaaatattggctacttccagtattttctagctttggtta
tggttgcagttttccagctcaccttgtggggatgaattcagaaaaaagttacaaattgaa
atgaacatgccagaagtattggctcaaatcaacgttgtcctattaagccacttagtgaat
caaaagaccgcttgttggactgttaatctcggtggccagagaaaggagctgaagaaggtg
ttgccagatcaggaacaaataattacagcggcaatagaaaatggaagaccacttgttcat
aaccatttgaataagggcaaggtgtatggaaacacattatgaactgatattttcagtttt
gtttgcaagaaaatgattaataaggtgaaataattgaagtatcacggaagatacattaaa
aaaaaaaaaagcctttgtacagtttgctggagccacagatgtcctactccagagcagaac
aatgcctgaatcttcagggtccatttctgccgcattcactagcaaccacaaatgtgactt
aattttactttggaaataatgcttacccattgtgagatgctgtaatatgaaccatcatta
catgttaacatggcacatggaattttgagtgtctaagttacatttttagagttgtttctt
agtagccatgtgagtttccactccaaaaacacaagctaaaaacttgttttgagtgaagga
catctagggcaaatggtggctgaaagtgaatgagatc
</pre>
</td></tr></table><p>

<p>

A merged sequence is written out.

<p>

Where there has been a mismatch between the two sequences, the merged
sequence is written out in uppercase and the sequence whose mismatch
region is furthest from the edges of the sequence is used in the merged
sequence. 

<p>

The name and description of the first input sequence is used for the
name and description of the output sequence. 

<p>

A report of the merger is written out.

<P>


<H2>
    Data files
</H2>

None.

<H2>
    Notes
</H2>

If you run out of memory, use a larger wordsize.

<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>


None.

<H2>
    Exit status
</H2>

It always exits with status 0.

<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr>
<td><a href="cons.html">cons</a></td>
<td>Creates a consensus from multiple alignments</td>
</tr>

<tr>
<td><a href="merger.html">merger</a></td>
<td>Merge two overlapping sequences</td>
</tr>

</table>
<P>
Compare this with the program <b>merger</b> which does a more accurate
alignment of more divergent sequences using the Needle and Wunsch
algorithm but which uses much more memory.
<P>

A graphical dotplot of the matches used in this merge can be displayed
using the program <b>dotpath</b>.

<H2>
    Author(s)
</H2>
Gary Williams (gwilliam&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK



<H2>
    History
</H2>

Written Aug 2000 by Gary Williams.

<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.


<H2>
    Comments
</H2>
None

</BODY>
</HTML>

