<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBOSS: PDBPLUS documentation. </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">
<!-- #include file="header1.inc" -->
<H2> PDBPLUS documentation </H2>
<!-- #include file="header2.inc" -->
<!-- END OF HEADER -->





<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>

<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<H2> 1.0   SUMMARY  </H2>

<!-- #include file="inc/pdbplus.ione" -->



<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<H2> 2.0   INPUTS & OUTPUTS          </H2>
PDBPLUS reads a directory of domain OR protein CCF files (clean coordinate files) and writes
a directory containing the same files as the input directory but in which 
values for records for secondary structure and residue accessibility are
given.  These data are calculated by using the STRIDE and NACCESS programs
and these must be installed on the system that is running PDBPLUS (see Section 8.0
below).  PDBPLUS also requires a directory containing files in PDB format
for each coordinate file in the input directory: either the original PDB file 
(for proteins) or domain PDB files generated by using DOMAINER (for domains).  The paths 
and extensions of the pdb (input) and coordinate (input and output) files are 
specified by the user (file extensions are given in the ACD file).  The names of the 
output CCF files will be identical to those that are read.

<br><br>PDBPLUS generates 3 log files containing diagnostic messages for various problems encountered by PDBPARSE, STRIDE or NACCESS respectively (see Section 12.0 below).  


<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.
     Use "<b>PDBPLUS</b> reads any normal sequence USAs." if
     appropriate.
-->
<a name="3.0"></a>
<H2> 3.0   INPUT FILE FORMAT         </H2>
The format of the CCF file is described in the <a href="pdbparse.html">PDBPARSE documentation.</a>

No qa tests were found for pdbplus


<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<H2> 4.0   OUTPUT FILE FORMAT         </H2>
The format of the CCF file is described in the <a href="pdbparse.html">PDBPARSE documentation.</a>

No qa tests were found for pdbplus


<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml" 
-->

<a name="5.0"></a>
<H2> 5.0   DATA FILES                </H2>
PDBPLUS does not use a data file.







<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<H2> 6.0   USAGE                     </H2>

<pre>
   Standard (Mandatory) qualifiers (* if not always prompted):
  [-ccfin]             dirlist    This option specifies the location of CCF
                                  files (clean coordinate files) (input). A
                                  'clean cordinate file' contains protein
                                  coordinate and derived data for a single PDB
                                  file ('protein clean coordinate file') or a
                                  single domain from SCOP or CATH ('domain
                                  clean coordinate file'), in CCF format
                                  (EMBL-like). The files, generated by using
                                  PDBPARSE (PDB files) or DOMAINER (domains),
                                  contain 'cleaned-up' data that is
                                  self-consistent and error-corrected. Records
                                  for residue solvent accessibility and
                                  secondary structure are added to the file by
                                  using PDBPLUS.
  [-pdbin]             directory  This option specifies the location of PDB
                                  files (input). A PDB file contains protein
                                  coordinate and other data. A detailed
                                  explanation of the PDB file format is
                                  available on the PDB web site
                                  http://www.rcsb.org/pdb/info.html
   -pdbprefix          string     This option specifies the file prefix of PDB
                                  files (if any)
   -mode               menu       This option specifies the mode of operation.
*  -thresholdsize      integer    This option specifies the threshold size for
                                  SSEs (secondary structure elements)
  [-ccfout]            directory  This option specifies the location of CCF
                                  files (clean coordinate files) (output). A
                                  'clean cordinate file' contains protein
                                  coordinate and derived data for a single PDB
                                  file ('protein clean coordinate file') or a
                                  single domain from SCOP or CATH ('domain
                                  clean coordinate file'), in CCF format
                                  (EMBL-like). The files, generated by using
                                  PDBPARSE (PDB files) or DOMAINER (domains),
                                  contain 'cleaned-up' data that is
                                  self-consistent and error-corrected. Records
                                  for residue solvent accessibility and
                                  secondary structure are added to the file by
                                  using PDBPLUS.
   -errfile            outfile    This option specifies the name of the log
                                  file for PDBPLUS
*  -serrfile           outfile    This option specifies the name of the log
                                  file for STRIDE
*  -nerrfile           outfile    This option specifies the name of the log
                                  file for NACCESS

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-errfile" associated qualifiers
   -odirectory         string     Output directory

   "-serrfile" associated qualifiers
   -odirectory         string     Output directory

   "-nerrfile" associated qualifiers
   -odirectory         string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write standard output
   -filter             boolean    Read standard input, write standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report deaths


</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-ccfin]<br>(Parameter 1)</td>
<td>This option specifies the location of CCF files (clean coordinate files) (input). A 'clean cordinate file' contains protein coordinate and derived data for a single PDB file ('protein clean coordinate file') or a single domain from SCOP or CATH ('domain clean coordinate file'), in CCF format (EMBL-like). The files, generated by using PDBPARSE (PDB files) or DOMAINER (domains), contain 'cleaned-up' data that is self-consistent and error-corrected. Records for residue solvent accessibility and secondary structure are added to the file by using PDBPLUS.</td>
<td>Directory with files</td>
<td>./</td>
</tr>

<tr>
<td>[-pdbin]<br>(Parameter 2)</td>
<td>This option specifies the location of PDB files (input). A PDB file contains protein coordinate and other data. A detailed explanation of the PDB file format is available on the PDB web site http://www.rcsb.org/pdb/info.html</td>
<td>Directory</td>
<td>./</td>
</tr>

<tr>
<td>-pdbprefix</td>
<td>This option specifies the file prefix of PDB files (if any)</td>
<td>Any string is accepted</td>
<td><i>An empty string is accepted</i></td>
</tr>

<tr>
<td>-mode</td>
<td>This option specifies the mode of operation.</td>
<td><table><tr><td>1</td> <td><i>(Use STRIDE only)</i></td></tr><tr><td>2</td> <td><i>(Use NACCESS only)</i></td></tr><tr><td>3</td> <td><i>(Use STRIDE and NACCESS)</i></td></tr></table></td>
<td>1</td>
</tr>

<tr>
<td>-thresholdsize</td>
<td>This option specifies the threshold size for SSEs (secondary structure elements)</td>
<td>Any integer value</td>
<td>4</td>
</tr>

<tr>
<td>[-ccfout]<br>(Parameter 3)</td>
<td>This option specifies the location of CCF files (clean coordinate files) (output). A 'clean cordinate file' contains protein coordinate and derived data for a single PDB file ('protein clean coordinate file') or a single domain from SCOP or CATH ('domain clean coordinate file'), in CCF format (EMBL-like). The files, generated by using PDBPARSE (PDB files) or DOMAINER (domains), contain 'cleaned-up' data that is self-consistent and error-corrected. Records for residue solvent accessibility and secondary structure are added to the file by using PDBPLUS.</td>
<td>Directory</td>
<td>./</td>
</tr>

<tr>
<td>-errfile</td>
<td>This option specifies the name of the log file for PDBPLUS</td>
<td>Output file</td>
<td>pdbplus.log</td>
</tr>

<tr>
<td>-serrfile</td>
<td>This option specifies the name of the log file for STRIDE</td>
<td>Output file</td>
<td>stride.log</td>
</tr>

<tr>
<td>-nerrfile</td>
<td>This option specifies the name of the log file for NACCESS</td>
<td>Output file</td>
<td>naccess.log</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>



<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of PDBPLUS is shown below.

No qa tests were found for pdbplus

All CCF files were read from testdata/pdbplus/ccfin and CCF files
(annotated with values for secondary structure and solvent
accessibility) were written to testdata/pdbplus/ccfout.  The STRIDE
and NACCESS programs ran on PDB files within testdata/. Three log
files (pdbplus.log, naccess.log and stride.log) were written. The
threshold size for secondary structure elements was 4 residues.  File
extension of all input and output files would be set in the ACD file.



<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
NACCESS and STRIDE do not always work as advertised owing, e.g. to arbitrary array limits in their code.  If the derived data appears to be missing from the PDDBPARSE output this is probably why.





<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<H2> 8.0   NOTES                     </H2>
PDBPLUS requires a working version of NACCESS and STRIDE.
When running PDBPLUS at the RFCGR it is essential that the commands 'use naccess' or 
'use stride' (which run the scripts /packages/menu/USE/naccess and /packages/menu/USE/stride respectively) 
are given before it is run.  
<br>
<br>


<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<tr>
<td><b> Clean coordinate file (for protein) </b></td>
<td> CCF format (EMBL-like).</td>
<td> Protein coordinate and derived data for a single PDB file.  The data are 'cleaned-up': self-consistent and error-corrected.  </td>
<td> <a href="pdbparse.html">PDBPARSE</a> </td>
<td> Records for residue solvent accessibility and secondary structure are added to the file by using PDBPLUS. </td>
</tr>
<tr>
<td><b> Clean coordinate file (for domain)</b></td>
<td> CCF format (EMBL-like).</td>
<td> Protein coordinate and derived data for a single domain from SCOP or CATH. The data are 'cleaned-up': self-consistent and error-corrected. </td>
<td> <a href="domainer.html">DOMAINER</a></td>
<td> Records for residue solvent accessibility and secondary structure are added to the file by using PDBPLUS.</td>
</tr>
<tr>
<td><b>PDB file </b></td>
<td> PDB format. </td>
<td> Protein coordinate data in PDB format. </td>
<td> N.A. </td>
<td> N.A. </td>
</tr>
<tr>
<td><b>Domain PDB file </b></td>
<td> PDB format. </td>
<td> Protein coordinate data for a single domain from SCOP or CATH. </td>
<td> <a href="domainer.html">DOMAINER</a> </td>
<td> N.A. </td>
</tr>
</table>

[an error occurred while processing this directive]
[an error occurred while processing this directive]


<a name="9.0"></a>
<H2> 9.0   DESCRIPTION               </H2>
PDBPLUS is a wrapper to the STRIDE [1] and NACCESS [2] programs that adds derived data (i - iv below) to a CCF file.  The values of the derived data for a given residue will of course depend on whether they were calculated from the coordinates for the entire structure or just the domain, i.e. whether PDBPLUS is provided with protein or domain CCF files.  Thus PDBPLUS provides useful flexibility in generating the derived data.
<br>
<UL>
<LI>  Absolute (Abs.) and relative (Rel.) accessible surface area of residues according to NACCESS.  Abs. is the summed accessible surface area of the atoms, whereas Rel. is expressed as a percentage relative to the accessibility of the atoms in an extended ALA-x-ALA tri-peptide conformation.  The NACCESS authors treat alpha carbons as side-chain atoms so that glycine can have a value for side-chain accessible surface area (see 'Database format' below).  They are therefore not included in the main-chain.  
<LI>  Phi and Psi angle and solvent accessible surface area of residues as calculated by using STRIDE.  
<LI>  Secondary structure assignment according to STRIDE, one of 'H' (alpha helix), 'G' (3-10 helix), 'I' (Pi-helix), 'E' (extended conformation), 'B' or 'b' (isolated bridge), 'T' (turn) or 'C' (coil, i.e. none of the above).  
<LI>  The number of helical or beta-strand secondary structure elements (SSEs) in the chain or domain.  An 'element' is defined as a run of a user-defined number (typically 4) of residues in the 'H', 'G' or 'I' conformation (helices) or the 'E' conformation (beta-strands).  
</UL>



<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<H2> 10.0  ALGORITHM                 </H2>
Values in the CCF file for the number of helices (NH) or beta-strands (NE) in a chain and columns 12-13 and 22-34 of the coordinate line record (CO) are given null values ('.' or 0) by PDBPARSE (see Section 4.0).  PDBPLUS is used to assign values to these these records.



<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
-->
<a name="11.0"></a>
<br><H2> 11.0  RELATED APPLICATIONS      </H2>


<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="../../apps/aaindexextract.html">aaindexextract</a></td><td>Extract data from AAINDEX</td></tr>
<tr><td><a href="allversusall.html">allversusall</a></td><td>Does an all-versus-all global alignment for each set of sequences in an input directory and writes files of sequence similarity values</td></tr>
<tr><td><a href="../../apps/cathparse.html">cathparse</a></td><td>Reads raw CATH classification files and writes DCF file (domain classification file)</td></tr>
<tr><td><a href="../../apps/cutgextract.html">cutgextract</a></td><td>Extract data from CUTG</td></tr>
<tr><td><a href="domainer.html">domainer</a></td><td>Reads CCF files (clean coordinate files) for proteins and writes CCF files for domains, taken from a DCF file (domain classification file)</td></tr>
<tr><td><a href="domainnr.html">domainnr</a></td><td>Removes redundant domains from a DCF file (domain classification file). The file must contain domain sequence information, which can be added by using DOMAINSEQS</td></tr>
<tr><td><a href="domainseqs.html">domainseqs</a></td><td>Adds sequence records to a DCF file (domain classification file)</td></tr>
<tr><td><a href="../../apps/domainsse.html">domainsse</a></td><td>Adds secondary structure records to a DCF file (domain classification file)</td></tr>
<tr><td><a href="hetparse.html">hetparse</a></td><td>Converts raw dictionary of heterogen groups to a file in EMBL-like format</td></tr>
<tr><td><a href="pdbparse.html">pdbparse</a></td><td>Parses PDB files and writes CCF files (clean coordinate files) for proteins</td></tr>
<tr><td><a href="pdbtosp.html">pdbtosp</a></td><td>Convert raw swissprot:PDB equivalence file to EMBL-like format</td></tr>
<tr><td><a href="../../apps/printsextract.html">printsextract</a></td><td>Extract data from PRINTS</td></tr>
<tr><td><a href="../../apps/prosextract.html">prosextract</a></td><td>Builds the PROSITE motif database for patmatmotifs to search</td></tr>
<tr><td><a href="../../apps/rebaseextract.html">rebaseextract</a></td><td>Extract data from REBASE</td></tr>
<tr><td><a href="scopparse.html">scopparse</a></td><td>Reads raw SCOP classification files and writes a DCF file (domain classification file)</td></tr>
<tr><td><a href="seqnr.html">seqnr</a></td><td>Removes redundancy from DHF files (domain hits files) or other files of sequences</td></tr>
<tr><td><a href="sites.html">sites</a></td><td>Reads CCF files (clean coordinate files) and writes CON files (contact files) of residue-ligand contact data for domains in a DCF file (domain classification file)</td></tr>
<tr><td><a href="../../apps/ssematch.html">ssematch</a></td><td>Searches a DCF file (domain classification file) for secondary structure matches</td></tr>
<tr><td><a href="../../apps/tfextract.html">tfextract</a></td><td>Extract data from TRANSFAC</td></tr>
</table>



<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="12.0"></a>
<H2> 12.0  DIAGNOSTIC ERROR MESSAGES </H2>
The following messages (in "quotes") may appear in the PDBPARSE log file (an explanation is given in parenethesis). 

<UL>
<LI>"clean coordinate file not found my.file" (a clean coordinate input file was not found for input)
<LI>"ERROR Clean coordinate file read error: my.file" (error in reading clean coordinate file)
<LI>"no stride output for: my.file" (No STRIDE output was generated for my.file)
<LI>"stride output for: my.file" (STRIDE output *was* generated for my.file)
<LI>"no naccess output for:" (No NACCESS output was generated for my.file)
<LI>"naccess output for:" (NACCESS output *was* generated for my.file)
<LI>"PDB file not found: my.file" (a pdb file was not found for input)
<LI>"Could not write results file: my.file" (output file could not be written)
<LI>"Could not convert chain id A to chain number in pdb file my.file" (chain identifier was not found in the clean coordinate file)
<LI>"STRIDE  failures: %12\nNACCESS failures: 123" (STRIDE failed for 12 pdb files and NACCESS failed for 123 pdb files.  The clean coordinate files will lack the appropriate data (see Section 10.0).
</UL>

<!-- AUTHORS
     The main author first then all others. 
-->
<a name="13.0"></a>
<H2> 13.0  AUTHORS                   </H2>
Jon Ison <a href="mailto:jison@rfcgr.mrc.ac.uk">(jison@rfcgr.mrc.ac.uk)</a>
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK



<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<H2> 14.0  REFERENCES                </H2>
Please cite the authors and EMBOSS.


<H3>14.1 Other useful references  </H3>
<pre>
1. Frishman, D. & Argos, P. (1995).  Knowledge-Based Secondary Structure Assignment.  Proteins 23:566-579.

2. Hubbard,S.J. & Thornton, J.M. (1993), 'NACCESS', Computer Program, Department of Biochemistry and Molecular Biology, University College London.
</pre>
</BODY>
</HTML>
