<HTML>

<HEAD>
  <TITLE>
  EMBOSS: pscan
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">






<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

pscan
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>







<H2>
    Function
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->
Scans proteins using PRINTS


<H2>
    Description
</H2>


PRINTS is a database of diagnostic protein signatures, or fingerprints.

<p>
Fingerprints are groups of conserved motifs or elements that together
form a diagnostic signature for particular protein families. 

<p>
An uncharacterised sequence matching all motifs or elements can then be
readily diagnosed as a true match to a particular family fingerprint. 

<p>
They can be used to diagnose family relationships in newly-determined
sequences (especially from genome projects). 

<p>
Usually the motifs or elements do not overlap, but are separated along a
sequence, though they may be contiguous in 3D-space.  Fingerprints can
encode protein folds and functionalities more flexibly and powerfully
than can single motifs, full diagnostic potency deriving from the mutual
context provided by motif neighbours. 

<p>
Diagnostically, this is more powerful than using single motifs by virtue
of the biological context afforded by matching motif neighbours. 

<p>
pscan finds matches between a query protein sequence and the motifs or
elements in the PRINTS database.  It reports various classes of matches:

<p>
<ul>

<li>Matches where all elements of a motif exist in the correct order
<li>Matches where all elements exist but some are in the incorrect order
<li>Matches where some elements match and are in the correct order
<li>Miscellaneous matches. 

</ul>

<p>
The home web page of the PRINTS database is:
<a href="http://www.bioinf.man.ac.uk/dbbrowser/PRINTS/">http://www.bioinf.man.ac.uk/dbbrowser/PRINTS/</a>

<H2>
    Usage
</H2>





<H2>
    Command line arguments
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Sequence database USA
   -emin               integer    Minimum number of elements per fingerprint
   -emax               integer    Maximum number of elements per fingerprint
  [-outfile]           outfile    Output file name

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1             integer    Start of each sequence to be used
   -send1               integer    End of each sequence to be used
   -sreverse1           boolean    Reverse (if DNA)
   -sask1               boolean    Ask for begin/end/reverse
   -snucleotide1        boolean    Sequence is nucleotide
   -sprotein1           boolean    Sequence is protein
   -slower1             boolean    Make lower case
   -supper1             boolean    Make upper case
   -sformat1            string     Input sequence format
   -sdbname1            string     Database name
   -sid1                string     Entryname
   -ufo1                string     UFO features
   -fformat1            string     Features format
   -fopenfile1          string     Features file name

   "-outfile" associated qualifiers
   -odirectory2         string     Output directory

   General qualifiers:
   -auto                boolean    Turn off prompts
   -stdout              boolean    Write standard output
   -filter              boolean    Read standard input, write standard output
   -options             boolean    Prompt for standard and additional values
   -debug               boolean    Write debug output to program.dbg
   -verbose             boolean    Report some/full command line options
   -help                boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning             boolean    Report warnings
   -error               boolean    Report errors
   -fatal               boolean    Report fatal errors
   -die                 boolean    Report deaths


</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>Sequence database USA</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-emin</td>
<td>Minimum number of elements per fingerprint</td>
<td>Integer from 1 to 20</td>
<td>2</td>
</tr>

<tr>
<td>-emax</td>
<td>Maximum number of elements per fingerprint</td>
<td>Integer up to 20</td>
<td>20</td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;sequence&gt;</i>.pscan</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>



<H2>
    Input file format
</H2>


<b>pscan</b> reads one or more protein sequence USAs.

<p>



<H2>
    Output file format
</H2>




<p>

The program reports hits in four classes.

<dl>


	<dt> Class1:</dt><dd>Matches where all elements of a motif
	exist in the correct order

        <dt> Class2:</dt><dd>Matches where all elements exist but some
        are in the incorrect order

        <dt> Class3:</dt><dd>Matches where some elements match and are
        in the correct order

        <dt> Class4:</dt><dd>Miscellaneous matches

</dl>


<H2>
    Data files
</H2>


The data file is stored in the PRINTS directory of the standard EMBOSS
data directory.

<ul>
   <li> prints.mat       Matrices derived from PRINTS
   <li> Pxxxxx           Information for each fingerprint
</ul>

The column information is described at the top of the matrix data file

<H2>
    Notes
</H2>


The matrix information used to scan a sequence is derived from the final
motif sets in the PRINTS database. The matrices are of the simple frequency
type and contain the number of times a residue occurs in each position of
the alignment. Each matrix therefore has a highest possible score, being
the sum of the maximum score of each column. A match to the sequence
window is obtained if it has a score equal to or greater than the percentage
of the maximum score of the lowest scoring sequence in the final motif
set.
<p>

The data files must have been created before running this program.  This
is done by running the <b>printsextract</b> program with the
"prints.dat" file from a PRINTS release.  You may have to ask your
system manager to do this. 

<H2>
    References
</H2>


<ol>
<li>
Attwood, T.K., Flower, D.R., Lewis, A.P., Mabey, J.E., Morgan, S.R., 
   Scordis, P., Selley, J. and Wright, W. (1999) PRINTS prepares for the new
   millennium. Nucleic Acids Research, 27(1), 220-225. 
<li>
Attwood, T.K., Beck, M.E., Flower, D.R., Scordis, P. and Selley, J. (1998)
   The PRINTS protein fingerprint database in its fifth year. Nucleic Acids 
   Research, 26(1), 304-308. 
<li>
Attwood, T.K., Beck, M.E., Bleasby, A.J., Degtyarenko, K., Michie, A.D. and 
   Parry-Smith, D.J. (1997) Novel developments with the PRINTS protein motif
   fingerprint database. Nucleic Acids Research, 25 (1), 212-216.
<li>
Attwood, T.K. and Beck, M.E. (1994) PRINTS - A protein motif fingerprint
   database. Protein Engineering, 7(7), 841-848.
<li>
Bleasby, A.J., Akrigg, D.A. and Attwood, T.K. (1994) OWL - A non-redundant
   composite protein sequence database. Nucleic Acids Research, 22(17), 3574-77.
<li>
Bleasby, A.J. and Wootton, J.C. (1990) Constructing validated, non-
   redundant composite protein sequence databases. Protein Engineering, 3(3), 
   153-159. 
<li>
Parry-Smith, D.J. and Attwood, T.K. (1992) ADSP - A new package for
   computational sequence analysis. CABIOS, 8(5), 451-459.
<li>
Attwood, T.K. and Findlay, J.B.C. (1994) Fingerprinting G-protein-coupled
   receptors. Prot.Engng. 7(2), 195-203.
<li>
Attwood, T.K. and Findlay, J.B.C. (1993) Design of a discriminating finger-
   print for G-protein-coupled receptors. Prot.Engng. 6(2) 167-176.
<li>
 Akrigg, D., Attwood, T.K., Bleasby, A.J., Findlay, J.B.C, North, A.C.T.,
   Maughan, N.A., Parry-Smith, D.J., Perkins, D.N. and Wootton, J.C. (1992)
   SERPENT - An information storage and analysis resource for protein
   sequences. CABIOS 8(3) 295-296.
<li>
 Parry-Smith, D.J. and Attwood, T.K. (1991) SOMAP - A novel interactive
    approach to multiple protein sequence aligment. CABIOS, 7(2), 233-235.
<li>
 Perkins, D.N. and Attwood, T.K. (1995) VISTAS - A package for VIsualising
    STructures And Sequences of proteins. J.Mol.Graph., 13, 73-75. 
<li>
 Parry-Smith, D.J., Payne, A.W.R, Michie, A.D. and Attwood, T.K. (1998) 
    CINEMA - A novel Colour INteractive Editor for Multiple Alignments.
    Gene, 211(2), GC45-56. 
</ol>

<H2>
    Warnings
</H2>


The program will warn you if a DNA sequence is given.

<H2>
    Diagnostic Error Messages
</H2>


If you get the following EMBOSS FATAL ERROR message: 

<pre>

"prints.mat file not found. Create it with printsextract."

</pre>

<p>
then your local PRINTS data has not been set up correctly in your
EMBOSS DATA directory.  Use <a
href="printsextract.html">'printsextract'</a> to do this.

<H2>
    Exit status
</H2>


It exits with status 0 unless an error is reported.

<H2>
    Known bugs
</H2>


<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="antigenic.shtml">antigenic</a></td><td>Finds antigenic sites in proteins</td></tr>
<tr><td><a href="digest.shtml">digest</a></td><td>Protein proteolytic enzyme or reagent cleavage digest</td></tr>
<tr><td><a href="epestfind.shtml">epestfind</a></td><td>Finds PEST motifs as potential proteolytic cleavage sites</td></tr>
<tr><td><a href="fuzzpro.shtml">fuzzpro</a></td><td>Protein pattern search</td></tr>
<tr><td><a href="fuzztran.shtml">fuzztran</a></td><td>Protein pattern search after translation</td></tr>
<tr><td><a href="helixturnhelix.shtml">helixturnhelix</a></td><td>Report nucleic acid binding motifs</td></tr>
<tr><td><a href="oddcomp.shtml">oddcomp</a></td><td>Finds protein sequence regions with a biased composition</td></tr>
<tr><td><a href="patmatdb.shtml">patmatdb</a></td><td>Search a protein sequence with a motif</td></tr>
<tr><td><a href="patmatmotifs.shtml">patmatmotifs</a></td><td>Search a PROSITE motif database with a protein sequence</td></tr>
<tr><td><a href="pepcoil.shtml">pepcoil</a></td><td>Predicts coiled coil regions</td></tr>
<tr><td><a href="preg.shtml">preg</a></td><td>Regular expression search of a protein sequence</td></tr>
<tr><td><a href="sigcleave.shtml">sigcleave</a></td><td>Reports protein signal cleavage sites</td></tr>
</table>

<!--
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find.
   -->

<p>

<ul>

<li>
<a href="printsextract.html">printsextract</a> - Extract data from
PRINTS.  pscan uses the PRINTS data.  Until printsextract has been run
to set up the PRINTS data, pscan will not run correctly.

</ul>

<H2>
    Author(s)
</H2>


Alan Bleasby (ableasby&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK




<H2>
    History
</H2>

<!--
<p>
[an error occurred while processing this directive]
-->

<H2>
    Target users
</H2>


This program is intended to be used by everyone and everything,
from naive users to embedded scripts. 

<H2>
    Comments
</H2>


<pre>

</pre>


</BODY>
</HTML>

