<HTML>

<HEAD>
  <TITLE>
  EMBOSS: restrict
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">






<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

restrict
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>






<H2>
    Function
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->
Finds restriction enzyme cleavage sites


<H2>
    Description
</H2>


Restrict uses the REBASE database of restriction enzymes to predict
cut sites in a DNA sequence. The program allows you to select a range
of cuts, whether the DNA is circular, whether IUB ambiguity codes are
used, whether blunt or sticky ends or both are reported. You may also
force the reporting of single cleavage sites.

<p>

By default, only one of any group of isoschizomers (enzymes that have the
same recognition site and cut positions) is reported (this behaviour can
be turned off by setting the qualifier '-limit' to be false.) The reported
enzyme from any one group of isoschizomers (the prototype) is specified in
the REBASE database and the information is held in the data file 'embossre.equ'.
You may edit this file to set your own preferred prototype,if you wish.

<H2>
    Usage
</H2>


<b>Here is a sample session with restrict</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>restrict </b>
Finds restriction enzyme cleavage sites
Input sequence(s): <b>tembl:hsfau</b>
Minimum recognition site length [4]: <b></b>
Comma separated enzyme list [all]: <b></b>
Output report [hsfau.restrict]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>
This gives the lengths of the restriction fragments created by cutting with just one of each of the specified enzymes in turn. 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>restrict -solofragment </b>
Finds restriction enzyme cleavage sites
Input sequence(s): <b>tembl:hsfau</b>
Minimum recognition site length [4]: <b></b>
Comma separated enzyme list [all]: <b></b>
Output report [hsfau.restrict]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>



<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Sequence database USA
   -sitelen            integer    This sets the minimum length of the
                                  restriction enzyme recognition site. Any
                                  enzymes with sites shorter than this will be
                                  ignored.
   -enzymes            string     The name 'all' reads in all enzyme names
                                  from the REBASE database. You can specify
                                  enzymes by giving their names with commas
                                  between then, such as:
                                  'HincII,hinfI,ppiI,hindiii'.
                                  The case of the names is not important. You
                                  can specify a file of enzyme names to read
                                  in by giving the name of the file holding
                                  the enzyme names with a '@' character in
                                  front of it, for example, '@enz.list'.
                                  Blank lines and lines starting with a hash
                                  character or '!' are ignored and all other
                                  lines are concatenated together with a comma
                                  character ',' and then treated as the list
                                  of enzymes to search for.
                                  An example of a file of enzyme names is:
                                  ! my enzymes
                                  HincII, ppiII
                                  ! other enzymes
                                  hindiii
                                  HinfI
                                  PpiI
  [-outfile]           report     Output report file name

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers:
   -datafile           datafile   Alternative RE data file
   -min                integer    This sets the minimum number of cuts for any
                                  restriction enzyme that will be considered.
                                  Any enzymes that cut fewer times than this
                                  will be ignored.
   -max                integer    This sets the maximum number of cuts for any
                                  restriction enzyme that will be considered.
                                  Any enzymes that cut more times than this
                                  will be ignored.
   -solofragment       boolean    This gives the fragment lengths of the
                                  forward sense strand produced by complete
                                  restriction by each restriction enzyme on it
                                  own. Results are added to the tail section
                                  of the report.
   -single             boolean    If this is set then this forces the values
                                  of the mincuts and maxcuts qualifiers to
                                  both be 1. Any other value you may have set
                                  them to will be ignored.
   -[no]blunt          boolean    This allows those enzymes which cut at the
                                  same position on the forward and reverse
                                  strands to be considered.
   -[no]sticky         boolean    This allows those enzymes which cut at
                                  different positions on the forward and
                                  reverse strands, leaving an overhang, to be
                                  considered.
   -[no]ambiguity      boolean    This allows those enzymes which have one or
                                  more 'N' ambiguity codes in their pattern to
                                  be considered
   -plasmid            boolean    If this is set then this allows searches for
                                  restriction enzyme recognition site and cut
                                  postions that span the end of the sequence
                                  to be considered.
   -[no]commercial     boolean    If this is set, then only those enzymes with
                                  a commercial supplier will be searched for.
                                  This qualifier is ignored if you have
                                  specified an explicit list of enzymes to
                                  search for, rather than searching through
                                  'all' the enzymes in the REBASE database. It
                                  is assumed that, if you are asking for an
                                  explicit enzyme, then you probably know
                                  where to get it from and so all enzymes
                                  names that you have asked to be searched
                                  for, and which cut, will be reported whether
                                  or not they have a commercial supplier.
   -[no]limit          boolean    This limits the reporting of enzymes to just
                                  one enzyme from each group of
                                  isoschizomers. The enzyme chosen to
                                  represent an isoschizomer group is the
                                  prototype indicated in the data file
                                  'embossre.equ', which is created by the
                                  program 'rebaseextract'. If you prefer
                                  different prototypes to be used, make a copy
                                  of embossre.equ in your home directory and
                                  edit it. If this value is set to be false
                                  then all of the input enzymes will be
                                  reported. You might like to set this to
                                  false if you are supplying an explicit set
                                  of enzymes rather than searching 'all' of
                                  them.
   -alphabetic         boolean    Sort output alphabetically
   -fragments          boolean    This gives the fragment lengths of the
                                  forward sense strand produced by complete
                                  restriction using all of the input enzymes
                                  together. Results are added to the tail
                                  section of the report.
   -name               boolean    Show sequence name

   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1             integer    Start of each sequence to be used
   -send1               integer    End of each sequence to be used
   -sreverse1           boolean    Reverse (if DNA)
   -sask1               boolean    Ask for begin/end/reverse
   -snucleotide1        boolean    Sequence is nucleotide
   -sprotein1           boolean    Sequence is protein
   -slower1             boolean    Make lower case
   -supper1             boolean    Make upper case
   -sformat1            string     Input sequence format
   -sdbname1            string     Database name
   -sid1                string     Entryname
   -ufo1                string     UFO features
   -fformat1            string     Features format
   -fopenfile1          string     Features file name

   "-outfile" associated qualifiers
   -rformat2            string     Report format
   -rname2              string     Base file name
   -rextension2         string     File name extension
   -rdirectory2         string     Output directory
   -raccshow2           boolean    Show accession number in the report
   -rdesshow2           boolean    Show description in the report
   -rscoreshow2         boolean    Show the score in the report
   -rusashow2           boolean    Show the full USA in the report

   General qualifiers:
   -auto                boolean    Turn off prompts
   -stdout              boolean    Write standard output
   -filter              boolean    Read standard input, write standard output
   -options             boolean    Prompt for standard and additional values
   -debug               boolean    Write debug output to program.dbg
   -verbose             boolean    Report some/full command line options
   -help                boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning             boolean    Report warnings
   -error               boolean    Report errors
   -fatal               boolean    Report fatal errors
   -die                 boolean    Report deaths


</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>Sequence database USA</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-sitelen</td>
<td>This sets the minimum length of the restriction enzyme recognition site. Any enzymes with sites shorter than this will be ignored.</td>
<td>Integer from 2 to 20</td>
<td>4</td>
</tr>

<tr>
<td>-enzymes</td>
<td>The name 'all' reads in all enzyme names from the REBASE database. You can specify enzymes by giving their names with commas between then, such as: 'HincII,hinfI,ppiI,hindiii'.
The case of the names is not important. You can specify a file of enzyme names to read in by giving the name of the file holding the enzyme names with a '@' character in front of it, for example, '@enz.list'.
Blank lines and lines starting with a hash character or '!' are ignored and all other lines are concatenated together with a comma character ',' and then treated as the list of enzymes to search for.
An example of a file of enzyme names is:
! my enzymes
HincII, ppiII
! other enzymes
hindiii
HinfI
PpiI</td>
<td>Any string is accepted</td>
<td>all</td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>Output report file name</td>
<td>Report output file</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-datafile</td>
<td>Alternative RE data file</td>
<td>Data file</td>
<td><i>File in the data file path</i></td>
</tr>

<tr>
<td>-min</td>
<td>This sets the minimum number of cuts for any restriction enzyme that will be considered. Any enzymes that cut fewer times than this will be ignored.</td>
<td>Integer from 1 to 1000</td>
<td>1</td>
</tr>

<tr>
<td>-max</td>
<td>This sets the maximum number of cuts for any restriction enzyme that will be considered. Any enzymes that cut more times than this will be ignored.</td>
<td>Integer up to 2000000000</td>
<td>2000000000</td>
</tr>

<tr>
<td>-solofragment</td>
<td>This gives the fragment lengths of the forward sense strand produced by complete restriction by each restriction enzyme on it own. Results are added to the tail section of the report.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-single</td>
<td>If this is set then this forces the values of the mincuts and maxcuts qualifiers to both be 1. Any other value you may have set them to will be ignored.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-[no]blunt</td>
<td>This allows those enzymes which cut at the same position on the forward and reverse strands to be considered.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-[no]sticky</td>
<td>This allows those enzymes which cut at different positions on the forward and reverse strands, leaving an overhang, to be considered.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-[no]ambiguity</td>
<td>This allows those enzymes which have one or more 'N' ambiguity codes in their pattern to be considered</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-plasmid</td>
<td>If this is set then this allows searches for restriction enzyme recognition site and cut postions that span the end of the sequence to be considered.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-[no]commercial</td>
<td>If this is set, then only those enzymes with a commercial supplier will be searched for. This qualifier is ignored if you have specified an explicit list of enzymes to search for, rather than searching through 'all' the enzymes in the REBASE database. It is assumed that, if you are asking for an explicit enzyme, then you probably know where to get it from and so all enzymes names that you have asked to be searched for, and which cut, will be reported whether or not they have a commercial supplier.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-[no]limit</td>
<td>This limits the reporting of enzymes to just one enzyme from each group of isoschizomers. The enzyme chosen to represent an isoschizomer group is the prototype indicated in the data file 'embossre.equ', which is created by the program 'rebaseextract'. If you prefer different prototypes to be used, make a copy of embossre.equ in your home directory and edit it. If this value is set to be false then all of the input enzymes will be reported. You might like to set this to false if you are supplying an explicit set of enzymes rather than searching 'all' of them.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-alphabetic</td>
<td>Sort output alphabetically</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-fragments</td>
<td>This gives the fragment lengths of the forward sense strand produced by complete restriction using all of the input enzymes together. Results are added to the tail section of the report.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-name</td>
<td>Show sequence name</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

</table>



<H2>
    Input file format
</H2>


<b>restrict</b> reads one or more DNA sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:hsfau' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:hsfau</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
ID   HSFAU      standard; RNA; HUM; 518 BP.
XX
AC   X65923;
XX
SV   X65923.1
XX
DT   13-MAY-1992 (Rel. 31, Created)
DT   23-SEP-1993 (Rel. 37, Last updated, Version 10)
XX
DE   H.sapiens fau mRNA
XX
KW   fau gene.
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Primates; Catarrhini; Hominidae; Homo.
XX
RN   [1]
RP   1-518
RA   Michiels L.M.R.;
RT   ;
RL   Submitted (29-APR-1992) to the EMBL/GenBank/DDBJ databases.
RL   L.M.R. Michiels, University of Antwerp, Dept of Biochemistry,
RL   Universiteisplein 1, 2610 Wilrijk, BELGIUM
XX
RN   [2]
RP   1-518
RX   MEDLINE; 93368957.
RA   Michiels L., Van der Rauwelaert E., Van Hasselt F., Kas K., Merregaert J.;
RT   " fau cDNA encodes a ubiquitin-like-S30 fusion protein and is expressed as
RT   an antisense sequences in the Finkel-Biskis-Reilly murine sarcoma virus";
RL   Oncogene 8:2537-2546(1993).
XX
DR   SWISS-PROT; P35544; UBIM_HUMAN.
DR   SWISS-PROT; Q05472; RS30_HUMAN.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..518
FT                   /chromosome="11q"
FT                   /db_xref="taxon:9606"
FT                   /organism="Homo sapiens"
FT                   /tissue_type="placenta"
FT                   /clone_lib="cDNA"
FT                   /clone="pUIA 631"
FT                   /map="13"
FT   misc_feature    57..278
FT                   /note="ubiquitin like part"
FT   CDS             57..458
FT                   /db_xref="SWISS-PROT:P35544"
FT                   /db_xref="SWISS-PROT:Q05472"
FT                   /gene="fau"
FT                   /protein_id="CAA46716.1"
FT                   /translation="MQLFVRAQELHTFEVTGQETVAQIKAHVASLEGIAPEDQVVLLAG
FT                   APLEDEATLGQCGVEALTTLEVAGRMLGGKVHGSLARAGKVRGQTPKVAKQEKKKKKTG
FT                   RAKRRMQYNRRFVNVVPTFGKKKGPNANS"
FT   misc_feature    98..102
FT                   /note="nucleolar localization signal"
FT   misc_feature    279..458
FT                   /note="S30 part"
FT   polyA_signal    484..489
FT   polyA_site      509
XX
SQ   Sequence 518 BP; 125 A; 139 C; 148 G; 106 T; 0 other;
     ttcctctttc tcgactccat cttcgcggta gctgggaccg ccgttcagtc gccaatatgc        60
     agctctttgt ccgcgcccag gagctacaca ccttcgaggt gaccggccag gaaacggtcg       120
     cccagatcaa ggctcatgta gcctcactgg agggcattgc cccggaagat caagtcgtgc       180
     tcctggcagg cgcgcccctg gaggatgagg ccactctggg ccagtgcggg gtggaggccc       240
     tgactaccct ggaagtagca ggccgcatgc ttggaggtaa agttcatggt tccctggccc       300
     gtgctggaaa agtgagaggt cagactccta aggtggccaa acaggagaag aagaagaaga       360
     agacaggtcg ggctaagcgg cggatgcagt acaaccggcg ctttgtcaac gttgtgccca       420
     cctttggcaa gaagaagggc cccaatgcca actcttaagt cttttgtaat tctggctttc       480
     tctaataaaa aagccactta gttcagtcaa aaaaaaaa                               518
//
</pre>
</td></tr></table><p>


<H2>
    Output file format
</H2>


<p>

The output is a standard EMBOSS report file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <b>-rformat xxx</b>, where 'xxx' is replaced by
the name of the required format.  The available format names are: embl,
genbank, gff, pir, swiss, trace, listfile, dbmotif, diffseq, excel,
feattable, motif, regions, seqtable, simple, srs, table, tagseq

<p>

See:
<A href="http://www.uk.embnet.org/Software/EMBOSS/Themes/ReportFormats.html">
http://www.uk.embnet.org/Software/EMBOSS/Themes/ReportFormats.html</A>
for further information on report formats.

<p>


<p>

By default <b>restrict</b> writes a 'table' report file.

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: hsfau.restrict</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: restrict
# Rundate: Wed Jul 21 11:57:08 2004
# Report_format: table
# Report_file: hsfau.restrict
########################################

#=======================================
#
# Sequence: HSFAU     from: 1   to: 518
# HitCount: 278
#
# Minimum cuts per enzyme: 1
# Maximum cuts per enzyme: 2000000000
# Minimum length of recognition site: 4
# Blunt ends allowed
# Sticky ends allowed
# DNA is linear
# Ambiguities allowed
#
#=======================================

  Start     End Enzyme_name Restriction_site 5prime 3prime 5primerev 3primerev
      3       6 MnlI        CCTC                 13     12         .         .
      9      14 Hpy178III   TCNNGA               10     12         .         .
     11      14 EsaBC3I     TCGA                 12     12         .         .
     11      14 TaqI        TCGA                 11     13         .         .
     13      17 HinfI       GANTC                13     16         .         .
     17      21 BccI        CCATC                25     26         .         .
     17      13 MlyI        GAGTC                 7      7         .         .
     17      13 PleI        GAGTC                 7      8         .         .
     24      27 ThaI        CGCG                 25     25         .         .
     24      27 SelI        CGCG                 23     27         .         .
     24      20 MboII       GAAGA                12     11         .         .
     28      25 AciI        CCGC                 25     27         .         .
     30      33 AluI        AGCT                 31     31         .         .
     30      33 CviJI       RGCY                 31     31         .         .
     31      41 Bce4I       GCNNNNNNNGC          37     34         .         .
     31      41 HpyF10VI    GCNNNNNNNGC          38     33         .         .
     34      39 NlaIV       GGNNCC               36     36         .         .
     34      38 BsmFI       GGGAC                48     52         .         .
     35      39 Sau96I      GGNCC                35     38         .         .
     35      39 FmuI        GGNCC                38     35         .         .
     35      39 Psp03I      GGWCC                38     35         .         .
     35      39 AvaII       GGWCC                35     38         .         .
     35      39 UnbI        GGNCC                34     39         .         .
     35      39 VpaK11AI    GGWCC                34     39         .         .
     36      31 BseYI       CCCAGC               31     35         .         .
     38      41 AciI        CCGC                 38     40         .         .
     44      40 BceAI       ACGGC                25     27         .         .


<font color=red>  [Part of this file has been deleted for brevity]</font>

    385     388 CviRI       TGCA                386    386         .         .
    387     383 BscAI       GCATC               376    378         .         .
    387     383 SfaNI       GCATC               373    377         .         .
    388     393 TatI        WGTACW              388    392         .         .
    389     392 RsaI        GTAC                390    390         .         .
    389     392 Csp6I       GTAC                389    391         .         .
    394     399 BsrFI       RCCGGY              394    398         .         .
    395     398 MspI        CCGG                395    397         .         .
    397     402 HaeII       RGCGCY              401    397         .         .
    397     402 Bme142I     RGCGCY              399    399         .         .
    398     401 HhaI        GCGC                400    398         .         .
    398     401 Hin6I       GCGC                398    400         .         .
    405     410 HincII      GTYRAC              407    407         .         .
    405     410 Hpy8I       GTNNAC              407    407         .         .
    408     413 AclI        AACGTT              409    411         .         .
    409     412 MaeII       ACGT                409    411         .         .
    409     412 TaiI        ACGT                412    408         .         .
    414     419 Bsp1286I    GDGCHC              418    414         .         .
    414     419 Bme1580I    GKGCMC              418    414         .         .
    417     427 BslI        CCNNNNNNNGG         423    420         .         .
    431     435 MboII       GAAGA               443    442         .         .
    436     442 EcoO109I    RGGNCCY             437    440         .         .
    436     442 Kaz48kI     RGGNCCY             440    437         .         .
    437     442 ApaI        GGGCCC              441    437         .         .
    437     442 NlaIV       GGNNCC              439    439         .         .
    437     441 Sau96I      GGNCC               437    440         .         .
    437     442 BanII       GRGCYC              441    437         .         .
    437     442 Bme1580I    GKGCMC              441    437         .         .
    437     442 Bsp120I     GGGCCC              437    441         .         .
    437     443 EcoO109I    RGGNCCY             438    441         .         .
    437     442 Bsp1286I    GDGCHC              441    437         .         .
    437     441 FmuI        GGNCC               440    437         .         .
    437     441 UnbI        GGNCC               436    441         .         .
    437     443 Kaz48kI     RGGNCCY             441    438         .         .
    438     442 Sau96I      GGNCC               438    441         .         .
    438     441 BanAI       GGCC                439    439         .         .
    438     443 NlaIV       GGNNCC              440    440         .         .
    438     441 CviJI       RGCY                439    439         .         .
    438     442 UnbI        GGNCC               437    442         .         .
    438     442 FmuI        GGNCC               441    438         .         .
    438     434 HpyAV       CCTTC               428    427         .         .
    454     459 AflII       CTTAAG              454    458         .         .
    454     459 SmlI        CTYRAG              454    458         .         .
    455     458 MseI        TTAA                455    457         .         .
    468     471 Tsp509I     AATT                467    471         .         .
    474     477 CviJI       RGCY                475    475         .         .
    492     495 CviJI       RGCY                493    493         .         .
    497     501 DdeI        CTNAG               497    500         .         .

#---------------------------------------
#---------------------------------------
</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: hsfau.restrict</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
########################################
# Program: restrict
# Rundate: Wed Jul 21 11:57:09 2004
# Report_format: table
# Report_file: hsfau.restrict
########################################

#=======================================
#
# Sequence: HSFAU     from: 1   to: 518
# HitCount: 278
#
# Minimum cuts per enzyme: 1
# Maximum cuts per enzyme: 2000000000
# Minimum length of recognition site: 4
# Blunt ends allowed
# Sticky ends allowed
# DNA is linear
# Ambiguities allowed
#
#=======================================

  Start     End Enzyme_name Restriction_site 5prime 3prime 5primerev 3primerev
      3       6 MnlI        CCTC                 13     12         .         .
      9      14 Hpy178III   TCNNGA               10     12         .         .
     11      14 EsaBC3I     TCGA                 12     12         .         .
     11      14 TaqI        TCGA                 11     13         .         .
     13      17 HinfI       GANTC                13     16         .         .
     17      21 BccI        CCATC                25     26         .         .
     17      13 MlyI        GAGTC                 7      7         .         .
     17      13 PleI        GAGTC                 7      8         .         .
     24      27 ThaI        CGCG                 25     25         .         .
     24      27 SelI        CGCG                 23     27         .         .
     24      20 MboII       GAAGA                12     11         .         .
     28      25 AciI        CCGC                 25     27         .         .
     30      33 AluI        AGCT                 31     31         .         .
     30      33 CviJI       RGCY                 31     31         .         .
     31      41 Bce4I       GCNNNNNNNGC          37     34         .         .
     31      41 HpyF10VI    GCNNNNNNNGC          38     33         .         .
     34      39 NlaIV       GGNNCC               36     36         .         .
     34      38 BsmFI       GGGAC                48     52         .         .
     35      39 Sau96I      GGNCC                35     38         .         .
     35      39 FmuI        GGNCC                38     35         .         .
     35      39 Psp03I      GGWCC                38     35         .         .
     35      39 AvaII       GGWCC                35     38         .         .
     35      39 UnbI        GGNCC                34     39         .         .
     35      39 VpaK11AI    GGWCC                34     39         .         .
     36      31 BseYI       CCCAGC               31     35         .         .
     38      41 AciI        CCGC                 38     40         .         .
     44      40 BceAI       ACGGC                25     27         .         .


<font color=red>  [Part of this file has been deleted for brevity]</font>

# 
# TaqI:
# [TCGA]
# 	11	83	424
# 
# TatI:
# [WGTACW]
# 	130	388
# 
# TauI:
# [GCSGC]
# 	115	138	265
# 
# ThaI:
# [CGCG]
# 	25	48	119	326
# 
# TseI:
# [GCWGC]
# 	59	459
# 
# Tsp45I:
# [GTSAC]
# 	98	420
# 
# Tsp509I:
# [AATT]
# 	51	467
# 
# TspDTI:
# [ATGAA]
# 	245	273
# 
# TspRI:
# [CASTGNN]
# 	77	151	290
# 
# Tth111II:
# [CAARCA]
# 	96	164	258
# 
# UnbI:
# [GGNCC]
# 	1	18	34	60	81	141
# 	183
# 
# VpaK11AI:
# [GGWCC]
# 	34	484
#
#---------------------------------------
</pre>
</td></tr></table><p>


<p>

The output from restrict is a simple text one. The base number,
restriction enzyme name, recognition site and cut positions are
shown. Note that cuts are always to the right of the residue
shown and that 5' cuts are referred to by their associated 3'
number sequence.

<p>

The program reports enzymes that cut at two or four sites. The
program also reports isoschizomers and enzymes having the same
recognition sequence but different cut sites.


<p>

When the "-fragments" or "-solofragments" qualifiers are given then the sizes of
the fragments produced by either all of the specified enzymes cutting, or
by each enzyme cutting individually, are given in the 'tail' section
at the end of the report file.

<H2>
    Data files
</H2>

This uses the EMBOSS REBASE data files in 'data/REBASE/*' under the
EMBOSS installation directory.

<p>
These files must first be set up using the program <a
href="rebaseextract.html">'rebaseextract'.</a> Running
'rebaseextract' may be the job of your system manager.  

<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>



The data files are stored in the REBASE directory of the standard EMBOSS
data directory. The names are:

<ul>
   <li> embossre.enz     Cleavage information
   <li> embossre.ref     Reference/methylation information
   <li> embossre.sup     Supplier information
</ul>

The column information is described at the top of the data files

<p>

The reported
enzyme from any one group of isoschizomers (the prototype) is specified in
the REBASE database and the information is held in the data file 'embossre.equ'.
You may edit this file to set your own preferred prototype, if you wish.

<p>

The format of the file "embossre.equ" is 
<br>
Enzyme-name Prototype-name

<p>

i.e. two columns of enzyme names separated by a space. The first name of the
pair of enzymes is the name that is not preferred and the second is the
preferred, prototype name.


<H2>
    Notes
</H2>


Output file size is related to the size of the recognition site and the
maximum number of allowed cutting positions. Setting the site length
to six and restricting the cuts to two is a common choice of parameters.
The size of the output can sometimes be reduced by specifying
the -noambiguity switch.
<p>

The data files must have been created before running this program. This is
done by running the <b>rebaseextract</b> program with the "withrefm" and "prot" 
files from an REBASE release. You may have to ask your system manager to do this.

<H2>
    References
</H2>


<ol>

<li>
Nucleic Acids Research 27: 312-313 (1999).

</ol>

<H2>
    Warnings
</H2>


The program will warn you if a protein sequence is given.

<H2>
    Diagnostic Error Messages
</H2>

None.


<H2>
    Exit status
</H2>


It exits with status 0 unless an error is reported.

<H2>
    Known bugs
</H2>


None.

<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="recoder.shtml">recoder</a></td><td>Remove restriction sites but maintain the same translation</td></tr>
<tr><td><a href="redata.shtml">redata</a></td><td>Search REBASE for enzyme name, references, suppliers etc</td></tr>
<tr><td><a href="remap.shtml">remap</a></td><td>Display a sequence with restriction cut sites, translation etc</td></tr>
<tr><td><a href="restover.shtml">restover</a></td><td>Finds restriction enzymes that produce a specific overhang</td></tr>
<tr><td><a href="showseq.shtml">showseq</a></td><td>Display a sequence with features, translation etc</td></tr>
<tr><td><a href="silent.shtml">silent</a></td><td>Silent mutation restriction enzyme scan</td></tr>
</table>

<!--
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find.
   -->




<H2>
    Author(s)
</H2>


Alan Bleasby (ableasby&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK



<H2>
    History
</H2>


Completed 16th April 1999

<!--
<p>
[an error occurred while processing this directive]
-->

<H2>
    Target users
</H2>


This program is intended to be used by everyone and everything,
from naive users to embedded scripts. 

<H2>
    Comments
</H2>




</BODY>
</HTML>

