<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBOSS: ROCON documentation. </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">
<H2> ROCON documentation </H2>
<!-- END OF HEADER -->





<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<br><H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>

<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<br><br><br><H2> 1.0   SUMMARY  </H2>
Reads a DHF file (domain hits file) of hits (sequences of unknown structural
classification) and a domain families file (validation sequences of known 
classification) and writes a "hits file" for the hits, which are classified 
and rank-ordered on the basis of score.






<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<br><br><br><H2> 2.0   INPUTS & OUTPUTS          </H2>
ROCON reads a DHF file (domain hits file) of hits generated for a single node from a classification hierarchy, e.g. SCOP family.  These sequences are putatively related to the node in question but are, in fact, of unknown classification.  ROCON also reads a domain families file (in DHF format), containing "validation" sequences (of known classification).  These sequences are used to classify the input hits. A "hits file" (suitable for input into the ROCPLOT application) is written, which contains the input hits, classified and rank-ordered on the basis of score.


<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.

     Cannot use the test data files because they might be empty or need 
     hand-editing 

     Use "<b>ROCON</b> reads any normal sequence USAs." if
     appropriate.
-->
<a name="3.0"></a>
<br><br><br><H2> 3.0   INPUT FILE FORMAT         </H2>
The format of the DHF is described in <a href="seqsearch.html">SEQSEARCH documentation.</a> See also the example of the DHF file for hit sequences (Figure 1) and validation sequences (Figure 2) below.

<br><br><b>Figure 1   Excerpt of DHF file (hit sequences)  </b>
<table><td bgcolor="#CFCCFF">
<pre>
> Q9YBD5^.^11^105^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^SPARSE^61.50^0.000e+00^4.000e-10
VRKIRSGVVIDHIPPGRAFTMLKALGLLPPRGYRWRIAVVINAESSKLGRKDILKIEGYKPRQRDLEVLGIIAPGATFNVIEDYKVVEKVKLKLP
> Q9YBD5^.^95^135^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^SPARSE^11.50^0.000e+00^4.000e-5
VRKIRSGVVIDHIPPGRAFTMLKALGLLPPRGYRWRIAVVINAESSKLGRKDILKIEGYKPRQRDLEVLGIIAPGATFNVIEDYKVVEKVKLKLP
> Q9YBD5^.^181^235^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^SPARSE^161.50^0.000e+00^4.000e-5
VRKIRSGVVIDHIPPGRAFTMLKALGLLPPRGYRWRIAVVINAESSKLGRKDILKIEGYKPRQRDLEVLGIIAPGATFNVIEDYKVVEKVKLKLP
> O26938^.^11^101^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^81.90^0.000e+00^3.000e-16
VKPIKNGTVIDHITANRSLNVLNILGLPDGRSKVTVAMNMDSSQLGSKDIVKIENRELKPSEVDQIALIAPRATINIVRDYKIVEKAKVRL
> Q8Z130^.^8^99^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^181.00^0.000e+00^0.000e+00
VEAIKCGTVIDHIPAQVGFKLLSLFKLTETDQRITIGLNLPSGEMGRKDLIKIENTFLTDEQVNQLALYAPQATVNRIDNYDVVGKSRPSLP
> Q7MX57^.^8^99^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^80.80^0.000e+00^7.000e-16
VAAIRNGIVIDHIPPTKLFKVATLLQLDDLDKRITIGNNLRSRSHGSKGVIKIEDKTFEEEELNRIALIAPNVRLNIIRDYEVVEKRQVEVP
> Q8TVB1^.^7^98^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^72.70^0.000e+00^2.000e-13
VKRIEMGTVLDHLPPGTAPQIMRILDIDPTETTLLVAINVESSKMGRKDILKIEGKILSEEEANKVALVAPNATVNIVRDYSVAEKFQVKPP
> P96175^.^8^99^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^107.00^0.000e+00^7.000e-24
VEAICNGYVIDHIPSGQGVKILRLFSLTDTKQRVTVGFNLPSHDGTTKDLIKVENTEITKSQANQLALLAPNATVNIIENFKVTDKHSLALP
</pre>
</table>

<br><b>Figure 1   Excerpt of domain families file (validation sequences) </b>
<table><td bgcolor="#CFCCFF">
<pre>
> Q9YBD5^.^11^105^.^1^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^61.50^0.000e+00^4.000e-10
VRKIRSGVVIDHIPPGRAFTMLKALGLLPPRGYRWRIAVVINAESSKLGRKDILKIEGYKPRQRDLEVLGIIAPGATFNVIEDYKVVEKVKLKLP
> Q9UX07^.^12^104^.^1^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^65.80^0.000e+00^2.000e-11
VSKIRNGTVIDHIPAGRALAVLRILGIRGSEGYRVALVMNVESKKIGRKDIVKIEDRVIDEKEASLITLIAPSATINIIRDYVVTEKRHLEVP
> Q9KP65^.^9^100^.^1^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^128.00^0.000e+00^3.000e-30
VEAIKNGTVIDHIPAKVGIKVLKLFDMHNSAQRVTIGLNLPSSALGSKDLLKIENVFISEAQANKLALYAPHATVNQIENYEVVKKLALQLP
> Q9K1K9^.^8^99^.^1^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^101.00^0.000e+00^5.000e-22
VEAIEKGTVIDHIPAGRGLTILRQFKLLHYGNAVTVGFNLPSKTQGSKDIIKIKGVCLDDKAADRLALFAPEAVVNTIDNFKVVQKRHLNLP
> Q9JWY6^.^8^99^.^1^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^98.90^0.000e+00^2.000e-21
VEAIEKGTVIDHIPAGRGLTILRQFKLLHYGNAVTVGFNLPSKTQGSKDIIKIKGVCLDDKAADRLALFAPEAVVNTIDHFKVVQKRHLNLP
> Q9HKM3^.^7^99^.^1^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^79.60^0.000e+00^2.000e-15
ISKIRDGTVIDHVPSGKGIRVIGVLGVHEDVNYTVSLAIHVPSNKMGFKDVIKIENRFLDRNELDMISLIAPNATISIIKNYEISEKFQVELP
> Q9HHN3^.^9^101^.^1^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^78.50^0.000e+00^4.000e-15
VSKIQAGTVIDHIPAGQALQVLQILGTNGASDDQITVGMNVTSERHHRKDIVKIEGRELSQDEVDVLSLIAPDATINIVRDYEVDEKRRVDRP
> Q97FS4^.^4^93^.^2^Class 1^.^.^Fold 1^Superfamily 1^Family 2^PSIBLAST^49.20^0.000e+00^2.000e-06
INSIKNGIVIDHIKAGHGIKIYNYLKLGEAEFPTALIMNAISKKNKAKDIIKIENVMDLDLAVLGFLDPNITVNIIEDEKIRQKIQLKLP
> Q97B28^.^8^100^.^2^Class 1^.^.^Fold 1^Superfamily 1^Family 2^PSIBLAST^79.20^0.000e+00^2.000e-15
ISKIKDGTVIDHIPSGKALRVLSILGIRDDVDYTVSVGMHVPSSKMEYKDVIKIENRSLDKNELDMISLTAPNATISIIKNYEISEKFKVELP
> Q970X3^.^11^101^.^2^Class 1^.^.^Fold 1^Superfamily 1^Family 2^PSIBLAST^78.50^0.000e+00^3.000e-15
VSKIKNGTVIDHIPAGRALAVLRILKIAEGYRIALVMNVESKKMGKKDIVKIENKEVDEKEANLITLIAPTATINIIRDYEVVEKKKLKIP
> Q8ZTG2^.^7^99^.^2^Class 1^.^.^Fold 1^Superfamily 1^Family 2^PSIBLAST^66.10^0.000e+00^2.000e-11
VSKIENGTVIDHIPAGRALTVLRILGISGKEGLRVALVMNVESKKLGKKDIVKIEGRELTPEEVNIISAVAPTATINIIRNFAVVKKFKVTPP
> Q8ZB38^.^9^100^.^2^Class 1^.^.^Fold 1^Superfamily 1^Family 2^PSIBLAST^156.00^0.000e+00^1.000e-38
VEAIKCGTVIDHIPAQIGFKLLSLFKLTATDQRITIGLNLPSKRSGRKDLIKIENTFLTEQQANQLAMYAPDATVNRIDNYEVVKKLTLSLP
> Q8Z130^.^8^99^.^2^Class 1^.^.^Fold 1^Superfamily 1^Family 2^PSIBLAST^181.00^0.000e+00^0.000e+00
VEAIKCGTVIDHIPAQVGFKLLSLFKLTETDQRITIGLNLPSGEMGRKDLIKIENTFLTDEQVNQLALYAPQATVNRIDNYDVVGKSRPSLP
> Q8U374^.^6^99^.^2^Class 1^.^.^Fold 1^Superfamily 1^Family 2^PSIBLAST^92.00^0.000e+00^3.000e-19
VSAIKEGTVIDHIPAGKGLKVIQILGLGELKNGGAVLLAMNVPSKKLGRKDIVKVEGKFLSEEEVNKIALVAPTATVNIIREYKVVEKFKVEIP
> Q8TVB1^.^7^98^.^3^Class 1^.^.^Fold 1^Superfamily 2^Family 1^PSIBLAST^72.70^0.000e+00^2.000e-13
VKRIEMGTVLDHLPPGTAPQIMRILDIDPTETTLLVAINVESSKMGRKDILKIEGKILSEEEANKVALVAPNATVNIVRDYSVAEKFQVKPP
> Q8THL3^.^9^100^.^3^Class 1^.^.^Fold 1^Superfamily 2^Family 1^PSIBLAST^69.20^0.000e+00^2.000e-12
IQAIENGTVIDHITAGQALNVLRILRISSAFRATVSFVMNAPGARGKKDVVKIEGKELSVEELNRIALISPKATINIIRDFEVVQKNKVVLP
> Q8PXK6^.^9^100^.^3^Class 1^.^.^Fold 1^Superfamily 2^Family 1^PSIBLAST^62.70^0.000e+00^2.000e-10
VQAIESGTVIDHIKSGQALNVLRILGISSAFRATISFVMNAPGAGGKKDVVKIEGKELSVEELNRIALISPKATINIIRDFVVVQKNNVVLP
> Q8K9H8^.^8^99^.^3^Class 1^.^.^Fold 1^Superfamily 2^Family 1^PSIBLAST^146.00^0.000e+00^1.000e-35
VEAIKSGSVIDHIPAHIGFKLLSLFRFTETEKRITIGLNLPSQKLDKKDIIKIENTFLSDDQINQLAIYAPCATVNYIEKYNLVGKIFPSLP
> Q8DCF7^.^9^100^.^3^Class 1^.^.^Fold 1^Superfamily 2^Family 1^PSIBLAST^127.00^0.000e+00^9.000e-30
VEAIKNGTVIDHIPAQVGIKVLKLFDMHNSSQRVTIGLNLPSSALGNKDLLKIENVFINEEQASKLALYAPHATVNQIEDYQVVKKLALELP
> Q8D1W6^.^9^100^.^4^Class 1^.^.^Fold 1^Superfamily 2^Family 2^PSIBLAST^123.00^0.000e+00^1.000e-28
VEAIFGGTVIDHIPAQVGLKLLSLFKWLHTKERITMGLNLPSNQQKKKDLIKLENVLLNEDQANQLSIYAPLATVNQIKNYIVIKKQKLKLP
> Q8A9S4^.^10^101^.^4^Class 1^.^.^Fold 1^Superfamily 2^Family 2^PSIBLAST^63.80^0.000e+00^9.000e-11
VAALKNGTVIDHIPSEKLFTVVQLLGVEQMKCNITIGFNLDSKKLGKKGIIKIADKFFCDEEINRISVVAPYVKLNIIRDYEVVEKKEVRMP
> Q891I9^.^4^94^.^4^Class 1^.^.^Fold 1^Superfamily 2^Family 2^PSIBLAST^52.30^0.000e+00^2.000e-07
ITSIKDGIVIDHIKSGYGIKIFNYLNLKNVEYSVALIMNVFSSKLGKKDIIKIANKEIDIDFTVLGLIDPTITINIIEDEKIKEKLNLELP
> Q87LF7^.^9^100^.^4^Class 1^.^.^Fold 1^Superfamily 2^Family 2^PSIBLAST^130.00^0.000e+00^7.000e-31
VEAIKNGTVIDHIPAQIGIKVLKLFDMHNSSQRVTIGLNLPSSALGHKDLLKIENVFINEEQASKLALYAPHATVNQIENYEVVKKLALELP
> Q83IL8^.^8^99^.^4^Class 1^.^.^Fold 1^Superfamily 2^Family 2^PSIBLAST^189.00^0.000e+00^0.000e+00
VEAIKRGTVIDHIPAQIGFKLLSLFKLTETDQRITIGLNLPSGEMGRKDLIKIENTFLSEEQVDQLALYAPQATVNRIDNYEVVGKSRPSLP
> Q7P144^.^7^98^.^4^Class 1^.^.^Fold 1^Superfamily 2^Family 2^PSIBLAST^128.00^0.000e+00^3.000e-30
VEALKQGTVIDHIPAGEGVKILRLFKLTETGERVTVGLNLVSRHMGSKDLIKVENVALTEEQANELALFAPKATVNVIDNFEVVKKHKLTLP
> Q7MZ14^.^9^100^.^4^Class 1^.^.^Fold 1^Superfamily 2^Family 2^PSIBLAST^150.00^0.000e+00^6.000e-37
VEAIRCGTVIDHIPAQVGFKLLSLFKLTETDQRITIGLNLPSNRLGKKDLIKIENTFLTEQQANQLAMYAPNATVNCIENYEVVKKLPINLP
> Q7MX57^.^8^99^.^5^Class 1^.^.^Fold 2^Superfamily 1^Family 1^PSIBLAST^80.80^0.000e+00^7.000e-16
VAAIRNGIVIDHIPPTKLFKVATLLQLDDLDKRITIGNNLRSRSHGSKGVIKIEDKTFEEEELNRIALIAPNVRLNIIRDYEVVEKRQVEVP
> Q7MHF0^.^9^100^.^5^Class 1^.^.^Fold 2^Superfamily 1^Family 1^PSIBLAST^127.00^0.000e+00^8.000e-30
VEAIKNGTVIDHIPAQVGIKVLKLFDMHNSSQRVTIGLNLPSSALGNKDLLKIENVFINEEQASKLALYAPHATVNQIEDYQVVKKLALELP
> Q58801^.^9^99^.^5^Class 1^.^.^Fold 2^Superfamily 1^Family 1^PSIBLAST^61.50^0.000e+00^5.000e-10
VKKITNGTVIDHIDAGKALMVFKVLNVPKETSVMIAINVPSKKKGKKDILKIEGIELKKEDVDKISLISPDVTINIIRNGKVVEKLKPQIP
> P96175^.^8^99^.^5^Class 1^.^.^Fold 2^Superfamily 1^Family 1^PSIBLAST^107.00^0.000e+00^7.000e-24
VEAICNGYVIDHIPSGQGVKILRLFSLTDTKQRVTVGFNLPSHDGTTKDLIKVENTEITKSQANQLALLAPNATVNIIENFKVTDKHSLALP
> P96111^.^375^472^.^5^Class 1^.^.^Fold 2^Superfamily 1^Family 1^PSIBLAST^47.30^0.000e+00^9.000e-06
GIKPIENGTVIDHIAKGKTPEEIYSTILKIRKILRLYDVDSADGIFRSSDGSFKGYISLPDRYLSKKEIKKLSAISPNTTVNIIKNSTVVEKYRIKLP
> P77919^.^6^99^.^6^Class 1^.^.^Fold 2^Superfamily 1^Family 2^PSIBLAST^93.50^0.000e+00^1.000e-19
VSAIKEGTVIDHIPAGKGLKVIEILKLGKLTNGGAVLLAMNVPSKKLGRKDIVKVEGRFLSEEEVNKIALVAPNATVNIIRDYKVVEKFKVEVP
> P74766^.^12^104^.^6^Class 1^.^.^Fold 2^Superfamily 1^Family 2^PSIBLAST^74.20^0.000e+00^7.000e-14
VSKIKNGTVIDHIPAGRAFAVLNVLGIKGHEGFRIALVINVDSKKMGKKDIVKIEDKEISDTEANLITLIAPTATINIVREYEVVKKTKLEVP
> P57451^.^8^99^.^7^Class 1^.^.^Fold 2^Superfamily 2^Family 1^PSIBLAST^143.00^0.000e+00^1.000e-34
VEAIKSGSVIDHIPEYIGFKLLSLFRFTETEKRITIGLNLPSKKLGRKDIIKIENTFLSDEQINQLAIYAPHATVNYINEYNLVRKVFPTLP
> P19936^.^8^99^.^7^Class 1^.^.^Fold 2^Superfamily 2^Family 1^PSIBLAST^159.00^0.000e+00^1.000e-39
VEAIKCGTVIDHIPAQIGFKLLTLFKLTATDQRITIGLNLPSNELGRKDLIKIENTFLTEQQANQLAMYAPKATVNRIDNYEVVRKLTLSLP
> P08421^.^8^99^.^7^Class 1^.^.^Fold 2^Superfamily 2^Family 1^PSIBLAST^183.00^0.000e+00^0.000e+00
VEAIKCGTVIDHIPAQVGFKLLSLFKLTETDQRITIGLNLPSGEMGRKDLIKIENTFLTEEQVNQLALYAPQATVNRIDNYDVVGKSRPSLP
> P00478^.^8^99^.^8^Class 1^.^.^Fold 2^Superfamily 2^Family 2^PSIBLAST^191.00^0.000e+00^0.000e+00
VEAIKRGTVIDHIPAQIGFKLLSLFKLTETDQRITIGLNLPSGEMGRKDLIKIENTFLSEDQVDQLALYAPQATVNRIDNYEVVGKSRPSLP
> O58452^.^6^99^.^8^Class 1^.^.^Fold 2^Superfamily 2^Family 2^PSIBLAST^94.30^0.000e+00^6.000e-20
VSAIKEGTVIDHIPAGKGLKVIEILGLSKLSNGGSVLLAMNVPSKKLGRKDIVKVEGKFLSEEEVNKIALVAPTATVNIIRNYKVVEKFKVEVP
> O30129^.^6^98^.^9^Class 2^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^79.60^0.000e+00^2.000e-15
VSKIKEGTVIDHINAGKALLVLKILKIQPGTDLTVSMAMNVPSSKMGKKDIVKVEGMFIRDEELNKIALISPNATINLIRDYEIERKFKVSPP
> O26938^.^11^101^.^54894^Class 1^.^.^Fold 1^Superfamily 1^Family 1^PSIBLAST^81.90^0.000e+00^3.000e-16
VKPIKNGTVIDHITANRSLNVLNILGLPDGRSKVTVAMNMDSSQLGSKDIVKIENRELKPSEVDQIALIAPRATINIVRDYKIVEKAKVRL
</pre>
</table>

<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<br><br><br><H2> 4.0   OUTPUT FILE FORMAT         </H2>
The format of the hits file is described in <a href="rocplot.html">ROCPLOT documentation.</a> See also Figure 3.

<br><br><b>Figure 3   Excerpt from a "hits file" (output) </b>
<table><td bgcolor="#CFCCFF">
<pre>
> RELATED 8 ; ROC 2
CROSS        Q8Z130    8     99    
UNKNOWN      Q9YBD5    181   235   
FALSE        P96175    8     99    
TRUE         O26938    11    101   
FALSE        Q7MX57    8     99    
CROSS        Q8TVB1    7     98    
TRUE         Q9YBD5    11    105   
TRUE         Q9YBD5    95    135   
</pre>
</table>




<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml" 
-->

<a name="5.0"></a>
<br><br><br><H2> 5.0   DATA FILES                </H2>
None.


<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<br><br><br><H2> 6.0   USAGE                     </H2>
<H3> 6.1   COMMAND LINE ARGUMENTS </H3>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td>[-hitsin]<br>(Parameter 1)</td>
<td>This option specifies the location of the DHF file (domain hits file) (input). A 'domain hits file' contains database hits (sequences) with domain classification information, in the DHF format (FASTA or EMBL-like). The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH.</td>
<td>Input file</td>
<td><b>Required</b></td>
</tr>
 
<tr>
<td>[-validin]<br>(Parameter 2)</td>
<td>This option specifies the name of domain families file (input). A 'domain families file' contains sequence relatives (hits) for each of a number of different SCOP or CATH families found from searching a sequence database, e.g. by using SEQSEARCH (psiblast). The file contains the collated search results for the indvidual families; only those hits of unambiguous family assignment are included. Hits of ambiguous family assignment are assigned as relatives to a SCOP or CATH superfamily or fold instead and are collated into a 'domain ambiguities file'. The domain families and ambiguities files are generated by using SEQSORT and use the same format as a DHF file (domain hits file).</td>
<td>Input file</td>
<td><b>Required</b></td>
</tr>
 
<tr>
<td>-thresh</td>
<td>This option specifies the overlap threshold for hits. This is the minimum length (residues) of overlap required for two hits with the accession number to be counted as the same hit. The accession number of the hit, and the start and end point respectively of the hit relative to full length sequence are provided in the lists of hits in the DHF input file. The overlap is determined from the start and end points of the hit. For example two hits with the start and end points of 1-100 and 91-190 respectively are considered to be the same hit if they have the same accession numbers and the overlap threshold is 10 or less.</td>
<td>Any integer value</td>
<td>10</td>
</tr>
 
<tr>
<td>-mode</td>
<td>This option specifies the classification scheme to use. See ROCON on-line documentation for more information.</td>
<td><table><tr><td>1</td> <td><i>(Family classification scheme)</i></td></tr><tr><td>2</td> <td><i>((Not yet available))</i></td></tr></table></td>
<td>1</td>
</tr>
 
<tr>
<td>[-hitsout]<br>(Parameter 3)</td>
<td>This option specifies the name of the hits files (output). A 'hits file'contains a list of hits (e.g. from a prediction method) that are classified and rank-ordered on the basis of score, p-value, E-value etc. The files generated by using SIGSCAN and LIBSCAN will contain the results of a search of a discriminating element (e.g. hidden Markov model, profile or signature) against a sequence database. The ROCPLOT application is run on the files to perform Receiver Operator Characteristic (ROC) analysis on the hits.</td>
<td>Output file</td>
<td>hits.dhf</td>
</tr>
 
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td colspan=4>(none)</td>
</tr>
 
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td colspan=4>(none)</td>
</tr>
 
</table>

<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of ROCON is shown below.
<table><td bgcolor="#FFCCFF">
<pre>
Unix % rocon
Reads a DHF file (domain hits file) of hits (sequences of unknown
structural classification) and a DHF file of validation sequences (known
classification) and writes a 'hits file' for the hits, which are classified and
rank-ordered on the basis of score.
Name of DHF file (domain hits file) (input).: test_data/rocon.hits
Name of domain families file (input).: test_data/rocon.valid
Overlap threshold for hits. [10]: 10
Classification scheme to use
         1 : Family classification scheme
         2 : (Not yet available)
Select number. [1]: 1
Directory of hits files (output). [hits.dhf]: rocon.dhf
Unix %
</pre>
</table>


<br> 
<br>The following command line would achieve the same result.
<br>
<table><td bgcolor="#FFCCFF">
<pre> 
rocon test_data/rocon.hits test_data/rocon.valid rocon.dhf -thresh 10 -mode 1
</pre> 
</table>

<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<br><br><br><H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
None.





<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<br><br><br><H2> 8.0   NOTES                     </H2>
None.


<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<tr>
<td><b>Domain hits file </b></td>
<td> DHF format (FASTA-like). </td>
<td> Database hits (sequences) with domain classification information.  The hits are relatives to a SCOP or CATH family (or other node in the structural hierarchies) and are found from a search of a sequence database. </td>
<td> <a href="seqsearch.html">SEQSEARCH</a> (hits retrieved by PSIBLAST)</td>
<td> N.A. </td>
</tr>
<tr>
<td><b> Hits file</b></td>
<td> Text file of classified hits </td>
<td> A list of hits (e.g. from a prediction method) that are classified and rank-ordered on the basis of score, p-value, E-value etc.  </td>
<td> <a href="sigscan.html">SIGSCAN</a> and <a href="libscan.html">LIBSCAN</a> (hits from searches of a discriminating element (hidden Markov model, profile or signature) against a sequence database).   </td>
<td> ROCPLOT is run on the files to perform Receiver Operator Characteristic (ROC) analysis on the hits.  </td>
</tr>
<tr>
<td><b> Domain families & ambiguities file </b></td>
<td> DHF format (FASTA-like). </td>
<td> Contains sequence relatives (hits) for each of a number of different SCOP or CATH families found from PSIBLAST searches of a sequence database.  The file contains the collated search results for the indvidual families; only those hits of unambiguous family assignment are included.  Hits of ambiguous family assignment are assigned as relatives to a SCOP or CATH superfamily or fold instead and are collated into a 'domain ambiguities file'. </td>
<td> <a href="seqsort.html">SEQSORT</a> </td>
<td> N.A. </td>
</tr>
</table>





<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="9.0"></a>
<br><br><br><H2> 9.0   DESCRIPTION               </H2>
Discrciminating elements such as hidden Markov models (HMM), sparse protein signatures and profiles can be generated for a set of proteins with related sequence, structural or functional properties.  These discriminators are characteristic of the property considered and can be used diagnostically, for instance, by screening a database of uncharacterised sequences.  

<br><br>Such screens can be performed by using the LIBSCAN and SIGSCAN applications, which generate a DHF file (domain hits file) of database hits (sequences).  The hits are relatives to a SCOP or CATH family (or other node in the structural hierarchies) and are found from a search of a sequence database.  The DHF file includes domain classification information of the family in question.

<br><br>When assessing the performance of a predictive method, a "gold standard" of truth is required.  This is a set of examples that are known to be related to the discriminating element, and, ideally, a further set that is known to be definitely not related.  For example, to assess a protein family HMM to detect true members of that family requires, at least, a list of the known family members.  If a method works well for the "gold standard" we can infer it will work well generally. Increasingly, use is made of databases such as SCOP, in which sequence, structural and functional relationships are classified. 
<br><br>
Such a "gold standard" can be generated for SCOP families by using the DOMAINATRIX package and particularly the SEQSORT application.  SEQSORT generates a "domain families file" containing sequence relatives (hits) for each of a number of different SCOP or CATH families found from PSIBLAST searches of a sequence database.  The file contains the collated search results for the indvidual families; only those hits of unambiguous family assignment are included.  
<br><br>A powerful measure of diagnostic performance is to use Receiver Operator Characteristic (ROC) curves to display graphically the sensitivity and specificity of a method.  ROC analysis is implemented in the ROCPLOT application. ROCPLOT requires a "hits file" containing a list of classified hits that are rank-ordered on the basis of score.

<br><br>The ROCON application was developed to take as input a DHF file of hits, and a domain families file of validation sequences (the gold standard) and generate a hits file for use with ROCPLOT.



<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<br><br><br><H2> 10.0  ALGORITHM                 </H2>
The domain families file uses the same format as a DHF file (domain hits file). Thus ROCON takes two DHF files as input, one containing hits (sequences of unknown classification) to be classified and the the domain families file containing sequences (of known classification) that are used to make the classification.  A DHF file includes 6 tokens (in <b>bold</b> in the example below) that collectively describe the classification of a sequence as follows: domain <b>class</b> (SCOP and CATH domains), domain <b>architecture</b> (CATH only), domain <b>topology</b> (CATH only), domain <b>fold</b> (SCOP domains only), domain <b>superfamily</b> and domain <b>family</b> (SCOP only) - see below.

<br><br>
<table><td bgcolor="#FFCCFF">
<pre>
> Q9WVI4^.^513^667^.^55074^<b>CLASS</b>^<b>ARCHITECTURE</b>^<b>TOPOLOGY</b>^<b>FOLD</b>^<b>SUPERFAMILY</b>^<b>FAMILY</b>^PSIBLAST^113.00^0.000e+00^2.000e-25
RKFDDVTMLFSDIVGFTAICAQCTPMQVISMLNELYTRFDHQCGFLDIYKVETIGDAYCVASGLHRKSLCHAKPIALMALKMMELSEEVLTPDGRPIQMRIGIHSG
</pre>
</table>


<br><br>
The value of the tokens (<b>CLASS</b> through to <b>FAMILY</b>, but, in the current implementation of classification scheme, excluding <b>ARCHITECTURE</b> and <b>TOPOLOGY</b>) determines the classification of a hit that is given in the ROCON output file as follows.  
<br><br>If a hit does not overlap significantly with any validation sequence then the hit is classified as <b>UNKNOWN</b>.  A hit and validation sequence are defined as overlapping if they have identical accesssion number and have a common region of at least a user-defined number of residues.  The overlap is determined from the start and end points (relative to the full-length sequences) of the hit and validation sequences.  For example a hit and validation sequence with the same accession numbers and with the start and end points of 1-100 and 91 - 190 respectively are defined as overlapping if the overlap threshold is 10 or less.  
<br><br>If a hit does overlap significantly with a validation sequence it is defined as one of <b>TRUE</b>, <b>CROSS</b> or <b>FALSE</b> depending on the value of the tokens (<b>CLASS</b> through to <b>FAMILY</b>) as per the table below.

<br><table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#FFCCFF" >
<tr>
<td><b>CLASS</b></td>
<td><b>FOLD</b></td>
<td><b>SUPERFAMILY</b></td>
<td><b>FAMILY</b></td>
<td><b>CLASSIFICATION</b></td>
</tr>
<tr>
<td>Not available</td>
<td>Not available</td>
<td>Not available</td>
<td>Not available</td>
<td><b>UNKNOWN</b></td>
</tr>
<tr>
<td>Different</td>
<td>Different</td>
<td>Different</td>
<td>Different</td>
<td><b>FALSE</b></td>
</tr>
<tr>
<td>Same</td>
<td>Different</td>
<td>Different</td>
<td>Different</td>
<td><b>FALSE</b></td>
</tr>
<tr>
<td>Same</td>
<td>Same</td>
<td>Different</td>
<td>Different</td>
<td><b>CROSS</b></td>
</tr>
<tr>
<td>Same</td>
<td>Same</td>
<td>Same</td>
<td>Different</td>
<td><b>CROSS</b></td>
</tr>
<tr>
<td>Same</td>
<td>Same</td>
<td>Same</td>
<td>Same</td>
<td><b>TRUE</b></td>
</tr>
</table>



<br><br>Putting this in context of a real example, imagine an input DHF file containing hits derived from searching a sequence database with a novel type of profile specific to a SCOP family.  In this case, the full SCOP classification (Class, Fold etc) of the hit are putatively assigned. To validate the novel method a validation file of manually curated sequences of known classification are used.  A <b>TRUE</b> hit would be one that overlaps with a validation sequence belonging to the same Family (and by implication Superfamily, Fold and Class) to the hit.  A <b>CROSS</b> hit overlaps with a sequence of the same fold,  but different family, as a validation sequence, and a <b>FALSE</b> hit overlaps with a sequence of a different fold to the hit.


<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
     (Take this from "Sister applications" in the old documentation)
-->
<a name="11.0"></a>
<br><br><br><H2> 11.0  RELATED APPLICATIONS      </H2>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>APPLICATION</b></td>
<td><b>DESCRIPTION</b></td>
</tr>
<tr>
<td><b><a href="seqsearch.html">SEQSEARCH</a> </b></td>
<td> Generate DHF files (domain hits files) of database hits (sequences) from a DAF file (domain alignment file) or other sequences by using PSI-BLAST.</td>
</tr>
<tr>
<td><b><a href="seqwords.html">SEQWORDS</a> </b></td>
<td>Generates DHF files (domain hits files) of database hits (sequences) for nodes in a DCF file (domain classification file) by keyword search of UniProt.</td>
</tr>
<tr>
<td><b><a href="seqsort.html">SEQSORT</a> </b></td>
<td>  Reads DHF files (domain hits files) of database hits (sequences) and removes hits of ambiguous classification. </td>
</tr>
<tr>
<td><b><a href="libgen.html">LIBGEN</a> </b></td>
<td>Generates various type of discriminator for each alignment in a
      directory.  </td>
</tr>
<tr>
<td><b><a href="libscan.html">LIBSCAN</a> </b></td>
<td> Generates hits (sequences) from searches of various types of
      discriminator (HMMs, profiles etc) against a sequence database.  
      Or generates hits from screening sequences against a library of 
      such discriminators. </td>
</tr>
<tr>
<td><b><a href="siggen.html">SIGGEN</a> </b></td>
<td>Generates a sparse protein signature from an alignment and residue
     contact data.  </td>
</tr>
<tr>
<td><b><a href="sigscan.html">SIGSCAN</a> </b></td>
<td>  Generates hits (sequences) from scanning a signature against a
     sequence database. </td>
</tr>
<tr>
<td><b><a href=".html">ROCPLOT</a> </b></td>
<td> Provides interpretation and graphical display of the 
                  performance of discriminating elements (e.g. profiles for
                  protein families).  rocplot reads file(s) of hits from 
	          discriminator-database search(es), performs ROC analysis 
	          on the hits, and writes graphs illustrating the diagnostic 
	          performance of the discriminating elements. </td>
</tr>
</table>





<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="12.0"></a>
<br><br><br><H2> 12.0  DIAGNOSTIC ERROR MESSAGES </H2>
None.






<!-- AUTHORS -->
<a name="13.0"><br><br><br><H2> 13.0  AUTHORS                   </H2></a>
Jon Ison <a href="mailto:jison@rfcgr.mrc.ac.uk">(jison@rfcgr.mrc.ac.uk)</a>
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK




<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  REFERENCES                </H2>
Please cite the authors and EMBOSS.
<br><br>
<i>Rice P, Bleasby A.J.  2000.  EMBOSS - The European Molecular Biology Open Software Suite.  Trends in Genetics, 15:276-278.  See also http://www.uk.embnet.org/Software/EMBOSS</i>

<H3>14.1 Other useful references  </H3>

<br>

</BODY>
</HTML>
