<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBOSS: SEQALIGN documentation. </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">
[an error occurred while processing this directive]
<H2> SEQALIGN documentation </H2>
[an error occurred while processing this directive]





<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<br><H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>

<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<br><br><br><H2> 1.0   SUMMARY  </H2>

[an error occurred while processing this directive]

<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<br><br><br><H2> 2.0   INPUTS & OUTPUTS          </H2>
SEQALIGN reads a directory of "seed" alignments and a directory of sets of protein sequences and writes a directory of "extended" alignments.  An extended alignment is a seed alignment that is extended with sequences from the appropriate sequence set.  The sets directory must contain a sequence set for each seed alignment directory and the files must have the same base name.  For example, sequences from "family.sets" are used to extend the seed alignment "family.aln". 
<br>Typically, the inputs are (i) DAF files (domain alignment files) containing structure-based sequence alignments and (ii) DHF files (domain hits files) containing sequence relatives (of unknown structure) of these alignments, and the output are DAF files containing the extended alignments. 
<br>The base names of the output alignments are the same as the input alignments.  The paths for input and output files are specified by the user and the file extensions are set in the ACD file.




<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.

     Cannot use the test data files because they might be empty or need 
     hand-editing 

     Use "<b>SEQALIGN</b> reads any normal sequence USAs." if
     appropriate.
-->
<a name="3.0"></a>
<br><br><br><H2> 3.0   INPUT FILE FORMAT         </H2>
The format of the domain alignment file is described in <a href="domainalign.html">DOMAINALIGN documentation</a>. 
<br>If other alignment files are used as input, all of the common file formats are supported.

<br><br>The format of the domain hits file is described in <a href="seqsearch.html">SEQSEARCH documentation</a>. 
<br>If other sequence sets are used as input, all of the common file formats are supported.




<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<br><br><br><H2> 4.0   OUTPUT FILE FORMAT         </H2>
The format of the domain alignment file is described in <a href="domainalign.html">DOMAINALIGN documentation</a>. 

Note however that the file will never contain 'Number' or 'Post_similar' records. Futhermore, the code before each sequence is of the form <b>Acc_Start_End </b>(where domain hits files were used as input sequences) or <b>Acc_n</b> (where other sequences were used as input), where 'Acc' is the accession number of the sequence, 'Start' and 'End' are the start and end point of the sequence in the full length sequence respectively, and 'n' is a sequential count of the sequence in the alignment.


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: 54894.daf</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
# TY   SCOP
# XX
# CL   Alpha and beta proteins (a+b)
# XX
# FO   Ferredoxin-like
# XX
# SF   Aspartate carbamoyltransferase, Regulatory-chain, N-terminal domain
# XX
# FA   Aspartate carbamoyltransferase, Regulatory-chain, N-terminal domain
# XX
# SI   54894
# XX
Q9YBD5_11_105             0 VRKIRSGVVIDHIPPGRAFTMLKALGLLPPRGYRWRIAVVINAESSKLGRKDILKIEGYK 0
Q9YBD4_11_105             0 VRKIRSGVVIDHIPPGRAFTMLKALGLLPPRGYRWRIAVVINAESSKLGRKDILKIEGYK 0
Q9UX07_12_104             0 VSKIRNGTVIDHIPAGRALAVLRILGIRGSEGY--RVALVMNVESKKIGRKDIVKIEDRV 0
Q9KP65_9_100              0 VEAIKNGTVIDHIPAKVGIKVLKLFDMHN---SAQRVTIGLNLPSSALGSKDLLKIENVF 0

Q9YBD5_11_105             0 PRQRDLEVLGIIAPGATFNVIEDYKVVEKVKLKLP 0
Q9YBD4_11_105             0 PRQRDLEVLGIIAPGATFNVIEDYKVVEKVKLKLP 0
Q9UX07_12_104             0 IDEKEASLITLIAPSATINIIRDYVVTEKRHLEVP 0
Q9KP65_9_100              0 ISEAQANKLALYAPHATVNQIENYEVVKKLALQLP 0

</pre>
</td></tr></table><p>
<p><h3>File: 55074.daf</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
# TY   SCOP
# XX
# CL   Alpha and beta proteins (a+b)
# XX
# FO   Ferredoxin-like
# XX
# SF   Adenylyl and guanylyl cyclase catalytic domain
# XX
# FA   Adenylyl and guanylyl cyclase catalytic domain
# XX
# SI   55074
# XX
Q9WVI4_513_667            0 --------RKFDDVTMLFSDIVGFTAICAQCTP--MQVISMLNELYTRFDHQCG---FLD 0
Q9WVI5_513_667            0 --------RKFDDVTMLFSDIVGFTAICAQCTP--MQVISMLNELYTRFDHQCG---FLD 0
Q9ERL9_475_626            0 --------KKFSEVTMLFSDIVGFTAICSQCSP--LQVITMLNALYTRFDQQCG---ELD 0
Q9DGG6_369_560            0 IIFRPFKMQQIEQVSILFADIVGFTKMSANKSA--HALVGLLNDLFGRFDRLCE---DTK 0
Q9DGG6_1032_1200          0 ------YSKNHDSGGVIFASIVNFSEFYEENYEGGKECYRVLNELIGDFDELLSKPHYSS 0

Q9WVI4_513_667            0 IYKVETIGDAYCVASGLHRKSL-------CHAKPIALMALKMMELSEEVLTPDGRP-IQM 0
Q9WVI5_513_667            0 IYKVETIGDAYCVASGLHRKSL-------CHAKPIALMALKMMELSEEVLTPDGRP-IQM 0
Q9ERL9_475_626            0 VYKVETIGDAYCVAGGLHRESD-------THAVQIALMALKMMELSNEVMSPHGEP-IKM 0
Q9DGG6_369_560            0 CEKISTLGDCYYCVAGCPEPRA-------DHAYCCIEMGLGMIKAIEQFCQEKKEM-VNM 0
Q9DGG6_1032_1200          0 IEKIKTIGATYMAASGLNTSQCQDSNHPHGHLQTLFEFAKEMMRVVDDFNNNMLWFNFKL 0

Q9WVI4_513_667            0 RIGIHSGSVLAGVVGVRMPRYCLFGNNVTLASKFESGSHPRRINISPTTYQLLKRE---- 0
Q9WVI5_513_667            0 RIGIHSGSVLAGVVGVRMPRYCLFGNNVTLASKFESGSHPRRINISPTTYQLLKRE---- 0
Q9ERL9_475_626            0 RIGLHSGSVFAGVVGVKMPRYCLFGNNVTLANKFESCSVPRKINVSPTTYRLL------- 0
Q9DGG6_369_560            0 RVGVHTGTVLCGILGMRRFKFDVWSNDVNLANLMEQLGVAGKVHISEATAKYLDDRYEME 0
Q9DGG6_1032_1200          0 RIGFNHGPLTAGVIGTTKLLYDIWGDTVNIASRMDTIGVECRIQVSEETYRILSK----- 0

Q9WVI4_513_667            0 ------------------------- 0
Q9WVI5_513_667            0 ------------------------- 0
Q9ERL9_475_626            0 ------------------------- 0
Q9DGG6_369_560            0 DGKVTERVGQSAVADQLKGLKTYLI 0
Q9DGG6_1032_1200          0 ------------------------- 0

</pre>
</td></tr></table><p>
<p><h3>File: seqalign.log</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
//
/ebi/services/idata/pmr/hgmp/test/data/structure/54894.dhf
//
/ebi/services/idata/pmr/hgmp/test/data/structure/55074.dhf
</pre>
</td></tr></table><p>


<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml" 
-->

<a name="5.0"></a>
<br><br><br><H2> 5.0   DATA FILES                </H2>
SEQALIGN does not use a data file.







<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<br><br><br><H2> 6.0   USAGE                     </H2>
<H3> 6.1   COMMAND LINE ARGUMENTS </H3>
 
<pre>
   Standard (Mandatory) qualifiers (* if not always prompted):
   -mode               menu       This option specifies the mode of SEQALIGN
                                  operation. SEQALIGN takes as input a
                                  directory of either i. single sequences, ii.
                                  set of sequences (unaligned or aligned, but
                                  typically aligned sequences within a domain
                                  alignment file)). The user has to specify
                                  which.
*  -dhfinpath          dirlist    This option specifies the location of
                                  sequences, e.g. DHF files (domain hits
                                  files) (input). SEQALIGN takes as input a
                                  database of either i. single sequences, ii.
                                  sets of unaligned sequences or iii. sets of
                                  aligned sequences, e.g. a domain alignment
                                  file. A 'domain alignment file' contains a
                                  sequence alignment of domains belonging to
                                  the same SCOP or CATH family. The file is in
                                  clustal format annotated with domain family
                                  classification information. The files
                                  generated by using SCOPALIGN will contain a
                                  structure-based sequence alignment of
                                  domains of known structure only. Such
                                  alignments can be extended with sequence
                                  relatives (of unknown structure) by using
                                  SEQALIGN.
*  -dafinpath          dirlist    This option specifies the location of
                                  sequences, e.g. DAF files (domain alignment
                                  files) (input). SEQALIGN takes as input a
                                  database of either i. single sequences, ii.
                                  sets of unaligned sequences or iii. sets of
                                  aligned sequences, e.g. a domain alignment
                                  file. A 'domain alignment file' contains a
                                  sequence alignment of domains belonging to
                                  the same SCOP or CATH family. The file is in
                                  clustal format annotated with domain family
                                  classification information. The files
                                  generated by using SCOPALIGN will contain a
                                  structure-based sequence alignment of
                                  domains of known structure only. Such
                                  alignments can be extended with sequence
                                  relatives (of unknown structure) by using
                                  SEQALIGN.
  [-dhfindir]          directory  This option specifies the location of DHF
                                  files (domain hits files) (input). A 'domain
                                  hits file' contains database hits
                                  (sequences) with domain classification
                                  information, in the DHF format (FASTA or
                                  EMBL-like). The hits are relatives to a SCOP
                                  or CATH family and are found from a search
                                  of a sequence database. Files containing
                                  hits retrieved by PSIBLAST are generated by
                                  using SEQSEARCH.
   -amode              menu       This option specifies which alignment
                                  algorithm to use.
  [-dafoutdir]         outdir     This option specifies the location of DAF
                                  files (domain alignment files) (output). A
                                  'domain alignment file' contains a sequence
                                  alignment of domains belonging to the same
                                  SCOP or CATH family. The file is in clustal
                                  format annotated with domain family
                                  classification information. The files
                                  generated by using SCOPALIGN will contain a
                                  structure-based sequence alignment of
                                  domains of known structure only. Such
                                  alignments can be extended with sequence
                                  relatives (of unknown structure) by using
                                  SEQALIGN.
   -logfile            outfile    This option specifies the name of log file
                                  for the build. The log file contains
                                  messages about any errors arising while
                                  SEQALIGN ran.

   Additional (Optional) qualifiers:
   -forcetype          boolean    This option specifies whether to force
                                  minimal domain classification data to be
                                  written to the output file in cases where
                                  singlet sequences were given as input file
                                  and no classification data was available

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-logfile" associated qualifiers
   -odirectory         string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write standard output
   -filter             boolean    Read standard input, write standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report deaths


</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-mode</td>
<td>This option specifies the mode of SEQALIGN operation. SEQALIGN takes as input a directory of either i. single sequences, ii. set of sequences (unaligned or aligned, but typically aligned sequences within a domain alignment file)). The user has to specify which.</td>
<td><table><tr><td>1</td> <td><i>(Single sequences)</i></td></tr><tr><td>2</td> <td><i>(Multiple sequences (e.g. sequence set or alignment))</i></td></tr></table></td>
<td>1</td>
</tr>

<tr>
<td>-dhfinpath</td>
<td>This option specifies the location of sequences, e.g. DHF files (domain hits files) (input). SEQALIGN takes as input a database of either i. single sequences, ii. sets of unaligned sequences or iii. sets of aligned sequences, e.g. a domain alignment file. A 'domain alignment file' contains a sequence alignment of domains belonging to the same SCOP or CATH family. The file is in clustal format annotated with domain family classification information. The files generated by using SCOPALIGN will contain a structure-based sequence alignment of domains of known structure only. Such alignments can be extended with sequence relatives (of unknown structure) by using SEQALIGN.</td>
<td>Directory with files</td>
<td>./</td>
</tr>

<tr>
<td>-dafinpath</td>
<td>This option specifies the location of sequences, e.g. DAF files (domain alignment files) (input). SEQALIGN takes as input a database of either i. single sequences, ii. sets of unaligned sequences or iii. sets of aligned sequences, e.g. a domain alignment file. A 'domain alignment file' contains a sequence alignment of domains belonging to the same SCOP or CATH family. The file is in clustal format annotated with domain family classification information. The files generated by using SCOPALIGN will contain a structure-based sequence alignment of domains of known structure only. Such alignments can be extended with sequence relatives (of unknown structure) by using SEQALIGN.</td>
<td>Directory with files</td>
<td>./</td>
</tr>

<tr>
<td>[-dhfindir]<br>(Parameter 1)</td>
<td>This option specifies the location of DHF files (domain hits files) (input). A 'domain hits file' contains database hits (sequences) with domain classification information, in the DHF format (FASTA or EMBL-like). The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH.</td>
<td>Directory</td>
<td>./</td>
</tr>

<tr>
<td>-amode</td>
<td>This option specifies which alignment algorithm to use.</td>
<td><table><tr><td>1</td> <td><i>(CLUSTALW)</i></td></tr><tr><td>2</td> <td><i>(TCOFFEE (not yet implemented))</i></td></tr></table></td>
<td>1</td>
</tr>

<tr>
<td>[-dafoutdir]<br>(Parameter 2)</td>
<td>This option specifies the location of DAF files (domain alignment files) (output). A 'domain alignment file' contains a sequence alignment of domains belonging to the same SCOP or CATH family. The file is in clustal format annotated with domain family classification information. The files generated by using SCOPALIGN will contain a structure-based sequence alignment of domains of known structure only. Such alignments can be extended with sequence relatives (of unknown structure) by using SEQALIGN.</td>
<td>Output directory</td>
<td>./</td>
</tr>

<tr>
<td>-logfile</td>
<td>This option specifies the name of log file for the build. The log file contains messages about any errors arising while SEQALIGN ran.</td>
<td>Output file</td>
<td>seqalign.log</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-forcetype</td>
<td>This option specifies whether to force minimal domain classification data to be written to the output file in cases where singlet sequences were given as input file and no classification data was available</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>


<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of SEQALIGN is shown below.

<b>Here is a sample session with seqalign</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>seqalign </b>
Extend alignments (DAF file) with sequences (DHF file).
Input mode
         1 : Single sequences
         2 : Multiple sequences (e.g. sequence set or alignment)
Select mode of operation. [1]: <b>1</b>
Location of sequences, e.g. DHF files (domain hits files) (input). [./]: <b>structure</b>
Location of DHF files (domain hits files) (input). [./]: <b>seqalign</b>
Alignment algorithm option
         1 : CLUSTALW
         2 : TCOFFEE (not yet implemented)
Select number. [1]: <b></b>
Location of DAF files (domain alignment files) (output). [./]: <b></b>
Name of log file for the build. [seqalign.log]: <b></b>




 CLUSTAL W (1.83) Multiple Sequence Alignments



Sequence format is Pearson
Sequence 1: Q9YBD5_11_105       95 aa
Sequence 2: Q9YBD4_11_105       95 aa
Sequence 3: Q9UX07_12_104       93 aa
Sequence 4: Q9KP65_9_100        92 aa
Start of Pairwise alignments
Aligning...
Sequences (1:2) Aligned. Score:  100
Sequences (1:3) Aligned. Score:  51
Sequences (1:4) Aligned. Score:  41
Sequences (2:3) Aligned. Score:  51
Sequences (2:4) Aligned. Score:  41
Sequences (3:4) Aligned. Score:  43
Guide tree        file created:   [./seqalign-1119283021.25929.dnd]
Start of Multiple Alignment
There are 3 groups
Aligning...
Group 1: Sequences:   2      Score:1553
Group 2: Sequences:   3      Score:739
Group 3: Sequences:   4      Score:655
Alignment Score 1720
CLUSTAL-Alignment file created  [./seqalign-1119283021.25929.out]



 CLUSTAL W (1.83) Multiple Sequence Alignments



Sequence format is Pearson
Sequence 1: Q9WVI4_513_667        155 aa
Sequence 2: Q9WVI5_513_667        155 aa
Sequence 3: Q9ERL9_475_626        152 aa
Sequence 4: Q9DGG6_1032_1200      169 aa
Sequence 5: Q9DGG6_369_560        192 aa
Start of Pairwise alignments
Aligning...
Sequences (1:2) Aligned. Score:  100
Sequences (1:3) Aligned. Score:  78
Sequences (1:4) Aligned. Score:  24
Sequences (1:5) Aligned. Score:  30
Sequences (2:3) Aligned. Score:  78
Sequences (2:4) Aligned. Score:  24
Sequences (2:5) Aligned. Score:  30
Sequences (3:4) Aligned. Score:  25
Sequences (3:5) Aligned. Score:  31
Sequences (4:5) Aligned. Score:  23
Guide tree        file created:   [./seqalign-1119283021.25929.dnd]
Start of Multiple Alignment
There are 4 groups
Aligning...
Group 1: Sequences:   2      Score:2567
Group 2: Sequences:   3      Score:2490
Group 3: Sequences:   4      Score:1061
Group 4:                     Delayed
Sequence:4     Score:926
Alignment Score 4025
CLUSTAL-Alignment file created  [./seqalign-1119283021.25929.out]
Processing /ebi/services/idata/pmr/hgmp/test/data/structure/54894.dhf

clustalw -infile=./seqalign-1119283021.25929.seqs -align -MATRIX=BLOSUM -GAPOPEN=10 -GAPEXT=0.5 -outfile=./seqalign-1119283021.25929.out

Processing /ebi/services/idata/pmr/hgmp/test/data/structure/55074.dhf

clustalw -infile=./seqalign-1119283021.25929.seqs -align -MATRIX=BLOSUM -GAPOPEN=10 -GAPEXT=0.5 -outfile=./seqalign-1119283021.25929.out


</pre></td></tr></table><p>
<p>
<a href="#output.1">Go to the output files for this example</a><p><p>


SEQALIGN read all domain alignment files with the file extension
.salign (specified in ACD) from the directory /test_data, and extended
these alignments using the sequences from domain hits files with file
extension .hits (specified in ACD) in the directory /test_data.
Extended domain alignment files (file extension .ealign specified in
ACD) were written to /test_data/seqalign.


<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<br><br><br><H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
None.





<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<br><br><br><H2> 8.0   NOTES                     </H2>
<b>  1. Use of CLUSTAL</b><br>
clustalw must be installed on the system that is running SEQSEARCH.

<br><br>When running SEQSEARCH at the HGMP it is essential that the command 'use clustal' (which runs the script /packages/menu/USE/clustal) is given before it is run. 


<br><br><b>  2. Empty domain hits file (input)</b>
<br>If just a single or no sequences are given in the domain hits file then no extended alignment will be generated.


<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<tr>
<td><b>Domain hits file </b></td>
<td> DHF format (FASTA-like format with domain classification information). </td>
<td> Database hits (sequences) with domain classification information.  The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. </td>
<td> <a href="seqsearch.html">SEQSEARCH</a> (hits retrieved by PSIBLAST)</td>
<td> N.A. </td>
</tr>
<tr>
<td><b>Domain alignment file </b></td>
<td> DAF format (CLUSTAL-like format with domain classification information). </td>
<td> Contains a sequence alignment of domains belonging to the same SCOP or CATH family.  The file is annotated with domain family classification information.</td>
<td> <a href="domainalign.html">DOMAINALIGN</a> (structure-based sequence alignment of domains of known structure). </td>
<td> DOMAINALIGN alignments can be extended with sequence relatives (of unknown structure) to the family in question by using SEQALIGN. </td>
</tr>
</table>






None


<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="9.0"></a>
<br><br><br><H2> 9.0   DESCRIPTION               </H2>
The generation of sequence alignments for large datasets such as SCOP or CATH
potentially requires a lot of time for preparation of datasets, running 
jobs and so on, in addition to the compute time required for the 
alignments themselves. 
SEQALIGN will generate an extended alignment for each seed alignment in a directory for which a file of sequences is available, by using CLUSTALW.  
Sequences are added one by one to the appropriate seed alignment.




<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<br><br><br><H2> 10.0  ALGORITHM                 </H2>
None.





<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
     (Take this from "Sister applications" in the old documentation)
-->
<a name="11.0"></a>
<br><br><br><H2> 11.0  RELATED APPLICATIONS      </H2>

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="../structure/contacts.html">contacts</a></td><td>Generate intra-chain CON files from CCF files</td></tr>
<tr><td><a href="domainalign.html">domainalign</a></td><td>Generate alignments (DAF file) for nodes in a DCF file</td></tr>
<tr><td><a href="domainrep.html">domainrep</a></td><td>Reorder DCF file to identify representative structures</td></tr>
<tr><td><a href="../domainatrix/domainreso.html">domainreso</a></td><td>Remove low resolution domains from a DCF file</td></tr>
<tr><td><a href="../structure/interface.html">interface</a></td><td>Generate inter-chain CON files from CCF files</td></tr>
<tr><td><a href="../signature/libgen.html">libgen</a></td><td>Generate discriminating elements from alignments</td></tr>
<tr><td><a href="../signature/matgen3d.html">matgen3d</a></td><td>Generate a 3D-1D scoring matrix from CCF files</td></tr>
<tr><td><a href="../../apps/psiphi.html">psiphi</a></td><td>Phi and psi torsion angles from protein coordinates</td></tr>
<tr><td><a href="../signature/rocon.html">rocon</a></td><td>Generates a hits file from comparing two DHF files</td></tr>
<tr><td><a href="../signature/rocplot.html">rocplot</a></td><td>Performs ROC analysis on hits files</td></tr>
<tr><td><a href="../domsearch/seqfraggle.html">seqfraggle</a></td><td>Removes fragment sequences from DHF files</td></tr>
<tr><td><a href="../domsearch/seqsearch.html">seqsearch</a></td><td>Generate PSI-BLAST hits (DHF file) from a DAF file</td></tr>
<tr><td><a href="../domsearch/seqsort.html">seqsort</a></td><td>Remove ambiguous classified sequences from DHF files</td></tr>
<tr><td><a href="../domsearch/seqwords.html">seqwords</a></td><td>Generates DHF files from keyword search of UniProt</td></tr>
<tr><td><a href="../signature/siggen.html">siggen</a></td><td>Generates a sparse protein signature from an alignment</td></tr>
<tr><td><a href="../signature/siggenlig.html">siggenlig</a></td><td>Generate ligand-binding signatures from a CON file</td></tr>
<tr><td><a href="../signature/sigscan.html">sigscan</a></td><td>Generate hits (DHF file) from a signature search</td></tr>
<tr><td><a href="../signature/sigscanlig.html">sigscanlig</a></td><td>Search ligand-signature library & write hits (LHF file)</td></tr>
</table>


<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="12.0"></a>
<br><br><br><H2> 12.0  DIAGNOSTIC ERROR MESSAGES </H2>
The log file might contain the following messages:
<br><br><i>Could not open seed sequence file my.file </i> (file read error on input sequence alignment or sequence file)
<br><i>Empty alignment in my.file </i> (input alignment was empty.)
<br><i>Empty singlet sequence for my.file </i> (singlet sequence file was empty.)
<br><i>Could not open domain hits file my.file</i>  (file read error on input sequence file.)
<br><i>Not enough sequences found in domain hits file </i> (insufficient sequences (i.e. < 2) in input sequence file.)


<!-- AUTHORS -->
<a name="13.0"><br><br><br><H2> 13.0  AUTHORS                   </H2></a>
Ranjeeva Ranasinghe <a href="mailto:rranasin@rfcgr.mrc.ac.uk">(rranasin@rfcgr.mrc.ac.uk)</a>
<br><br>
Jon Ison <a href="mailto:jison@rfcgr.mrc.ac.uk">(jison@rfcgr.mrc.ac.uk)</a>
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK




<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  REFERENCES                </H2>

Please cite the authors and EMBOSS.
<br><br>
<i>Rice P, Longden I and Bleasby A (2000) "EMBOSS - The European
Molecular Biology Open Software Suite"  Trends in Genetics,
15:276-278.

<p>
See also <a href="http://emboss.sourceforge.net/">http://emboss.sourceforge.net/</a></i>



<H3>14.1 Other useful references  </H3>

<br>

</BODY>
</HTML>
