<html>
<head>
<title>EMBASSY domainatrix: seqalign</title>
<base href="http://emboss.sourceforge.net/embassy/domainatrix/">
</head>
<body>
<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBOSS: SEQALIGN documentation. </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">
<H2> SEQALIGN documentation </H2>
<!-- END OF HEADER -->





<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<br><H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>

<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<br><br><br><H2> 1.0   SUMMARY  </H2>
Reads a DAF file (domain alignment file) and a DHF (domain hits file) and writes a DAF file extended with the hits.



<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<br><br><br><H2> 2.0   INPUTS & OUTPUTS          </H2>
SEQALIGN reads a directory of "seed" alignments and a directory of sets of protein sequences and writes a directory of "extended" alignments.  An extended alignment is a seed alignment that is extended with sequences from the appropriate sequence set.  The sets directory must contain a sequence set for each seed alignment directory and the files must have the same base name.  For example, sequences from "family.sets" are used to extend the seed alignment "family.aln". 
<br>Typically, the inputs are (i) DAF files (domain alignment files) containing structure-based sequence alignments and (ii) DHF files (domain hits files) containing sequence relatives (of unknown structure) of these alignments, and the output are DAF files containing the extended alignments. 
<br>The base names of the output alignments are the same as the input alignments.  The paths for input and output files are specified by the user and the file extensions are set in the ACD file.




<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.

     Cannot use the test data files because they might be empty or need 
     hand-editing 

     Use "<b>SEQALIGN</b> reads any normal sequence USAs." if
     appropriate.
-->
<a name="3.0"></a>
<br><br><br><H2> 3.0   INPUT FILE FORMAT         </H2>
The format of the domain alignment file is described in <a href="domainalign.html">DOMAINALIGN documentation</a>. 
<br>If other alignment files are used as input, all of the common file formats are supported.

<br><br>The format of the domain hits file is described in <a href="seqsearch.html">SEQSEARCH documentation</a>. 
<br>If other sequence sets are used as input, all of the common file formats are supported.



<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<br><br><br><H2> 4.0   OUTPUT FILE FORMAT         </H2>
The format of the domain alignment file is described in <a href="domainalign.html">DOMAINALIGN documentation</a>. 

Note however that the file will never contain 'Number' or 'Post_similar' records. Futhermore, the code before each sequence is of the form <b>Acc_Start_End </b>(where domain hits files were used as input sequences) or <b>Acc_n</b> (where other sequences were used as input), where 'Acc' is the accession number of the sequence, 'Start' and 'End' are the start and end point of the sequence in the full length sequence respectively, and 'n' is a sequential count of the sequence in the alignment.


<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml" 
-->

<a name="5.0"></a>
<br><br><br><H2> 5.0   DATA FILES                </H2>
SEQALIGN does not use a data file.







<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<br><br><br><H2> 6.0   USAGE                     </H2>
<H3> 6.1   COMMAND LINE ARGUMENTS </H3>
 
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td>-mode</td>
<td>This option specifies the mode of SEQSEARCH operation. SEQSEARCH takes as input a directory of either i. single sequences, ii. set of sequences (unaligned or aligned, but typically aligned sequences within a domain alignment file)). The user has to specify which.</td>
<td><table><tr><td>1</td> <td><i>(Single sequences)</i></td></tr><tr><td>2</td> <td><i>(Multiple sequences (e.g. sequence set or alignment))</i></td></tr></table></td>
<td>1</td>
</tr>
 
<tr>
<td>[-inseqs]<br>(Parameter 1)</td>
<td>This option specifies the location of sequences, e.g. DAF files (domain alignment files) (input). SEQSEARCH takes as input a database of either i. single sequences, ii. sets of unaligned sequences or iii. sets of aligned sequences, e.g. a domain alignment file. A 'domain alignment file' contains a sequence alignment of domains belonging to the same SCOP or CATH family. The file is in clustal format annotated with domain family classification information. The files generated by using SCOPALIGN will contain a structure-based sequence alignment of domains of known structure only. Such alignments can be extended with sequence relatives (of unknown structure) by using SEQALIGN.</td>
<td>Directory with files</td>
<td>./</td>
</tr>
 
<tr>
<td>[-dhfin]<br>(Parameter 2)</td>
<td>This option specifies the location of DHF files (domain hits files) (input). A 'domain hits file' contains database hits (sequences) with domain classification information, in the DHF format (FASTA or EMBL-like). The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. Files containing hits retrieved by PSIBLAST are generated by using SEQSEARCH.</td>
<td>Directory</td>
<td>./</td>
</tr>
 
<tr>
<td>-amode</td>
<td>This option specifies which alignment algorithm to use.</td>
<td><table><tr><td>1</td> <td><i>(CLUSTALW)</i></td></tr><tr><td>2</td> <td><i>(TCOFFEE (not yet implemented))</i></td></tr></table></td>
<td>1</td>
</tr>
 
<tr>
<td>[-dafout]<br>(Parameter 3)</td>
<td>This option specifies the location of DAF files (domain alignment files) (output). A 'domain alignment file' contains a sequence alignment of domains belonging to the same SCOP or CATH family. The file is in clustal format annotated with domain family classification information. The files generated by using SCOPALIGN will contain a structure-based sequence alignment of domains of known structure only. Such alignments can be extended with sequence relatives (of unknown structure) by using SEQALIGN.</td>
<td>Directory</td>
<td>./</td>
</tr>
 
<tr>
<td>-logf</td>
<td>This option specifies the name of log file for the build. The log file contains messages about any errors arising while SEQALIGN ran.</td>
<td>Output file</td>
<td>seqalign.log</td>
</tr>
 
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td colspan=4>(none)</td>
</tr>
 
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>
 
<tr>
<td colspan=4>(none)</td>
</tr>
 
</table>


<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of SEQALIGN is shown below.
<table><td bgcolor="#FFCCFF">
<pre>

Unix % seqalign
Reads a DAF file (domain alignment file) and a DHF (domain hits file) and
writes a DAF file extended with the hits.
Location of DAF files (domain alignment files) (input) [./]: /test_data
Location of DHF files (domain hits files) (input) [./]: /test_data
Alignment algorithm option
         1 : STAMP
         2 : TCOFFEE (not yet implemented)
Select number [1]: 1
Location of DAF files (domain alignment files) (output) [./]: /test_data/seqalign
Name of log file for the build [seqalign.log]: /test_data/seqalign/seqalign.log
Processing /test_data/54894.salign
 
clustalw -type=protein -profile1=./seqalign-1093871933.5448.aln -sequences -profile2=./seqalign-1093871933.5448.seqs -MATRIX=BLOSUM -GAPOPEN=10 -GAPEXT=0.5 -outfile=./seqalign-1093871933.5448.out
 
 
 
 
 CLUSTAL W (1.83) Multiple Sequence Alignments
 
 
 
Sequence type explicitly set to Protein
Sequence format is Pearson
No. of seqs=2
Sequences assumed to be PROTEIN
Sequence 1: d4at1b1_0        94 aa
Sequence 2: d4at1d1_1        94 aa
Sequence format is Pearson
No. of seqs in profile=40
Total no. of seqs     =42
Sequences assumed to be PROTEIN
Sequence 3: Q9YBD5_11_105        95 aa
Sequence 4: Q9UX07_12_104        93 aa
Sequence 5: Q9KP65_9_100         92 aa
Sequence 6: Q9K1K9_8_99          92 aa
Sequence 7: Q9JWY6_8_99          92 aa
Sequence 8: Q9HKM3_7_99          93 aa
Sequence 9: Q9HHN3_9_101         93 aa

< Further clustal output omitted for clarity >

Unix % 
</pre>
</table>

SEQALIGN read all domain alignment files with the file extension .salign (specified in ACD) 
from the directory /test_data, and extended these alignments using the sequences from domain 
hits files with file extension .hits (specified in ACD) in the directory /test_data.  
Extended domain alignment files (file extension .ealign specified in ACD) were written to 
/test_data/seqalign.
<br> 
<br>The following command line would achieve the same result.
<br>
<table><td bgcolor="#FFCCFF">
<pre> 
seqalign test_data test_data test_data/seqalign -mode 1  -amode 1 -logf test_data/seqalign/seqalign.log
</pre> 
</table>

<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<br><br><br><H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
None.





<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<br><br><br><H2> 8.0   NOTES                     </H2>
<b>  1. Use of CLUSTAL</b><br>
clustalw must be installed on the system that is running SEQSEARCH.

<br><br>When running SEQSEARCH at the HGMP it is essential that the command 'use clustal' (which runs the script /packages/menu/USE/clustal) is given before it is run. 


<br><br><b>  2. Empty domain hits file (input)</b>
<br>If just a single or no sequences are given in the domain hits file then no extended alignment will be generated.


<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<tr>
<td><b>Domain hits file </b></td>
<td> DHF format (FASTA-like format with domain classification information). </td>
<td> Database hits (sequences) with domain classification information.  The hits are relatives to a SCOP or CATH family and are found from a search of a sequence database. </td>
<td> <a href="seqsearch.html">SEQSEARCH</a> (hits retrieved by PSIBLAST)</td>
<td> N.A. </td>
</tr>
<tr>
<td><b>Domain alignment file </b></td>
<td> DAF format (CLUSTAL-like format with domain classification information). </td>
<td> Contains a sequence alignment of domains belonging to the same SCOP or CATH family.  The file is annotated with domain family classification information.</td>
<td> <a href="domainalign.html">DOMAINALIGN</a> (structure-based sequence alignment of domains of known structure). </td>
<td> DOMAINALIGN alignments can be extended with sequence relatives (of unknown structure) to the family in question by using SEQALIGN. </td>
</tr>
</table>





<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="9.0"></a>
<br><br><br><H2> 9.0   DESCRIPTION               </H2>
The generation of sequence alignments for large datasets such as SCOP or CATH
potentially requires a lot of time for preparation of datasets, running 
jobs and so on, in addition to the compute time required for the 
alignments themselves. 
SEQALIGN will generate an extended alignment for each seed alignment in a directory for which a file of sequences is available, by using CLUSTALW.  
Sequences are added one by one to the appropriate seed alignment.







<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<br><br><br><H2> 10.0  ALGORITHM                 </H2>
None.





<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
     (Take this from "Sister applications" in the old documentation)
-->
<a name="11.0"></a>
<br><br><br><H2> 11.0  RELATED APPLICATIONS      </H2>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>APPLICATION</b></td>
<td><b>DESCRIPTION</b></td>
</tr>
<tr>
<td><b><a href="domainalign.html">DOMAINALIGN</a> </b></td>
<td> Generates structure-based sequence alignments for nodes in a DCF
      file (domain classification file). </td>
</tr>
<tr>
<td><b><a href="seqsearch.html">SEQSEARCH</a> </b></td>
<td> Generate DHF files (domain hits files) of database hits (sequences) from a DAF file (domain alignment file) or other sequences by using PSI-BLAST.</td>
</tr>
<tr>
<td><b><a href="seqwords.html">SEQWORDS</a> </b></td>
<td>Generates DHF files (domain hits files) of database hits (sequences) for nodes in a DCF file (domain classification file) by keyword search of UniProt.</td>
</tr>
<tr>
<td><b><a href="seqalign.html">SEQALIGN</a> </b></td>
<td> Reads a DAF file (domain alignment file) and a DHF (domain hits file) and writes a DAF file extended with the hits.</td>
</tr>
</table>





<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="12.0"></a>
<br><br><br><H2> 12.0  DIAGNOSTIC ERROR MESSAGES </H2>
The log file might contain the following messages:
<br><br><i>Could not open seed sequence file my.file </i> (file read error on input sequence alignment or sequence file)
<br><i>Empty alignment in my.file </i> (input alignment was empty.)
<br><i>Empty singlet sequence for my.file </i> (singlet sequence file was empty.)
<br><i>Could not open domain hits file my.file</i>  (file read error on input sequence file.)
<br><i>Not enough sequences found in domain hits file </i> (insufficient sequences (i.e. < 2) in input sequence file.)


<!-- AUTHORS -->
<a name="13.0"><br><br><br><H2> 13.0  AUTHORS                   </H2></a>
Ranjeeva Ranasinghe <a href="mailto:rranasin@rfcgr.mrc.ac.uk">(rranasin@rfcgr.mrc.ac.uk)</a>
<br><br>
Jon Ison <a href="mailto:jison@rfcgr.mrc.ac.uk">(jison@rfcgr.mrc.ac.uk)</a>
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK




<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  REFERENCES                </H2>
Please cite the authors and EMBOSS.
<br><br>
<i>Rice P, Bleasby A.J.  2000.  EMBOSS - The European Molecular Biology Open Software Suite.  Trends in Genetics, 15:276-278.  See also http://www.uk.embnet.org/Software/EMBOSS</i>

<H3>14.1 Other useful references  </H3>

<br>

</BODY>
</HTML>

</body>
</html>
