<!--START OF HEADER - DON'T ALTER -->

<HTML>
<HEAD>
  <TITLE>
  EMBOSS: showalign
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">






<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="index.html" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

showalign
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>






<!--END OF HEADER-->

<H2>
    Function
</H2>
Displays a multiple sequence alignment

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<!-- 
	This is the main part of the document.
        This is what the user looks at to find out if this program will do what he wants. 
        Long description.
        Why it is useful.
        When it should/shouldn't be used.
        Algorithms
  -->

<b>showalign</b> displays an aligned set of protein or a nucleic acid
sequences in a style suitable for publication. 

<p>

The output is sent to the screen by default for the user to view, but it
can write the results to a file.

<p>


The output highlights various differences or similarities between each
of the sequences and a reference sequence by setting selected types of
matches to a reference sequence to be '.' characters.

<p>


The reference sequence can be either the calculated consensus sequence
(the default) or it can be one of the set of aligned sequences,
specified by either the ordinal number of that sequence in the input
file, or by its name. 

<p>

The output sequences can be displayed in either the input order (the
default) or they can be sorted in order of their similarity to the
reference sequence or sorted alphabetically by their names. 

<p>

By using the '-show' option, the displayed sequences can either be shown as:
<UL>
<LI>complete (-show=All), 
<LI>only identical matches between
the sequence and the reference sequence, all other positions being
replaced by '.' characters (-show=Identities)
<LI>non-identical matches, with identical matches being replaced by '.'
characters, similar matches are shown in lower case (-show=Non-identities)
<LI>similar matches, with non-similar matches being replaced by '.'  
characters, similar matches are shown in lower case (-show=Similarities)
<LI>dissimilar matches, with identical or similar matches being replaced by 
'.' characters (-show=Dissimilarities)
</UL>

<p>

A small table of the way these alignments are displayed illustrates this.

<br>

If we have a reference protein sequence of "III" and a sequence aligned
to this of "ILW", then we have an identical matching residue, then a
similar one, then a dissimilar one. 

<br>

The different methods of display would give the following:

<p>

<hr>
<pre>
Reference       III

All             ILw
Identical       I..
Non-id          .lW
Similar         Il.
Dissimilar      ..W
</pre>
<hr>

<p>

Changing the similar matches to lowercase can optionally be disable by
using the option -nosimilarcase.

<p>

The displayed sequence can be numbered by placing a ruler with ticks
above the sequence. 

<p>

The width of a line can be set.  The width of a margin to the left of
the sequences that shows the sequence names can be set. 

<p>

Specified regions of the sequence can be displayed in uppercase to
highlight them.

<p>

The output can be formatted for HTML.

<p>

If the output is being formatted for HTML, then specified regions of the
sequence can be displayed in any valid HTML colours.

<p>

<H3>The consensus line</H3>

The consensus line can be displayed in a mixture of uppercase and
lowercase symbols. 

<p>

The uppercase consensus symbol is indicates that the consensus
is strong and lowercase indicates that it is weak.

<p>

The cutoff for setting the case of the consensus is set by the qualifier
'-setcase'.  If the number of residues at that position that match the
consensus value is greater than this, then the symbol is in uppercase,
otherwise the symbol is in lowercase.  By default, the value of setcase
is set so that if there are more than 50% of residues identical to the
consunsus at that position, then the consensus is in uppercase. 

<p>

To put all of the consensus symbols into uppercase or lowercase, make
-setcase zero or very large (try 100000 ?).


<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->


<b>Here is a sample session with showalign</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>
Display the sequences in order of similarity to the reference sequence 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -order=s </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>
<p>
<b>Example 3</b>
<p>
Format for HTML and highlight some interesting regions in different colours: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -html -high '4-13 green 43-43 red 51-56 blue' </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.3">Go to the output files for this example</a><p><p>
<p>
<b>Example 4</b>
<p>
No consensus line at the bottom No ruler line No numbers line Don't repeat the reference sequence at the bottom of the sequences Use sequence 1 as the reference sequence Display residues from position 10 to 30 only  
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -nocon -norule -nonum -nobot -ref=1 -sb=10 -send=30 </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.4">Go to the output files for this example</a><p><p>
<p>
<b>Example 5</b>
<p>
Show non-identities between the sequences 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -show=n </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<p>
<p>
<b>Example 6</b>
<p>
Show all of the sequences 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -show=a </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.6">Go to the output files for this example</a><p><p>
<p>
<b>Example 7</b>
<p>
Show identities between the sequences 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -show=i </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.7">Go to the output files for this example</a><p><p>
<p>
<b>Example 8</b>
<p>
Show similarities between the sequences 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -show=s </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.8">Go to the output files for this example</a><p><p>
<p>
<b>Example 9</b>
<p>
Show dissimilarities between the sequences 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -show=d </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.9">Go to the output files for this example</a><p><p>
<p>
<b>Example 10</b>
<p>
Use the first sequence as the reference to compare to: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -ref=1 </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.10">Go to the output files for this example</a><p><p>
<p>
<b>Example 11</b>
<p>
Show a range of sequences in uppercase, everything else in lowercase 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -nocon -ref=1 -sl -upper 9-15 -nosimilarcase </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.11">Go to the output files for this example</a><p><p>
<p>
<b>Example 12</b>
<p>
Display the sequences in alphabetic order: 
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>showalign -order=a </b>
Displays a multiple sequence alignment
The sequence alignment to be displayed.: <b>../../data/globins.msf</b>
Output sequence details to a file [globins.showalign]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#output.12">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Mandatory qualifiers:
  [-sequence]          seqset     The sequence alignment to be displayed.
  [-outfile]           outfile    If you enter the name of a file here then
                                  this program will write the sequence details
                                  into that file.

   Optional qualifiers:
   -refseq             string     If you give the number in the alignment or
                                  the name of a sequence, it will be taken to
                                  be the reference sequence. The reference
                                  sequence is always show in full and is the
                                  one against which all the other sequences
                                  are compared. If this is set to 0 then the
                                  consensus sequence will be used as the
                                  reference sequence. By default the consensus
                                  sequence is used as the reference sequence.
   -[no]bottom         boolean    If this is true then the reference sequence
                                  is displayed at the bottom of the alignment
                                  as well as at the top.
   -show               menu       What to show
   -order              menu       Output order of the sequences
   -[no]similarcase    boolean    If this is set True, then when -show is set
                                  to 'Similarities' or 'Non-identities' and a
                                  residue is similar but not identical to the
                                  reference sequence residue, it will be
                                  changed to lower-case. If -show is set to
                                  'All' then non-identical, non-similar
                                  residues will be changed to lower-case. If
                                  this is False then no change to the case of
                                  the residues is made on the basis of their
                                  similarity to the reference sequence.
   -matrix             matrix     This is the scoring matrix file used when
                                  comparing sequences. By default it is the
                                  file 'EBLOSUM62' (for proteins) or the file
                                  'EDNAFULL' (for nucleic sequences). These
                                  files are found in the 'data' directory of
                                  the EMBOSS installation.
   -[no]consensus      boolean    If this is true then the consensus line is
                                  displayed.

   Advanced qualifiers:
   -uppercase          range      Regions to put in uppercase.
                                  If this is left blank, then the sequence
                                  case is left alone.
                                  A set of regions is specified by a set of
                                  pairs of positions.
                                  The positions are integers.
                                  They are separated by any non-digit,
                                  non-alpha character.
                                  Examples of region specifications are:
                                  24-45, 56-78
                                  1:45, 67=99;765..888
                                  1,5,8,10,23,45,57,99
   -[no]number         boolean    If this option is true then a line giving
                                  the positions in the alignment is displayed
                                  every 10 characters above the alignment.
   -[no]ruler          boolean    If this option is true then a ruler line
                                  marking every 5th and 10th character in the
                                  alignment is displayed.
   -width              integer    Width of sequence to display
   -margin             integer    This sets the length of the left-hand margin
                                  for sequence names. If the margin is set at
                                  0 then no margin and no names are
                                  displayed. If the margin is set to a value
                                  that is less than the length of a sequence
                                  name then the sequence name is displayed
                                  truncated to the length of the margin. If
                                  the margin is set to -1 then the minimum
                                  margin width that will allow all the
                                  sequence names to be displayed in full plus
                                  a space at the end of the name will
                                  automatically be selected.
   -html               boolean    Use HTML formatting
   -highlight          range      Regions to colour if formatting for HTML.
                                  If this is left blank, then the sequence is
                                  left alone.
                                  A set of regions is specified by a set of
                                  pairs of positions.
                                  The positions are integers.
                                  They are followed by any valid HTML font
                                  colour.
                                  Examples of region specifications are:
                                  24-45 blue 56-78 orange
                                  1-100 green 120-156 red
                                  A file of ranges to colour (one range per
                                  line) can be specified as '@filename'.
   -plurality          float      Set a cut-off for the % of positive scoring
                                  matches below which there is no consensus.
                                  The default plurality is taken as 50% of the
                                  total weight of all the sequences in the
                                  alignment.
   -setcase            float      Sets the threshold for the scores of the
                                  positive matches above which the consensus
                                  is in upper-case and below which the
                                  consensus is in lower-case. By default this
                                  is set to be half of the (weight-adjusted)
                                  number of sequences in the alignment.
   -identity           float      Provides the facility of setting the
                                  required number of identities at a position
                                  for it to give a consensus. Therefore, if
                                  this is set to 100% only columns of
                                  identities contribute to the consensus.

   Associated qualifiers:
  "-sequence" related qualifiers
  -sbegin1             integer    First base used
  -send1               integer    Last base used, def=seq length
  -sreverse1           boolean    Reverse (if DNA)
  -sask1               boolean    Ask for begin/end/reverse
  -snucleotide1        boolean    Sequence is nucleotide
  -sprotein1           boolean    Sequence is protein
  -slower1             boolean    Make lower case
  -supper1             boolean    Make upper case
  -sformat1            string     Input sequence format
  -sopenfile1          string     Input filename
  -sdbname1            string     Database name
  -sid1                string     Entryname
  -ufo1                string     UFO features
  -fformat1            string     Features format
  -fopenfile1          string     Features file name
  "-outfile" related qualifiers
  -odirectory2         string     Output directory

   General qualifiers:
  -auto                boolean    Turn off prompts
  -stdout              boolean    Write standard output
  -filter              boolean    Read standard input, write standard output
  -options             boolean    Prompt for required and optional values
  -debug               boolean    Write debug output to program.dbg
  -verbose             boolean    Report some/full command line options
  -help                boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
  -warning             boolean    Report warnings
  -error               boolean    Report errors
  -fatal               boolean    Report fatal errors
  -die                 boolean    Report deaths


</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Mandatory qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>The sequence alignment to be displayed.</td>
<td>Readable sequences</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>If you enter the name of a file here then this program will write the sequence details into that file.</td>
<td>Output file</td>
<td><i>&lt;sequence&gt;</i>.showalign</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Optional qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-refseq</td>
<td>If you give the number in the alignment or the name of a sequence, it will be taken to be the reference sequence. The reference sequence is always show in full and is the one against which all the other sequences are compared. If this is set to 0 then the consensus sequence will be used as the reference sequence. By default the consensus sequence is used as the reference sequence.</td>
<td>Any string is accepted</td>
<td>0</td>
</tr>

<tr>
<td>-[no]bottom</td>
<td>If this is true then the reference sequence is displayed at the bottom of the alignment as well as at the top.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-show</td>
<td>What to show</td>
<td><table><tr><td>A</td> <td><i>(All of the sequences)</i></td></tr><tr><td>I</td> <td><i>(Identities between the sequences)</i></td></tr><tr><td>N</td> <td><i>(Non-identities between the sequences)</i></td></tr><tr><td>S</td> <td><i>(Similarities between the sequences)</i></td></tr><tr><td>D</td> <td><i>(Dissimilarities between the sequences)</i></td></tr></table></td>
<td>N</td>
</tr>

<tr>
<td>-order</td>
<td>Output order of the sequences</td>
<td><table><tr><td>I</td> <td><i>(Input order - no change)</i></td></tr><tr><td>A</td> <td><i>(Alphabetical order of the names)</i></td></tr><tr><td>S</td> <td><i>(Similarity to the reference sequence)</i></td></tr></table></td>
<td>I</td>
</tr>

<tr>
<td>-[no]similarcase</td>
<td>If this is set True, then when -show is set to 'Similarities' or 'Non-identities' and a residue is similar but not identical to the reference sequence residue, it will be changed to lower-case. If -show is set to 'All' then non-identical, non-similar residues will be changed to lower-case. If this is False then no change to the case of the residues is made on the basis of their similarity to the reference sequence.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-matrix</td>
<td>This is the scoring matrix file used when comparing sequences. By default it is the file 'EBLOSUM62' (for proteins) or the file 'EDNAFULL' (for nucleic sequences). These files are found in the 'data' directory of the EMBOSS installation.</td>
<td>Comparison matrix file in EMBOSS data path</td>
<td>EBLOSUM62 for protein<br>EDNAFULL for DNA</td>
</tr>

<tr>
<td>-[no]consensus</td>
<td>If this is true then the consensus line is displayed.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-uppercase</td>
<td>Regions to put in uppercase.
If this is left blank, then the sequence case is left alone.
A set of regions is specified by a set of pairs of positions.
The positions are integers.
They are separated by any non-digit, non-alpha character.
Examples of region specifications are:
24-45, 56-78
1:45, 67=99;765..888
1,5,8,10,23,45,57,99</td>
<td>Sequence range</td>
<td>If this is left blank, then the sequence case is left alone.</td>
</tr>

<tr>
<td>-[no]number</td>
<td>If this option is true then a line giving the positions in the alignment is displayed every 10 characters above the alignment.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-[no]ruler</td>
<td>If this option is true then a ruler line marking every 5th and 10th character in the alignment is displayed.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-width</td>
<td>Width of sequence to display</td>
<td>Integer 1 or more</td>
<td>60</td>
</tr>

<tr>
<td>-margin</td>
<td>This sets the length of the left-hand margin for sequence names. If the margin is set at 0 then no margin and no names are displayed. If the margin is set to a value that is less than the length of a sequence name then the sequence name is displayed truncated to the length of the margin. If the margin is set to -1 then the minimum margin width that will allow all the sequence names to be displayed in full plus a space at the end of the name will automatically be selected.</td>
<td>Integer -1 or more</td>
<td>-1</td>
</tr>

<tr>
<td>-html</td>
<td>Use HTML formatting</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-highlight</td>
<td>Regions to colour if formatting for HTML.
If this is left blank, then the sequence is left alone.
A set of regions is specified by a set of pairs of positions.
The positions are integers.
They are followed by any valid HTML font colour.
Examples of region specifications are:
24-45 blue 56-78 orange
1-100 green 120-156 red
A file of ranges to colour (one range per line) can be specified as '@filename'.</td>
<td>Sequence range</td>
<td><i>full sequence</i></td>
</tr>

<tr>
<td>-plurality</td>
<td>Set a cut-off for the % of positive scoring matches below which there is no consensus. The default plurality is taken as 50% of the total weight of all the sequences in the alignment.</td>
<td>Number from 0.000 to 100.000</td>
<td>50.0</td>
</tr>

<tr>
<td>-setcase</td>
<td>Sets the threshold for the scores of the positive matches above which the consensus is in upper-case and below which the consensus is in lower-case. By default this is set to be half of the (weight-adjusted) number of sequences in the alignment.</td>
<td>Any numeric value</td>
<td>@( $(sequence.totweight) / 2)</td>
</tr>

<tr>
<td>-identity</td>
<td>Provides the facility of setting the required number of identities at a position for it to give a consensus. Therefore, if this is set to 100% only columns of identities contribute to the consensus.</td>
<td>Number from 0.000 to 100.000</td>
<td>0.0</td>
</tr>

</table>


<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>showalign</b> reads in a set of aligned protein or nucleic sequences. 

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: ../../data/globins.msf</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
!!AA_MULTIPLE_ALIGNMENT 1.0

  ../data/globins.msf MSF:  164 Type: P 25/06/01 CompCheck: 4278 ..

  Name: HBB_HUMAN Len: 164  Check: 6914 Weight: 0.14
  Name: HBB_HORSE Len: 164  Check: 6007 Weight: 0.15
  Name: HBA_HUMAN Len: 164  Check: 3921 Weight: 0.15
  Name: HBA_HORSE Len: 164  Check: 4770 Weight: 0.19
  Name: MYG_PHYCA Len: 164  Check: 7930 Weight: 0.23
  Name: GLB5_PETMA Len: 164  Check: 1857 Weight: 0.21
  Name: LGB2_LUPLU Len: 164  Check: 2879 Weight: 0.10

//

           1                                               50
HBB_HUMAN  ~~~~~~~~VHLTPEEKSAVTALWGKVN.VDEVGGEALGR.LLVVYPWTQR
HBB_HORSE  ~~~~~~~~VQLSGEEKAAVLALWDKVN.EEEVGGEALGR.LLVVYPWTQR
HBA_HUMAN  ~~~~~~~~~~~~~~VLSPADKTNVKAA.WGKVGAHAGEYGAEALERMFLS
HBA_HORSE  ~~~~~~~~~~~~~~VLSAADKTNVKAA.WSKVGGHAGEYGAEALERMFLG
MYG_PHYCA  ~~~~~~~VLSEGEWQLVLHVWAKVEAD.VAGHGQDILIR.LFKSHPETLE
GLB5_PETMA PIVDTGSVAPLSAAEKTKIRSAWAPVYSTYETSGVDILVKFFTSTPAAQE
LGB2_LUPLU ~~~~~~~~GALTESQAALVKSSWEEFNANIPKHTHRFFILVLEIAPAAKD

           51                                             100
HBB_HUMAN  FFESFGDLSTPDAVMGNPKVKAHGKKVLGAFSDGLAHLDNLKGTFATLSE
HBB_HORSE  FFDSFGDLSNPGAVMGNPKVKAHGKKVLHSFGEGVHHLDNLKGTFAALSE
HBA_HUMAN  FPTTKTYFPHFDLSHGSAQVKGHGKKVADALTNAVAHVDDMPNALSALSD
HBA_HORSE  FPTTKTYFPHFDLSHGSAQVKAHGKKVGDALTLAVGHLDDLPGALSNLSD
MYG_PHYCA  KFDRFKHLKTEAEMKASEDLKKHGVTVLTALGAILKKKGHHEAELKPLAQ
GLB5_PETMA FFPKFKGLTTADQLKKSADVRWHAERIINAVNDAVASMDDTEKMSMKLRD
LGB2_LUPLU LFSFLKGTSEVPQNNPELQAHAGKVFKLVYEAAIQLQVTGVVVTDATLKN

           101                                            150
HBB_HUMAN  LHCDKLH..VDPENFRLLGNVLVCVLAHHFGKEFTPPVQAAYQKVVAGVA
HBB_HORSE  LHCDKLH..VDPENFRLLGNVLVVVLARHFGKDFTPELQASYQKVVAGVA
HBA_HUMAN  LHAHKLR..VDPVNFKLLSHCLLVTLAAHLPAEFTPAVHASLDKFLASVS
HBA_HORSE  LHAHKLR..VDPVNFKLLSHCLLSTLAVHLPNDFTPAVHASLDKFLSSVS
MYG_PHYCA  SHATKHK..IPIKYLEFISEAIIHVLHSRHPGDFGADAQGAMNKALELFR
GLB5_PETMA LSGKHAK..SFQVDPQYFKVLAAVIADTVAAGDAGFEKLMSMICILLRSA
LGB2_LUPLU LGSVHVSKGVADAHFPVVKEAILKTIKEVVGAKWSEELNSAWTIAYDELA

           151        164
HBB_HUMAN  NALAHKYH~~~~~~
HBB_HORSE  NALAHKYH~~~~~~
HBA_HUMAN  TVLTSKYR~~~~~~
HBA_HORSE  TVLTSKYR~~~~~~
MYG_PHYCA  KDIAAKYKELGYQG
GLB5_PETMA Y~~~~~~~~~~~~~
LGB2_LUPLU IVIKKEMNDAA~~~

</pre>
</td></tr></table><p>


<p>

You can specifiy a file of ranges to display in uppercase by giving the
'-uppercase' qualifier the value '@' followed by the name of the file
containing the ranges.  (eg: '-upper @myfile').
<p>
          
The format of the range file is:
<p>
<ul>
      <li>Comment lines start with '#' in the first column.
      <li>Comment lines and blank lines are ignored.
      <li>The line may start with white-space.
      <li>There are two positive (integer) numbers per line separated by one
                or more space or
      <li>TAB characters.
      <li>The second number must be greater or equal to the first number.  
      <li>There can be optional text after the two numbers to annotate the line.
      <li>White-space before or after the text is removed.
</ul>
<p>
An example range file is:
<p>

<pre>          
# this is my set of ranges
12   23                           
 4   5       this is like 12-23, but smaller
67   10348   interesting region
</pre>
<p>

You can specifiy a file of ranges to highlight in a different colour
when outputting in HTML format (using the '-html' qualifier) by giving
the '-highlight' qualifier the value '@' followed by the name of the
file containing the ranges.  (eg: '-highlight @myfile').
<p>

The format of this file is very similar to the format of the above
uppercase range file, except that the text after the start and end
positions is used as the HTML colour name.  This colour name is used 'as
is' when specifying the colour in HTML in a '<FONT COLOR=xxx>'
construct, (where 'xxx' is the name of the colour).
<p>

The standard names of HTML font colours are given in:
<br>

<A HREF="http://http://www.w3.org/TR/REC-html40/types.html">
http://http://www.w3.org/TR/REC-html40/types.html</A>
and
<A HREF="http://www.ausmall.com.au/freegraf/ncolour2.htm">
http://www.ausmall.com.au/freegraf/ncolour2.htm</A>
and
<A HREF="http://mindprod.com/htmlcolours.html">
http://mindprod.com/htmlcolours.html</A>
(amongst other places).

<p>

An example highlight range file is:
<p>

<pre>          
# this is my set of ranges
12   23         red
 4   5          darkturquoise
67   10348      #FFE4E1
</pre>
<p>













<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 
   -->


<b>showalign</b> writes out a text file, optionally formatted for HTML. 

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx
HBB_HUMAN  --------VHLTPEEKS.VTALWGKVN-VDEV.GEALG.-.LVVY.WTqR..ES.GD..T
HBB_HORSE  --------VQLSGEEKA.VLALWDKVN-EEEV.GEALG.-.LVVY.WTqR..DS.GD..N
HBA_HUMAN  --------------VLSPADKTNVKAA-WGKV.AHAGEYGaEALErMF.S.PTTKTYFpH
HBA_HORSE  --------------VLS.ADKTNVKAA-WSKV.GHAGEYGaEALErMF.G.PTTKTYFpH
MYG_PHYCA  -------VLSEGEWQLVlHVWAKVEAD-VAGH.QDILI.-.FKSH.ET.EK.DR.KH.KT
GLB5_PETMA PIVDTGSVAPLSAAEKTKIRSAWAPVYSTYETSGVDILvKFFTST.AAqE..PK.KG.TT
LGB2_LUPLU --------GALTESQAAlVKSSWEEFNANIPKHTHRFFILVLEIA.AAKDL.SFLKGT.E
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx
HBB_HUMAN  PDaVMGNPK..A..KK..G.FSDGLA.L.NLKGTFAT.SE..cD..H--.DPEN.RLLGN
HBB_HORSE  PGaVMGNPK..A..KK..HSFGEG.H.L.NLKGTFAA.SE..cD..H--.DPEN.RLLGN
HBA_HUMAN  FD.SHG.AQ..G..KK.aD.LTNA.A.V.DMPNALSA.SD...H..R--.DPVN.KLLSH
HBA_HORSE  FD.SHG.AQ..A..KK.GD.LTLA.G.L.DLPGALSN.SD...H..R--.DPVN.KLLSH
MYG_PHYCA  EAEMKA.EDL.K..VT..T.LGAILKkKGHHEAELKP.AQS..T.HK--IPIKYLEFISE
GLB5_PETMA ADqLKK.AD.RW.AERIIN.VNDA.ASM.DTEKMSMK.RD.SGKhaK--SFQVDPQYFKV
LGB2_LUPLU VPqNNPELQAhAGKVFK.VYEAAIqLQVTGVVVTDAT.KN.GSVhVSKG.ADAH.PVVKE
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx
HBB_HUMAN  V.VCV.AHHFGKE.TPPV.AAYQ.VVAGVANALAH..H------
HBB_HORSE  V.VVV.ARHFGK..TPEL.ASYQ.VVAGVANALAH..H------
HBA_HUMAN  c.LVT.AAHLPAE.TPAVHASLD.F.ASVSTVLTS..R------
HBA_HORSE  c.LST.AVHLPN..TPAVHASLD.F.SSVSTVLTS..R------
MYG_PHYCA  .IIHV.HSRHPG..GADA.GAMN.A.ELFRKDIAA..KELGYQG
GLB5_PETMA laAVIaDTVAAG.AGFEKlMSMICI.LRSAY-------------
LGB2_LUPLU .ILKTIKEVVGAKWSEELnSAWTiAYDELAIVIKKEMNDAA---
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx
HBB_HORSE  --------VQLSGEEKA.VLALWDKVN-EEEV.GEALG.-.LVVY.WTqR..DS.GD..N
HBB_HUMAN  --------VHLTPEEKS.VTALWGKVN-VDEV.GEALG.-.LVVY.WTqR..ES.GD..T
HBA_HORSE  --------------VLS.ADKTNVKAA-WSKV.GHAGEYGaEALErMF.G.PTTKTYFpH
HBA_HUMAN  --------------VLSPADKTNVKAA-WGKV.AHAGEYGaEALErMF.S.PTTKTYFpH
MYG_PHYCA  -------VLSEGEWQLVlHVWAKVEAD-VAGH.QDILI.-.FKSH.ET.EK.DR.KH.KT
GLB5_PETMA PIVDTGSVAPLSAAEKTKIRSAWAPVYSTYETSGVDILvKFFTST.AAqE..PK.KG.TT
LGB2_LUPLU --------GALTESQAAlVKSSWEEFNANIPKHTHRFFILVLEIA.AAKDL.SFLKGT.E
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx
HBB_HORSE  PGaVMGNPK..A..KK..HSFGEG.H.L.NLKGTFAA.SE..cD..H--.DPEN.RLLGN
HBB_HUMAN  PDaVMGNPK..A..KK..G.FSDGLA.L.NLKGTFAT.SE..cD..H--.DPEN.RLLGN
HBA_HORSE  FD.SHG.AQ..A..KK.GD.LTLA.G.L.DLPGALSN.SD...H..R--.DPVN.KLLSH
HBA_HUMAN  FD.SHG.AQ..G..KK.aD.LTNA.A.V.DMPNALSA.SD...H..R--.DPVN.KLLSH
MYG_PHYCA  EAEMKA.EDL.K..VT..T.LGAILKkKGHHEAELKP.AQS..T.HK--IPIKYLEFISE
GLB5_PETMA ADqLKK.AD.RW.AERIIN.VNDA.ASM.DTEKMSMK.RD.SGKhaK--SFQVDPQYFKV
LGB2_LUPLU VPqNNPELQAhAGKVFK.VYEAAIqLQVTGVVVTDAT.KN.GSVhVSKG.ADAH.PVVKE
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx
HBB_HORSE  V.VVV.ARHFGK..TPEL.ASYQ.VVAGVANALAH..H------
HBB_HUMAN  V.VCV.AHHFGKE.TPPV.AAYQ.VVAGVANALAH..H------
HBA_HORSE  c.LST.AVHLPN..TPAVHASLD.F.SSVSTVLTS..R------
HBA_HUMAN  c.LVT.AAHLPAE.TPAVHASLD.F.ASVSTVLTS..R------
MYG_PHYCA  .IIHV.HSRHPG..GADA.GAMN.A.ELFRKDIAA..KELGYQG
GLB5_PETMA laAVIaDTVAAG.AGFEKlMSMICI.LRSAY-------------
LGB2_LUPLU .ILKTIKEVVGAKWSEELnSAWTiAYDELAIVIKKEMNDAA---
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

</pre>
</td></tr></table><p>

<a name="output.3"></a>
<h3>Output files for usage example 3</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&lt;pre&gt;
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxx&lt;font color=green&gt;xxxxxxxxxx&lt;/font&gt;xxxxAxxxxxxxxxxxxxxGxxxxxRxLx&lt;font color=red&gt;x&lt;/font&gt;xxPxxLx&lt;font color=blue&gt;FFxxFx&lt;/font&gt;xLSx
HBB_HUMAN  ---&lt;font color=green&gt;-----VHLTP&lt;/font&gt;EEKS.VTALWGKVN-VDEV.GEALG.-.L&lt;font color=red&gt;V&lt;/font&gt;VY.WTqR&lt;font color=blue&gt;..ES.G&lt;/font&gt;D..T
HBB_HORSE  ---&lt;font color=green&gt;-----VQLSG&lt;/font&gt;EEKA.VLALWDKVN-EEEV.GEALG.-.L&lt;font color=red&gt;V&lt;/font&gt;VY.WTqR&lt;font color=blue&gt;..DS.G&lt;/font&gt;D..N
HBA_HUMAN  ---&lt;font color=green&gt;----------&lt;/font&gt;-VLSPADKTNVKAA-WGKV.AHAGEYGaE&lt;font color=red&gt;A&lt;/font&gt;LErMF.S&lt;font color=blue&gt;.PTTKT&lt;/font&gt;YFpH
HBA_HORSE  ---&lt;font color=green&gt;----------&lt;/font&gt;-VLS.ADKTNVKAA-WSKV.GHAGEYGaE&lt;font color=red&gt;A&lt;/font&gt;LErMF.G&lt;font color=blue&gt;.PTTKT&lt;/font&gt;YFpH
MYG_PHYCA  ---&lt;font color=green&gt;----VLSEGE&lt;/font&gt;WQLVlHVWAKVEAD-VAGH.QDILI.-.F&lt;font color=red&gt;K&lt;/font&gt;SH.ET.E&lt;font color=blue&gt;K.DR.K&lt;/font&gt;H.KT
GLB5_PETMA PIV&lt;font color=green&gt;DTGSVAPLSA&lt;/font&gt;AEKTKIRSAWAPVYSTYETSGVDILvKFF&lt;font color=red&gt;T&lt;/font&gt;ST.AAqE&lt;font color=blue&gt;..PK.K&lt;/font&gt;G.TT
LGB2_LUPLU ---&lt;font color=green&gt;-----GALTE&lt;/font&gt;SQAAlVKSSWEEFNANIPKHTHRFFILVL&lt;font color=red&gt;E&lt;/font&gt;IA.AAKD&lt;font color=blue&gt;L.SFLK&lt;/font&gt;GT.E
Consensus  xxx&lt;font color=green&gt;xxxxxxxxxx&lt;/font&gt;xxxxAxxxxxxxxxxxxxxGxxxxxRxLx&lt;font color=red&gt;x&lt;/font&gt;xxPxxLx&lt;font color=blue&gt;FFxxFx&lt;/font&gt;xLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx
HBB_HUMAN  PDaVMGNPK..A..KK..G.FSDGLA.L.NLKGTFAT.SE..cD..H--.DPEN.RLLGN
HBB_HORSE  PGaVMGNPK..A..KK..HSFGEG.H.L.NLKGTFAA.SE..cD..H--.DPEN.RLLGN
HBA_HUMAN  FD.SHG.AQ..G..KK.aD.LTNA.A.V.DMPNALSA.SD...H..R--.DPVN.KLLSH
HBA_HORSE  FD.SHG.AQ..A..KK.GD.LTLA.G.L.DLPGALSN.SD...H..R--.DPVN.KLLSH
MYG_PHYCA  EAEMKA.EDL.K..VT..T.LGAILKkKGHHEAELKP.AQS..T.HK--IPIKYLEFISE
GLB5_PETMA ADqLKK.AD.RW.AERIIN.VNDA.ASM.DTEKMSMK.RD.SGKhaK--SFQVDPQYFKV
LGB2_LUPLU VPqNNPELQAhAGKVFK.VYEAAIqLQVTGVVVTDAT.KN.GSVhVSKG.ADAH.PVVKE
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx
HBB_HUMAN  V.VCV.AHHFGKE.TPPV.AAYQ.VVAGVANALAH..H------
HBB_HORSE  V.VVV.ARHFGK..TPEL.ASYQ.VVAGVANALAH..H------
HBA_HUMAN  c.LVT.AAHLPAE.TPAVHASLD.F.ASVSTVLTS..R------
HBA_HORSE  c.LST.AVHLPN..TPAVHASLD.F.SSVSTVLTS..R------
MYG_PHYCA  .IIHV.HSRHPG..GADA.GAMN.A.ELFRKDIAA..KELGYQG
GLB5_PETMA laAVIaDTVAAG.AGFEKlMSMICI.LRSAY-------------
LGB2_LUPLU .ILKTIKEVVGAKWSEELnSAWTiAYDELAIVIKKEMNDAA---
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

&lt;/pre&gt;
</pre>
</td></tr></table><p>

<a name="output.4"></a>
<h3>Output files for usage example 4</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
HBB_HUMAN  HLTPEEKSAVTALWGKVN-VD
HBB_HORSE  Q.SG...A..L...D...-EE
HBA_HUMAN  -----VL.PADKTNV.AA-WG
HBA_HORSE  -----VL..ADKTNV.AA-WS
MYG_PHYCA  SEGEWQLVlHVWaKVEAD-.A
GLB5_PETMA P.SAA..TKIRSa.AP.YSTY
LGB2_LUPLU A..ESQAAl.KSS.EEF.AnI

</pre>
</td></tr></table><p>

<a name="output.6"></a>
<h3>Output files for usage example 6</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx
HBB_HUMAN  --------vhltpeeksAvtalwgkvn-vdevGgealgR-LlvvyPwtQrFFesFgdLSt
HBB_HORSE  --------vqlsgeekaAvlalwdkvn-eeevGgealgR-LlvvyPwtQrFFdsFgdLSn
HBA_HUMAN  --------------vlspadktnvkaa-wgkvGahageygAealeRmfLsFpttktyfPh
HBA_HORSE  --------------vlsAadktnvkaa-wskvGghageygAealeRmfLgFpttktyfPh
MYG_PHYCA  -------vlsegewqlvLhvwakvead-vaghGqdiliR-LfkshPetLekFdrFkhLkt
GLB5_PETMA pivdtgsvaplsaaektkirsawapvystyetsgvdilVkfftstPaaQeFFpkFkgLtt
LGB2_LUPLU --------galtesqaaLvkssweefnanipkhthrffilvleiaPaakdlFsflkgtSe
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx
HBB_HUMAN  pdAvmgnpkVKaHGkkVLgAfsdglaHlDnlkgtfatLseLHCdKLh--VdpenFrllgn
HBB_HORSE  pgAvmgnpkVKaHGkkVLhsfgegVhHlDnlkgtfaaLseLHCdKLh--VdpenFrllgn
HBA_HUMAN  fdLshgSaqVKgHGkkVAdAltnaVaHvDdmpnalsaLsdLHAhKLr--VdpvnFkllsh
HBA_HORSE  fdLshgSaqVKaHGkkVgdAltlaVgHlDdlpgalsnLsdLHAhKLr--VdpvnFkllsh
MYG_PHYCA  eaemkaSedlKkHGvtVLtAlgailkKkghheaelkpLaqsHAtKhk--ipikylefise
GLB5_PETMA adQlkkSadVrwHaeriinAvndaVasmDdtekmsmkLrdLsgkHAk--sfqvdpqyfkv
LGB2_LUPLU vpQnnpelqaHagkvfkLvyeaaiQlqvtgvvvtdatLknLgsvHvskgVadahFpvvke
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx
HBB_HUMAN  vLvcvLahhfgkeFtppvQaayqKvvagvanalahKYh------
HBB_HORSE  vLvvvLarhfgkDFtpelQasyqKvvagvanalahKYh------
HBA_HUMAN  CLlvtLaahlpaeFtpavhasldKfLasvstvltsKYr------
HBA_HORSE  CLlstLavhlpnDFtpavhasldKfLssvstvltsKYr------
MYG_PHYCA  AiihvLhsrhpgDFgadaQgamnKaLelfrkdiaaKYkelgyqg
GLB5_PETMA LAaviAdtvaagDagfekLmsmiciLlrsay-------------
LGB2_LUPLU AilktikevvgakwseelNsawtIaydelaivikkemndaa---
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

</pre>
</td></tr></table><p>

<a name="output.7"></a>
<h3>Output files for usage example 7</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx
HBB_HUMAN  --------.........A.........-....G.....R-L....P....FF..F..LS.
HBB_HORSE  --------.........A.........-....G.....R-L....P....FF..F..LS.
HBA_HUMAN  --------------.............-....G...............L.F.........
HBA_HORSE  --------------...A.........-....G...............L.F.........
MYG_PHYCA  -------....................-....G.....R-L....P..L..F..F..L..
GLB5_PETMA .............................................P....FF..F..L..
LGB2_LUPLU --------.....................................P.....F......S.
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx
HBB_HUMAN  .........VK.HG..VL.A......H.D........L..LH..KL.--V....F.....
HBB_HORSE  .........VK.HG..VL......V.H.D........L..LH..KL.--V....F.....
HBA_HUMAN  ..L...S..VK.HG..V..A....V.H.D........L..LHA.KL.--V....F.....
HBA_HORSE  ..L...S..VK.HG..V..A....V.H.D........L..LHA.KL.--V....F.....
MYG_PHYCA  ......S...K.HG..VL.A.................L...HA.K..--...........
GLB5_PETMA ......S..V..H......A....V...D........L..L......--...........
LGB2_LUPLU .................L...................L..L........V....F.....
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx
HBB_HUMAN  .L...L.......F....Q....K...........KY.------
HBB_HORSE  .L...L......DF....Q....K...........KY.------
HBA_HUMAN  .L...L.......F.........K.L.........KY.------
HBA_HORSE  .L...L......DF.........K.L.........KY.------
MYG_PHYCA  A....L......DF....Q....K.L.........KY.......
GLB5_PETMA ............D............L.....-------------
LGB2_LUPLU A........................................---
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

</pre>
</td></tr></table><p>

<a name="output.8"></a>
<h3>Output files for usage example 8</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx
HBB_HUMAN  --------.........A.........-....G.....R-L....P..q.FF..F..LS.
HBB_HORSE  --------.........A.........-....G.....R-L....P..q.FF..F..LS.
HBA_HUMAN  --------------.............-....G.......a....r..L.F.......p.
HBA_HORSE  --------------...A.........-....G.......a....r..L.F.......p.
MYG_PHYCA  -------..........l.........-....G.....R-L....P..L..F..F..L..
GLB5_PETMA ......................................v......P..q.FF..F..L..
LGB2_LUPLU --------.........l...........................P.....F......S.
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx
HBB_HUMAN  ..a......VK.HG..VL.A......H.D........L..LHc.KL.--V....F.....
HBB_HORSE  ..a......VK.HG..VL......V.H.D........L..LHc.KL.--V....F.....
HBA_HUMAN  ..L...S..VK.HG..Va.A....V.H.D........L..LHA.KL.--V....F.....
HBA_HORSE  ..L...S..VK.HG..V..A....V.H.D........L..LHA.KL.--V....F.....
MYG_PHYCA  ......S...K.HG..VL.A......k..........L...HA.K..--...........
GLB5_PETMA ..q...S..V..H......A....V...D........L..L...ha.--...........
LGB2_LUPLU ..q.......h......L......q............L..L...h....V....F.....
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx
HBB_HUMAN  .L...L.......F....Q....K...........KY.------
HBB_HORSE  .L...L......DF....Q....K...........KY.------
HBA_HUMAN  cL...L.......F.........K.L.........KY.------
HBA_HORSE  cL...L......DF.........K.L.........KY.------
MYG_PHYCA  A....L......DF....Q....K.L.........KY.......
GLB5_PETMA la...a......D.....l......L.....-------------
LGB2_LUPLU A.................n....i.................---
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

</pre>
</td></tr></table><p>

<a name="output.9"></a>
<h3>Output files for usage example 9</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx
HBB_HUMAN  --------VHLTPEEKS.VTALWGKVN-VDEV.GEALG.-.LVVY.WT.R..ES.GD..T
HBB_HORSE  --------VQLSGEEKA.VLALWDKVN-EEEV.GEALG.-.LVVY.WT.R..DS.GD..N
HBA_HUMAN  --------------VLSPADKTNVKAA-WGKV.AHAGEYG.EALE.MF.S.PTTKTYF.H
HBA_HORSE  --------------VLS.ADKTNVKAA-WSKV.GHAGEYG.EALE.MF.G.PTTKTYF.H
MYG_PHYCA  -------VLSEGEWQLV.HVWAKVEAD-VAGH.QDILI.-.FKSH.ET.EK.DR.KH.KT
GLB5_PETMA PIVDTGSVAPLSAAEKTKIRSAWAPVYSTYETSGVDIL.KFFTST.AA.E..PK.KG.TT
LGB2_LUPLU --------GALTESQAA.VKSSWEEFNANIPKHTHRFFILVLEIA.AAKDL.SFLKGT.E
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx
HBB_HUMAN  PD.VMGNPK..A..KK..G.FSDGLA.L.NLKGTFAT.SE...D..H--.DPEN.RLLGN
HBB_HORSE  PG.VMGNPK..A..KK..HSFGEG.H.L.NLKGTFAA.SE...D..H--.DPEN.RLLGN
HBA_HUMAN  FD.SHG.AQ..G..KK..D.LTNA.A.V.DMPNALSA.SD...H..R--.DPVN.KLLSH
HBA_HORSE  FD.SHG.AQ..A..KK.GD.LTLA.G.L.DLPGALSN.SD...H..R--.DPVN.KLLSH
MYG_PHYCA  EAEMKA.EDL.K..VT..T.LGAILK.KGHHEAELKP.AQS..T.HK--IPIKYLEFISE
GLB5_PETMA AD.LKK.AD.RW.AERIIN.VNDA.ASM.DTEKMSMK.RD.SGK..K--SFQVDPQYFKV
LGB2_LUPLU VP.NNPELQA.AGKVFK.VYEAAI.LQVTGVVVTDAT.KN.GSV.VSKG.ADAH.PVVKE
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx
HBB_HUMAN  V.VCV.AHHFGKE.TPPV.AAYQ.VVAGVANALAH..H------
HBB_HORSE  V.VVV.ARHFGK..TPEL.ASYQ.VVAGVANALAH..H------
HBA_HUMAN  ..LVT.AAHLPAE.TPAVHASLD.F.ASVSTVLTS..R------
HBA_HORSE  ..LST.AVHLPN..TPAVHASLD.F.SSVSTVLTS..R------
MYG_PHYCA  .IIHV.HSRHPG..GADA.GAMN.A.ELFRKDIAA..KELGYQG
GLB5_PETMA ..AVI.DTVAAG.AGFEK.MSMICI.LRSAY-------------
LGB2_LUPLU .ILKTIKEVVGAKWSEEL.SAWT.AYDELAIVIKKEMNDAA---
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

</pre>
</td></tr></table><p>

<a name="output.10"></a>
<h3>Output files for usage example 10</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
HBB_HUMAN  --------VHLTPEEKSAVTALWGKVN-VDEVGGEALGR-LLVVYPWTQRFFESFGDLST
HBB_HORSE  --------.Q.SG...A..L...D...-EE.........-............D......N
HBA_HUMAN  --------------VL.PADKTNV.AA-WGK..AH.GEYGaEALErmFlS.PTTKTYFpH
HBA_HORSE  --------------VL..ADKTNV.AA-WSK...H.GEYGaEALErmFlg.PTTKTYFpH
MYG_PHYCA  -------VLSEGEWQLVlHVWaKVEAD-.AGH.QDI.I.-.FKSh.E.lEK.DR.KH.K.
GLB5_PETMA pIVDTGSVAP.SAA..TKIRSa.AP.YSTY.TS.VDILvKFFTST.AA.E..PK.KG.T.
LGB2_LUPLU --------GA..ESQAAl.KSS.EEF.AnIPKHTHRFFILV.EIA.AAKDL.SFLKGT.E
HBB_HUMAN  --------VHLTPEEKSAVTALWGKVN-VDEVGGEALGR-LLVVYPWTQRFFESFGDLST
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
HBB_HUMAN  PDAVMGNPKVKAHGKKVLGAFSDGLAHLDNLKGTFATLSELHCDKLH--VDPENFRLLGN
HBB_HORSE  .G................HS.GE.VH..........A..........--...........
HBA_HUMAN  F.lSH.SAQ..G.....aD.LTNAV..V.DMPNALSA..D..aH..R--...V..K..SH
HBA_HORSE  F.lSH.SAQ........GD.LTLAVG...D.P.ALSN..D..aH..R--...V..K..SH
MYG_PHYCA  EAEMKASEDL.K..VT..T.LGAI.KkKGHHEAELKP.AQS.aT.Hk--IPIKYLEFISE
GLB5_PETMA A.QLKKSAD.RW.AERIIN.VN.AV.SM.DTEKmSMK.RD.SGKhak--SFQVDPQYFKv
LGB2_LUPLU VPQnNPELQAh.GKVFK.VYEAAIqlQVTGVVV.D...KN.GSVhVSKG.ADAH.pVVKE
HBB_HUMAN  PDAVMGNPKVKAHGKKVLGAFSDGLAHLDNLKGTFATLSELHCDKLH--VDPENFRLLGN
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
HBB_HUMAN  VLVCVLAHHFGKEFTPPVQAAYQKVVAGVANALAHKYH------
HBB_HORSE  ...V...R....D...EL..S.................------
HBA_HUMAN  C.LVT..A.LPA....A.H.SLD.FL.S.STV.TS..R------
HBA_HORSE  C.LST..V.LPND...A.H.SLD.FLSS.STV.TS..R------
MYG_PHYCA  AIIH..HSRHPGD.GADA.G.Mn.ALELFRKDI.A..kELGYQG
GLB5_PETMA LaAVIaDTVAAGDAGFEKlMSMICILlrS.Y-------------
LGB2_LUPLU AILKTIKEVV.AKWSEELnS.WTiAYDEL.IVIKkEMNDAA---
HBB_HUMAN  VLVCVLAHHFGKEFTPPVQAAYQKVVAGVANALAHKYH------
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

</pre>
</td></tr></table><p>

<a name="output.11"></a>
<h3>Output files for usage example 11</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
HBB_HUMAN  --------VHLTPEEksavtalwgkvn-vdevggealgr-llvvypwtqrffesfgdlst
HBB_HORSE  --------.Q.SG...a..l...d...-ee.........-............d......n
HBA_HUMAN  --------------Vl.padktnv.aa-wgk..ah.geygaealermfls.pttktyfph
HBA_HORSE  --------------Vl..adktnv.aa-wsk...h.geygaealermflg.pttktyfph
MYG_PHYCA  -------vLSEGEWQlvlhvwakvead-.agh.qdi.i.-.fksh.e.lek.dr.kh.k.
GLB5_PETMA pivdtgsvAP.SAA..tkirsa.ap.ysty.ts.vdilvkfftst.aa.e..pk.kg.t.
LGB2_LUPLU --------GA..ESQaal.kss.eef.anipkhthrffilv.eia.aakdl.sflkgt.e
HBB_HUMAN  --------VHLTPEEksavtalwgkvn-vdevggealgr-llvvypwtqrffesfgdlst

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
HBB_HUMAN  pdavmgnpkvkahgkkvlgafsdglahldnlkgtfatlselhcdklh--vdpenfrllgn
HBB_HORSE  .g................hs.ge.vh..........a..........--...........
HBA_HUMAN  f.lsh.saq..g.....ad.ltnav..v.dmpnalsa..d..ah..r--...v..k..sh
HBA_HORSE  f.lsh.saq........gd.ltlavg...d.p.alsn..d..ah..r--...v..k..sh
MYG_PHYCA  eaemkasedl.k..vt..t.lgai.kkkghheaelkp.aqs.at.hk--ipikylefise
GLB5_PETMA a.qlkksad.rw.aeriin.vn.av.sm.dtekmsmk.rd.sgkhak--sfqvdpqyfkv
LGB2_LUPLU vpqnnpelqah.gkvfk.vyeaaiqlqvtgvvv.d...kn.gsvhvskg.adah.pvvke
HBB_HUMAN  pdavmgnpkvkahgkkvlgafsdglahldnlkgtfatlselhcdklh--vdpenfrllgn

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
HBB_HUMAN  vlvcvlahhfgkeftppvqaayqkvvagvanalahkyh------
HBB_HORSE  ...v...r....d...el..s.................------
HBA_HUMAN  c.lvt..a.lpa....a.h.sld.fl.s.stv.ts..r------
HBA_HORSE  c.lst..v.lpnd...a.h.sld.flss.stv.ts..r------
MYG_PHYCA  aiih..hsrhpgd.gada.g.mn.alelfrkdi.a..kelgyqg
GLB5_PETMA laaviadtvaagdagfeklmsmicillrs.y-------------
LGB2_LUPLU ailktikevv.akwseelns.wtiaydel.ivikkemndaa---
HBB_HUMAN  vlvcvlahhfgkeftppvqaayqkvvagvanalahkyh------

</pre>
</td></tr></table><p>

<a name="output.12"></a>
<h3>Output files for usage example 12</h3>
<p><h3>File: globins.showalign</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
                   10        20        30        40        50        60
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx
GLB5_PETMA PIVDTGSVAPLSAAEKTKIRSAWAPVYSTYETSGVDILvKFFTST.AAqE..PK.KG.TT
HBA_HORSE  --------------VLS.ADKTNVKAA-WSKV.GHAGEYGaEALErMF.G.PTTKTYFpH
HBA_HUMAN  --------------VLSPADKTNVKAA-WGKV.AHAGEYGaEALErMF.S.PTTKTYFpH
HBB_HORSE  --------VQLSGEEKA.VLALWDKVN-EEEV.GEALG.-.LVVY.WTqR..DS.GD..N
HBB_HUMAN  --------VHLTPEEKS.VTALWGKVN-VDEV.GEALG.-.LVVY.WTqR..ES.GD..T
LGB2_LUPLU --------GALTESQAAlVKSSWEEFNANIPKHTHRFFILVLEIA.AAKDL.SFLKGT.E
MYG_PHYCA  -------VLSEGEWQLVlHVWAKVEAD-VAGH.QDILI.-.FKSH.ET.EK.DR.KH.KT
Consensus  xxxxxxxxxxxxxxxxxAxxxxxxxxxxxxxxGxxxxxRxLxxxxPxxLxFFxxFxxLSx

                   70        80        90       100       110       120
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx
GLB5_PETMA ADqLKK.AD.RW.AERIIN.VNDA.ASM.DTEKMSMK.RD.SGKhaK--SFQVDPQYFKV
HBA_HORSE  FD.SHG.AQ..A..KK.GD.LTLA.G.L.DLPGALSN.SD...H..R--.DPVN.KLLSH
HBA_HUMAN  FD.SHG.AQ..G..KK.aD.LTNA.A.V.DMPNALSA.SD...H..R--.DPVN.KLLSH
HBB_HORSE  PGaVMGNPK..A..KK..HSFGEG.H.L.NLKGTFAA.SE..cD..H--.DPEN.RLLGN
HBB_HUMAN  PDaVMGNPK..A..KK..G.FSDGLA.L.NLKGTFAT.SE..cD..H--.DPEN.RLLGN
LGB2_LUPLU VPqNNPELQAhAGKVFK.VYEAAIqLQVTGVVVTDAT.KN.GSVhVSKG.ADAH.PVVKE
MYG_PHYCA  EAEMKA.EDL.K..VT..T.LGAILKkKGHHEAELKP.AQS..T.HK--IPIKYLEFISE
Consensus  xxLxxxSxxVKxHGxxVLxAxxxxVxHxDxxxxxxxxLxxLHAxKLxxxVxxxxFxxxxx

                  130       140       150       160       170       180
           ----:----|----:----|----:----|----:----|----:----|----:----|
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx
GLB5_PETMA laAVIaDTVAAG.AGFEKlMSMICI.LRSAY-------------
HBA_HORSE  c.LST.AVHLPN..TPAVHASLD.F.SSVSTVLTS..R------
HBA_HUMAN  c.LVT.AAHLPAE.TPAVHASLD.F.ASVSTVLTS..R------
HBB_HORSE  V.VVV.ARHFGK..TPEL.ASYQ.VVAGVANALAH..H------
HBB_HUMAN  V.VCV.AHHFGKE.TPPV.AAYQ.VVAGVANALAH..H------
LGB2_LUPLU .ILKTIKEVVGAKWSEELnSAWTiAYDELAIVIKKEMNDAA---
MYG_PHYCA  .IIHV.HSRHPG..GADA.GAMN.A.ELFRKDIAA..KELGYQG
Consensus  ALxxxLxxxxxxDFxxxxQxxxxKxLxxxxxxxxxKYxxxxxxx

</pre>
</td></tr></table><p>







<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.
	If you wish to include the standard description of what data
	files are and how to use embossdata to inspect and retrieve
	them, use:
   -->

<b>showalign</b> reads in scoring matrices to determine the consesnus
sequence and to determine which matches are similar or not. 

<p>

<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>















<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="abiview.html">abiview</a></td><td>Reads ABI file and display the trace</td></tr>
<tr><td><a href="cirdna.html">cirdna</a></td><td>Draws circular maps of DNA constructs</td></tr>
<tr><td><a href="emma.html">emma</a></td><td>Multiple alignment program - interface to ClustalW program</td></tr>
<tr><td><a href="infoalign.html">infoalign</a></td><td>Information on a multiple sequence alignment</td></tr>
<tr><td><a href="lindna.html">lindna</a></td><td>Draws linear maps of DNA constructs</td></tr>
<tr><td><a href="pepnet.html">pepnet</a></td><td>Displays proteins as a helical net</td></tr>
<tr><td><a href="pepwheel.html">pepwheel</a></td><td>Shows protein sequences as helices</td></tr>
<tr><td><a href="plotcon.html">plotcon</a></td><td>Plots the quality of conservation of a sequence alignment</td></tr>
<tr><td><a href="prettyplot.html">prettyplot</a></td><td>Displays aligned sequences, with colouring and boxing</td></tr>
<tr><td><a href="prettyseq.html">prettyseq</a></td><td>Output sequence with translated ranges</td></tr>
<tr><td><a href="remap.html">remap</a></td><td>Display a sequence with restriction cut sites, translation etc</td></tr>
<tr><td><a href="seealso.html">seealso</a></td><td>Finds programs sharing group names</td></tr>
<tr><td><a href="showdb.html">showdb</a></td><td>Displays information on the currently available databases</td></tr>
<tr><td><a href="showfeat.html">showfeat</a></td><td>Show features of a sequence</td></tr>
<tr><td><a href="showseq.html">showseq</a></td><td>Display a sequence with features, translation etc</td></tr>
<tr><td><a href="sixpack.html">sixpack</a></td><td>Display a DNA sequence with 6-frame translation and ORFs</td></tr>
<tr><td><a href="textsearch.html">textsearch</a></td><td>Search sequence documentation text. SRS and Entrez are faster!</td></tr>
<tr><td><a href="tranalign.html">tranalign</a></td><td>Align nucleic coding regions given the aligned proteins</td></tr>
</table>

<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>
<!--
        Who has worked on the program in the past.
   -->

Gary Williams (gwilliam&nbsp;&copy;&nbsp;hgmp.mrc.ac.uk)
<br>
HGMP-RC, Genome Campus, Hinxton, Cambridge CB10 1SB, UK






<H2>
    History
</H2>

<!--
        What changes have been made.
   -->

Written (23 May 2001) - Gary Williams

<!--
<p>
[an error occurred while processing this directive]
-->











<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.












<H2>
    Comments
</H2>

<!--
	Any comments, requests etc. that people have about the program
   -->








</BODY>
</HTML>

