<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBASSY: DOMAINATRIX: sigscanlig documentation</TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">
[an error occurred while processing this directive]
<H2> sigscanlig documentation </H2>
[an error occurred while processing this directive]
<!-- END OF HEADER -->





<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     COMMAND-LINE ARGUMENTS    </a></b><br>    
<b> <a href="#8.0">8.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#9.0">9.0     NOTES                     </a></b><br>
<b> <a href="#10.0">10.0     DESCRIPTION               </a></b><br>
<b> <a href="#11.0">11.0   ALGORITHM                 </a></b><br>
<b> <a href="#12.0">12.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#13.0">13.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#14.0">14.0   AUTHORS                   </a></b><br>
<b> <a href="#15.0">15.0   REFERENCES                </a></b><br>

<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"><H2> 1.0   SUMMARY  </H2></a>
Search ligand-signature library & write hits (LHF file)

<p>

 <b>sigscanlig</b> is a generic tool for 




<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"><H2> 2.0   INPUTS & OUTPUTS          </H2></a>

<b>sigscanlig</b> reads a 


 
 




<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.
     Use "<b>sigscanlig</b> reads any normal sequence USAs." if
     appropriate.
-->
<a name="3.0"><H2> 3.0   INPUT FILE FORMAT         </H2></a>



<p>







<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"><H2> 4.0   OUTPUT FILE FORMAT         </H2></a>






<p>





<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml"
-->

<a name="5.0"><H2> 5.0   DATA FILES                </H2></a>
<b>sigscanlig</b> does not use a data file.







<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"><H2> 6.0   USAGE                     </H2></a>



<p>







<a name="7.0"><H2>7.0     COMMAND-LINE ARGUMENTS    </H2></a>
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->     
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" >
<tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-siginfiles]        dirlist    This option specifies the directory of
                                  signature files (input). A 'signature file'
                                  contains a sparse sequence signature
                                  suitable for use with the SIGSCAN and
                                  SIGSCANLIG programs. The files are generated
                                  by using SIGGEN and SIGGENLIG.
  [-dbsequences]       seqall     This option specifies the name of the
                                  database of query sequences.
   -sub                matrixf    This option specifies the scoring matrix to
                                  use, e.g. residue substitution matrix (1D
                                  signatures) or matrix of 3D:1D
                                  (environment:residue) scores (3D).
   -mode               menu       This option specifies the mode of scoring
                                  ligands. In 'Patch Score Mode' (mode 1) the
                                  score for a ligand is the mean of its patch
                                  scores. In 'Site Score Mode' the score for a
                                  ligand is the mean of its site scores. A
                                  site score is the mean of patch score for
                                  all patches for a given site.
   -gapo               float      This option specifies the gap insertion
                                  penalty. The gap insertion penalty is the
                                  score taken away when a gap is created. The
                                  best value depends on the choice of
                                  comparison matrix. The default value assumes
                                  you are using the EBLOSUM62 matrix for
                                  protein sequences, and the EDNAMAT matrix
                                  for nucleotide sequences.
   -gape               float      This option specifies the gap extension
                                  penalty. The gap extension penalty is added
                                  to the standard gap penalty for each base or
                                  residue in the gap. This is how long gaps
                                  are penalized. Usually you will expect a few
                                  long gaps rather than many short gaps, so
                                  the gap extension penalty should be lower
                                  than the gap penalty.
   -nterm              menu       This option specifies the N-terminal
                                  matching option. This determines how the
                                  first signature position is aligned to a
                                  sequence from the database.
  [-hitsdir]           outdir     This option specifies the directory of the
                                  LHF files (ligand hits files) (output). A
                                  'ligand hits file' contains database hits
                                  (sequences) with ligand classification
                                  information, in the LHF format (FASTA-like).
                                  The hits are putative ligand-binding
                                  sequences and are found from a search of a
                                  sequence database, in this case, by using
                                  SIGSCANLIG.
  [-aligndir]          outdir     This option specifies the name of the SAF
                                  (signature alignment file) (output).A
                                  'signature alignment file' contains one or
                                  more signnature-sequence alignments. The
                                  file is in DAF format (CLUSTAL-like) and is
                                  annotated with bibliographic information,
                                  either the domain family classification (for
                                  SIGSCAN output) or ligand classification
                                  (for SIGSCANLIG output). The files generated
                                  by SIGSCAN will contain a
                                  signature-sequence alignment for a single
                                  signature against a library of one or more
                                  sequences. The files generated by using
                                  SIGSCANLIG will contain a signature-sequence
                                  alignment for a single query sequence
                                  against a library of one or more signatures.
  [-resultsdir]        outdir     This option specifies the directory for the
                                  results files. A results file is generated
                                  for each query sequence and contains a list
                                  of hits (ligands), rank-ordered on the basis
                                  of score.

   Additional (Optional) qualifiers (* if not always prompted):
   -nhits              integer    This option specifies the maximum number of
                                  hits to output.
   -domax              toggle     This option specifies whether to consider a
                                  max. number of top-scoring hits only for
                                  each ligand.
*  -maxhits            integer    This option specifies the maximum number of
                                  top-scoring hits to consider for each
                                  ligand.

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-dbsequences" associated qualifiers
   -sbegin2            integer    Start of each sequence to be used
   -send2              integer    End of each sequence to be used
   -sreverse2          boolean    Reverse (if DNA)
   -sask2              boolean    Ask for begin/end/reverse
   -snucleotide2       boolean    Sequence is nucleotide
   -sprotein2          boolean    Sequence is protein
   -slower2            boolean    Make lower case
   -supper2            boolean    Make upper case
   -sformat2           string     Input sequence format
   -sdbname2           string     Database name
   -sid2               string     Entryname
   -ufo2               string     UFO features
   -fformat2           string     Features format
   -fopenfile2         string     Features file name

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write standard output
   -filter             boolean    Read standard input, write standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report deaths

</pre>
</pre>
</td></tr>
</table>
<p> 

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-siginfiles]<br>(Parameter 1)</td>
<td>This option specifies the directory of signature files (input). A 'signature file' contains a sparse sequence signature suitable for use with the SIGSCAN and SIGSCANLIG programs. The files are generated by using SIGGEN and SIGGENLIG.</td>
<td>Directory with files</td>
<td>./</td>
</tr>

<tr>
<td>[-dbsequences]<br>(Parameter 2)</td>
<td>This option specifies the name of the database of query sequences.</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-sub</td>
<td>This option specifies the scoring matrix to use, e.g. residue substitution matrix (1D signatures) or matrix of 3D:1D (environment:residue) scores (3D).</td>
<td>Comparison matrix file in EMBOSS data path</td>
<td>EBLOSUM62</td>
</tr>

<tr>
<td>-mode</td>
<td>This option specifies the mode of scoring ligands. In 'Patch Score Mode' (mode 1) the score for a ligand is the mean of its patch scores. In 'Site Score Mode' the score for a ligand is the mean of its site scores. A site score is the mean of patch score for all patches for a given site.</td>
<td><table><tr><td>1</td> <td><i>(Patch score mode)</i></td></tr><tr><td>2</td> <td><i>(Site score mode)</i></td></tr></table></td>
<td>1</td>
</tr>

<tr>
<td>-gapo</td>
<td>This option specifies the gap insertion penalty. The gap insertion penalty is the score taken away when a gap is created. The best value depends on the choice of comparison matrix. The default value assumes you are using the EBLOSUM62 matrix for protein sequences, and the EDNAMAT matrix for nucleotide sequences.</td>
<td>Floating point number from 1.0 to 100.0</td>
<td>10.0 for any sequence</td>
</tr>

<tr>
<td>-gape</td>
<td>This option specifies the gap extension penalty. The gap extension penalty is added to the standard gap penalty for each base or residue in the gap. This is how long gaps are penalized. Usually you will expect a few long gaps rather than many short gaps, so the gap extension penalty should be lower than the gap penalty.</td>
<td>Floating point number from 0.0 to 10.0</td>
<td>0.5 for any sequence</td>
</tr>

<tr>
<td>-nterm</td>
<td>This option specifies the N-terminal matching option. This determines how the first signature position is aligned to a sequence from the database.</td>
<td><table><tr><td>1</td> <td><i>(Align anywhere and allow only complete signature-sequence fit)</i></td></tr><tr><td>2</td> <td><i>(Align anywhere and allow partial signature-sequence fit)</i></td></tr><tr><td>3</td> <td><i>(Use empirical gaps only)</i></td></tr></table></td>
<td>1</td>
</tr>

<tr>
<td>[-hitsdir]<br>(Parameter 3)</td>
<td>This option specifies the directory of the LHF files (ligand hits files) (output). A 'ligand hits file' contains database hits (sequences) with ligand classification information, in the LHF format (FASTA-like). The hits are putative ligand-binding sequences and are found from a search of a sequence database, in this case, by using SIGSCANLIG.</td>
<td>Output directory</td>
<td>./</td>
</tr>

<tr>
<td>[-aligndir]<br>(Parameter 4)</td>
<td>This option specifies the name of the SAF (signature alignment file) (output).A 'signature alignment file' contains one or more signnature-sequence alignments. The file is in DAF format (CLUSTAL-like) and is annotated with bibliographic information, either the domain family classification (for SIGSCAN output) or ligand classification (for SIGSCANLIG output). The files generated by SIGSCAN will contain a signature-sequence alignment for a single signature against a library of one or more sequences. The files generated by using SIGSCANLIG will contain a signature-sequence alignment for a single query sequence against a library of one or more signatures.</td>
<td>Output directory</td>
<td>./</td>
</tr>

<tr>
<td>[-resultsdir]<br>(Parameter 5)</td>
<td>This option specifies the directory for the results files. A results file is generated for each query sequence and contains a list of hits (ligands), rank-ordered on the basis of score.</td>
<td>Output directory</td>
<td>./</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-nhits</td>
<td>This option specifies the maximum number of hits to output.</td>
<td>Any integer value</td>
<td>100</td>
</tr>

<tr>
<td>-domax</td>
<td>This option specifies whether to consider a max. number of top-scoring hits only for each ligand.</td>
<td>Toggle value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-maxhits</td>
<td>This option specifies the maximum number of top-scoring hits to consider for each ligand.</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>





<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="8.0"><H2> 8.0   KNOWN BUGS & WARNINGS     </H2></a>






<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="9.0"><H2> 9.0   NOTES                     </H2></a>

None.



<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="10.0"></a>
<H2> 10.0   DESCRIPTION               </H2>





<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="11.0"><H2> 11.0  ALGORITHM                 </H2></a>






<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
-->
<a name="12.0"><H2> 12.0  RELATED APPLICATIONS      </H2></a>

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="../../apps/contactcount.html">contactcount</a></td><td>Count specific versus non-specific contacts</td></tr>
<tr><td><a href="../structure/contacts.html">contacts</a></td><td>Generate intra-chain CON files from CCF files</td></tr>
<tr><td><a href="../domalign/domainalign.html">domainalign</a></td><td>Generate alignments (DAF file) for nodes in a DCF file</td></tr>
<tr><td><a href="../domalign/domainrep.html">domainrep</a></td><td>Reorder DCF file to identify representative structures</td></tr>
<tr><td><a href="../domainatrix/domainreso.html">domainreso</a></td><td>Remove low resolution domains from a DCF file</td></tr>
<tr><td><a href="../structure/interface.html">interface</a></td><td>Generate inter-chain CON files from CCF files</td></tr>
<tr><td><a href="libgen.html">libgen</a></td><td>Generate discriminating elements from alignments</td></tr>
<tr><td><a href="matgen3d.html">matgen3d</a></td><td>Generate a 3D-1D scoring matrix from CCF files</td></tr>
<tr><td><a href="../../apps/psiphi.html">psiphi</a></td><td>Phi and psi torsion angles from protein coordinates</td></tr>
<tr><td><a href="rocon.html">rocon</a></td><td>Generates a hits file from comparing two DHF files</td></tr>
<tr><td><a href="rocplot.html">rocplot</a></td><td>Performs ROC analysis on hits files</td></tr>
<tr><td><a href="../reconstruct/scorecmapdir.html">scorecmapdir</a></td><td>Contact scores for cleaned protein chain contact files</td></tr>
<tr><td><a href="../domalign/seqalign.html">seqalign</a></td><td>Extend alignments (DAF file) with sequences (DHF file)</td></tr>
<tr><td><a href="../domsearch/seqfraggle.html">seqfraggle</a></td><td>Removes fragment sequences from DHF files</td></tr>
<tr><td><a href="../domsearch/seqsearch.html">seqsearch</a></td><td>Generate PSI-BLAST hits (DHF file) from a DAF file</td></tr>
<tr><td><a href="../domsearch/seqsort.html">seqsort</a></td><td>Remove ambiguous classified sequences from DHF files</td></tr>
<tr><td><a href="../domsearch/seqwords.html">seqwords</a></td><td>Generates DHF files from keyword search of UniProt</td></tr>
<tr><td><a href="siggen.html">siggen</a></td><td>Generates a sparse protein signature from an alignment</td></tr>
<tr><td><a href="siggenlig.html">siggenlig</a></td><td>Generate ligand-binding signatures from a CON file</td></tr>
<tr><td><a href="sigscan.html">sigscan</a></td><td>Generate hits (DHF file) from a signature search</td></tr>
</table>








<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="13.0"><H2> 13.0  DIAGNOSTIC ERROR MESSAGES </H2></a>

None.







<!-- AUTHORS
     The main author first then all others. 
-->
<a name="14.0"><H2> 14.0  AUTHORS                   </H2></a>

Jon Ison (jison&nbsp;&copy;&nbsp;hgmp.mrc.ac.uk)
<br>
HGMP-RC, Genome Campus, Hinxton, Cambridge CB10 1SB, UK




<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"><H2> 14.0  REFERENCES                </H2></a>

Please cite the authors and EMBOSS.
<br><br>
<i>Rice P, Longden I and Bleasby A (2000) "EMBOSS - The European
Molecular Biology Open Software Suite"  Trends in Genetics,
15:276-278.

<p>
See also <a href="http://emboss.sourceforge.net/">http://emboss.sourceforge.net/</a></i>



<H3>14.1 Other useful references  </H3>
 

</BODY>
</HTML>
