<!--START OF HEADER - DON'T ALTER -->

<HTML>
<HEAD>
  <TITLE>
  EMBOSS: sirna
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="index.html" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">

sirna
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>



<!--END OF HEADER-->






<H2>
    Function
</H2>
Finds siRNA duplexes in mRNA

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<!-- 
	This is the main part of the document.
        This is what the user looks at to find out if this program will do what he wants. 
        Long description.
        Why it is useful.
        When it should/shouldn't be used.
        Algorithms
  -->

RNA interference, or RNAi, is a phenomenon in which double stranded RNA
(dsRNA) effects silencing of the expression of genes that are highly
homologous to either of the RNA strands in the duplex.  Gene silencing
in RNAi results from the degradation of mRNA sequences, and the effect
has been used to determine the function of many genes in Drosophilia, C. 
elegans, and many plant species. 

<p>

The duration of knockdown by siRNA can typically last for 7-10 days, and
has been shown to transfer to daughter cells.  Of further note, siRNAs
are effective at quantities much lower than alternative gene silencing
methodologies, including antisense and ribozyme based strategies. 

<p>
                       
Due to various mechanisms of antiviral response to long dsRNA, RNAi at
first proved more difficult to establish in mammalian species.  Then,
Tuschl, Elbashir, and others discovered that RNAi can be elicited very
effectively by well-defined 21-base duplex RNAs.  When these small
interfering RNA, or siRNA, are added in duplex form with a transfection
agent to mammalian cell cultures, the 21-base-pair RNA acts in concert
with cellular components to silence the gene with sequence homology to
one of the siRNA sequences. 

Strategies for the design of effective siRNA sequences have been
recently documented, most notably by Sayda Elbashir, Thomas Tuschl, et
al. 

<p>

Their studies of mammalian RNAi suggest that the most efficient
gene-silencing effect is achieved using double-stranded siRNA having a
19-nucleotide complementary region and a 2-nucleotide 3' overhang at
each end.  Current models of the RNAi mechanism suggest that the
antisense siRNA strand recognizes the specific gene target. 

<p>

In gene-specific RNAi, the coding region (CDS) of the mRNA is usually
targeted.  The search for an appropriate target sequence should begin
50-100 nucleotides downstream of the start codon.  UTR-binding proteins
and/or translation initiation complexes may interfere with the binding
of the siRNP endonuclease complex.  Tuschl, Elbashir et al.  say that
they have successfully used siRNAs targetting the 3' UTR. 


To avoid interference
from mRNA regulatory proteins, sequences in the 5' untranslated region
or near the start codon should not be targeted. 

<p>

A set of rules for the design of siRNA has been suggested <A href
="http://www.mpibpc.gwdg.de/abteilungen/100/105/sirna.html
">http://www.mpibpc.gwdg.de/abteilungen/100/105/sirna.html</a> based on
the work of Tuschl, Elbashir et al. 

<p>

They suggest searching for 23-nt sequence motif AA(N19)TT (N, any
nucleotide) and select hits with approx.  50% G/C-content (30% to 70%
has also worked in for them).  If no suitable sequences are found, the
search is extended using the motif NA(N21).  The sequence of the sense
siRNA corresponds to (N19)TT or N21 (position 3 to 23 of the 23-nt
motif), respectively.  In the latter case, they convert the 3' end of
the sense siRNA to TT. 

<p>

The rationale for this sequence conversion is to generate a symmetric
duplex with respect to the sequence composition of the sense and
antisense 3' overhangs.  The antisense siRNA is synthesized as the
complement to position 1 to 21 of the 23-nt motif.  Because position 1
of the 23-nt motif is not recognized sequence-specifically by the
antisense siRNA, the 3'-most nucleotide residue of the antisense siRNA,
can be chosen deliberately.  However, the penultimate nucleotide of the
antisense siRNA (complementary to position 2 of the 23-nt motif) should
always be complementary to the targeted sequence.  For simplifying
chemical synthesis, they always use TT.  

<p>

More recently, they preferentially select siRNAs corresponding to the
target motif NAR(N17)YNN, where R is purine (A, G) and Y is pyrimidine
(C, U).  The respective 21-nt sense and antisense siRNAs therefore begin
with a purine nucleotide and can also be expressed from pol III
expression vectors without a change in targeting site; expression of
RNAs from pol III promoters is only efficient when the first transcribed
nucleotide is a purine. 

<p>

They always design siRNAs with symmetric 3' TT overhangs, believing that
symmetric 3' overhangs help to ensure that the siRNPs are formed with
approximately equal ratios of sense and antisense target RNA-cleaving
siRNPs Please note that the modification of the overhang of the sense
sequence of the siRNA duplex is not expected to affect targeted mRNA
recognition, as the antisense siRNA strand guides target recognition. 
In summary, no matter what you do to your overhangs, siRNAs should still
function to a reasonable extent.  However, using TT in the 3' overhang
will always help your RNA synthesis company to let you know when you
accidentally order a siRNA sequences 3' to 5' rather than in the
recommended format of 5' to 3'.  

<p>

<b>sirna</b> reports both the sense and antisense siRNAs as 5' to 3'. 

<p>

<a href="http://www.xeragon.com/siRNA_support.html">Xeragon.com</a> also
suggest that choosing a 21 nt region of the mRNA with a GC content as
close as possible to 50% is a more important consideration than choosing
a target sequence that begins with AA.  They also suggest that a key
consideration in target selection is to avoid having more than three
guanosines in a row, since poly G sequences can hyperstack and form
agglomerates that potentially interfere with the siRNA silencing
mechanism.

<p>

siRNAs appear to effectively silence genes in more than 80% of cases. 
Current data indicate that there are regions of some mRNAs where gene
silencing does not work.  To help ensure that a given target gene is
silenced, it is advised that at least two target sequences as far apart
on the gene as possible be chosen. 

<p>

mRNA secondary structure does not appear to have a significant effect on
gene silencing. 

<p>

<b>sirna</b> looks at the feature table of the input mRNA sequence to
find the coding regions (CDS).  It will ignore the 5' UTR and the first
50 bases of the CDS.  It will assign a penalty of 2 points to any siRNA
in positions 51 to 100 in the CDS.  If there is no CDS in the feature
table, you can specify the CDS by using the command-line qualifier
'-sbegin' to indicate where the CDS should start.  If there is no CDS in
the feature table and you do not use the command-line qualifier
'-sbegin', then <b>sirna</b> will assume that the CDS region is not
known and will look for siRNAs in the whole of the sequence with no
penalties. 




<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with sirna</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>sirna </b>
Finds siRNA duplexes in mRNA
Input sequence(s): <b>tembl:hsfau</b>
Output report [hsfau.sirna]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>




<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Mandatory qualifiers:
  [-sequence]          seqall     Sequence database USA
  [-outfile]           report     Output data file

   Optional qualifiers:
   -[no]poliii         boolean    This option allows you to select the 21 base
                                  probes that start with a purine and so can
                                  be expressed from Pol III expression
                                  vectors.
   -aa                 boolean    This option allows you to select the 23 base
                                  regions that start with AA.

   Advanced qualifiers: (none)
   General qualifiers:
  -help                boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose


</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#f5f5ff">
<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Mandatory qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>Sequence database USA</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>Output data file</td>
<td>Report output file</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Optional qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-[no]poliii</td>
<td>This option allows you to select the 21 base probes that start with a purine and so can be expressed from Pol III expression vectors.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr>
<td>-aa</td>
<td>This option allows you to select the 23 base regions that start with AA.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFD0">
<th align="left" colspan=2>Advanced qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>


<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>sirna</b> reads any normal sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>

'tembl:hsfau' is a sequence entry in the example nucleic acid database 'tembl'
<p>
<p><h3>Database entry: tembl:hsfau</h3>
<table width="90%"><tr><td bgcolor="#FFDDFF">
<pre>
ID   HSFAU      standard; RNA; HUM; 518 BP.
XX
AC   X65923;
XX
SV   X65923.1
XX
DT   13-MAY-1992 (Rel. 31, Created)
DT   23-SEP-1993 (Rel. 37, Last updated, Version 10)
XX
DE   H.sapiens fau mRNA
XX
KW   fau gene.
XX
OS   Homo sapiens (human)
OC   Eukaryota; Metazoa; Chordata; Craniata; Vertebrata; Euteleostomi; Mammalia;
OC   Eutheria; Primates; Catarrhini; Hominidae; Homo.
XX
RN   [1]
RP   1-518
RA   Michiels L.M.R.;
RT   ;
RL   Submitted (29-APR-1992) to the EMBL/GenBank/DDBJ databases.
RL   L.M.R. Michiels, University of Antwerp, Dept of Biochemistry,
RL   Universiteisplein 1, 2610 Wilrijk, BELGIUM
XX
RN   [2]
RP   1-518
RX   MEDLINE; 93368957.
RA   Michiels L., Van der Rauwelaert E., Van Hasselt F., Kas K., Merregaert J.;
RT   " fau cDNA encodes a ubiquitin-like-S30 fusion protein and is expressed as
RT   an antisense sequences in the Finkel-Biskis-Reilly murine sarcoma virus";
RL   Oncogene 8:2537-2546(1993).
XX
DR   SWISS-PROT; P35544; UBIM_HUMAN.
DR   SWISS-PROT; Q05472; RS30_HUMAN.
XX
FH   Key             Location/Qualifiers
FH
FT   source          1..518
FT                   /chromosome="11q"
FT                   /db_xref="taxon:9606"
FT                   /organism="Homo sapiens"
FT                   /tissue_type="placenta"
FT                   /clone_lib="cDNA"
FT                   /clone="pUIA 631"
FT                   /map="13"
FT   misc_feature    57..278
FT                   /note="ubiquitin like part"
FT   CDS             57..458
FT                   /db_xref="SWISS-PROT:P35544"
FT                   /db_xref="SWISS-PROT:Q05472"
FT                   /gene="fau"
FT                   /protein_id="CAA46716.1"
FT                   /translation="MQLFVRAQELHTFEVTGQETVAQIKAHVASLEGIAPEDQVVLLAG
FT                   APLEDEATLGQCGVEALTTLEVAGRMLGGKVHGSLARAGKVRGQTPKVAKQEKKKKKTG
FT                   RAKRRMQYNRRFVNVVPTFGKKKGPNANS"
FT   misc_feature    98..102
FT                   /note="nucleolar localization signal"
FT   misc_feature    279..458
FT                   /note="S30 part"
FT   polyA_signal    484..489
FT   polyA_site      509
XX
SQ   Sequence 518 BP; 125 A; 139 C; 148 G; 106 T; 0 other;
     ttcctctttc tcgactccat cttcgcggta gctgggaccg ccgttcagtc gccaatatgc        60
     agctctttgt ccgcgcccag gagctacaca ccttcgaggt gaccggccag gaaacggtcg       120
     cccagatcaa ggctcatgta gcctcactgg agggcattgc cccggaagat caagtcgtgc       180
     tcctggcagg cgcgcccctg gaggatgagg ccactctggg ccagtgcggg gtggaggccc       240
     tgactaccct ggaagtagca ggccgcatgc ttggaggtaa agttcatggt tccctggccc       300
     gtgctggaaa agtgagaggt cagactccta aggtggccaa acaggagaag aagaagaaga       360
     agacaggtcg ggctaagcgg cggatgcagt acaaccggcg ctttgtcaac gttgtgccca       420
     cctttggcaa gaagaagggc cccaatgcca actcttaagt cttttgtaat tctggctttc       480
     tctaataaaa aagccactta gttcagtcaa aaaaaaaa                               518
//
</pre>
</td></tr></table><p>






<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
   -->
<p>

The output is a standard EMBOSS report file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <b>-rformat xxx</b>, where 'xxx' is replaced by
the name of the required format.  The available format names are: embl,
genbank, gff, pir, swiss, trace, listfile, dbmotif, diffseq, excel,
feattable, motif, regions, seqtable, simple, srs, table, tagseq

<p>

See:
<A href="http://www.uk.embnet.org/Software/EMBOSS/Themes/ReportFormats.html">
http://www.uk.embnet.org/Software/EMBOSS/Themes/ReportFormats.html</A>
for further information on report formats.

<p>

<p>

<b>sirna</b> 
outputs a report format file. The default format is 'table'

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: hsfau.sirna</h3>
<table width="90%"><tr><td bgcolor="#DDFFDD">
<pre>
########################################
# Program: sirna
# Rundate: Wed Nov 27 16:45:49 2002
# Report_format: table
# Report_file: hsfau.sirna
########################################

#=======================================
#
# Sequence: HSFAU     from: 1   to: 518
# HitCount: 26
#
# Selecting only regions suitable for PolIII expression vectors
# CDS region found in feature table starting at 57
#
#=======================================

  Start     End   Score    GC% Sense_siRNA Antisense_siRNA
    354     376   8.000   47.8 GAAGAAGACAGGUCGGGCUdTdT AGCCCGACCUGUCUUCUUCdTdT
    429     451   8.000   52.2 GAAGAAGGGCCCCAAUGCCdTdT GGCAUUGGGGCCCUUCUUCdTdT
    435     457   8.000   52.2 GGGCCCCAAUGCCAACUCUdTdT AGAGUUGGCAUUGGGGCCCdTdT
    279     301   7.000   56.5 AGUUCAUGGUUCCCUGGCCdTdT GGCCAGGGAACCAUGAACUdTdT
    280     302   7.000   56.5 GUUCAUGGUUCCCUGGCCCdTdT GGGCCAGGGAACCAUGAACdTdT
    308     330   7.000   43.5 AAGUGAGAGGUCAGACUCCdTdT GGAGUCUGACCUCUCACUUdTdT
    309     331   7.000   43.5 AGUGAGAGGUCAGACUCCUdTdT AGGAGUCUGACCUCUCACUdTdT
    348     370   7.000   43.5 GAAGAAGAAGAAGACAGGUdTdT ACCUGUCUUCUUCUUCUUCdTdT
    432     454   7.000   56.5 GAAGGGCCCCAAUGCCAACdTdT GUUGGCAUUGGGGCCCUUCdTdT
    274     296   5.000   47.8 GGUAAAGUUCAUGGUUCCCdTdT GGGAACCAUGAACUUUACCdTdT
    278     300   5.000   52.2 AAGUUCAUGGUUCCCUGGCdTdT GCCAGGGAACCAUGAACUUdTdT
    307     329   5.000   47.8 AAAGUGAGAGGUCAGACUCdTdT GAGUCUGACCUCUCACUUUdTdT
    353     375   5.000   52.2 AGAAGAAGACAGGUCGGGCdTdT GCCCGACCUGUCUUCUUCUdTdT
    375     397   5.000   60.9 GCGGCGGAUGCAGUACAACdTdT GUUGUACUGCAUCCGCCGCdTdT
    387     409   5.000   52.2 GUACAACCGGCGCUUUGUCdTdT GACAAAGCGCCGGUUGUACdTdT
    434     456   5.000   56.5 AGGGCCCCAAUGCCAACUCdTdT GAGUUGGCAUUGGGGCCCUdTdT
    314     336   4.000   56.5 GAGGUCAGACUCCUAAGGUdTdT ACCUUAGGAGUCUGACCUCdTdT
    392     414   4.000   56.5 ACCGGCGCUUUGUCAACGUdTdT ACGUUGACAAAGCGCCGGUdTdT
    428     450   4.000   56.5 AGAAGAAGGGCCCCAAUGCdTdT GCAUUGGGGCCCUUCUUCUdTdT
    123     145   3.000   52.2 GAUCAAGGCUCAUGUAGCCdTdT GGCUACAUGAGCCUUGAUCdTdT
    487     509   3.000   34.8 AAAAGCCACUUAGUUCAGUdTdT ACUGAACUAAGUGGCUUUUdTdT
    488     510   3.000   34.8 AAAGCCACUUAGUUCAGUCdTdT GACUGAACUAAGUGGCUUUdTdT
    108     130   2.000   56.5 GGAAACGGUCGCCCAGAUCdTdT GAUCUGGGCGACCGUUUCCdTdT
    128     150   2.000   56.5 AGGCUCAUGUAGCCUCACUdTdT AGUGAGGCUACAUGAGCCUdTdT
    252     274   2.000   60.9 AGUAGCAGGCCGCAUGCUUdTdT AAGCAUGCGGCCUGCUACUdTdT
    443     465   1.000   34.8 AUGCCAACUCUUAAGUCUUdTdT AAGACUUAAGAGUUGGCAUdTdT

#---------------------------------------
#---------------------------------------
</pre>
</td></tr></table><p>


<p>

The siRNAs are reported in order of best score first.

<p>

<b>sirna</b> reports both the sense and antisense siRNAs as 5' to 3'. 




<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.

	If you wish to include the standard description of what data
	files are and how to use embossdata to inspect and retrieve
	them, use:
<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>

   -->

None.














<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

<ol>
<li>Elbashir, S. M., et al. (2001a). Duplexes of 21-nucleotide RNAs mediate RNA interference in
    mammalian cell culture. Nature 411: 494-498.
<li>
    Elbashir, S. M., W. Lendeckel and T. Tuschl (2001b). RNA interference is mediated by 21 and 22
    nt RNAs. Genes & Dev. 15: 188-200.
</ol>








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

It is assumed that the input sequence is mRNA.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr><td><a href="banana.html">banana</a></td><td>Bending and curvature plot in B-DNA</td></tr>
<tr><td><a href="btwisted.html">btwisted</a></td><td>Calculates the twisting in a B-DNA sequence</td></tr>
<tr><td><a href="chaos.html">chaos</a></td><td>Create a chaos game representation plot for a sequence</td></tr>
<tr><td><a href="compseq.html">compseq</a></td><td>Counts the composition of dimer/trimer/etc words in a sequence</td></tr>
<tr><td><a href="dan.html">dan</a></td><td>Calculates DNA RNA/DNA melting temperature</td></tr>
<tr><td><a href="freak.html">freak</a></td><td>Residue/base frequency table or plot</td></tr>
<tr><td><a href="isochore.html">isochore</a></td><td>Plots isochores in large DNA sequences</td></tr>
<tr><td><a href="wordcount.html">wordcount</a></td><td>Counts words of a specified size in a DNA sequence</td></tr>
</table>

<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>
<!--
        Who has worked on the program in the past.
   -->

This application was written by Gary Williams <a
href="mailto:gwilliam@hgmp.mrc.ac.uk">(gwilliam@hgmp.mrc.ac.uk)</a>
















<H2>
    History
</H2>

<!--
        What changes have been made.
   -->

Written (November 2002) - Gary Williams.













<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.












<H2>
    Comments
</H2>

<!--
	Any comments, requests etc. that people have about the program
   -->








</BODY>
</HTML>

