<HTML>

<HEAD>
  <TITLE>
  EMBOSS: wordfinder
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
wordfinder
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<H2>
    Function
</H2>
Match large sequences against one or more other sequences

<H2>
    Description
</H2>

This is a rough and ready local alignment program for large
sequences. The reason it is rough and ready is that wordmatch is used
to find all the word matches between the first sequence and another
sequence. Then by calculating the highest score for a diagonal we can
then use this as the centre point for a Smith-Waterman type
calculation of a width given by the user.  So a narrow diagonal
Smith-Waterman is calculated hence the results will be rough but due
to the space saving much larger sequences can be aligned.

<H2>
    Usage
</H2>
[an error has occurred processing this directive]


<H2>
    Command line arguments
</H2>
<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-asequence]         seqset     Sequence set filename and optional format,
                                  or reference (input USA)
  [-bsequence]         seqall     Sequence(s) filename and optional format, or
                                  reference (input USA)
   -gapopen            float      [10.0 for any sequence type] Gap opening
                                  penalty (Number from 0.000 to 1000.000)
   -gapextend          float      [0.5 for any sequence type] Gap extension
                                  penalty (Number from 0.000 to 10.000)
  [-outfile]           align      [*.wordfinder] Output alignment file name

   Additional (Optional) qualifiers:
   -datafile           matrixf    [EBLOSUM62 for protein, EDNAFULL for DNA]
                                  This is the scoring matrix file used when
                                  comparing sequences. By default it is the
                                  file 'EBLOSUM62' (for proteins) or the file
                                  'EDNAFULL' (for nucleic sequences). These
                                  files are found in the 'data' directory of
                                  the EMBOSS installation.
   -width              integer    [16] Alignment width (Integer 1 or more)
   -wordlen            integer    [6] Word length for initial matching
                                  (Integer 3 or more)
   -limitmatch         integer    [0] Limiting match score (zero for no limit)
                                  (Integer 0 or more)
   -limitalign         integer    [0] Limiting alignment length (zero for no
                                  limit) (Integer 0 or more)
   -lowmatch           integer    [0] Limiting match score (zero for no limit)
                                  (Integer 0 or more)
   -lowalign           integer    [0] Limiting alignment length (zero for no
                                  limit) (Integer 0 or more)
   -errorfile          outfile    [wordfinder.error] Error file to be written
                                  to

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-asequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-bsequence" associated qualifiers
   -sbegin2            integer    Start of each sequence to be used
   -send2              integer    End of each sequence to be used
   -sreverse2          boolean    Reverse (if DNA)
   -sask2              boolean    Ask for begin/end/reverse
   -snucleotide2       boolean    Sequence is nucleotide
   -sprotein2          boolean    Sequence is protein
   -slower2            boolean    Make lower case
   -supper2            boolean    Make upper case
   -sformat2           string     Input sequence format
   -sdbname2           string     Database name
   -sid2               string     Entryname
   -ufo2               string     UFO features
   -fformat2           string     Features format
   -fopenfile2         string     Features file name

   "-outfile" associated qualifiers
   -aformat3           string     Alignment format
   -aextension3        string     File name extension
   -adirectory3        string     Output directory
   -aname3             string     Base file name
   -awidth3            integer    Alignment width
   -aaccshow3          boolean    Show accession number in the header
   -adesshow3          boolean    Show description in the header
   -ausashow3          boolean    Show the full USA in the alignment
   -aglobal3           boolean    Show the full sequence in alignment

   "-errorfile" associated qualifiers
   -odirectory         string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write standard output
   -filter             boolean    Read standard input, write standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>

<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-asequence]<br>(Parameter 1)</td>
<td>Sequence set filename and optional format, or reference (input USA)</td>
<td>Readable set of sequences</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-bsequence]<br>(Parameter 2)</td>
<td>Sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-gapopen</td>
<td>Gap opening penalty</td>
<td>Number from 0.000 to 1000.000</td>
<td>10.0 for any sequence type</td>
</tr>

<tr>
<td>-gapextend</td>
<td>Gap extension penalty</td>
<td>Number from 0.000 to 10.000</td>
<td>0.5 for any sequence type</td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 3)</td>
<td>Output alignment file name</td>
<td>Alignment output file</td>
<td><i>&lt;*&gt;</i>.wordfinder</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-datafile</td>
<td>This is the scoring matrix file used when comparing sequences. By default it is the file 'EBLOSUM62' (for proteins) or the file 'EDNAFULL' (for nucleic sequences). These files are found in the 'data' directory of the EMBOSS installation.</td>
<td>Comparison matrix file in EMBOSS data path</td>
<td>EBLOSUM62 for protein<br>EDNAFULL for DNA</td>
</tr>

<tr>
<td>-width</td>
<td>Alignment width</td>
<td>Integer 1 or more</td>
<td>16</td>
</tr>

<tr>
<td>-wordlen</td>
<td>Word length for initial matching</td>
<td>Integer 3 or more</td>
<td>6</td>
</tr>

<tr>
<td>-limitmatch</td>
<td>Limiting match score (zero for no limit)</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr>
<td>-limitalign</td>
<td>Limiting alignment length (zero for no limit)</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr>
<td>-lowmatch</td>
<td>Limiting match score (zero for no limit)</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr>
<td>-lowalign</td>
<td>Limiting alignment length (zero for no limit)</td>
<td>Integer 0 or more</td>
<td>0</td>
</tr>

<tr>
<td>-errorfile</td>
<td>Error file to be written to</td>
<td>Output file</td>
<td>wordfinder.error</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

</table>


<H2>
    Input file format
</H2>

<b>wordfinder</b> reads two sequence USAs of the same type.  They must
both be protein or both be nucleic acid sequences. 

<p>

[an error has occurred processing this directive]

<H2>
    Output file format
</H2>

<p>

The output is a standard EMBOSS alignment file. 

<p>

The results can be output in one of several styles by using the
command-line qualifier <b>-aformat xxx</b>, where 'xxx' is replaced by
the name of the required format.  Some of the alignment formats can cope
with an unlimited number of sequences, while others are only for pairs
of sequences.  

<p>

The available multiple alignment format names are: unknown, multiple,
simple, fasta, msf, trace, srs

<p>

The available pairwise alignment format names are: pair, markx0, markx1,
markx2, markx3, markx10, srspair, score

<p>

See:
<A href="http://emboss.sf.net/docs/themes/AlignFormats.html">
http://emboss.sf.net/docs/themes/AlignFormats.html</A>
for further information on alignment formats.

<p>

<p>

The output alignment is in <b>simple</b> format by default.

<p>

[an error has occurred processing this directive]

<p>

The file 'wordfinder.error' will contain any errors that occured
during the program.  This may be that wordmatch could not find any
matches hence no suitable start point is found for the smith-waterman
calculation. 


<H2>
    Data files
</H2>

For protein sequences EBLOSUM62 is used for the substitution
    matrix. For nucleotide sequence, EDNAMAT is used. Others can be specified.

<p>
<p>
EMBOSS data files are distributed with the application and stored
in the standard EMBOSS data directory, which is defined
by the EMBOSS environment variable EMBOSS_DATA.

<p>

To see the available EMBOSS data files, run:
<p>
<pre>
% embossdata -showall
</pre>
<p>
To fetch one of the data files (for example 'Exxx.dat') into your
current directory for you to inspect or modify, run:

<pre>

% embossdata -fetch -file Exxx.dat

</pre>
<p>

Users can provide their own data files in their own directories.
Project specific files can be put in the current directory, or for
tidier directory listings in a subdirectory called
".embossdata". Files for all EMBOSS runs can be put in the user's home
directory, or again in a subdirectory called ".embossdata".

<p>
The directories are searched in the following order:

<ul>
   <li> . (your current directory)
   <li> .embossdata (under your current directory)
   <li> ~/ (your home directory)
   <li> ~/.embossdata
</ul>
<p>

<H2>
    Notes
</H2>

The time this program takes to do an alignment depends very much on the
word size.  For short sequences a short word size (e.g.  4) can make it
take a very long time.  Large word sizes (e.g.  30) for sequences that
are very similar give a very quick result.  The default of 16 should
give reasonable fast alignments. 

<p>

Because it does a Smith & Waterman alignment (albeit in a narrow region
around the diagonal shown to be the 'best' by a word match), this
program can use huge amounts of memory if the sequences are large. 

<p>

Because the alignment is made within a narrow area each side of the
'best' diagonal, if there are sufficient indels between the two
sequences, then the path of the Smith & Waterman alignment can wander
outside of this area.  Making the width larger can avoid this problem,
but you then use more memory. 

<p>

The longer the sequences and the wider the specified alignment width,
the more memory will be used. 

<p>

If the program terminates due to lack of memory you can try the
following:

<p>

Run the UNIX command 'limit' to see if your stack or memory usage have
been limited and if so, run 'unlimit', (e.g.: '% unlimit stacksize').


<H2>
    References
</H2>

None.

<H2>
    Warnings
</H2>

None.

<H2>
    Diagnostic Error Messages
</H2>

None.

<H2>
    Exit status
</H2>

It always exits with a status of 0.

<H2>
    Known bugs
</H2>

None.

<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr>
<td><a href="matcher.html">matcher</a></td>
<td>Finds the best local alignments between two sequences</td>
</tr>

<tr>
<td><a href="seqmatchall.html">seqmatchall</a></td>
<td>All-against-all comparison of a set of sequences</td>
</tr>

<tr>
<td><a href="supermatcher.html">supermatcher</a></td>
<td>Match large sequences against one or more other sequences</td>
</tr>

<tr>
<td><a href="water.html">water</a></td>
<td>Smith-Waterman local alignment</td>
</tr>

<tr>
<td><a href="wordmatch.html">wordmatch</a></td>
<td>Finds all exact matches of a given size between 2 sequences</td>
</tr>

</table>

<H2>
    Author(s)
</H2>


Ian Longden (il&nbsp;&copy;&nbsp;sanger.ac.uk)
<br>
Sanger Institute, Wellcome Trust Genome Campus, Hinxton,
Cambridge, CB10 1SA, UK.                      


<H2>
    History
</H2>


<H2>
    Target users
</H2>
This program is intended to be used by everyone and everything, from naive users to embedded scripts.



<H2>
    Comments
</H2>
None

</BODY>
</HTML>

