<?xml version="1.0" encoding="iso-8859-1"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <title>PHYLIPNEW Application</title> 
    <meta name="keywords" lang="en-uk" content="EMBOSS, European
    Molecular Biology Open Source Software Suite, bioinformatics, 
    EBI, European Bioinformatics Institute, Molecular Biology, 
    Computational Biology, Open Source, Software, Database" />
  </head>

<body bgcolor="#ffffff" style="background-color=#ffffff; body-background-color=#ffffff">
<link rel="stylesheet" href="../../stylesheets/EMBOSS.css" type="text/css" >

<!-- BEGIN PAGE HEADING -->
<div id="container">
<div id="header">  <a href="../index.html"> <img border="0" src="../../images/emboss.jpg" > </a> </div>
<!--#include virtual="../../header.inc" -->
<!-- END PAGE HEADING -->
    
<!--BEGIN MAIN CONTENT -->
<h1>PHYLIPNEW Applications</h1>

<hr> 



The PHYLIPNEW programs are EMBOSS conversions of the programs in Joe
Felsenstein's PHYLIP package, version 3.69.

<p>
The PHYLIPNEW versions of these programs all have the prefix "f" to
distinguish them from the original programs. 

<p>

<h3><A NAME="phylipnew">Applications</A> in the <a
href="ftp://emboss.open-bio.org/pub/EMBOSS/">current phylipnew release</a></h3>

<table border cellpadding=4 bgcolor="#FFFFF0">

<tr>
<th>Program name</th>
<th>Description</th>
</tr>

<tr>
<td><a href="fclique.html">fclique</a></td>
<td>
Largest clique program
</td>
</tr>
<tr>
<td><a href="fconsense.html">fconsense</a></td>
<td>
Majority-rule and strict consensus tree
</td>
</tr>
<tr>
<td><a href="fcontml.html">fcontml</a></td>
<td>
Gene frequency and continuous character Maximum Likelihood
</td>
</tr>
<tr>
<td><a href="fcontrast.html">fcontrast</a></td>
<td>
Continuous character Contrasts
</td>
</tr>
<tr>
<td><a href="fdiscboot.html">fdiscboot</a></td>
<td>
Bootstrapped discrete sites algorithm
</td>
</tr>
<tr>
<td><a href="fdnacomp.html">fdnacomp</a></td>
<td>
DNA compatibility algorithm
</td>
</tr>
<tr>
<td><a href="fdnadist.html">fdnadist</a></td>
<td>
Nucleic acid sequence Distance Matrix program
</td>
</tr>
<tr>
<td><a href="fdnainvar.html">fdnainvar</a></td>
<td>
Nucleic acid sequence Invariants method
</td>
</tr>
<tr>
<td><a href="fdnaml.html">fdnaml</a></td>
<td>
Estimates nucleotide phylogeny by maximum likelihood
</td>
</tr>
<tr>
<td><a href="fdnamlk.html">fdnamlk</a></td>
<td>
Estimates nucleotide phylogeny by maximum likelihood
</td>
</tr>
<tr>
<td><a href="fdnamove.html">fdnamove</a></td>
<td>
Interactive DNA parsimony
</td>
</tr>
<tr>
<td><a href="fdnapars.html">fdnapars</a></td>
<td>
DNA parsimony algorithm
</td>
</tr>
<tr>
<td><a href="fdnapenny.html">fdnapenny</a></td>
<td>
Penny algorithm for DNA
</td>
</tr>
<tr>
<td><a href="fdollop.html">fdollop</a></td>
<td>
Dollo and polymorphism parsimony algorithm
</td>
</tr>
<tr>
<td><a href="fdolmove.html">fdolmove</a></td>
<td>
Interactive Dollo or Polymorphism Parsimony
</td>
</tr>
<tr>
<td><a href="fdolpenny.html">fdolpenny</a></td>
<td>
Penny algorithm Dollo or polymorphism
</td>
</tr>
<tr>
<td><a href="fdrawgram.html">fdrawgram</a></td>
<td>
Plots a cladogram- or phenogram-like rooted tree diagram
</td>
</tr>
<tr>
<td><a href="fdrawtree.html">fdrawtree</a></td>
<td>
Plots an unrooted tree diagram
</td>
</tr>
<tr>
<td><a href="ffactor.html">ffactor</a></td>
<td>
Multistate to binary recoding program
</td>
</tr>
<tr>
<td><a href="ffitch.html">ffitch</a></td>
<td>
Fitch-Margoliash and Least-Squares Distance Methods
</td>
</tr>
<tr>
<td><a href="ffreqboot.html">ffreqboot</a></td>
<td>
Bootstrapped genetic frequencies algorithm
</td>
</tr>
<tr>
<td><a href="fgendist.html">fgendist</a></td>
<td>
Compute genetic distances from gene frequencies
</td>
</tr>
<tr>
<td><a href="fkitsch.html">fkitsch</a></td>
<td>
Fitch-Margoliash method with contemporary tips
</td>
</tr>
<tr>
<td><a href="fmix.html">fmix</a></td>
<td>
Mixed parsimony algorithm
</td>
</tr>
<tr>
<td><a href="fmove.html">fmove</a></td>
<td>
Interactive mixed method parsimony
</td>
</tr>
<tr>
<td><a href="fneighbor.html">fneighbor</a></td>
<td>
Phylogenies from distance matrix by N-J or UPGMA method
</td>
</tr>
<tr>
<td><a href="fpars.html">fpars</a></td>
<td>
Discrete character parsimony
</td>
</tr>
<tr>
<td><a href="fpenny.html">fpenny</a></td>
<td>
Penny algorithm, branch-and-bound
</td>
</tr>
<tr>
<td><a href="fproml.html">fproml</a></td>
<td>
Protein phylogeny by maximum likelihood
</td>
</tr>
<tr>
<td><a href="fpromlk.html">fpromlk</a></td>
<td>
Protein phylogeny by maximum likelihood
</td>
</tr>
<tr>
<td><a href="fprotdist.html">fprotdist</a></td>
<td>
Protein distance algorithm
</td>
</tr>
<tr>
<td><a href="fprotpars.html">fprotpars</a></td>
<td>
Protein parsimony algorithm
</td>
</tr>
<tr>
<td><a href="frestboot.html">frestboot</a></td>
<td>
Bootstrapped restriction sites algorithm
</td>
</tr>
<tr>
<td><a href="frestdist.html">frestdist</a></td>
<td>
Distance matrix from restriction sites or fragments
</td>
</tr>
<tr>
<td><a href="frestml.html">frestml</a></td>
<td>
Restriction site maximum Likelihood method
</td>
</tr>
<tr>
<td><a href="fretree.html">fretree</a></td>
<td>
Interactive tree rearrangement
</td>
</tr>
<tr>
<td><a href="fseqboot.html">fseqboot</a></td>
<td>
Bootstrapped sequences algorithm
</td>
</tr>
<tr>
<td><a href="fseqbootall.html">fseqbootall</a></td>
<td>
Bootstrapped sequences algorithm
</td>
</tr>
<tr>
<td><a href="ftreedist.html">ftreedist</a></td>
<td>
Distances between trees
</td>
</tr>
<tr>
<td><a href="ftreedistpair.html">ftreedistpair</a></td>
<td>
Distances between two sets of trees
</td>
</tr>
</table>

<hr>
</div>
</BODY>
</HTML>
