<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBASSY: MATGEN3D documentation</TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">
<!--#include virtual="/apps/header1.inc" -->
<H2> MATGEN3D documentation
<!--#include virtual="/apps/header2.inc" -->
<!-- END OF HEADER -->





<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>

<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<br><br><br><H2> 1.0   SUMMARY  </H2>
<!--#include file="inc/matgen3d.ione" -->

<p>
Generates a 3D-1D scoring matrix from CCF files (clean coordinate files).




<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<br><br><br><H2> 2.0   INPUTS & OUTPUTS          </H2>
MATGEN3D reads a DCF file (domain classification file) and a directory of domain CCF files (clean coordinate files) which have been processed by using PDBPLUS so that they contain solvent accessibility and secondary structure information.  The directory must contain a CCF file for the first domain from each family represented in the DCF file.  A matrix of 3D:1D scores (environment:residue scoring matrix), reflecting the probability of the amino acids occuring in different tertiary environments, is calculated from the CCF files of the first domain from each family only. The path of the CCF files is specified by the user and the file extensions is specified in the ACD file. Two log files of informative messages are also written. 


 
 

<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.
     Use "<b>matgen3d</b> reads any normal sequence USAs." if
     appropriate.
-->
<a name="3.0"></a>
<br><br><br><H2> 3.0   INPUT FILE FORMAT         </H2>
The format of domain CCF files is described in the <a href="domainer.html">DOMAINER documentation</a>.
<br>
The format of the DCF file (domain classification file) is described in the <a href="scopparse.html">SCOPPARSE documentation</a> and the <a href="cathparse.html">CATHPARSE documentation</a>.




<p>

<!--#include file="inc/matgen3d.input" -->





<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<br><br><br><H2> 4.0   OUTPUT FILE FORMAT         </H2>
The matrix of 3D:1D scores (environment:residue scoring matrix, Figure 1) follows the standard EMBOSS format.  Single letter amino acid codes are column labels and environments are row labels.  The environment labels are strings of 2 characters, beginning with AA, AB, AC, through to AZ, BA, BB etc.  The final row and column have the label '*' and give default substitution values (the minimum from the entire matrix).  In the example shown (Figure 1), only two environments AA through AX are defined but only AA to AB are given scores owing to the sparse input data for this example (typically, all environments would receive scores).





<p>

<!--#include file="inc/matgen3d.output" -->



<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml"
-->

<a name="5.0"></a>
<br><br><br><H2> 5.0   DATA FILES                </H2>
MATGEN3D does not use any data files.







<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<br><br><br><H2> 6.0   USAGE                     </H2>


<H3> 6.1   COMMAND LINE ARGUMENTS </H3>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" >
<tr><td>
<pre>
<!--#include file="inc/matgen3d.ihelp" --></pre>
</pre>
</td></tr>
</table>
<p> 

<!--#include file="inc/matgen3d.itable" -->


<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of MATGEN3D is shown below.
<!--#include file="inc/matgen3d.usage" -->




<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<br><br><br><H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
The CCF files read <B>must</B> contain secondary structure and solvent accessibility data.  These can be added to the file by using <a href="pdbplus.html">PDBPLUS</a> 





<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<br><br><br><H2> 8.0   NOTES                     </H2>


<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<tr>
<td><b> Domain classification file (for SCOP)</b></td>
<td> DCF format (EMBL-like). </td>
<td> Classification and other data for domains from SCOP.  </td>
<td> <a href="scopparse.html">SCOPPARSE</a> </td>
<td> Domain sequence information can be added to the file by using DOMAINSEQS.  </td>
</tr>
<tr>
<td><b> Domain classification file (for CATH)</b></td>
<td> DCF format (EMBL-like). </td>
<td> Classification and other data for domains from CATH.  </td>
<td> <a href="cathparse.html">CATHPARSE</a> </td>
<td> Domain sequence information can be added to the file by using DOMAINSEQS.  </td>
</tr>
<tr>
<td><b> Clean coordinate file (for domain)</b></td>
<td> CCF format (EMBL-like).</td>
<td> Protein coordinate and derived data for a single domain from SCOP or CATH. The data are 'cleaned-up': self-consistent and error-corrected. </td>
<td> <a href="domainer.html">DOMAINER</a></td>
<td> Records for residue solvent accessibility and secondary structure are added to the file by using PDBPLUS.</td>
</tr>
</table>




<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="9.0"></a>
<br><br><br><H2> 9.0   DESCRIPTION               </H2>
MATGEN3D generates a matrix of 3D:1D scores which give the probability of finding a certain amino acid residue (1D) in a certain environment in space (3D).  The environments are defined on the basis of secondary structure and solvent accessibility. The 3D:1D scores are calculated from the first domain only from each family represented in the DCF file (input).  This ensures the scores are not biased to any particular family of proteins. 





<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<br><br><br><H2> 10.0  ALGORITHM                 </H2>

<B><Environment definitions. </B>
<I><b>Environment definitions will be described here. </b></I>




<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
-->
<a name="11.0"></a>
<br><br><br><H2> 11.0  RELATED APPLICATIONS      </H2>

<!--#include file="inc/matgen3d.isee" -->







<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="12.0"></a>
<br><br><br><H2> 12.0  DIAGNOSTIC ERROR MESSAGES </H2>

None.







<!-- AUTHORS
     The main author first then all others. 
-->
<a name="13.0"></a>
<br><br><br><H2> 13.0  AUTHORS                   </H2>
Waqas Awan
<br>

Jon Ison <a href="mailto:jison@ebi.ac.uk">(jison@ebi.ac.uk)</a>
<br>

The European Bioinformatics Institute 
Wellcome Trust Genome Campus 
Cambridge CB10 1SD 
UK 


<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  REFERENCES                </H2>

<!--#include file="inc/embossref.ihtml" -->

<H3>14.1 Other useful references  </H3>
 

</BODY>
</HTML>
