<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBASSY: SIGGENLIG documentation</TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">
<!--#include virtual="/apps/header1.inc" -->
<H2> SIGGENLIG documentation
<!--#include virtual="/apps/header2.inc" -->
<!-- END OF HEADER -->





<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>

<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<br><br><br><H2> 1.0   SUMMARY  </H2>
Generates ligand-binding signatures from a CON file (contacts file) of residue-ligand contacts.

<!--#include file="inc/siggenlig.ione" -->


<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<br><br><br><H2> 2.0   INPUTS & OUTPUTS          </H2>
SIGGENLIG reads a CON file of residue-ligand contacts generated by using SITES and a directory of CCF files (clean coordinate files) containing a CCF file for each protein or domain in the CON file.  One or more signature files, each containing a ligand-binding signature, are generated for <i>each</i> ligand-binding site in the CON file.  The user specifies whether 1D (sequence) or 3D (structural) signatures are generated and whether they are 'full-length' (signature corresponds to entire ligand-binding site) or 'patch' (signature corresponds to part of ligand-binding site).  For 3D signatures, the environment definition is specified and for patch signatures, a 'Minimum patch size' and 'Maximum gap distance' are specified. A 'Window size' is specified for all signatures. The paths of the CCF files (input) and signature files (output) are specified by the user and the file extensions are specified in the ACD file.  A log file is also written.



<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.
     Use "<b>siggenlig</b> reads any normal sequence USAs." if
     appropriate.
-->
<a name="3.0"></a>
<br><br><br><H2> 3.0   INPUT FILE FORMAT         </H2>

The format of the CON file (contacts file) is described in <a href="sites.html">SITES documentation</a>.
The format of the CCF files is described in <a href="pdbparse.html">PDBPARSE documentation</a> (proteins) and the <a href="domainer.html">DOMAINER documentation</a> (domains).

<p>
<!--#include file="inc/siggenlig.input" -->





<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<br><br><br><H2> 4.0   OUTPUT FILE FORMAT         </H2>

The output file (Figure 1) uses the standard signature file format and is described in the <a href="siggen.html">SIGGEN documentation</a>.  For the ligand-binding signatures generated by SIGGENLIG, however, four additional lines of bibliographic information taken from the CON (input) file are written to a signature file.  The records have the following meaning:

<UL>
<LI> ID - identifier codes: (1) PDB: 4-character PDB identifier code. (2) DOM: 7-character domain identifier code from SCOP or CATH. (3) LIG: Ligand identifier (an abbreviation of its full name).
<LI> DE - full name of the ligand, see <a href="hetparse.html">HETPARSE</a> documentation. 
<LI> IS - ligand-binding site information: (1) SN: Site number.  This uniquely identifies a ligand-binding site in the CON file generated by running SITES. (2) NS: The total number of binding sites for a given ligand.
<LI> IP - path information:  (1) PN: Patch number.  The sequential order (from the N-terminus) of the patch in the ligand-binding site. (2) NP: Number of patches. The total number of patches in this ligand-binding site.  (3) MP: Minimum patch size.  (4) MG: Maximum gap distance.  MP and MG were specified in the generation of the signature (see 'Algorithm' below). 
</UL>


In addition, where 3D signatures are generated, the following records have different meanings than for 1D signatures (e.g. those generated by using <a href="siggen.html">SIGGEN</a>):
<UL>
<LI> IN - Informative line about signature position. The number of different observed amino acid <i>environments</i> (rather than residue identities) is given after 'NRES'
<LI> AA - The identifier of an <i>environment</i> (rather than residue) seen in this position and the frequency of its occurence are delimited by ';'.
</UL> 







<p>

<!--#include file="inc/siggenlig.output" -->



<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml"
-->

<a name="5.0"></a>
<br><br><br><H2> 5.0   DATA FILES                </H2>
SIGGENLIG does not use any data files.



<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<br><br><br><H2> 6.0   USAGE                     </H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" >
<tr><td>
<pre>
<!--#include file="inc/siggenlig.ihelp" --></pre>
</pre>
</td></tr>
</table>

<a name="6.1"></a>
<H3>6.1     COMMAND-LINE ARGUMENTS    </H3>
<!--#include file="inc/siggenlig.itable" -->

<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of SIGGENLIG is shown below.

<!--#include file="inc/siggenlig.usage" -->


<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<br><br><br><H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
<B>Window size</B>
<br>The user is prompted for a window size that is used for every position in the signature.  Likely this is not optimal. A future implementation will provide a range of methods for generating values of window size depending upon the alignment (window size is identified by the WSIZ record in the signature output file).


<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<br><br><br><H2> 8.0   NOTES                     </H2>

<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<tr>
<td><b> Clean coordinate file (for domain)</b></td>
<td> CCF format (EMBL-like).</td>
<td> Protein coordinate and derived data for a single domain from SCOP or CATH. The data are 'cleaned-up': self-consistent and error-corrected. </td>
<td> <a href="domainer.html">DOMAINER</a></td>
<td> Records for residue solvent accessibility and secondary structure are added to the file by using PDBPLUS.</td>
</tr>
<tr>
<td><b> Clean coordinate file (for domain)</b></td>
<td> CCF format (EMBL-like).</td>
<td> Protein coordinate and derived data for a single domain from SCOP or CATH. The data are 'cleaned-up': self-consistent and error-corrected. </td>
<td> <a href="domainer.html">DOMAINER</a></td>
<td> Records for residue solvent accessibility and secondary structure are added to the file by using PDBPLUS.</td>
</tr>
<tr>
<td><b> Contact file (residue-ligand contacts)</b></td>
<td> CON format (EMBL-like.) </td>
<td> Residue-ligand contact data for a protein or a domain from SCOP or CATH. </td>
<td> <a href="sites.html">SITES</a> </td>
<td> N.A. </td>
</tr>
<tr>
<td><b>Signature file </b></td>
<td> SIG format </td>
<td> Contains a sparse sequence signature suitable for use with the SIGSCAN program. Contains a sparse sequence signature. </td>
<td> <a href="SIGGEN.html">SIGGEN</a>, <a href="libgen.html">LIBGEN</a> </td>
<td> The files are generated by using SIGGENLIG. </td>
</tr>
</table>





<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="9.0"></a>
<br><br><br><H2> 9.0   DESCRIPTION               </H2>
A protein or a single domain of a protein may contain one or more ligand-binding sites.  SIGGENLIG provides an automated means to generate signatures of ligand-binding from a CON file (contacts file) of residue-ligand contacts.  The signatures generated may be of two types, 1D or 3D.  1D signatures represent protein sites as residue identitites whereas 3D signatures represent sites as residue environments in space.



<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<br><br><br><H2> 10.0  ALGORITHM                 </H2>

The user specifies whether 1D (sequence) or 3D (structural) signatures are generated and whether they are 'full-length' (signature corresponds to entire ligand-binding site) or 'patch' (signature corresponds to part of ligand-binding site).  For 3D signatures, the environment definition is specified and for patch signatures, a 'Minimum patch size' and 'Maximum gap distance' are specified. A 'Window size' is specified for all signatures. 

<br><br><b>Definition of full-length and patch signatures</b><br>
For each ligand-binding site represented in the CON file (input), one or more signatures are generated as follows: (1) a single 'full-length' signature incorporating all residue positions that contact the ligand plus intervening gaps, or (2) one or more signatures corresponding to 'patches' of residue positions.

<br>
 A patch is a set of residues that are near-neighbours in sequence and is described by two user-defined parameters as follows. (1) Minimum patch size; the minimum number of contact positions that must be incorporated for a patch to be defined.   (2) Maximum gap distance.  The maximum allowable gap (residues) between two residue in a patch.  If two contact residues are further than this distance apart in sequence, they would not belong to the same patch.

<br><br><B>Environment definitions. </B>
<br>
See <a href="matgen3d.html">MATGEN3d</a> documentation for environment definitions.


<br><br><b>Naming of output files.</b>
The naming convention of the signature (output) files is as follows:<br>

<i>Ligand identifier</i>.<i>Site number</i>.<i>F</i> <b>or</b> <i>P</i>.<i>Patch number</i>-<i>Total patches</i>.<i>PDB identifier</i>.<i>Domain identifier</i>. 

<UL>
<LI>The <i>ligand identifier</i> is a 3-letter code that uniquely identifies the ligand.
<LI>The <i>site number</i> uniquely identifies a ligand-binding site in the CON file generated by running SITES.
<LI><i>F</i> signifies a 'Full-length' and <i>P</i> a 'Patch' signature. 
<LI><i>Patch number</i> is the sequential order (from the N-terminus) of the patch in the ligand-binding site. 
<LI><i>Total patches</i> is the total number of patches for this ligand-binding site.
The patch number and total number of patches are only specified for a 'patch'.
<LI>PDB identifier code.
<LI>Domain identifier code. Only specified if the binding site is mapped to SCOP.
</UL>

For example, 
<br>
101.1.P.1-1.1cs4.d1cs4a_.sig





<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
-->
<a name="11.0"></a>
<br><br><br><H2> 11.0  RELATED APPLICATIONS      </H2>

<!--#include file="inc/siggenlig.isee" -->







<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="13.0"></a>
<br><br><br><H2> 13.0  DIAGNOSTIC ERROR MESSAGES </H2>

None.







<!-- AUTHORS
     The main author first then all others. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  AUTHORS                   </H2>

<!--#include file="inc/jison.address" -->

<br><br>
Waqas Awan <a href="mailto:wawan@rfcgr.mrc.ac.uk">(mblades@rfcgr.mrc.ac.uk)</a>
<br>MRC Rosalind Franklin Centre for Genomics Research, Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK


<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  REFERENCES                </H2>

<!--#include file="inc/embossref.ihtml" -->



<H3>14.1 Other useful references  </H3>
 

</BODY>
</HTML>

