<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBASSY: SIGSCAN documentation. </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include virtual="/apps/header1.inc" -->
<H2> sigscan documentation
<!--#include virtual="/apps/header2.inc" -->

<!-- END OF HEADER -->




<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>


<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<br><br><br><H2> 1.0   SUMMARY  </H2>
Generates a DHF (domain hits file) of hits (sequences) from scanning a signature against a sequence database.
<!--#include file="inc/sigscan.ione" -->



<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<br><br><br><H2> 2.0   INPUTS & OUTPUTS          </H2>
SIGSCAN reads a signature from a protein signature file, scans the signature against a protein sequence database and generates a DHF file (domain hits file) of hits to database sequences and a DAF file (domain alignment file) of corresponding signature-sequence alignments. The names of the signature file, DHF file and DAF file are provided by the user. The user specifies a maximum number of high-scoring hits that will be generated.


<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.
     Use "<b>sigscan</b> reads any normal sequence USAs." if
     appropriate.
-->



<a name="3.0"></a>
<br><br><br><H2> 3.0   INPUT FILE FORMAT         </H2>
The format of the signature file is described in <a href="siggen.html">SIGGEN documentation</a>.

<!--#include file="inc/sigscan.input" -->






<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<br><br><br><H2> 4.0   OUTPUT FILE FORMAT         </H2>

<b>DHF file (domain hits file)</b>
<br>The format of the DHF file (domain hits file) of hit sequences generated by SIGSCAN (Figure 1)
is described fully in <a href="seqsearch.html">SEQSEARCH documentation</a> and only summarised here.
The file contains two lines per hit, the first is a description of the hit in 16 text tokens delimited by '^'. The second line contains the protein sequence.
The first 4 tokens refer to the hit (sequence) itself, the tokens are 
<UL>
<LI>(i) Accession number
<LI>(ii)  Database code, 
<LI>(iii - iv) Start and end positions of the hit relative to the full length sequence. 
</UL>

<br>The next 9 tokens refer to the domain family, superfamily etc for which the signature was derived and are as follows: 
<UL>
<LI>(v)  Type of domain (one of 'SCOP' or 'CATH'), 
<LI>(vi)  SCOP or CATH domain identifier.  
<LI>(vii) SCOP or CATH node unique identifier, e.g. SCOP Family Sunid.
<LI>(viii) Domain class.  Textual description of the 'Class' (SCOP and CATH domains). 
<LI>(ix) Domain architecture. Textual description of the 'Architecture' (CATH only). 
<LI>(x) Domain topology. Textual description of the 'Topology' (CATH only).  
<LI>(xi) Domain fold. Textual description of the 'Fold' (SCOP domains only).  
<LI>(xii) Domain superfamily. Textual description of the 'Superfamily' (SCOP and CATH domains).  
<LI>(xiii) Domain family. Textual description of the 'Fold' (SCOP only). 
</UL>

<br>The next 4 tokens refer to the hit, specifically, information about the search result as follows: 
<UL>
<LI>(xiv) Model type.  The type of model that was used to generate the hit. For DHF files generated by using SIGSCAN a value of SPARSE (sparse protein signature) is given.  Several other values are possible, however, see <a href="seqsearch.html">SEQSEARCH documentation</a>.
<LI>(xv) SC - Score of hit from search algoritm  (not written by SIGSCAN).
<LI>(xvi) P-value of hit (not written by SIGSCAN).
<LI>(xvii) E-value of hit (not written by SIGSCAN).
</UL>



<br><br>
<b>DAF file (domain alignment file)</b>
<br>
The format of the DAF file (domain alignment file, Figure 2) generated by SIGSCAN 
is described fully in <a href="domainalign.html">DOMAINALIGN documentation</a> and
is only summarised here. 

<br>
It conforms to EMBOSS "simple" multiple sequence alignment format and includes domain classification records (in comment lines beginning with '#') for the node for which the signature was generated. The classification records are TY (domain type, either SCOP or CATH), CL (class), FO (fold), SF (superfamily) and FA (family).  For CATH domains, AR (architecture) and TP (topology) may also be given.  A <b>unique identifier</b> for the node is given after <b>SI</b>. 
<br>There are multiple blocks that contain the accession numbers, positions and aligned sequences. An accession number is given for each hit.  The positions are the start and end residue positions of the appropriate section of sequence. The sequence uses '-' as a gap character.  A 'SIGNATURE' line is given as a markup line underneath the sequence (signature positions are marked with a '*').


<!--#include file="inc/sigscan.output" -->



<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml" 
-->

<a name="5.0"></a>
<br><br><br><H2> 5.0   DATA FILES                </H2>
SIGSCAN requires a residue substitution matrix.

<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<br><br><br><H2> 6.0   USAGE                     </H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" >
<tr><td>
<pre>
<!--#include file="inc/sigscan.ihelp" --></pre>
</pre>
</td></tr>
</table>


<H3> 6.1   COMMAND LINE ARGUMENTS </H3>
 
<!--#include file="inc/sigscan.itable" -->

<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of sigscan is shown below.

<!--#include file="inc/sigscan.usage" -->

<!-- <br><br>
The signature file /test_data/54894.sig was scanned against the swissprot database /test_data/swsmall to generate a DHF file (database hits) called /test_data/54894.dhf and a DAF file (signature-sequence alignments) called /test_data/54894.daf.  The output files contained no more than 100 hits in total.  When aligning and scoring the signature to a protein sequence, the residue substitution matrix EBLOSUM62, a gap insertion penalty of 10 and a gap extension penalty of 0.5 was used.  The first signature position was permitted to be aligned anywhere in the sequence without attracing a gap penalty and only complete signature-sequence fits were allowed, i.e. every signature position had to be aligned or otherwise the sequence was rejected (sequences smaller than the sum of the gap distances in the signature were rejected).
<br><br>


<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<br><br><br><H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
None.








<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<br><br><br><H2> 8.0   NOTES                     </H2>
SIGSCAN does not generate p-values or E-values.  DHF files of hits for which p-values or E-values are calculated may be generated by using <a href="libscan.html">LIBSCAN </a>. LIBSCAN provides searches for sparse protein signatures as well as various types of hidden Markov models and other profiles. 

<br><br>In the case a signature file is generated by hand, it is essential that the 
gap data given is listed in order of increasing gap size (see <a href="siggen.html">SIGGEN documentation </a>).


<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<td><b>Domain hits file </b></td>
<td> DHF format (FASTA-like). </td>
<td> Database hits (sequences) with domain classification information.  The hits are relatives to a SCOP or CATH family (or other node in the structural hierarchies) and are found from a search of a discriminating element (e.g. a protein signature, hidden Markov model, simple frequency matrix, Gribskov profile or Hennikoff profile) against a sequence database. </td>
<td> <a href="seqsearch.html">SEQSEARCH</a> (hits retrieved by PSIBLAST). <a href="sigscan.html">SIGSCAN</a> (hits retrieved by sparse protein signature). <a href="libscan.html">LIBSCAN</a> (hits retrieved by various types of HMM and profile). </td>
<td> N.A. </td>
</tr>
<tr>
<td><b>Domain alignment file </b></td>
<td> DAF format (CLUSTAL-like). </td>
<td> Sequence alignment of domains belonging to the same SCOP or CATH family (or other node in the structural hierarchies).  The file is annotated with domain family classification information.</td>
<td> <a href="domainalign.html">DOMAINALIGN</a> (structure-based sequence alignment of domains of known structure). </td>
<td> DOMAINALIGN alignments can be extended with sequence relatives (of unknown structure) to the family in question by using SEQALIGN. </td>
</tr>
<tr>
<td><b> Hits file</b></td>
<td> Text file of classified hits </td>
<td> A list of hits (e.g. from a prediction method) that are classified and rank-ordered on the basis of score, p-value, E-value etc.  </td>
<td> <a href="rocon.html">ROCON</a> and <a href="libscan.html">LIBSCAN</a> (hits from searches of a discriminating element (hidden Markov model, profile or signature) against a sequence database).   </td>
<td> ROCPLOT is run on the files to perform Receiver Operator Characteristic (ROC) analysis on the hits.  </td>
</tr>
<tr>
<td><b>Signature file </b></td>
<td> SIG format </td>
<td> Contains a sparse sequence signature suitable for use with the SIGSCAN program. Contains a sparse sequence signature. </td>
<td> <a href="siggen.html">SIGGEN</a>, <a href="siggenlig.html">SIGGENLIG</a>, <a href="libgen.html">LIBGEN</a> </td>
<td> The files are generated by using SIGGEN. </td>
</tr>
</table>



<!--#include file="inc/sigscan.history" -->
<!--#include file="inc/sigscan.comment" -->

<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="9.0"></a>
<br><br><br><H2> 9.0   DESCRIPTION               </H2>
See Blades et al., Ison et al. and Daniel et al. for a description of protein signatures and their application.


<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<br><br><br><H2> 10.0  ALGORITHM                 </H2>
The algorithm is based on approach first described in Daniel et al (1999) that was applied to the definition of protein families (Ison et al, 2000) and later to automatically-generated signatures (Blades et al, 2005).



<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
-->

<a name="11.0"></a>
<br><br><br><H2> 11.0  RELATED APPLICATIONS      </H2>

<!--#include file="inc/sigscan.isee" -->



<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="12.0"></a>
<br><br><br><H2> 12.0  DIAGNOSTIC ERROR MESSAGES </H2>
None.



<!-- AUTHORS
     The main author first then all others. 
-->
<a name="13.0"></a>
<br><br><br><H2> 13.0  AUTHORS                   </H2>
Jon Ison <a href="mailto:jison@rfcgr.mrc.ac.uk">(jison@rfcgr.mrc.ac.uk)</a>
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK




<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  REFERENCES                </H2>
<!--#include file="inc/embossref.ihtml" -->

Automatic generation and evaluation of sparse protein signatures for families of protein 
structural domains. MJ Blades, JC Ison, R Ranasinghe, and JBC Findlay. Protein Science. 2005 (accepted)

<br><br>
A key residues approach to the definition of protein families and analysis
of sparse family signatures.  JC Ison, AJ Bleasby, MJ Blades, SC Daniel, 
JH Parish, JBC Findlay.  PROTEINS: Structure, Function & Genetics.  2000, 
40:330-341
<br><br>
Alignment of a sparse protein signature with protein sequences: application
to fold prediction for three small globulins.  SC Daniel, JH Parish, 
JC Ison, MJ Blades & JBC Findlay.  FEBS Letters.  1999, 459:349-352. 





<H3>14.1 Other useful references  </H3>


</BODY>
</HTML>



