<!--START OF HEADER - DON'T ALTER -->

<HTML>
<HEAD>
  <TITLE>
  EMBOSS: ehmmindex
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
ehmmindex
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<!--END OF HEADER-->






<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Create a binary SSI index for an HMM database
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<!-- 
	This is the main part of the document.
        This is what the user looks at to find out if this program will do what he wants. 
        Long description.
        Why it is useful.
        When it should/shouldn't be used.
  -->
EMBASSY HMMER is a suite of application wrappers to the original hmmer v2.3.2 applications written by Sean Eddy. hmmer v2.3.2 must be installed on the same system as EMBOSS and the location of the hmmer executables must be defined in your path for EMBASSY HMMER to work.

<p><b>Usage:</b>
<br>ehmmindex [options] database

<p>hmmindex is a utility that creates a binary SSI ('squid sequence index' format) index for an HMM database file called <database>. The new index file is named database.ssi. An SSI index file is required for hmmfetch to work, and also for the PVM implementation of hmmpfam.













<H2>
    Algorithm
</H2>

<!-- 
        Algorithms
	  Pseudocode/Vague description/Handwaving/web links to useful pages
  -->

Please read the Userguide.pdf distributed with the original HMMER and included in the EMBASSY HMMER distribution under the DOCS directory. 













<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with ehmmindex</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ehmmindex myhmmso </b>
Create a binary SSI index for an HMM database.

hmmindex -- create SSI index for an HMM database
HMMER 2.3.2 (Oct 2003)
Copyright (C) 1992-2003 HHMI/Washington University School of Medicine
Freely distributed under the GNU General Public License (GPL)
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
HMM file:                 ./myhmmso
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

Determining offsets for ./myhmmso, please be patient...
Complete.
HMM file:       ./myhmmso
SSI index:      ./myhmmso.ssi
# of HMMS:      3
HMM names:      3
HMM accessions: 2

/shared/software/bin/hmmindex ./myhmmso


</pre></td></tr></table><p>
<p>
<a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>
Where possible, the same command-line qualifier names and parameter order is used as in the original hmmer.  There are however several unavoidable differences and these are clearly documented in the "Notes" section below.
<p>More or less all options documented as "expert" in the original hmmer user guide are given in ACD as "advanced" options (-options must be specified on the command-line in order to be prompted for a value for them).  
<p>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Create a binary SSI index for an HMM database.
Version: EMBOSS:6.2.0

   Standard (Mandatory) qualifiers:
  [-database]          infile     HMM database.

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers: (none)
   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-database]<br>(Parameter 1)</td>
<td>infile</td>
<td>HMM database.</td>
<td>Input file</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>
<H3>Alignment and sequence formats </H3>
Input and output of alignments and sequences is limited to the formats that the original hmmer supports. These include stockholm, SELEX, MSF, Clustal, Phylip and A2M /aligned FASTA (alignments) and FASTA, GENBANK, EMBL, GCG, PIR (sequences).  It would be fairly straightforward to adapt the code to support all EMBOSS-supported formats. 

<H3>Compressed input files </H3>
Automatic processing of gzipped files is not supported. 
<p>


<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>ehmmindex</b> reads any normal sequence USAs.

<p>






<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
include file="inc/reportformats.ihtml"
   -->

<b>ehmmindex</b> 
outputs a graph to the specified graphics device. 
outputs a report format file. The default format is ...


<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: myhmmso.ssi</h3>
<p>This file contains non-printing characters and so cannot be displayed here.
<p><h3>File: myhmmso</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
HMMER2.0  [2.3.2]
NAME  rrm
LENG  77
ALPH  Amino
RF    no
CS    no
MAP   yes
COM   /shared/software/bin/hmmbuild -n rrm --pbswitch 1000 --archpri 0.850000 --idlevel 0.620000 --swentry 0.500000 --swexit 0.500000 --wgsc -A -F myhmms ../../data/hmmnew/rrm.sto
COM   /shared/software/bin/hmmcalibrate --mean 350.000000 --num 5000 --sd 350.000000 --seed 1 ../ehmmbuild-ex4-keep/myhmms
NSEQ  90
DATE  Thu Jan  7 16:51:40 2010
CKSUM 8325
XT      -8455     -4  -1000  -1000  -8455     -4  -8455     -4 
NULT      -4  -8455
NULE     595  -1558     85    338   -294    453  -1158    197    249    902  -1085   -142    -21   -313     45    531    201    384  -1998   -644 
EVD   -45.860321   0.213107
HMM        A      C      D      E      F      G      H      I      K      L      M      N      P      Q      R      S      T      V      W      Y    
         m-&gt;m   m-&gt;i   m-&gt;d   i-&gt;m   i-&gt;i   d-&gt;m   d-&gt;d   b-&gt;m   m-&gt;e
          -16      *  -6492
     1  -1084    390  -8597  -8255  -5793  -8424  -8268   2395  -8202   2081  -1197  -8080  -8115  -8020  -8297  -7789  -5911   1827  -7525  -7140     1
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378    -16      * 
     2  -2140  -3785  -6293  -2251   3226  -2495   -727   -638  -2421   -545   -675  -5146  -5554  -4879  -1183  -2536  -1928    267     76   3171     2
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378      *      * 
     3  -2542    458  -8584  -8273  -6055  -8452  -8531   2304  -8255   -324    101  -8104  -8170  -8221  -8440  -7840  -5878   3145  -7857  -7333     3
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378      *      * 
     4  -1505  -5144  -1922   -558  -1842   2472  -3303  -2213   1099  -5160  -4233    372  -4738   -530   1147    168    498  -4766  -5327  -1476     4
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378      *      * 
     5  -3724  -5184    300  -3013  -1655   1803  -3353  -5245  -1569  -2686  -4276   3495  -1963  -1331  -1054  -1472  -3664  -4803  -5369     -2     5
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378      *      * 
     6  -1569  -6106  -8967  -8363    555  -8531  -7279    654  -8092   2953    -94  -8220  -7908  -1643  -7682  -7771  -6460    -59  -6191  -6284     6
     -   -151   -504    230     45   -380    399    101   -621    211   -470   -713    278    399     48     91    360    113   -364   -299   -254 
     -   -178  -3113 -12684  -1600   -578   -701  -1378      *      * 
     7   -409  -5130   -215  -2987  -1709   -956    690  -5188   -395  -5144  -4224    729   3054  -2862  -3409    354   1293  -1381  -5321  -4644    13
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378      *      * 
     8  -3674  -5118  -1004    639    420  -4652    176  -2050    404  -1039   -935     16   1755    168    147   -275    198  -1472   1889   1977    14
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378      *      * 
     9   -408  -5134   2415   1299   -950    -66   -767  -1296  -2889  -1843  -4224   1084   -968  -1439  -1854    540   -314  -2304  -5320    -60    15
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378      *      * 
    10    586   1804  -6294   -631  -1627  -1671  -4374   1029  -2223   -162   1172  -5147  -5554  -1870  -5058  -2327   1741   1687  -4242    687    16
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11642 -12684   -894  -1115   -701  -1378      *      * 
    11  -2134  -5144    845  -1187  -1652  -1667  -3303  -5216   -513   -801  -4233   1026  -1873   -543   -619    575   2956  -4766  -5327  -4644    17


<font color=red>  [Part of this file has been deleted for brevity]</font>

     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11253 -12295   -894  -1115   -701  -1378      *      * 
   279  -7207  -7306  -8076  -6588  -8459  -7223  -5448  -7982  -1500  -7531  -6953  -6369  -7277  -5081   4236  -7139  -6862  -7777  -7053  -7277   454
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11253 -12295   -894  -1115   -701  -1378      *      * 
   280   -694   -163  -5922  -5286  -1204  -2048   -610   1082  -1800   1434  -2618  -4776   2951  -4509  -4688  -1216  -1648  -2829    202     21   455
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -  -3168 -11253   -171   -894  -1115   -701  -1378      *      * 
   281  -1412  -2132  -2007  -2293  -4366   3113  -2847  -4225  -3107  -4377  -3503   1660  -2881  -2661  -3396    961  -1821  -3134  -4516  -4119   456
     -   -150   -489    232     42   -382    400    104   -627    211   -465   -722    274    393     51     95    359    116   -370   -296   -245 
     -  -2121   -637  -2975   -831  -1191  -6099    -21      *      * 
   282   -968  -1818  -1787  -1351  -3112    953  -1494  -2818  -1122  -2911  -2044  -1365  -2340  -1133   1510   1816   2121  -2205  -3137  -2649   459
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -   -228  -7899  -2816   -894  -1115  -4964    -47      *      * 
   283    840  -1663   -994    969   1159    503   -604  -1413   -325  -1594   -814   -688  -1996   -267   1103   -851   -755  -1179   2900  -1437   460
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -9  -7938  -8980   -894  -1115    -89  -4060      *      * 
   284  -3257  -4642   -697  -2590  -1218   -252  -2907  -4655  -1306  -2353   -529    482  -1607  -2459  -1398   2112   2745  -4246  -4848  -4187   461
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11183 -12226   -894  -1115   -186  -3045      *      * 
   285   2163    763  -1619  -5296   2250  -2060  -4007   1241  -4891   -489    484  -4781   -226  -4515  -4692   -678  -1688   -813    264  -3530   462
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11253 -12295   -894  -1115   -701  -1378      *      * 
   286   -268   -329   -158    917   -541  -1990    350  -4851   1273  -1075    388  -1130    233    840    993   -602    801   -595  -4964   -857   463
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11253 -12295   -894  -1115   -701  -1378      *      * 
   287    109   -243    672   2304  -5103  -4283    488  -4854  -1317  -2269   -656   -492  -1519   2679   -655   -618  -3248  -4404  -4965  -1114   464
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11253 -12295   -894  -1115   -701  -1378      *      * 
   288   1312   1294  -6215  -5593   -206  -1244  -4339   2188  -5201   1409    395  -5091  -5478  -4828  -5009  -4538  -3794   1162  -4187  -3846   465
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -    -23 -11253  -6022   -894  -1115   -701  -1378      *      * 
   289  -3562    799  -5767  -2054  -1235  -2075    318    138    237   2164   1713  -1454  -5145  -1272   -730  -4172  -1640   1071  -3865    -34   466
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11231 -12273   -894  -1115  -1470   -646      *      * 
   290     73   1351   -674   1236  -1549  -2008   1350  -4834   1049  -2498  -3851   1801  -4356   1813   -115   -223  -1582  -1052  -4945  -4262   467
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11231 -12273   -894  -1115   -369  -2147      *      * 
   291  -1739   -320    777  -2654  -1419  -2051   4360  -4707  -1358  -2412   -689  -1300  -4399   -224    537    531   -289  -2010  -4905  -1057   468
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -11253 -12295   -894  -1115   -701  -1378      *      * 
   292  -3345  -4494   -233   -332   -563  -1986  -3051    333     99   1063  -3616  -3072   2953  -1026  -1490   -943  -1528  -1070  -4753  -4151   469
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -10815 -11857   -894  -1115   -701  -1378      *      * 
   293  -6409  -5751  -7614  -7636   2593  -7311  -4003  -5084  -7219   -150   -151  -6210  -7172   -849  -6723  -6510  -6299  -1387   4881   2807   470
     -   -149   -500    233     43   -381    399    106   -626    210   -466   -720    275    394     45     96    359    117   -369   -294   -249 
     -     -1 -10749 -11791   -894  -1115   -701  -1378      *      * 
   294  -4057  -3817  -6415  -5791   3203  -1638  -4541   1679  -5412    765   1434  -5333  -5617  -4930  -5182  -4791  -3987   1226    750  -3959   471
     -      *      *      *      *      *      *      *      *      *      *      *      *      *      *      *      *      *      *      *      * 
     -      *      *      *      *      *      *      *      *      0 
//
</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.

	If you wish to include the standard description of what data
	files are and how to use embossdata to inspect and retrieve
	them, use:
include file="inc/localfiles.ihtml"
   -->

None.














<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->
<H3>1. Command-line arguments</H3>
The following original HMMER options are not supported:
<br><pre>
-h         : Use -help to get help information instead.
</pre>


<H3>2. Installing EMBASSY HMMER</H3>
The EMBASSY HMMER package contains "wrapper" applications providing an EMBOSS-style interface to the applications in the original HMMER package version 2.3.2 developed by Sean Eddy.  Please read the file INSTALL in the EMBASSY HMMER package distribution for installation instructions.

<H3>3. Installing original HMMER</H3>
To use EMBASSY HMMER, you will first need to download and install the original HMMER package.  Please read the file 00README in the the original HMMER package distribution for installation instructions:
<pre>
WWW home:       http://hmmer.wustl.edu/
Distribution:   ftp://ftp.genetics.wustl.edu/pub/eddy/hmmer/
</pre>

<H3>4. Setting up HMMER</H3>
For the EMBASSY HMMER package to work, the directory containing the original HMMER executables *must* be in your path. For example if you executables were installed to "/usr/local/hmmer/bin", then type:
<pre>
set path=(/usr/local/hmmer/bin/ $path)
rehash
</pre>

<H3> 5. Getting help </H3>
Please read the Userguide.pdf distributed with the original HMMER and included in the EMBASSY HMMER distribution under the DOCS directory. The first 3 chapters (Introduction, Installation and Tutorial) are particularly useful.
<p>Please read the 'Notes' section below for a description of the differences between the original and EMBASSY HMMER, particularly which application command line options are supported. 











<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->
<H3>Types of input data</H3>
hmmer v3.2.1 and therefore EMBASSY HMMER is only recommended for use with protein sequences.  If you provide a non-protein sequence you will be reprompted for a protein sequence.  To accept nucleic acid sequences you must replace instances of < type: "protein" > in the application ACD files with <type: "nucleic" >.

<H3>Environment variables</H3>
The original hmmer uses BLAST environment variables (below), if defined, to locate files. The EMBASSY HMMER does not.
<pre>
BLASTDB   location of sequence databases to be searched
BLASMAT   location of substitution matrices
HMMERDB   location of HMMs
</pre>

<H3> Output file </H3>
The index file is hardcoded to database.ssi (where database is the name of the database that was indexed) and is written to the same directory as the input file.  This behaviour could be changed in the future.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="ehmmalign.html">ehmmalign</a></td>
<td>Align sequences to an HMM profile</td>
</tr>

<tr>
<td><a href="ehmmbuild.html">ehmmbuild</a></td>
<td>Build a profile HMM from an alignment</td>
</tr>

<tr>
<td><a href="ehmmcalibrate.html">ehmmcalibrate</a></td>
<td>Calibrate HMM search statistics</td>
</tr>

<tr>
<td><a href="ehmmconvert.html">ehmmconvert</a></td>
<td>Convert between profile HMM file formats</td>
</tr>

<tr>
<td><a href="ehmmemit.html">ehmmemit</a></td>
<td>Generate sequences from a profile HMM</td>
</tr>

<tr>
<td><a href="ehmmfetch.html">ehmmfetch</a></td>
<td>Retrieve an HMM from an HMM database</td>
</tr>

<tr>
<td><a href="ehmmpfam.html">ehmmpfam</a></td>
<td>Search one or more sequences against an HMM database</td>
</tr>

<tr>
<td><a href="ehmmsearch.html">ehmmsearch</a></td>
<td>Search a sequence database with a profile HMM</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/oalistat.html">oalistat</a></td>
<td>Statistics for multiple alignment files</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmalign.html">ohmmalign</a></td>
<td>Align sequences with an HMM</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmbuild.html">ohmmbuild</a></td>
<td>Build HMM</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmcalibrate.html">ohmmcalibrate</a></td>
<td>Calibrate a hidden Markov model</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmconvert.html">ohmmconvert</a></td>
<td>Convert between HMM formats</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmemit.html">ohmmemit</a></td>
<td>Extract HMM sequences</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmfetch.html">ohmmfetch</a></td>
<td>Extract HMM from a database</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmindex.html">ohmmindex</a></td>
<td>Index an HMM database</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmpfam.html">ohmmpfam</a></td>
<td>Align single sequence with an HMM</td>
</tr>

<tr>
<td><a href="/embassy/hmmer/ohmmsearch.html">ohmmsearch</a></td>
<td>Search sequence database with an HMM</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

This program is an EMBOSS conversion of a program written by Sean Eddy
as part of his HMMER package.

<p>
Although we take every care to ensure that the results of the EMBOSS
version are identical to those from the original package, we recommend
that you check your inputs give the same results in both versions
before publication.

<p>
Please report all bugs in the EMBOSS version to the EMBOSS bug team,
not to the original author.
Jon Ison (jison&nbsp;&copy;&nbsp;ebi.ac.uk)
<br>
European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge CB10 1SD, UK

<p>This program is an EMBASSY wrapper to a program written by Sean Eddy as part of his hmmer package.
<p>Please report any bugs to the EMBOSS bug team in the first instance, not to Sean Eddy.


<H2>
    History
</H2>





<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.











</BODY>
</HTML>

