<!--START OF HEADER - DON'T ALTER -->

<HTML>
<HEAD>
  <TITLE>
  EMBOSS: ffitch
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<!--#include virtual="/apps/header1.inc" -->
ffitch
<!--#include virtual="/apps/header2.inc" -->

<!--END OF HEADER-->






<H2>
    Function
</H2>
<!--#include file="inc/ffitch.ione" -->

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>


Estimates phylogenies from distance matrix data under the "additive
tree model" according to which the distances are expected to equal the
sums of branch lengths between the species. Uses the Fitch-Margoliash
criterion and some related least squares criteria, or the Minimum
Evolution distance matrix method. Does not assume an evolutionary
clock. This program will be useful with distances computed from
molecular sequences, restriction sites or fragments distances, with
DNA hybridization measurements, and with genetic distances computed
from gene frequencies.


<H2>
    Algorithm
</H2>


The programs FITCH, KITSCH, and NEIGHBOR are for dealing with data
which comes in the form of a matrix of pairwise distances between all
pairs of taxa, such as distances based on molecular sequence data,
gene frequency genetic distances, amounts of DNA hybridization, or
immunological distances. In analyzing these data, distance matrix
programs implicitly assume that:

<ul>

<li>
Each distance is measured independently from the others: no item of
data contributes to more than one distance.

<li>
The distance between each pair of taxa is drawn from a distribution
with an expectation which is the sum of values (in effect amounts of
evolution) along the tree from one tip to the other. The variance of
the distribution is proportional to a power p of the expectation.

</ul>

<p>
These assumptions can be traced in the least squares methods of
programs FITCH and KITSCH but it is not quite so easy to see them in
operation in the Neighbor-Joining method of NEIGHBOR, where the
independence assumptions is less obvious.

<p>
THESE TWO ASSUMPTIONS ARE DUBIOUS IN MOST CASES: independence will not
be expected to be true in most kinds of data, such as genetic
distances from gene frequency data. For genetic distance data in which
pure genetic drift without mutation can be assumed to be the mechanism
of change CONTML may be more appropriate. However, FITCH, KITSCH, and
NEIGHBOR will not give positively misleading results (they will not
make a statistically inconsistent estimate) provided that additivity
holds, which it will if the distance is computed from the original
data by a method which corrects for reversals and parallelisms in
evolution. If additivity is not expected to hold, problems are more
severe. A short discussion of these matters will be found in a review
article of mine (1984a). For detailed, if sometimes irrelevant,
controversy see the papers by Farris (1981, 1985, 1986) and myself
(1986, 1988b).

<p>
For genetic distances from gene frequencies, FITCH, KITSCH, and
NEIGHBOR may be appropriate if a neutral mutation model can be assumed
and Nei's genetic distance is used, or if pure drift can be assumed
and either Cavalli-Sforza's chord measure or Reynolds, Weir, and
Cockerham's (1983) genetic distance is used. However, in the latter
case (pure drift) CONTML should be better.

<p>
Restriction site and restriction fragment data can be treated by
distance matrix methods if a distance such as that of Nei and Li
(1979) is used. Distances of this sort can be computed in PHYLIp by
the program RESTDIST.

<p>
For nucleic acid sequences, the distances computed in DNADIST allow
correction for multiple hits (in different ways) and should allow one
to analyse the data under the presumption of additivity. In all of
these cases independence will not be expected to hold. DNA
hybridization and immunological distances may be additive and
independent if transformed properly and if (and only if) the standards
against which each value is measured are independent. (This is rarely
exactly true).

<p>
FITCH and the Neighbor-Joining option of NEIGHBOR fit a tree which has
the branch lengths unconstrained. KITSCH and the UPGMA option of
NEIGHBOR, by contrast, assume that an "evolutionary clock" is valid,
according to which the true branch lengths from the root of the tree
to each tip are the same: the expected amount of evolution in any
lineage is proportional to elapsed time.

<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<!--#include file="inc/ffitch.usage" -->



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
<!--#include file="inc/ffitch.ihelp" -->
</pre>
</td></tr></table>
<P>
<!--#include file="inc/ffitch.itable" -->

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>ffitch</b> reads any normal sequence USAs.

<p>

<!--#include file="inc/ffitch.input" -->





<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
include file="inc/reportformats.ihtml"
   -->

<b>ffitch</b> output consists of an unrooted tree and the lengths of
the interior segments. The sum of squares is printed out, and if P =
2.0 Fitch and Margoliash's "average percent standard deviation" is
also computed and printed out. This is the sum of squares, divided by
N-2, and then square-rooted and then multiplied by 100 (n is the
number of species on the tree):

<p>
<pre>
     APSD = ( SSQ / (N-2) )1/2 x 100. 
</pre>
<p>
where N is the total number of off-diagonal distance measurements that
are in the (square) distance matrix. If the S (subreplication) option
is in force it is instead the sum of the numbers of replicates in all
the non-diagonal cells of the distance matrix. But if the L or R
option is also in effect, so that the distance matrix read in is
lower- or upper-triangular, then the sum of replicates is only over
those cells actually read in. If S is not in force, the number of
replicates in each cell is assumed to be 1, so that N is n(n-1), where
n is the number of species. The APSD gives an indication of the
average percentage error. The number of trees examined is also printed
out.



<p>

<!--#include file="inc/ffitch.output" -->





<H2>
    Data files
</H2>

None

<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<!--#include file="inc/ffitch.isee" -->
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

<!--#include virtual="/apps/inc/jfelsenstein.address" -->

<H2>
    History
</H2>

<!--#include file="inc/ffitch.history" -->



<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

<!--#include virtual="/apps/inc/target.itxt" -->











</BODY>
</HTML>

