<!--START OF HEADER - DON'T ALTER -->

<HTML>
<HEAD>
  <TITLE>
  EMBOSS: ememe
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
ememe
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<!--END OF HEADER-->






<H2>
    Function
</H2>
Motif detection
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<!-- 
	This is the main part of the document.
        This is what the user looks at to find out if this program will do what he wants. 
        Long description.
        Why it is useful.
        When it should/shouldn't be used.
  -->
EMBASSY MEME is a suite of application wrappers to the original meme v3.0.14 applications written by  Timothy Bailey. meme v3.0.14 must be installed on the same system as EMBOSS and the location of the meme executables must be defined in your path for EMBASSY MEME to work.

<p><b>Usage:</b>
<br>ememe [options] dataset outfile

<p>The <outfile> parameter is new to EMBASSY MEME. The output is always written to <outfile>. The name of the input sequences may be specified with the -dataset option as normal.

<p>  MEME -- Multiple EM for Motif Elicitation
   
<p>  MEME is a tool for discovering motifs in a group of related DNA or protein
  sequences.
   
<p>  A motif is a sequence pattern that occurs repeatedly in a group of related
  protein or DNA sequences. MEME represents motifs as position-dependent
  letter-probability matrices which describe the probability of each possible
  letter at each position in the pattern. Individual MEME motifs do not 
  contain gaps. Patterns with variable-length gaps are split by MEME into two 
  or more separate motifs.
   
<p>  MEME takes as input a group of DNA or protein sequences (the training set)
  and outputs as many motifs as requested. MEME uses statistical modeling
  techniques to automatically choose the best width, number of occurrences,
  and description for each motif.
   
<p>  MEME outputs its results as a hypertext (HTML) document.



<H2>
    Algorithm
</H2>

<!-- 
        Algorithms
	  Pseudocode/Vague description/Handwaving/web links to useful pages
  -->

Please read the file README distributed with the original MEME. 


<h3>  REQUIRED ARGUMENTS:</h3>
<b>  	< dataset ><b>
<br>       The name of the file containing the training set 
  			sequences.  If < dataset > is the word "stdin", MEME
  			reads from standard input.  
  
<p>  			The sequences in the dataset should be in 
  			Pearson/FASTA format.  For example:
<pre>  
  			>ICYA_MANSE INSECTICYANIN A FORM (BLUE BILIPROTEIN)
  			GDIFYPGYCPDVKPVNDFDLSAFAGAWHEIAK
  			LPLENENQGKCTIAEYKYDGKKASVYNSFVSNGVKEYMEGDLEIAPDA
  			>LACB_BOVIN BETA-LACTOGLOBULIN PRECURSOR (BETA-LG) 
  			MKCLLLALALTCGAQALIVTQTMKGLDI
  			QKVAGTWYSLAMAASDISLLDAQSAPLRVYVEELKPTPEGDLEILLQKW
</pre>  				
  			Sequences start with a header line followed by
  			sequence lines.  A header line has
  			the character ">" in position one, followed by
  			an unique name without any spaces, followed by
  			(optional) descriptive text.  After the header line 
  			come the actual sequence lines.  Spaces and blank 
  			lines are ignored.  Sequences may be in capital or 
  			lowercase or both.  
  
<p>  			MEME uses the first word in the header line of each 
  			sequence, truncated to 24 characters if necessary,
  			as the name of the sequence. This name must be unique. 
  			Sequences with duplicate names will be ignored. 
  			(The first word in the title line is 
  			everything following the ">" up to the first blank.)
  
<p>  			Sequence weights may be specified in the dataset
  			file by special header lines where the unique name
  			is "WEIGHTS" (all caps) and the descriptive 
  			text is a list of sequence weights. 
  			Sequence weights are numbers in the range 0 < w <=1.
  			All weights are assigned in order to the
  			sequences in the file. If there are more sequences
  			than weights, the remainder are given weight one.
  			Weights must be greater than zero and less than
  			or equal to one.  Weights may be specified by
  			more than one "WEIGHT" entry which may appear
  			anywhere in the file.  When weights are used, 
  			sequences will contribute to motifs in proportion
  			to their weights.  Here is an example for a file
  			of three sequences where the first two sequences are 
  			very similar and it is desired to down-weight them:
<pre>  
  			>WEIGHTS 0.5 .5 1.0 
  			>seq1
  			GDIFYPGYCPDVKPVNDFDLSAFAGAWHEIAK
  			>seq2
  			GDMFCPGYCPDVKPVGDFDLSAFAGAWHELAK
  			>seq3
  			QKVAGTWYSLAMAASDISLLDAQSAPLRVYVEELKPTPEGDLEILLQKW
</pre>  

<h3>  OPTIONAL ARGUMENTS:</h3>
   
  MEME has a large number of optional inputs that can be used
  to fine-tune its behavior.  To make these easier to understand
  they are divided into the following categories:
   
<p><b>  		ALPHABET</b>	- control the alphabet for the motifs
  				  (patterns) that MEME will search for
   
<p><b>  		DISTRIBUTION</b>	- control how MEME assumes the occurrences
  				  of the motifs are distributed throughout
  				  the training set sequences
   
<p><b>  		SEARCH</b>		- control how MEME searches for motifs
   
<p><b>                  SYSTEM</b>          - the -p <np> argument causes a version of MEME
                                    compiled for a parallel CPU architecture
                                    to be run.  (By placing < np > in quotes you
                                    may pass installation specific switches to
  				  the 'mpirun' command.  The number of 
                                    processors to run on must be the first 
  				  argument following -p).
  
<p>   
  In what follows, < n > is an integer, < a > is a decimal number, and < string > 
  is a string of characters.
   
<h3>  ALPHABET</h3>

  MEME accepts either DNA or protein sequences, but not both in the same run.
  By default, sequences are assumed to be protein.  The sequences must be in 
  FASTA format.
  
<p>  DNA sequences must contain only the letters "ACGT", plus the ambiguous
  letters "BDHKMNRSUVWY*-". 
<p>  Protein sequences must contain only the letters "ACDEFGHIKLMNPQRSTVWY",
  plus the ambiguous letters "BUXZ*-".
  
<p>  MEME converts all ambiguous letters to "X", which is treated as "unknown".
   
<p><b>  	-dna</b>		Assume sequences are DNA; default: protein sequences
<p><b>  	-protein</b>	Assume sequences are protein
  
   
<h3>  DISTRIBUTION</h3>

  If you know how occurrences of motifs are distributed in the training set 
  sequences, you can specify it with the following optional switches.  The 
  default distribution of motif occurrences is assumed to be zero or one 
  occurrence of per sequence.
   
<p><b>  	-mod < string > </b>   The type of distribution to assume.
<p>  			oops<br>    One Occurrence Per Sequence
<br>  				MEME assumes that each sequence in the dataset
  				contains exactly one occurrence of each motif.
  				This option is the fastest and most sensitive
  				but the motifs returned by MEME may be 
  				"blurry" if any of the sequences is missing
  				them. 	
   
<p>  			zoops<br> Zero or One Occurrence Per Sequence
<br>  				MEME assumes that each sequence may contain at
  				most one occurrence of each motif. This option
  				is useful when you suspect that some motifs
  				may be missing from some of the sequences. In
  				that case, the motifs found will be more
  				accurate than using the first option. This
  				option takes more computer time than the
  				first option (about twice as much) and is
  				slightly less sensitive to weak motifs present
  				in all of the sequences.
   
<p>  			anr<br> 	Any Number of Repetitions
<br>  				MEME assumes each sequence may contain any
  				number of non-overlapping occurrences of each
  				motif. This option is useful when you suspect
  				that motifs repeat multiple times within a
  				single sequence. In that case, the motifs 
  				found will be much more accurate than using 
  				one of the other options. This option can also
  				be used to discover repeats within a single
  				sequence. This option takes the much more
  				computer time than the first option (about ten
  				times as much) and is somewhat less sensitive
  				to weak motifs which do not repeat within a
  				single sequence than the other two options.
   
   
<h3>  SEARCH</h3>
  ------
  
<b>  A) OBJECTIVE FUNCTION </b>
  
<p>  MEME uses an objective function on motifs to select the "best" motif.
  The objective function is based on the statistical significance of the 
  log likelihood ratio (LLR) of the occurrences of the motif.  
  The E-value of the motif is an estimate of the number of motifs (with the 
  same width and number of occurrences) that would have equal or higher log 
  likelihood ratio if the training set sequences had been generated randomly 
  according to the (0-order portion of the) background model. 
  
<p>  MEME searches for the motif with the smallest E-value.
  It searches over different motif widths, numbers of occurrences, and
  positions in the training set for the motif occurrences.
  The user may limit the range of motif widths and number of occurrences
  that MEME tries using the switches described below.  In addition,
  MEME trims the motif (using a dynamic programming multiple alignment) to 
  eliminate any positions where there is a gap in any of the occurrences.  
  
<p>  The log likelihood ratio of a motif is
<pre>
  	llr = log (Pr(sites | motif) / Pr(sites | back))
</pre>
  and is a measure of how different the sites are from the background model.
  Pr(sites | motif) is the probability of the occurrences given the a model
  consisting of the position-specific probability matrix (PSPM) of the motif.
  (The PSPM is output by MEME).
<p>  Pr(sites | back) is the  probability of the occurrences given the background
  model.  The background model is an n-order Markov model.  By default,
  it is a 0-order model consisting of the frequencies of the letters in
  the training set.  A different 0-order Markov model or higher order Markov 
  models can be specified to MEME using the -bfile option described below.
  
<p>  The E-value reported by MEME is actually an approximation of the E-value
  of the log likelihood ratio.  (An approximation is used because it is far
  more efficient to compute.)  The approximation is based on the fact that
  the log likelihood ratio of a motif is the sum of the log 
  likelihood ratios of each column of the motif.  Instead of computing the 
  statistical significance of this sum (its p-value), MEME computes the 
  p-value of each column and then computes the significance of their product.  
  Although not identical to the significance of the log likelihood ratio, this 
  easier to compute objective function works very similarly in practice.
  
<p>  The motif significance is reported as the E-value of the motif.  
<p>  The statistical signficance of a motif is computed based on:
<ol>
<li>  	 the log likelihood ratio,
<li>  	 the width of the motif,
<li>  	 the number of occurrences,
<li>  	 the 0-order portion of the background model,
<li>  	 the size of the training set, and
<li>  	 the type of model (oops, zoops, or anr, which determines the
  	   number of possible different motifs of the given width and
  	   number of occurrences).
</ol>
  
  MEME searches for motifs by performing Expectation Maximization (EM) on a 
  motif model of a fixed width and using an initial estimate of the number of 
  sites.  It then sorts the possible sites according to their probability 
  according to EM.  MEME then and calculates the E-values of the first n sites 
  in the sorted list for different values of n.  This procedure (first EM, 
  followed by computing E-values for different numbers of sites) is repeated 
  with different widths and different initial estimates of the number of 
  sites.  MEME outputs the motif with the lowest E-value.
  
   
<b>  B) NUMBER OF MOTIFS</b>
   
<b>  	-nmotifs < n > </b>   The number of *different* motifs to search
  			for.  MEME will search for and output < n > motifs.
  			Default: 1
   
<p><b>  	-evt < p > </b>	Quit looking for motifs if E-value exceeds < p >.
  			Default: infinite (so by default MEME never quits
  			before -nmotifs < n > have been found.)
   
   
<b>  C) NUMBER OF MOTIF OCCURENCES  </b>
   
<b>  	-nsites < n > </b>
<b>  	-minsites < n > </b>
<b>  	-maxsites < n > </b>
  			The (expected) number of occurrences of each motif.
  			If -nsites is given, only that number of occurrences
  			is tried.  Otherwise, numbers of occurrences between
  			-minsites and -maxsites are tried as initial guesses
  			for the number of motif occurrences.  These
  			switches are ignored if mod = oops.
   
<p>  			Default:

        <p> -minsites sqrt(number sequences)
  	<p>			 -maxsites Default:
  	<br>				zoops 	# of sequences
  	<br>				anr	MIN(5*#sequences, 50)
  
<b>  	-wnsites < n > </b>	The weight on the prior on nsites.  This controls
  			how strong the bias towards motifs with exactly
  			nsites sites (or between minsites and maxsites sites)
  			is.  It is a number in the range [0..1).  The
  			larger it is, the stronger the bias towards 
  			motifs with exactly nsites occurrences is.
<p>  			Default: 0.8
   
<b>  D) MOTIF WIDTH </b>
   
<br><b>  	-w < n >  </b>
<br><b>  	-minw < n > </b>
<br><b>  	-maxw < n >  </b>
<br>  
  			The width of the motif(s) to search for.
  			If -w is given, only that width is tried.
  			Otherwise, widths between -minw and -maxw are tried.
  			Default: -minw  8, -maxw 50 (defined in user.h)
<br>  
  			Note: If < n > is less than the length of the shortest 
  			sequence in the dataset, < n > is reset by MEME to 
  			that value. 
  
<b>  	-nomatrim </b>
<b>  	-wg < a > </b> 
<b>  	-ws < a > </b>
<b>  	-noendgaps </b>
<br>  			These switches control trimming (shortening) of
  			motifs using the multiple alignment method.
  			Specifying -nomatrim causes MEME to skip this and
  			causes the other switches to be ignored.
  			MEME finds the best motif
  			found and then trims (shortens) it using the multiple 
  			alignment method (described below). The number of 
  			occurrences is then adjusted to maximize the motif 
  			E-value, and then the motif width is further
  			shortened to optimize the E-value.
  
<p>  			The multiple alignment method performs a separate 
  			pairwise alignment of the site with the highest
  			probability and each other possible site.
  			(The alignment includes width/2 positions on either 
  			side of the sites.) The pairwise alignment
  			is controlled by the switches:
<br>  				-wg < a > (gap cost; default: 11), 
<br>  				-ws < a > (space cost; default 1), and, 
<br>  				-noendgaps (do not penalize endgaps; default: 
  					penalize endgaps).  
<p>  			The pairwise alignments are then combined and the 
  			method determines the widest section of the motif with 
  			no insertions or deletions.  If this alignment
  		        is shorter than < minw >, it tries to find an alignment
  			allowing up to one insertion/deletion per motif
  			column.  This continues (allowing up to 2, 3 ...
  			insertions/deletions per motif column) until an 
  			alignment of width at least < minw > is found. 
  
  
<b>  E) BACKGROUND MODEL </b>
<b>  	-bfile < bfile >  </b>
<br>	The name of the file containing the background model
  			for sequences.  The background model is the model
  			of random sequences used by MEME.  The background 
  			model is used by MEME 
<br>
<ol>
<li>  				1) during EM as the "null model",
<li>  				2) for calculating the log likelihood ratio
  				   of a motif,
<li>  				3) for calculating the significance (E-value) 
  				   of a motif, and, 
<li>  				4) for creating the position-specific scoring
  				   matrix (log-odds matrix).
</ol>
  
  			By default, the background model is a 0-order Markov 
  			model based on the letter frequencies in the training 
  			set.  
  
<p>  			Markov models of any order can be specified in < bfile > 
  			by listing frequencies of all possible tuples of 
  			length up to order+1.  
  
<p>  			Note that MEME uses only the 0-order portion (single
  			letter frequencies) of the background model for
  			purposes 3) and 4), but uses the full-order model
  			for purposes 1) and 2), above.
  
<p>  			Example: To specify a 1-order Markov background model
  		 		 for DNA, < bfile > might contain the following
  				 lines.  Note that optional comment lines are
  				 by "#" and are ignored by MEME.

<pre>  
  				# tuple   frequency_non_coding
  				a       0.324
  				c       0.176
  				g       0.176
  				t       0.324
  				# tuple   frequency_non_coding
  				aa      0.119
  				ac      0.052
  				ag      0.056
  				at      0.097
  				ca      0.058
  				cc      0.033
  				cg      0.028
  				ct      0.056
  				ga      0.056
  				gc      0.035
  				gg      0.033
  				gt      0.052
  				ta      0.091
  				tc      0.056
  				tg      0.058
  				tt      0.119
</pre>


  
  Sample -bfile files are given in directory tests: 
<br>  	tests/nt.freq (DNA), and 
<br>  	tests/na.freq (amino acid).
  

<b>  F) DNA PALINDROMES AND STRANDS </b>
   
<b>  	-revcomp</b>	motifs occurrences may be on the given DNA strand
  			or on its reverse complement.
<br>  			Default: look for DNA motifs only on the strand given 
  			in the training set.
   
<p><b>  	-pal</b>		
<br>  			Choosing -pal causes MEME to look for palindromes in 
  			DNA datasets.  
<p>  
  			MEME averages the letter frequencies in corresponding 
  			columns of the motif (PSPM) together. For instance, 
  			if the width of the motif is 10, columns 1 and 10, 2 
  			and 9, 3 and 8, etc., are averaged together.  The 
  			averaging combines the frequency of A in one column 
  			with T in the other, and the frequency of C in one 
  			column with G in the other.  
  			If neither option is not chosen, MEME does not 
  			search for DNA palindromes.
  
  
<p> <b>  G) EM ALGORITHM </b>
   
 <p><b> 	-maxiter < n > </b> 
<br>   The number of iterations of EM to run from
  			any starting point.
  			EM is run for < n > iterations or until convergence
  			(see -distance, below) from each starting point.
<br>  			Default: 50
   
<p><b>  	-distance < a >   </b>
<br> The convergence criterion.  MEME stops
  			iterating EM when the change in the
  			motif frequency matrix is less than < a >.
  			(Change is the euclidean distance between
  			two successive frequency matrices.)
<br>  			Default: 0.001
   
<p><b>  	-prior < string > </p>
<br>The prior distribution on the model parameters:
  	<br>	<i>	dirichlet </i>      simple Dirichlet prior
  					This is the default for -dna and 
  					-alph.  It is based on the 
  					non-redundant database letter
  					frequencies.
  	<br>	<i>	dmix</i>		mixture of Dirichlets prior
  					This is the default for -protein. 
  	<br>	<i>	mega</i>		extremely low variance dmix;
  					variance is scaled inversely with
  					the size of the dataset.
  	<br>	<i>	megap</i>		mega for all but last iteration
  					of EM; dmix on last iteration.
  	<br>	<i>	addone</i>		add +1 to each observed count
   

<p><b>   	-b < a > </p>
<br>	  The strength of the prior on model parameters:
  				< a > = 0 means use intrinsic strength of prior
  					for prior = dmix.
<br>  			Defaults:
  				0.01 if prior = dirichlet
  				0 if prior = dmix
   
<p><b>   	-plib < string >  </b>
<br>The name of the file containing the Dirichlet prior
  			in the format of file prior30.plib.
   
   
<p><b>  H) SELECTING STARTS FOR EM </b>
   
<br>  The default is for MEME to search the dataset for good starts for EM.  How 
  the starting points are derived from the dataset is specified by the 
  following switches.
   
<p>  The default type of mapping MEME uses is:
<br>  		-spmap uni for -dna and -alph < string >
<br>  		-spmap pam for -protein
   
<br>  	-spfuzz < a >     The fuzziness of the mapping.
  			Possible values are greater than 0.  Meaning
  			depends on -spmap, see below.
   
<br>  	-spmap < string >  The type of mapping function to use.
<br>  		<i>	uni</i>     Use add-< a > prior when converting a substring
  				to an estimate of theta.
  				Default -spfuzz < a >: 0.5
  		<i>	pam</i>     Use columns of PAM < a > matrix when converting
  				a substring to an estimate of theta.
  				Default -spfuzz < a >: 120 (PAM 120)
   
<p>  			Other types of starting points
  			can be specified using the following switches.
   
<br>  	-cons < string >  Override the sampling of starting points
  			and just use a starting point derived from
  			< string >.
<br>  			This is useful when an actual occurrence of
  			a motif is known and can be used as the
  			starting point for finding the motif.











<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with ememe</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe crp0.s  -mod oops -revcomp ex2.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe crp0.s -mod oops -revcomp -w 20 ex3.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#output.2">Go to the output files for this example</a><p><p>
<p>
<b>Example 3</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe INO_up800.s -mod anr -revcomp -bfile memenew/yeast.nc.6.freq ex4.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#input.3">Go to the input files for this example</a><br><a href="#output.3">Go to the output files for this example</a><p><p>
<p>
<b>Example 4</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe lipocalin.s -mod oops -maxw 20 -nmotifs 2 ex5.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#input.4">Go to the input files for this example</a><br><a href="#output.4">Go to the output files for this example</a><p><p>
<p>
<b>Example 5</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe farntrans5.s -mod anr -maxw 40 -maxsites 50 ex6.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#input.5">Go to the input files for this example</a><br><a href="#output.5">Go to the output files for this example</a><p><p>
<p>
<b>Example 6</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe farntrans5.s -mod anr -w 10 -maxsites 30 -nmotifs 3 ex7.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#output.6">Go to the output files for this example</a><p><p>
<p>
<b>Example 7</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe farntrans5.s -mod anr -maxw 12 -nsites 24 -nmotifs 3 ex8.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#output.7">Go to the output files for this example</a><p><p>
<p>
<b>Example 8</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe adh.s -mod zoops -nmotifs 20 -evt 0.01  ex9.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#input.8">Go to the input files for this example</a><br><a href="#output.8">Go to the output files for this example</a><p><p>
<p>
<b>Example 9</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>ememe crp0.s -mod oops -pal ex1.html </b>
Motif detection

</pre></td></tr></table><p>
<p>
<a href="#output.9">Go to the output files for this example</a><p><p>


<h3>  EXAMPLES: </h3>
  
<p><b>Please note</b> the examples below are unedited excerpts of the original MEME documentation.  Bear in mind the EMBASSY and original MEME options may differ in practice (see "1. Command-line arguments").

<p>  The following examples use data files provided in this release of MEME.  
  MEME writes its output to standard output, so you will want to redirect it 
  to a file in order for use with MAST.
   
<p>  1) A simple DNA example:
<br>  	 meme crp0.s -dna -mod oops -pal > ex1.html
   
<p>  MEME looks for a single motif in the file crp0.s which contains DNA 
  sequences in FASTA format.  The OOPS model is used so MEME assumes that 
  every sequence contains exactly one occurrence of the motif.  The 
  palindrome switch is given so the motif model (PSPM) is converted into a 
  palindrome by combining corresponding frequency columns.  MEME automatically 
  chooses the best width for the motif in this example since no width was 
  specified.
   
<p>  2) Searching for motifs on both DNA strands:
<br>           meme crp0.s -dna -mod oops -revcomp > ex2.html

<p>    This is like the previous example except that the -revcomp switch tells
  MEME to consider both DNA strands, and the -pal switch is absent so the
  palindrome conversion is omitted.  When DNA uses both DNA strands, motif
  occurrences on the two strands may not overlap.  That is, any position
  in the sequence given in the training set may be contained in an occurrence
  of a motif on the positive strand or the negative strand, but not both.
  
<p>  3) A fast DNA example:
<br>  	meme crp0.s -dna -mod oops -revcomp -w 20 > ex3.html
   
<p>  This example differs from example 1) in that MEME is told to only 
  consider motifs of width 20.  This causes MEME to execute about 10 
  times faster.  The -w switch can also be used with protein datasets if 
  the width of the motifs are known in advance.
  
<p>  4) Using a higher-order background model:
<br>  	meme INO_up800.s -dna -mod anr -revcomp -bfile yeast.nc.6.freq > ex4.html
  
<p>  In this example we use -mod anr and -bfile yeast.nc.6.freq.  This specifies 
  that
<br>  	a) the motif may have any number of occurrences in each sequence, and,
<br>  	b) the Markov model specified in yeast.nc.6.freq is used as the 
  	   background model.  This file contains a fifth-order Markov model 
             for the non-coding regions in the yeast genome.
<br>  Using a higher order background model can often result in more sensitive
  detection of motifs.  This is because the background model more accurately
  models non-motif sequence, allowing MEME to discriminate against it and find 
  the true motifs.
  
<p>  5) A simple protein example:
<br>  	meme lipocalin.s -mod oops -maxw 20 -nmotifs 2 > ex5.html

<p>  The -dna switch is absent, so MEME assumes the file lipocalin.s contains 
  protein sequences.  MEME searches for two motifs each of width less than or 
  equal to 20.
  (Specifying -maxw 20 makes MEME run faster since it does not have to 
  consider motifs longer than 20.) Each motif is assumed to occur in each 
  of the sequences because the OOPS model is specified.
   
<p>  6) Another simple protein example:
<br>     	meme farntrans5.s -mod anr -maxw 40 -maxsites 50 > ex6.html

<p>     MEME searches for a motif of width up to 40 with up to 50 occurrences in
  the entire training set.  The ANR sequence model is specified,
  which allows each motif to have any number of occurrences in each sequence.  
  This dataset contains motifs with multiple repeats of motifs in each 
  sequence.  This example is fairly time consuming due to the fact that the 
  time required to initiale the motif probability tables is proportional 
  to < maxw > times < maxsites >.  By default, MEME only looks for motifs up to 
  29 letters wide with a maximum total of number of occurrences equal to twice 
  the number of sequences or 30, whichever is less.
  
<p>  7) A much faster protein example:
<br>  	meme farntrans5.s -mod anr -w 10 -maxsites 30 -nmotifs 3 > ex7.html
  
<p>  This time MEME is constrained to search for three motifs of width exactly 
  ten.  The effect is to break up the long motif found in the previous 
  example.  The -w switch forces motifs to be *exactly* ten letters wide.
  This example is much faster because, since only one width is considered, the
  time to build the motif probability tables is only proportional to 
  < maxsites >.
  
<p>  8) Splitting the sites into three:
<br>  	meme farntrans5.s -mod anr -maxw 12 -nsites 24 -nmotifs 3 > ex8.html
  
<p>  This forces each motif to have 24 occurrences, exactly, and be up to 12 
  letters wide.
  
<p>  9) A larger protein example with E-value cutoff:
<br>  	meme adh.s -mod zoops -nmotifs 20 -evt 0.01 > ex9.html

<p>  In this example, MEME looks for up to 20 motifs, but stops when a motif is
  found with E-value greater than 0.01.  Motifs with large E-values are likely
  to be statistical artifacts rather than biologically significant.







<H2>
    Command line arguments
</H2>
Where possible, the same command-line qualifier names and parameter order is used as in the original meme. There are however several unavoidable differences and these are clearly documented in the "Notes" section below.

<p>Most of the options in the original meme are given in ACD as "advanced" or
"additional" options. -options must be specified on the command-line in order 
to be prompted for a value for "additional" options but "advanced" options 
will never be prompted for.  
<p>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-dataset]           seqset     User must provide the full filename of a set
                                  of sequences, not an indirect reference,
                                  e.g. a USA is NOT acceptable.
  [-outfile]           outfile    [*.ememe] MEME program output file

   Additional (Optional) qualifiers:
   -bfile              infile     The name of the file containing the
                                  background model for sequences. The
                                  background model is the model of random
                                  sequences used by MEME. The background model
                                  is used by MEME 1) during EM as the 'null
                                  model', 2) for calculating the log
                                  likelihood ratio of a motif, 3) for
                                  calculating the significance (E-value) of a
                                  motif, and, 4) for creating the
                                  position-specific scoring matrix (log-odds
                                  matrix). See application documentation for
                                  more information.
   -plibfile           infile     The name of the file containing the
                                  Dirichlet prior in the format of file
                                  prior30.plib
   -mod                selection  [zoops] If you know how occurrences of
                                  motifs are distributed in the training set
                                  sequences, you can specify it with these
                                  options. The default distribution of motif
                                  occurrences is assumed to be zero or one
                                  occurrence of per sequence. oops : One
                                  Occurrence Per Sequence. MEME assumes that
                                  each sequence in the dataset contains
                                  exactly one occurrence of each motif. This
                                  option is the fastest and most sensitive but
                                  the motifs returned by MEME may be 'blurry'
                                  if any of the sequences is missing them.
                                  zoops : Zero or One Occurrence Per Sequence.
                                  MEME assumes that each sequence may contain
                                  at most one occurrence of each motif. This
                                  option is useful when you suspect that some
                                  motifs may be missing from some of the
                                  sequences. In that case, the motifs found
                                  will be more accurate than using the first
                                  option. This option takes more computer time
                                  than the first option (about twice as much)
                                  and is slightly less sensitive to weak
                                  motifs present in all of the sequences. anr
                                  : Any Number of Repetitions. MEME assumes
                                  each sequence may contain any number of
                                  non-overlapping occurrences of each motif.
                                  This option is useful when you suspect that
                                  motifs repeat multiple times within a single
                                  sequence. In that case, the motifs found
                                  will be much more accurate than using one of
                                  the other options. This option can also be
                                  used to discover repeats within a single
                                  sequence. This option takes the much more
                                  computer time than the first option (about
                                  ten times as much) and is somewhat less
                                  sensitive to weak motifs which do not repeat
                                  within a single sequence than the other two
                                  options.
   -nmotifs            integer    [1] The number of *different* motifs to
                                  search for. MEME will search for and output
                                  <n> motifs. (Any integer value)
   -text               boolean    [N] Default output is in HTML
   -prior              selection  [dirichlet] The prior distribution on the
                                  model parameters. dirichlet: Simple
                                  Dirichlet prior. This is the default for
                                  -dna and -alph. It is based on the
                                  non-redundant database letter frequencies.
                                  dmix: Mixture of Dirichlets prior. This is
                                  the default for -protein. mega: Extremely
                                  low variance dmix; variance is scaled
                                  inversely with the size of the dataset.
                                  megap: Mega for all but last iteration of
                                  EM; dmix on last iteration. addone: Add +1
                                  to each observed count.
   -evt                float      [-1] Quit looking for motifs if E-value
                                  exceeds this value. Has an extremely high
                                  default so by default MEME never quits
                                  before -nmotifs <n> have been found. (Any
                                  numeric value)
   -nsites             integer    [-1] These switches are ignored if mod =
                                  oops. The (expected) number of occurrences
                                  of each motif. If a value for -nsites is
                                  specified, only that number of occurrences
                                  is tried. Otherwise, numbers of occurrences
                                  between -minsites and -maxsites are tried as
                                  initial guesses for the number of motif
                                  occurrences. If a value is not specified for
                                  -minsites and maxsites then the default
                                  hardcoded into MEME, as opposed to the
                                  default value given in the ACD file, is
                                  used. The hardcoded default value of
                                  -minsites is equal to sqrt(number
                                  sequences). The hardcoded default value of
                                  -maxsites is equal to the number of
                                  sequences (zoops) or MIN(5* num.sequences,
                                  50) (anr). (Any integer value)
   -minsites           integer    [-1] These switches are ignored if mod =
                                  oops. The (expected) number of occurrences
                                  of each motif. If a value for -nsites is
                                  specified, only that number of occurrences
                                  is tried. Otherwise, numbers of occurrences
                                  between -minsites and -maxsites are tried as
                                  initial guesses for the number of motif
                                  occurrences. If a value is not specified for
                                  -minsites and maxsites then the default
                                  hardcoded into MEME, as opposed to the
                                  default value given in the ACD file, is
                                  used. The hardcoded default value of
                                  -minsites is equal to sqrt(number
                                  sequences). The hardcoded default value of
                                  -maxsites is equal to the number of
                                  sequences (zoops) or MIN(5 * num.sequences,
                                  50) (anr). (Any integer value)
   -maxsites           integer    [-1] These switches are ignored if mod =
                                  oops. The (expected) number of occurrences
                                  of each motif. If a value for -nsites is
                                  specified, only that number of occurrences
                                  is tried. Otherwise, numbers of occurrences
                                  between -minsites and -maxsites are tried as
                                  initial guesses for the number of motif
                                  occurrences. If a value is not specified for
                                  -minsites and maxsites then the default
                                  hardcoded into MEME, as opposed to the
                                  default value given in the ACD file, is
                                  used. The hardcoded default value of
                                  -minsites is equal to sqrt(number
                                  sequences). The hardcoded default value of
                                  -maxsites is equal to the number of
                                  sequences (zoops) or MIN(5 * num.sequences,
                                  50) (anr). (Any integer value)
   -wnsites            float      [.8] The weight on the prior on nsites. This
                                  controls how strong the bias towards motifs
                                  with exactly nsites sites (or between
                                  minsites and maxsites sites) is. It is a
                                  number in the range [0..1). The larger it
                                  is, the stronger the bias towards motifs
                                  with exactly nsites occurrences is. (Any
                                  numeric value)
   -w                  integer    [-1] The width of the motif(s) to search
                                  for. If -w is given, only that width is
                                  tried. Otherwise, widths between -minw and
                                  -maxw are tried. Note: if width is less than
                                  the length of the shortest sequence in the
                                  dataset, width is reset by MEME to that
                                  value. (Any integer value)
   -minw               integer    [8] The width of the motif(s) to search for.
                                  If -w is given, only that width is tried.
                                  Otherwise, widths between -minw and -maxw
                                  are tried. Note: if width is less than the
                                  length of the shortest sequence in the
                                  dataset, width is reset by MEME to that
                                  value. (Any integer value)
   -maxw               integer    [50] The width of the motif(s) to search
                                  for. If -w is given, only that width is
                                  tried. Otherwise, widths between -minw and
                                  -maxw are tried. Note: if width is less than
                                  the length of the shortest sequence in the
                                  dataset, width is reset by MEME to that
                                  value. (Any integer value)
   -nomatrim           boolean    [N] The -nomatrim, -wg, -ws and -noendgaps
                                  switches control trimming (shortening) of
                                  motifs using the multiple alignment method.
                                  Specifying -nomatrim causes MEME to skip
                                  this and causes the other switches to be
                                  ignored. The pairwise alignment is
                                  controlled by the switches -wg (gap cost),
                                  -ws (space cost) and -noendgaps (do not
                                  penalize endgaps). See application
                                  documentation for further information.
   -wg                 integer    [11] The -nomatrim, -wg, -ws and -noendgaps
                                  switches control trimming (shortening) of
                                  motifs using the multiple alignment method.
                                  Specifying -nomatrim causes MEME to skip
                                  this and causes the other switches to be
                                  ignored. The pairwise alignment is
                                  controlled by the switches -wg (gap cost),
                                  -ws (space cost) and -noendgaps (do not
                                  penalize endgaps). See application
                                  documentation for further information. (Any
                                  integer value)
   -ws                 integer    [1] The -nomatrim, -wg, -ws and -noendgaps
                                  switches control trimming (shortening) of
                                  motifs using the multiple alignment method.
                                  Specifying -nomatrim causes MEME to skip
                                  this and causes the other switches to be
                                  ignored. The pairwise alignment is
                                  controlled by the switches -wg (gap cost),
                                  -ws (space cost) and -noendgaps (do not
                                  penalize endgaps). See application
                                  documentation for further information. (Any
                                  integer value)
   -noendgaps          boolean    [N] The -nomatrim, -wg, -ws and -noendgaps
                                  switches control trimming (shortening) of
                                  motifs using the multiple alignment method.
                                  Specifying -nomatrim causes MEME to skip
                                  this and causes the other switches to be
                                  ignored. The pairwise alignment is
                                  controlled by the switches -wg (gap cost),
                                  -ws (space cost) and -noendgaps (do not
                                  penalize endgaps). See application
                                  documentation for further information.
   -revcomp            boolean    [N] Motifs occurrences may be on the given
                                  DNA strand or on its reverse complement. The
                                  default is to look for DNA motifs only on
                                  the strand given in the training set.
   -pal                boolean    [N] Choosing -pal causes MEME to look for
                                  palindromes in DNA datasets. MEME averages
                                  the letter frequencies in corresponding
                                  columns of the motif (PSPM) together. For
                                  instance, if the width of the motif is 10,
                                  columns 1 and 10, 2 and 9, 3 and 8, etc.,
                                  are averaged together. The averaging
                                  combines the frequency of A in one column
                                  with T in the other, and the frequency of C
                                  in one column with G in the other.
   -[no]nostatus       boolean    [Y] Set this option to print a progress
                                  report to the terminal.

   Advanced (Unprompted) qualifiers:
   -maxiter            integer    [50] The number of iterations of EM to run
                                  from any starting point. EM is run for <n>
                                  iterations or until convergence (see
                                  -distance, below) from each starting point.
                                  (Any integer value)
   -distance           float      [0.001] The convergence criterion. MEME
                                  stops iterating EM when the change in the
                                  motif frequency matrix is less than <a>.
                                  (Change is the euclidean distance between
                                  two successive frequency matrices.) (Any
                                  numeric value)
   -b                  float      [-1.0] The strength of the prior on model
                                  parameters. A value of 0 means use intrinsic
                                  strength of prior if prior = dmix. The
                                  default values are 0.01 if prior = dirichlet
                                  or 0 if prior = dmix. These defaults are
                                  hardcoded into MEME (the value of the
                                  default in the ACD file is not used). (Any
                                  numeric value)
   -spfuzz             float      [-1.0] The fuzziness of the mapping.
                                  Possible values are greater than 0. Meaning
                                  depends on -spmap, see below. See the
                                  application documentation for more
                                  information. (Any numeric value)
   -spmap              selection  [uni] The type of mapping function to use.
                                  uni: Use prior when converting a substring
                                  to an estimate of theta. Default -spfuzz
                                  <a>: 0.5. pam: Use columns of PAM <a> matrix
                                  when converting a substring to an estimate
                                  of theta. Default -spfuzz <a>: 120 (PAM
                                  120). See the application documentation for
                                  more information.
   -cons               string     Override the sampling of starting points and
                                  just use a starting point derived from
                                  <string>. This is useful when an actual
                                  occurrence of a motif is known and can be
                                  used as the starting point for finding the
                                  motif. See the application documentation for
                                  more information. (Any string is accepted)
   -maxsize            integer    [-1] Maximum dataset size in characters (Any
                                  integer value)
   -p                  integer    [0] Only values of >0 will be applied. The
                                  -p <np> argument causes a version of MEME
                                  compiled for a parallel CPU architecture to
                                  be run. (By placing <np> in quotes you may
                                  pass installation specific switches to the
                                  'mpirun' command. The number of processors
                                  to run on must be the first argument
                                  following -p). (Any integer value)
   -time               integer    [0] Only values of more than 0 will be
                                  applied. (Any integer value)
   -sf                 string     Print <sf> as name of sequence file (Any
                                  string is accepted)

   Associated qualifiers:

   "-dataset" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-dataset]<br>(Parameter 1)</td>
<td>User must provide the full filename of a set of sequences, not an indirect reference, e.g. a USA is NOT acceptable.</td>
<td>Readable set of sequences</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>MEME program output file</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.ememe</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-bfile</td>
<td>The name of the file containing the background model for sequences. The background model is the model of random sequences used by MEME. The background model is used by MEME 1) during EM as the 'null model', 2) for calculating the log likelihood ratio of a motif, 3) for calculating the significance (E-value) of a motif, and, 4) for creating the position-specific scoring matrix (log-odds matrix). See application documentation for more information.</td>
<td>Input file</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-plibfile</td>
<td>The name of the file containing the Dirichlet prior in the format of file prior30.plib</td>
<td>Input file</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>-mod</td>
<td>If you know how occurrences of motifs are distributed in the training set sequences, you can specify it with these options. The default distribution of motif occurrences is assumed to be zero or one occurrence of per sequence. oops : One Occurrence Per Sequence. MEME assumes that each sequence in the dataset contains exactly one occurrence of each motif. This option is the fastest and most sensitive but the motifs returned by MEME may be 'blurry' if any of the sequences is missing them. zoops : Zero or One Occurrence Per Sequence. MEME assumes that each sequence may contain at most one occurrence of each motif. This option is useful when you suspect that some motifs may be missing from some of the sequences. In that case, the motifs found will be more accurate than using the first option. This option takes more computer time than the first option (about twice as much) and is slightly less sensitive to weak motifs present in all of the sequences. anr : Any Number of Repetitions. MEME assumes each sequence may contain any number of non-overlapping occurrences of each motif. This option is useful when you suspect that motifs repeat multiple times within a single sequence. In that case, the motifs found will be much more accurate than using one of the other options. This option can also be used to discover repeats within a single sequence. This option takes the much more computer time than the first option (about ten times as much) and is somewhat less sensitive to weak motifs which do not repeat within a single sequence than the other two options.</td>
<td>Choose from selection list of values</td>
<td>zoops</td>
</tr>

<tr>
<td>-nmotifs</td>
<td>The number of *different* motifs to search for. MEME will search for and output &lt;n&gt; motifs.</td>
<td>Any integer value</td>
<td>1</td>
</tr>

<tr>
<td>-text</td>
<td>Default output is in HTML</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-prior</td>
<td>The prior distribution on the model parameters. dirichlet: Simple Dirichlet prior. This is the default for -dna and -alph. It is based on the non-redundant database letter frequencies. dmix: Mixture of Dirichlets prior. This is the default for -protein. mega: Extremely low variance dmix; variance is scaled inversely with the size of the dataset. megap: Mega for all but last iteration of EM; dmix on last iteration. addone: Add +1 to each observed count.</td>
<td>Choose from selection list of values</td>
<td>dirichlet</td>
</tr>

<tr>
<td>-evt</td>
<td>Quit looking for motifs if E-value exceeds this value. Has an extremely high default so by default MEME never quits before -nmotifs &lt;n&gt; have been found.</td>
<td>Any numeric value</td>
<td>-1</td>
</tr>

<tr>
<td>-nsites</td>
<td>These switches are ignored if mod = oops. The (expected) number of occurrences of each motif. If a value for -nsites is specified, only that number of occurrences is tried. Otherwise, numbers of occurrences between -minsites and -maxsites are tried as initial guesses for the number of motif occurrences. If a value is not specified for -minsites and maxsites then the default hardcoded into MEME, as opposed to the default value given in the ACD file, is used. The hardcoded default value of -minsites is equal to sqrt(number sequences). The hardcoded default value of -maxsites is equal to the number of sequences (zoops) or MIN(5* num.sequences, 50) (anr).</td>
<td>Any integer value</td>
<td>-1</td>
</tr>

<tr>
<td>-minsites</td>
<td>These switches are ignored if mod = oops. The (expected) number of occurrences of each motif. If a value for -nsites is specified, only that number of occurrences is tried. Otherwise, numbers of occurrences between -minsites and -maxsites are tried as initial guesses for the number of motif occurrences. If a value is not specified for -minsites and maxsites then the default hardcoded into MEME, as opposed to the default value given in the ACD file, is used. The hardcoded default value of -minsites is equal to sqrt(number sequences). The hardcoded default value of -maxsites is equal to the number of sequences (zoops) or MIN(5 * num.sequences, 50) (anr).</td>
<td>Any integer value</td>
<td>-1</td>
</tr>

<tr>
<td>-maxsites</td>
<td>These switches are ignored if mod = oops. The (expected) number of occurrences of each motif. If a value for -nsites is specified, only that number of occurrences is tried. Otherwise, numbers of occurrences between -minsites and -maxsites are tried as initial guesses for the number of motif occurrences. If a value is not specified for -minsites and maxsites then the default hardcoded into MEME, as opposed to the default value given in the ACD file, is used. The hardcoded default value of -minsites is equal to sqrt(number sequences). The hardcoded default value of -maxsites is equal to the number of sequences (zoops) or MIN(5 * num.sequences, 50) (anr).</td>
<td>Any integer value</td>
<td>-1</td>
</tr>

<tr>
<td>-wnsites</td>
<td>The weight on the prior on nsites. This controls how strong the bias towards motifs with exactly nsites sites (or between minsites and maxsites sites) is. It is a number in the range [0..1). The larger it is, the stronger the bias towards motifs with exactly nsites occurrences is.</td>
<td>Any numeric value</td>
<td>.8</td>
</tr>

<tr>
<td>-w</td>
<td>The width of the motif(s) to search for. If -w is given, only that width is tried. Otherwise, widths between -minw and -maxw are tried. Note: if width is less than the length of the shortest sequence in the dataset, width is reset by MEME to that value.</td>
<td>Any integer value</td>
<td>-1</td>
</tr>

<tr>
<td>-minw</td>
<td>The width of the motif(s) to search for. If -w is given, only that width is tried. Otherwise, widths between -minw and -maxw are tried. Note: if width is less than the length of the shortest sequence in the dataset, width is reset by MEME to that value.</td>
<td>Any integer value</td>
<td>8</td>
</tr>

<tr>
<td>-maxw</td>
<td>The width of the motif(s) to search for. If -w is given, only that width is tried. Otherwise, widths between -minw and -maxw are tried. Note: if width is less than the length of the shortest sequence in the dataset, width is reset by MEME to that value.</td>
<td>Any integer value</td>
<td>50</td>
</tr>

<tr>
<td>-nomatrim</td>
<td>The -nomatrim, -wg, -ws and -noendgaps switches control trimming (shortening) of motifs using the multiple alignment method. Specifying -nomatrim causes MEME to skip this and causes the other switches to be ignored. The pairwise alignment is controlled by the switches -wg (gap cost), -ws (space cost) and -noendgaps (do not penalize endgaps). See application documentation for further information.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-wg</td>
<td>The -nomatrim, -wg, -ws and -noendgaps switches control trimming (shortening) of motifs using the multiple alignment method. Specifying -nomatrim causes MEME to skip this and causes the other switches to be ignored. The pairwise alignment is controlled by the switches -wg (gap cost), -ws (space cost) and -noendgaps (do not penalize endgaps). See application documentation for further information.</td>
<td>Any integer value</td>
<td>11</td>
</tr>

<tr>
<td>-ws</td>
<td>The -nomatrim, -wg, -ws and -noendgaps switches control trimming (shortening) of motifs using the multiple alignment method. Specifying -nomatrim causes MEME to skip this and causes the other switches to be ignored. The pairwise alignment is controlled by the switches -wg (gap cost), -ws (space cost) and -noendgaps (do not penalize endgaps). See application documentation for further information.</td>
<td>Any integer value</td>
<td>1</td>
</tr>

<tr>
<td>-noendgaps</td>
<td>The -nomatrim, -wg, -ws and -noendgaps switches control trimming (shortening) of motifs using the multiple alignment method. Specifying -nomatrim causes MEME to skip this and causes the other switches to be ignored. The pairwise alignment is controlled by the switches -wg (gap cost), -ws (space cost) and -noendgaps (do not penalize endgaps). See application documentation for further information.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-revcomp</td>
<td>Motifs occurrences may be on the given DNA strand or on its reverse complement. The default is to look for DNA motifs only on the strand given in the training set.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-pal</td>
<td>Choosing -pal causes MEME to look for palindromes in DNA datasets. MEME averages the letter frequencies in corresponding columns of the motif (PSPM) together. For instance, if the width of the motif is 10, columns 1 and 10, 2 and 9, 3 and 8, etc., are averaged together. The averaging combines the frequency of A in one column with T in the other, and the frequency of C in one column with G in the other.</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-[no]nostatus</td>
<td>Set this option to print a progress report to the terminal.</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-maxiter</td>
<td>The number of iterations of EM to run from any starting point. EM is run for &lt;n&gt; iterations or until convergence (see -distance, below) from each starting point.</td>
<td>Any integer value</td>
<td>50</td>
</tr>

<tr>
<td>-distance</td>
<td>The convergence criterion. MEME stops iterating EM when the change in the motif frequency matrix is less than &lt;a&gt;. (Change is the euclidean distance between two successive frequency matrices.)</td>
<td>Any numeric value</td>
<td>0.001</td>
</tr>

<tr>
<td>-b</td>
<td>The strength of the prior on model parameters. A value of 0 means use intrinsic strength of prior if prior = dmix. The default values are 0.01 if prior = dirichlet or 0 if prior = dmix. These defaults are hardcoded into MEME (the value of the default in the ACD file is not used).</td>
<td>Any numeric value</td>
<td>-1.0</td>
</tr>

<tr>
<td>-spfuzz</td>
<td>The fuzziness of the mapping. Possible values are greater than 0. Meaning depends on -spmap, see below. See the application documentation for more information.</td>
<td>Any numeric value</td>
<td>-1.0</td>
</tr>

<tr>
<td>-spmap</td>
<td>The type of mapping function to use. uni: Use prior when converting a substring to an estimate of theta. Default -spfuzz &lt;a&gt;: 0.5. pam: Use columns of PAM &lt;a&gt; matrix when converting a substring to an estimate of theta. Default -spfuzz &lt;a&gt;: 120 (PAM 120). See the application documentation for more information.</td>
<td>Choose from selection list of values</td>
<td>uni</td>
</tr>

<tr>
<td>-cons</td>
<td>Override the sampling of starting points and just use a starting point derived from &lt;string&gt;. This is useful when an actual occurrence of a motif is known and can be used as the starting point for finding the motif. See the application documentation for more information.</td>
<td>Any string is accepted</td>
<td><i>An empty string is accepted</i></td>
</tr>

<tr>
<td>-maxsize</td>
<td>Maximum dataset size in characters</td>
<td>Any integer value</td>
<td>-1</td>
</tr>

<tr>
<td>-p</td>
<td>Only values of &gt;0 will be applied. The -p &lt;np&gt; argument causes a version of MEME compiled for a parallel CPU architecture to be run. (By placing &lt;np&gt; in quotes you may pass installation specific switches to the 'mpirun' command. The number of processors to run on must be the first argument following -p).</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr>
<td>-time</td>
<td>Only values of more than 0 will be applied.</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr>
<td>-sf</td>
<td>Print &lt;sf&gt; as name of sequence file</td>
<td>Any string is accepted</td>
<td><i>An empty string is accepted</i></td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>
<H3>Sequence formats </H3>
The original MEME only supported input sequences in FASTA format. EMBASSY MEME supports all EMBOSS-supported sequence formats. 

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>meme</b> reads any normal sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: crp0.s</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;ce1cg
TAATGTTTGTGCTGGTTTTTGTGGCATCGGGCGAGAATAGCGCGTGGTGTGAAAGACTGTTTTTTTGATCGTTTTCACAA
AAATGGAAGTCCACAGTCTTGACAG
&gt;ara
GACAAAAACGCGTAACAAAAGTGTCTATAATCACGGCAGAAAAGTCCACATTGATTATTTGCACGGCGTCACACTTTGCT
ATGCCATAGCATTTTTATCCATAAG
&gt;bglr1
ACAAATCCCAATAACTTAATTATTGGGATTTGTTATATATAACTTTATAAATTCCTAAAATTACACAAAGTTAATAACTG
TGAGCATGGTCATATTTTTATCAAT
&gt;crp
CACAAAGCGAAAGCTATGCTAAAACAGTCAGGATGCTACAGTAATACATTGATGTACTGCATGTATGCAAAGGACGTCAC
ATTACCGTGCAGTACAGTTGATAGC
&gt;cya
ACGGTGCTACACTTGTATGTAGCGCATCTTTCTTTACGGTCAATCAGCAAGGTGTTAAATTGATCACGTTTTAGACCATT
TTTTCGTCGTGAAACTAAAAAAACC
&gt;deop2
AGTGAATTATTTGAACCAGATCGCATTACAGTGATGCAAACTTGTAAGTAGATTTCCTTAATTGTGATGTGTATCGAAGT
GTGTTGCGGAGTAGATGTTAGAATA
&gt;gale
GCGCATAAAAAACGGCTAAATTCTTGTGTAAACGATTCCACTAATTTATTCCATGTCACACTTTTCGCATCTTTGTTATG
CTATGGTTATTTCATACCATAAGCC
&gt;ilv
GCTCCGGCGGGGTTTTTTGTTATCTGCAATTCAGTACAAAACGTGATCAACCCCTCAATTTTCCCTTTGCTGAAAAATTT
TCCATTGTCTCCCCTGTAAAGCTGT
&gt;lac
AACGCAATTAATGTGAGTTAGCTCACTCATTAGGCACCCCAGGCTTTACACTTTATGCTTCCGGCTCGTATGTTGTGTGG
AATTGTGAGCGGATAACAATTTCAC
&gt;male
ACATTACCGCCAATTCTGTAACAGAGATCACACAAAGCGACGGTGGGGCGTAGGGGCAAGGAGGATGGAAAGAGGTTGCC
GTATAAAGAAACTAGAGTCCGTTTA
&gt;malk
GGAGGAGGCGGGAGGATGAGAACACGGCTTCTGTGAACTAAACCGAGGTCATGTAAGGAATTTCGTGATGTTGCTTGCAA
AAATCGTGGCGATTTTATGTGCGCA
&gt;malt
GATCAGCGTCGTTTTAGGTGAGTTGTTAATAAAGATTTGGAATTGTGACACAGTGCAAATTCAGACACATAAAAAAACGT
CATCGCTTGCATTAGAAAGGTTTCT
&gt;ompa
GCTGACAAAAAAGATTAAACATACCTTATACAAGACTTTTTTTTCATATGCCTGACGGAGTTCACACTTGTAAGTTTTCA
ACTACGTTGTAGACTTTACATCGCC
&gt;tnaa
TTTTTTAAACATTAAAATTCTTACGTAATTTATAATCTTTAAAAAAAGCATTTAATATTGCTCCCCGAACGATTGTGATT
CGATTCACATTTAAACAATTTCAGA
&gt;uxu1
CCCATGAGAGTGAAATTGTTGTGATGTGGTTAACCCAATTAGAATTCGGGATTGACATGTCTTACCAAAAGGTAGAACTT
ATACGCCATCTCATCCGATGCAAGC
&gt;pbr322
CTGGCTTAACTATGCGGCATCAGAGCAGATTGTACTGAGAGTGCACCATATGCGGTGTGAAATACCGCACAGATGCGTAA
GGAGAAAATACCGCATCAGGCGCTC
&gt;trn9cat
CTGTGACGGAAGATCACTTCGCAGAATAAATAAATCCTGGTGTCCCTGTTGATACCGGGAAGCCCTGGGCCAACTTTTGG
CGAAAATGAGACGTTGATCGGCACG
&gt;tdc
GATTTTTATACTTTAACTTGTTGATATTTAAAGGTATTTAATTGTAATAACGATACTCTGGAAAGTATTGAAAGTTAATT
TGTGAGTGGTCGCACATATCCTGTT
</pre>
</td></tr></table><p>

<a name="input.3"></a>
<h3>Input files for usage example 3</h3>
<p><h3>File: INO_up800.s</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;CHO1	 sequence of the region upstream from YER026C
CCGACCCAAATGTAATGGAACAATATTATTTGACACTTGATCAGCAGCAAAATAATCACC
AAAATATGGCCTGGTTGACTCCTCCACAACTGCCACCTCATTTAGAAAACGTCATTTTGA
ATAGTTACTCAAACGCGCAAACTGATAATACGTCTGGCGCCCTTCCCATTCCGAACCATG
TTATATTGAACCATCTGGCGACAAGCAGTATTAAGCATAATACATTATGTGTCGCATCCA
TTGTTAGGTATAAACAAAAATACGTGACCCAAATACTGTATACACCATTGCAATAGATAT
GATTATAGAGCTTATAGCTACATCTTTTTAGATAAAAGCGAAGATGTTTCTGCGATTTTT
CCATTATAGCTCTCCATGATACTAAATATCAAGGTCTACATGTAAGTATTTGTATATATG
GGTTGGAATGTATATACGTATATACGTACGTACGTACGTATATGCACATAATTGTTACGG
GATGTATATATAAATTAGTAGCATTATAGAAGATATCCCTAACATCAATCCCCACTCCTT
CTCAATGTGTGCAGACTTCTGTGCCAGACACTGAATATATATCAGTAATTGGTCAAAATC
ACTTTGAACGTTCACACGGCACCCTCACGCCTTTGAGCTTTCACATGGACCCATCTAAAG
ATGAAGATCCGTATTTTATAGGAAACATTATAAATAAGGAAAGAGAGATACACCTATTTT
TTTCATTTTGTGGGTGATTGTCATTTTTAGTTGTCTATTTGATTCAATCAAAAAACAAAA
ATAAAACTATATATTAAAAA
&gt;CHO2	 sequence of the region upstream from YGR157W
ACCCTCTAACGCGAATAAAGCGAATGACAGCGGCACCATTAATATGGCGAAACTGCAATT
ACTACCTGAAAACCAACAAGATATGATCAAACAAGTTCTTACTTTGACACCTGCCCAGAT
CCAAAGTTTACCAAGTGACCAGCAACTTATGGTGGAAAACTTTAGAAAAGAATATATAAT
CTAAGTAATCAGAGCCATAGCGTATCAGAAAACCACACCTAATTAGATGGTTCTTGCATC
TGTACCTCTTATCACTAAAAGCGGCACTAAACTTCCAACATTAAATGTTTGCCTTGTTAA
ATATATATTTTTGCCTTGGTTTAAATTGGTCAAGACAGTCAATTGCCACACTTTTCTCAT
GCCGCATTCATTATTCGCGAAGTTTTCCACACAAAACTGTGAAAATGAACGGCGATGCCA
GAAACGGCAAAACCTCAAATGTTAGATAACGTGGATCTCCGACACATGTGAATTTATAAG
TAGGCATATGAAAATACAGATTCTTTCCACTGTGTTCCCTTTTATTCCCTTCTCATGTGA
AGAGTTCACACCAAATCTTCAAAATATAACTAATATAGTAGAGTTTGATTCAAAGGACCT
TTTTTTTTGCCTCTTTGATTAGTTTATCTTCTTTTCTTCATTTTATCCCCTAATTTTATA
CGTTAGTTCAACCTAACAATCCAGGATTTCATTAACAAGAAAGGTAAAAGTAACCTATCA
AGGCTATTTTGAAAAAAAAAATTCCGCCCTGAATATTTCGAGTGATTTTCTTAGTGACAA
AGCTTTTTCTTCATCTGTAG
&gt;FAS1	 sequence of the region upstream from YKL182W
CCGGGTTATAGCAGCGTCTGCTCCGCATCACGATACACGAGGTGCAGGCACGGTTCACTA
CTCCCCTGGCCTCCAACAAACGACGGCCAAAAACTTCACATGCCGCCCAGCCAAGCATAA
TTACGCAACAGCGATCTTTCCGTCGCACAAGTTAAAAGAAATTGTTGAAAAATACAAATA
ATCGCGAACAATACGTTGTTGCTATTTAACGCTTTTGGTCTGACAGTAAGTGTGCCTTTC
CCAATCACCGAAAAGTGTTGAACGATTCACTGCGACAATAATCAGAGATTACAGTCGGCA
TTTTGGCATTTTTGGCATACTTTTTATCGATTGAACCATCTTCTCCAAACACTTTTCCTT
TTTCCTTCTATTCTGCAGGACCAACTAAAACTGGGTATATATATCATTATCTATATATAT
AAACGGCTTTCAACAAAGTTATAGGGGAAAACTAAAAATATAAGAAAAAAAAAGGTATTG
ATTGATAAGGAAAAAGAACCAAGGGAAAAATATAAAAAAGTACATTGGGCCTTTTCATAC
TTGTTATCACTTACATTACAAAGAAGAACAAACAACTTTTTTAAACGAATTTTCTTTCTT
CCTTTTTCAATTTATTAATTCTTTTTTTCCATACAATTCAAGGTCAAATATATTCTTATA
TGCTCTTTGAATATTTCTGAAAAATATATAAAGAAAAGAAACTACAAGAACATCATCCGG
AAAATCAGATTATAGACTAGGATTCCGCTCTTTTTAGTATATTTATTCGCCACACCTAAC
TGCTCTATTATTCGCTCATT
&gt;FAS2	 sequence of the region upstream from YPL231W
TCCAGGCAAGGCACCAAGAGTTATTGAAACTAGAAAAATCCATGGCAGAACTTACTCAAT
TGTTTAATGACATGGAAGAACTGGTAATAGAACAACAAGAAAACGTAGACGTCATCGACA
AGAACGTTGAAGACGCTCAACTCGACGTAGAACAGGGTGTCGGTCATACCGATAAAGCCG
TCAAGAGTGCCAGAAAAGCAAGAAAGAACAAGATTAGATGTTGGTTGATTGTATTCGCCA


<font color=red>  [Part of this file has been deleted for brevity]</font>

CTCTTCCTAAAAATACATTGGGCATTACCCGCAAACTAACCCATCGCTTAGCAAAATCCA
ACCATTTTTTTTTTATCTCCCGCGTTTTCACATGCTACCTCATTCGCCTCGTAACGTTAC
GACCGAAATCTCACTAAGGCACGGTTTGTTGGGCAGTTTACAGATGTTGGATAACCAGTT
GTTTCTAAACGGTTATGCCTCATATATAACTTGTTAACTGAAGGTTACACAAGACCACAT
CACCACTGTCGTGCTTTTCTAATAACCGCTATATTAGACGTTTAAAGGGCTACAGCAACA
CCAATTGAAATACCATCATT
&gt;ACC1	 sequence of the region upstream from YNR016C
TATCCAAAGGGGAATGCTTCATCTTGTTGAACAACGCCCAACAATTTCCACTGCCCACCG
AATCGTTGCGCCCGTTAAAATCTTCACATGGCCCGGCCGCGCGCGCGTTGTGCCAACAAG
TCGCAGTCGAAATTCAACCGCTCATTGCCACTCTCTCTACTGCTTGGTGAACTAGGCTAT
ACGCTCAATCAGCGCCAAGATATATAAGAAGAACAGCACTCCCAGTCGTATTCTGGCACA
GTATAGCCTAGCACAATCACTGTCACAATTGTTATCGGTTCTACAATTGTTCTGCTCTCT
TCAATTTTCCTTTCCTTATTCTACTCTTTTTATCCCTTTCGTACAGTTTACCTGAAGATA
AAAAACAACAAAGCCAATTCCCTAATTTGCAATCGCCATTTGCATCTATATATATATATT
TGTTGTGCCATTTTTTTATCCTCTGTGAGTGATCGGTGCATGTGTTTATAAAAGTTTATT
CATTCTACTATACGAACTTTTCCCTCTGCCCTTCCCTCCCGCTTCATCCTTATTTTTGGA
CAATAAACTAGAGAACAATTTGAACTTGAATTGGAATTCAGATTCAGAGCAAGAGACAAG
AAACTTCCCTTTTTCTTCTCCACATATTATTATTTATTCGTGTATTTTCTTTTAACGATA
CGATACGATACGACACGATACGATACGACACGCTACTATACTATACAAATATAATAGTAT
AATAACCGATTCGTCTTCTAGCTTAATTTTTTTCCGTTCCCGAAACAGCGCAGAAAATTA
GAAAAAATCAAGTTTCTACC
&gt;INO1	 sequence of the region upstream from YJL153C
AGCAAACAACCAAATATAATTTAGAAATGGACAGAGACCATATTAATGACCATGACCATC
GAATGAGCTATTCCATCAACAAGGACGACTTGTTGTTAATGGTTTTGGCGGTTTTCATTC
CCCCAGTGGCCGTCTGGAAGCGTAAGGGTATGTTCAACAGGGATACACTATTGAACTTAC
TTCTCTTCCTACTGTTATTCTTCCCAGCAATCATTCACGCTTGCTACGTTGTATATGAAA
CGAGTAGTGAACGTTCGTACGATCTTTCACGCAGACATGCGACTGCGCCCGCCGTAGACC
GTGACCTGGAAGCTCACCCTGCAGAGGAATCTCAAGCACAGCCTCCAGCATATGATGAAG
ACGATGAGGCCGGTGCCGATGTGCCCTTGATGGACAACAAACAACAGCTCTCTTCCGGCC
GTACTTAGTGATCGGAACGAGCTCTTTATCACCGTAGTTCTAAATAACACATAGAGTAAA
TTATTGCCTTTTTCTTCGTTCCTTTTGTTCTTCACGTCCTTTTTATGAAATACGTGCCGG
TGTTCCGGGGTTGGATGCGGAATCGAAAGTGTTGAATGTGAAATATGCGGAGGCCAAGTA
TGCGCTTCGGCGGCTAAATGCGGCATGTGAAAAGTATTGTCTATTTTATCTTCATCCTTC
TTTCCCAGAATATTGAACTTATTTAATTCACATGGAGCAGAGAAAGCGCACCTCTGCGTT
GGCGGCAATGTTAATTTGAGACGTATATAAATTGGAGCTTTCGTCACCTTTTTTTGGCTT
GTTCTGTTGTCGGGTTCCTA
&gt;OPI3	 sequence of the region upstream from YJR073C
GTGTCCACAACGTGAAACTTCCGTACCATTTCTTGCAACAATTGGTAAACAGCATGACAT
CTTGCAGGCAACTCTTTGTTGCTTGCTTGCGACGCCTCCTCCTTTGTCAAAGGTACATTA
ATGGAGATGACCACATCCGTGTCAAACTGGGTTAATCTGATCAACGCTACGCCGATGACA
ACGGTCTGTGCCAGATCTGGTTTTCCCCACTTATTTGCTACTTCCATAACGAGTCCGGTG
AACTTGGTTCCTTGCTGAACAGTGTCTTCTTGTAAAGCTTCCCATTTGGTGGTCCCGTTC
AACTCCGTCAGGTCTTCCACGTGGAACTGCCAAGCCTCCTTCAGATCGCTCTTGTCGACC
GTCTCCAAGAGATCCACGATAATGCTTTCATTGGTGGCTAGTCCATCTTCGAATTCTTCT
TCATCGCGACGGGAATTGACGTACACCTCCTGTGTATCGGGGACTTCTCTTAGAGTAGAA
GCGTCTATAAACCCAGGTGGGACGACAGTAGTGATGGCGCCGCCGTATAATTCGACTTCC
TTGTTGTTCATGCTTCCTTGATGACCAGGGTAGGTGTCAATGAGAGTGCATGTGGAAAGT
TGCACCGGTTGTGAAATATGAGAAGCCTTTTCAATCTTCATATGCAAACCCACACATGCA
TCGTTGGTTTCTGTCCACTGCCACTGCAATGACCACTGGATAAGGGGTCTTTATAAGAGA
ACACATATGAAGAACATGAACGTTCTTGGACAGAGCCATAAACAGCAATTGAAGACAACA
AGAATAGCGCAAGTCAAGCG
</pre>
</td></tr></table><p>
<p><h3>File: yeast.nc.6.freq</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
# seq	frequency_non_coding
a	0.32442758667668
c	0.175572413323319
g	0.175572413323319
t	0.32442758667668
# seq	frequency_non_coding
aa	0.118982244161714
ac	0.0521182743409142
ag	0.0559273922850834
at	0.0973159523835682
ca	0.0584827538751812
cc	0.0326990007534392
cg	0.0284473890701011
ct	0.0559273922850834
ga	0.0559247902310797
gc	0.0348909421343666
gg	0.0326990007534392
gt	0.0521182743409142
ta	0.0910768051171416
tc	0.0559247902310797
tg	0.0584827538751812
tt	0.118982244161714
# seq	observed_freq
aaa	0.049152768651441
aac	0.0174036386740962
aag	0.0213094373095717
aat	0.0313483273294989
aca	0.0183651016732642
acc	0.00948257362793872
acg	0.00868125792953577
act	0.0156686613162602
aga	0.0191771324713567
agc	0.0105445268863571
agg	0.0105978127875158
agt	0.0157042817827957
ata	0.0333561053334843
atc	0.0152910264515268
atg	0.0174586621589883
att	0.0311913655989118
caa	0.0201461250000362
cac	0.0104918201797762
cag	0.0104046513958155
cat	0.0175637859748612
cca	0.0105905728552932
ccc	0.0063256735815742
ccg	0.00537550487667355
cct	0.0106563114398748
cga	0.00831404856720293
cgc	0.00609312695858266
cgg	0.00532859011587077


<font color=red>  [Part of this file has been deleted for brevity]</font>

tttatc	0.000598827491406134
tttatg	0.000612506661319178
tttatt	0.00158183592505095
tttcaa	0.000947937370357122
tttcac	0.000474696300599468
tttcag	0.000478625423872363
tttcat	0.000873720597424649
tttcca	0.000523301010716029
tttccc	0.000362352479611488
tttccg	0.00028871779901574
tttcct	0.000716701189593004
tttcga	0.000341251632405197
tttcgc	0.000242004888993536
tttcgg	0.000211736087483821
tttcgt	0.000410229574307143
tttcta	0.000718884035855724
tttctc	0.000684977157241476
tttctg	0.00052009950286404
tttctt	0.00171891867034976
tttgaa	0.000813910609826126
tttgac	0.000305161907528229
tttgag	0.000387236927006494
tttgat	0.000670424848823344
tttgca	0.000441080468153583
tttgcc	0.000306471615285861
tttgcg	0.000215228641504173
tttgct	0.000500599409583743
tttgga	0.000346635986519906
tttggc	0.000271400551998163
tttggg	0.000238366811889003
tttggt	0.000427110252072176
tttgta	0.000642920985913074
tttgtc	0.000363807710453302
tttgtg	0.000376613741861258
tttgtt	0.00102200862020541
ttttaa	0.00107774396144686
ttttac	0.00076588799204629
ttttag	0.000618473107770613
ttttat	0.00164935863611109
ttttca	0.00119867364440154
ttttcc	0.000846944349935286
ttttcg	0.000516897995012051
ttttct	0.00167235128341174
ttttga	0.00088157884397044
ttttgc	0.000600137199163766
ttttgg	0.000542364534743782
ttttgt	0.00103670645170773
ttttta	0.00171950076268648
tttttc	0.00190678897202784
tttttg	0.00124276713890848
tttttt	0.00570057577663487
</pre>
</td></tr></table><p>

<a name="input.4"></a>
<h3>Input files for usage example 4</h3>
<p><h3>File: lipocalin.s</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;ICYA_MANSE
GDIFYPGYCPDVKPVNDFDLSAFAGAWHEIAKLPLENENQGKCTIAEYKYDGKKASVYNSFVSNGVKEYMEGDLEIAPDA
KYTKQGKYVMTFKFGQRVVNLVPWVLATDYKNYAINYNCDYHPDKKAHSIHAWILSKSKVLEGNTKEVVDNVLKTFSHLI
DASKFISNDFSEAACQYSTTYSLTGPDRH
&gt;LACB_BOVIN
MKCLLLALALTCGAQALIVTQTMKGLDIQKVAGTWYSLAMAASDISLLDAQSAPLRVYVEELKPTPEGDLEILLQKWENG
ECAQKKIIAEKTKIPAVFKIDALNENKVLVLDTDYKKYLLFCMENSAEPEQSLACQCLVRTPEVDDEALEKFDKALKALP
MHIRLSFNPTQLEEQCHI
&gt;BBP_PIEBR
NVYHDGACPEVKPVDNFDWSNYHGKWWEVAKYPNSVEKYGKCGWAEYTPEGKSVKVSNYHVIHGKEYFIEGTAYPVGDSK
IGKIYHKLTYGGVTKENVFNVLSTDNKNYIIGYYCKYDEDKKGHQDFVWVLSRSKVLTGEAKTAVENYLIGSPVVDSQKL
VYSDFSEAACKVN
&gt;RETB_BOVIN
ERDCRVSSFRVKENFDKARFAGTWYAMAKKDPEGLFLQDNIVAEFSVDENGHMSATAKGRVRLLNNWDVCADMVGTFTDT
EDPAKFKMKYWGVASFLQKGNDDHWIIDTDYETFAVQYSCRLLNLDGTCADSYSFVFARDPSGFSPEVQKIVRQRQEELC
LARQYRLIPHNGYCDGKSERNIL
&gt;MUP2_MOUSE
MKMLLLLCLGLTLVCVHAEEASSTGRNFNVEKINGEWHTIILASDKREKIEDNGNFRLFLEQIHVLEKSLVLKFHTVRDE
ECSELSMVADKTEKAGEYSVTYDGFNTFTIPKTDYDNFLMAHLINEKDGETFQLMGLYGREPDLSSDIKERFAKLCEEHG
ILRENIIDLSNANRCLQARE
</pre>
</td></tr></table><p>

<a name="input.5"></a>
<h3>Input files for usage example 5</h3>
<p><h3>File: farntrans5.s</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;RAM1_YEAST PROTEIN FARNESYLTRANSFERASE BETA SUBUNIT (EC 2.5.1.-) (CAAX FARN
MRQRVGRSIA RAKFINTALL GRKRPVMERV VDIAHVDSSK AIQPLMKELE TDTTEARYKV
LQSVLEIYDD EKNIEPALTK EFHKMYLDVA FEISLPPQMT ALDASQPWML YWIANSLKVM
DRDWLSDDTK RKIVVKLFTI SPSGGPFGGG PGQLSHLAST YAAINALSLC DNIDGCWDRI
DRKGIYQWLI SLKEPNGGFK TCLEVGEVDT RGIYCALSIA TLLNILTEEL TEGVLNYLKN
CQNYEGGFGS CPHVDEAHGG YTFCATASLA ILRSMDQINV EKLLEWSSAR QLQEERGFCG
RSNKLVDGCY SFWVGGSAAI LEAFGYGQCF NKHALRDYIL YCCQEKEQPG LRDKPGAHSD
FYHTNYCLLG LAVAESSYSC TPNDSPHNIK CTPDRLIGSS KLTDVNPVYG LPIENVRKII
HYFKSNLSSP S

&gt;PFTB_RAT PROTEIN FARNESYLTRANSFERASE BETA SUBUNIT (EC 2.5.1.-) (CAAX FARNES
MASSSSFTYY CPPSSSPVWS EPLYSLRPEH ARERLQDDSV ETVTSIEQAK VEEKIQEVFS
SYKFNHLVPR LVLQREKHFH YLKRGLRQLT DAYECLDASR PWLCYWILHS LELLDEPIPQ
IVATDVCQFL ELCQSPDGGF GGGPGQYPHL APTYAAVNAL CIIGTEEAYN VINREKLLQY
LYSLKQPDGS FLMHVGGEVD VRSAYCAASV ASLTNIITPD LFEGTAEWIA RCQNWEGGIG
GVPGMEAHGG YTFCGLAALV ILKKERSLNL KSLLQWVTSR QMRFEGGFQG RCNKLVDGCY
SFWQAGLLPL LHRALHAQGD PALSMSHWMF HQQALQEYIL MCCQCPAGGL LDKPGKSRDF
YHTCYCLSGL SIAQHFGSGA MLHDVVMGVP ENVLQPTHPV YNIGPDKVIQ ATTHFLQKPV
PGFEECEDAV TSDPATD

&gt;BET2_YEAST YPT1/SEC4 PROTEINS GERANYLGERANYLTRANSFERASE BETA SUBUNIT (EC 2.
MSGSLTLLKE KHIRYIESLD TNKHNFEYWL TEHLRLNGIY WGLTALCVLD SPETFVKEEV
ISFVLSCWDD KYGAFAPFPR HDAHLLTTLS AVQILATYDA LDVLGKDRKV RLISFIRGNQ
LEDGSFQGDR FGEVDTRFVY TALSALSILG ELTSEVVDPA VDFVLKCYNF DGGFGLCPNA
ESHAAQAFTC LGALAIANKL DMLSDDQLEE IGWWLCERQL PEGGLNGRPS KLPDVCYSWW
VLSSLAIIGR LDWINYEKLT EFILKCQDEK KGGISDRPEN EVDVFHTVFG VAGLSLMGYD
NLVPIDPIYC MPKSVTSKFK KYPYK

&gt;RATRABGERB Rat rab geranylgeranyl transferase beta-subunit
MGTQQKDVTIKSDAPDTLLLEKHADYIASYGSKKDDYEYCMSEY
LRMSGVYWGLTVMDLMGQLHRMNKEEILVFIKSCQHECGGVSASIGHDPHLLYTLSAV
QILTLYDSIHVINVDKVVAYVQSLQKEDGSFAGDIWGEIDTRFSFCAVATLALLGKLD
AINVEKAIEFVLSCMNFDGGFGCRPGSESHAGQIYCCTGFLAITSQLHQVNSDLLGWW
LCERQLPSGGLNGRPEKLPDVCYSWWVLASLKIIGRLHWIDREKLRSFILACQDEETG
GFADRPGDMVDPFHTLFGIAGLSLLGEEQIKPVSPVFCMPEEVLQRVNVQPELVS

&gt;CAL1_YEAST RAS PROTEINS GERANYLGERANYLTRANSFERASE (EC 2.5.1.-) (PROTEIN GER
MCQATNGPSR VVTKKHRKFF ERHLQLLPSS HQGHDVNRMA IIFYSISGLS IFDVNVSAKY
GDHLGWMRKH YIKTVLDDTE NTVISGFVGS LVMNIPHATT INLPNTLFAL LSMIMLRDYE
YFETILDKRS LARFVSKCQR PDRGSFVSCL DYKTNCGSSV DSDDLRFCYI AVAILYICGC
RSKEDFDEYI DTEKLLGYIM SQQCYNGAFG AHNEPHSGYT SCALSTLALL SSLEKLSDKF
KEDTITWLLH RQVSSHGCMK FESELNASYD QSDDGGFQGR ENKFADTCYA FWCLNSLHLL
TKDWKMLCQT ELVTNYLLDR TQKTLTGGFS KNDEEDADLY HSCLGSAALA LIEGKFNGEL
CIPQEIFNDF SKRCCF

</pre>
</td></tr></table><p>

<a name="input.8"></a>
<h3>Input files for usage example 8</h3>
<p><h3>File: adh.s</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;2BHD_STREX 20-BETA-HYDROXYSTEROID DEHYDROGENASE (EC 1.1.1.53)
MNDLSGKTVIITGGARGLGAEAARQAVAAGARVVLADVLDEEGAATARELGDAARYQHLDVTIEEDWQRVVAYAREEFGSVDGLVNNAGISTGMFLETESVERFRKVVDINLTGVFIGMKTVIPAMKDAGGGSIVNISSAAGLMGLALTSSYGASKWGVRGLSKLAAVELGTDRIRVNSVHPGMTYTPMTAETGIRQGEGNYPNTPMGRVGNEPGEIAGAVVKLLSDTSSYVTGAELAVDGGWTTGPTVKYVMGQ
&gt;3BHD_COMTE 3-BETA-HYDROXYSTEROID DEHYDROGENASE (EC 1.1.1.51)
TNRLQGKVALVTGGASGVGLEVVKLLLGEGAKVAFSDINEAAGQQLAAELGERSMFVRHDVSSEADWTLVMAAVQRRLGTLNVLVNNAGILLPGDMETGRLEDFSRLLKINTESVFIGCQQGIAAMKETGGSIINMASVSSWLPIEQYAGYSASKAAVSALTRAAALSCRKQGYAIRVNSIHPDGIYTPMMQASLPKGVSKEMVLHDPKLNRAGRAYMPERIAQLVLFLASDESSVMSGGELHADNSILGMGL
&gt;ADH_DROME ALCOHOL DEHYDROGENASE (EC 1.1.1.1)
SFTLTNKNVIFVAGLGGIGLDTSKELLKRDLKNLVILDRIENPAAIAELKAINPKVTVTFYPYDVTVPIAETTKLLKTIFAQLKTVDVLINGAGILDDHQIERTIAVNYTGLVNTTTAILDFWDKRKGGPGGIICNIGSVTGFNAIYQVPVYSGTKAAVVNFTSSLAKLAPITGVTAYTVNPGITRTTLVHKFNSWLDVEPQVAEKLLAHPTQPSLACAENFVKAIELNQNGAIWKLDLGTLEAIQWTKHWDSGI
&gt;AP27_MOUSE ADIPOCYTE P27 PROTEIN (AP27)
MKLNFSGLRALVTGAGKGIGRDTVKALHASGAKVVAVTRTNSDLVSLAKECPGIEPVCVDLGDWDATEKALGGIGPVDLLVNNAALVIMQPFLEVTKEAFDRSFSVNLRSVFQVSQMVARDMINRGVPGSIVNVSSMVAHVTFPNLITYSSTKGAMTMLTKAMAMELGPHKIRVNSVNPTVVLTDMGKKVSADPEFARKLKERHPLRKFAEVEDVVNSILFLLSDRSASTSGGGILVDAGYLAS
&gt;BA72_EUBSP 7-ALPHA-HYDROXYSTEROID DEHYDROGENASE (EC 1.1.1.159) (BILE ACID 7-DEHYDROXYLASE) (BILE ACID-INDUCIBLE PROTEIN)
MNLVQDKVTIITGGTRGIGFAAAKIFIDNGAKVSIFGETQEEVDTALAQLKELYPEEEVLGFAPDLTSRDAVMAAVGQVAQKYGRLDVMINNAGITSNNVFSRVSEEEFKHIMDINVTGVFNGAWCAYQCMKDAKKGVIINTASVTGIFGSLSGVGYPASKASVIGLTHGLGREIIRKNIRVVGVAPGVVNTDMTNGNPPEIMEGYLKALPMKRMLEPEEIANVYLFLASDLASGITATTVSVDGAYRP
&gt;BDH_HUMAN D-BETA-HYDROXYBUTYRATE DEHYDROGENASE PRECURSOR (EC 1.1.1.30) (BDH) (3-HYDROXYBUTYRATE DEHYDROGENASE) (FRAGMENT)
GLRPPPPGRFSRLPGKTLSACDRENGARRPLLLGSTSFIPIGRRTYASAAEPVGSKAVLVTGCDSGFGFSLAKHLHSKGFLVFAGCLMKDKGHDGVKELDSLNSDRLRTVQLNVFRSEEVEKVVGDCPFEPEGPEKGMWGLVNNAGISTFGEVEFTSLETYKQVAEVNLWGTVRMTKSFLPLIRRAKGRVVNISSMLGRMANPARSPYCITKFGVEAFSDCLRYEMYPLGVKVSVVEPGNFIAATSLYNPESIQAIAKKMWEELPEVVRKDYGKKYFDEKIAKMETYCSSGSTDTSPVIDAVTHALTATTPYTRYHPMDYYWWLRMQIMTHLPGAISDMIYIR
&gt;BPHB_PSEPS BIPHENYL-CIS-DIOL DEHYDROGENASE (EC 1.3.1.-)
MKLKGEAVLITGGASGLGRALVDRFVAEAKVAVLDKSAERLAELETDLGDNVLGIVGDVRSLEDQKQAASRCVARFGKIDTLIPNAGIWDYSTALVDLPEESLDAAFDEVFHINVKGYIHAVKALPALVASRGNVIFTISNAGFYPNGGGPLYTAAKQAIVGLVRELAFELAPYVRVNGVGPGGMNSDMRGPSSLGMGSKAISTVPLADMLKSVLPIGRMPEVEEYTGAYVFFATRGDAAPASGALVNYDGGLGVRGFFSGAGGNDLLEQLNIHP
&gt;BUDC_KLETE ACETOIN(DIACETYL) REDUCTASE (EC 1.1.1.5) (ACETOIN DEHYDROGENASE)
MQKVALVTGAGQGIGKAIALRLVKDGFAVAIADYNDATATAVAAEINQAGGRAVAIKVDVSRRDQVFAAVEQARKALGGFNVIVNNAGIAPSTPIESITEEIVDRVYNINVKGVIWGMQAAVEAFKKEGHGGKIVNACSQAGHVGNPELAVYSSSKFAVRGLTQTAARDLAPLGITVNGFCPGIVKTPMWAEIDRQCRKRRANRWATARLNLPNASPLAACRSLKTSPPACRSSPARIPTI
&gt;DHES_HUMAN ESTRADIOL 17 BETA-DEHYDROGENASE (EC 1.1.1.62) (20 ALPHA-HYDROXYSTEROID DEHYDROGENASE) (E2DH) (17-BETA-HSD) (PLACENTAL 17-BETA-HYDROXYSTEROID DEHYDROGENASE)
ARTVVLITGCSSGIGLHLAVRLASDPSQSFKVYATLRDLKTQGRLWEAARALACPPGSLETLQLDVRDSKSVAAARERVTEGRVDVLVCNAGLGLLGPLEALGEDAVASVLDVNVVGTVRMLQAFLPDMKRRGSGRVLVTGSVGGLMGLPFNDVYCASKFALEGLCESLAVLLLPFGVHLSLIECGPVHTAFMEKVLGSPEEVLDRTDIHTFHRFYQYLAHSKQVFREAAQNPEEVAEVFLTALRAPKPTLRYFTTERFLPLLRMRLDDPSGSNYVTAMHREVFGDVPAKAEAGAEAGGGAGPGAEDEAGRSAVGDPELGDPPAAPQ
&gt;DHGB_BACME GLUCOSE 1-DEHYDROGENASE B (EC 1.1.1.47)
MYKDLEGKVVVITGSSTGLGKSMAIRFATEKAKVVVNYRSKEDEANSVLEEEIKKVGGEAIAVKGDVTVESDVINLVQSAIKEFGKLDVMINNAGMENPVSSHEMSLSDWNKVIDTNLTGAFLGSREAIKYFVENDIKGTVINMSSVHEWKIPWPLFVHYAASKGGMKLMTETLALEYAPKGIRVNNIGPGAINTPINAEKFADPEQRADVESMIPMGYIGEPEEIAAVAWLASSEASYVTGITLFADGGMTQYPSFQAGRG
&gt;DHII_HUMAN CORTICOSTEROID 11-BETA-DEHYDROGENASE (EC 1.1.1.146) (11-DH) (11-BETA- HYDROXYSTEROID DEHYDROGENASE) (11-BETA-HSD)
MAFMKKYLLPILGLFMAYYYYSANEEFRPEMLQGKKVIVTGASKGIGREMAYHLAKMGAHVVVTARSKETLQKVVSHCLELGAASAHYIAGTMEDMTFAEQFVAQAGKLMGGLDMLILNHITNTSLNLFHDDIHHVRKSMEVNFLSYVVLTVAALPMLKQSNGSIVVVSSLAGKVAYPMVAAYSASKFALDGFFSSIRKEYSVSRVNVSITLCVLGLIDTETAMKAVSGIVHMQAAPKEECALEIIKGGALRQEEVYYDSSLWTTLLIRNPCRKILEFLYSTSYNMDRFINK
&gt;DHMA_FLAS1 N-ACYLMANNOSAMINE 1-DEHYDROGENASE (EC 1.1.1.233) (NAM-DH) 
TTAGVSRRPGRLAGKAAIVTGAAGGIGRATVEAYLREGASVVAMDLAPRLAATRYEEPGAIPIACDLADRAAIDAAMADAVARLGGLDILVAGGALKGGTGNFLDLSDADWDRYVDVNMTGTFLTCRAGARMAVAAGAGKDGRSARIITIGSVNSFMAEPEAAAYVAAKGGVAMLTRAMAVDLARHGILVNMIAPGPVDVTGNNTGYSEPRLAEQVLDEVALGRPGLPEEVATAAVFLAEDGSSFITGSTITIDGGLSAMIFGGMREGRR
&gt;ENTA_ECOLI 2,3-DIHYDRO-2,3-DIHYDROXYBENZOATE DEHYDROGENASE (EC 1.3.1.28)
MDFSGKNVWVTGAGKGIGYATALAFVEAGAKVTGFDQAFTQEQYPFATEVMDVADAAQVAQVCQRLLAETERLDALVNAAGILRMGATDQLSKEDWQQTFAVNVGGAFNLFQQTMNQFRRQRGGAIVTVASDAAHTPRIGMSAYGASKAALKSLALSVGLELAGSGVRCNVVSPGSTDTDMQRTLWVSDDAEEQRIRGFGEQFKLGIPLGKIARPQEIANTILFLASDLASHITLQDIVVDGGSTLGA
&gt;FIXR_BRAJA FIXR PROTEIN
MGLDLPNDNLIRGPLPEAHLDRLVDAVNARVDRGEPKVMLLTGASRGIGHATAKLFSEAGWRIISCARQPFDGERCPWEAGNDDHFQVDLGDHRMLPRAITEVKKRLAGAPLHALVNNAGVSPKTPTGDRMTSLTTSTDTWMRVFHLNLVAPILLAQGLFDELRAASGSIVNVTSIAGSRVHPFAGSAYATSKAALASLTRELAHDYAPHGIRVNAIAPGEIRTDMLSPDAEARVVASIPLRRVGTPDEVAKVIFFLCSDAASYVTGAEVPINGGQHL
&gt;GUTD_ECOLI SORBITOL-6-PHOSPHATE 2-DEHYDROGENASE (EC 1.1.1.140) (GLUCITOL-6- PHOSPHATE DEHYDROGENASE) (KETOSEPHOSPHATE REDUCTASE)
MNQVAVVIGGGQTLGAFLCHGLAAEGYRVAVVDIQSDKAANVAQEINAEYGESMAYGFGADATSEQSCLALSRGVDEIFGRVDLLVYSAGIAKAAFISDFQLGDFDRSLQVNLVGYFLCAREFSRLMIRDGIQGRIIQINSKSGKVGSKHNSGYSAAKFGGVGLTQSLALDLAEYGITVHSLMLGNLLKSPMFQSLLPQYATKLGIKPDQVEQYYIDKVPLKRGCDYQDVLNMLLFYASPKASYCTGQSINVTGGQVMF
&gt;HDE_CANTR HYDRATASE-DEHYDROGENASE-EPIMERASE (HDE)
MSPVDFKDKVVIITGAGGGLGKYYSLEFAKLGAKVVVNDLGGALNGQGGNSKAADVVVDEIVKNGGVAVADYNNVLDGDKIVETAVKNFGTVHVIINNAGILRDASMKKMTEKDYKLVIDVHLNGAFAVTKAAWPYFQKQKYGRIVNTSSPAGLYGNFGQANYASAKSALLGFAETLAKEGAKYNIKANAIAPLARSRMTESILPPPMLEKLGPEKVAPLVLYLSSAENELTGQFFEVAAGFYAQIRWERSGGVLFKPDQSFTAEVVAKRFSEILDYDDSRKPEYLKNQYPFMLNDYATLTNE
ARKLPANDASGAPTVSLKDKVVLITGAGAGLGKEYAKWFAKYGAKVVVNDFKDATKTVDEIKAAGGEAWPDQHDVAKDSEAIIKNVIDKYGTIDILVNNAGILRDRSFAKMSKQEWDSVQQVHLIGTFNLSRLAWPYFVEKQFGRIINITSTSGIYGNFGQANYSSSKAGILGLSKTMAIEGAKNNIKVNIVAPHAETAMTLTIFREQDKNLYHADQVAPLLVYLGTDDVPVTGETSEIGGGWIGNTRWQRAKGAVSHDEHTTVEFIKEHLNEITDFTTDTENPKSTTESSMAILSAVGGDDD
DDDEDEEEDEGDEEEDEEDEEEDDPVWRFDDRDVILYNIALGATTKQLKYVYENDSDFQVIPTFGHLITFNSGKSQNSFAKLLRNFNPMLLLHGEHYLKVHSWPPPTEGEIKTTFEPIATTPKGTNVVIVHGSKSVDNKSGELIYSNEATYFIRNCQADNKVYADRPAFATNQFLAPKRAPDYQVDVPVSEDLAALYRLSGDRNPLHIDPNFAKGAKFPKPILHGMCTYGLSAKALIDKFGMFNEIKARFTGIVFPGETLRVLAWKESDDTIVFQTHVVDRGTIAINNAAIKLVGDKAKI
&gt;HDHA_ECOLI 7-ALPHA-HYDROXYSTEROID DEHYDROGENASE (EC 1.1.1.159) (HSDH)
MFNSDNLRLDGKCAIITGAGAGIGKEIAITFATAGASVVVSDINADAANHVVDEIQQLGGQAFACRCDITSEQELSALADFAISKLGKVDILVNNAGGGGPKPFDMPMADFRRAYELNVFSFFHLSQLVAPEMEKNGGGVILTITSMAAENKNINMTSYASSKAAASHLVRNMAFDLGEKNIRVNGIAPGAILTDALKSVITPEIEQKMLQHTPIRRLGQPQDIANAALFLCSPAASWVSGQILTVSGGGVQELN
&gt;LIGD_PSEPA C ALPHA-DEHYDROGENASE (EC -.-.-.-)
MKDFQDQVAFITGGASGAGFGQAKVFGQAGAKIVVADVRAEAVEKAVAELEGLGITAHGIVLDIMDREAYARAADEVEAVFGQAPTLLSNTAGVNSFGPIEKTTYDDFDWIIGVNLNGVINGMVTFVPRMIASGRPGHIVTVSSLGGFMGSALAGPYSAAKAASINLMEGYRQGLEKYGIGVSVCTPANIKSNIAEASRLRPAKYGTSGYVENEESIASLHSIHQHGLEPEKLAEAIKKGVEDNALYIIPYPEVREGLEKHFQAIIDSVAPMESDPEGARQRVEALMAWGRDRTRVFAEGDKKGA
&gt;NODG_RHIME NODULATION PROTEIN G (HOST-SPECIFICITY OF NODULATION PROTEIN C)
MFELTGRKALVTGASGAIGGAIARVLHAQGAIVGLHGTQIEKLETLATELGDRVKLFPANLANRDEVKALGQRAEADLEGVDILVNNAGITKDGLFLHMADPDWDIVLEVNLTAMFRLTREITQQMIRRRNGRIINVTSVAGAIGNPGQTNYCASKAGMIGFSKSLAQEIATRNITVNCVAPGFIESAMTDKLNHKQKEKIMVAIPIHRMGTGTEVASAVAYLASDHAAYVTGQTIHVNGGMAMI
&gt;RIDH_KLEAE RIBITOL 2-DEHYDROGENASE (EC 1.1.1.56) (RDH)
MKHSVSSMNTSLSGKVAAITGAASGIGLECARTLLGAGAKVVLIDREGEKLNKLVAELGENAFALQVDLMQADQVDNLLQGILQLTGRLDIFHANAGAYIGGPVAEGDPDVWDRVLHLNINAAFRCVRSVLPHLIAQKSGDIIFTAVIAGVVPVIWEPVYTASKFAVQAFVHTTRRQVAQYGVRVGAVLPGPVVTALLDDWPKAKMDEALANGSLMQPIEVAESVLFMVTRSKNVTVRDIVILPNSVDL
&gt;YINL_LISMO HYPOTHETICAL 26.8 KD PROTEIN IN INLA 5'REGION (ORFA)
MTIKNKVIIITGASSGIGKATALLLAEKGAKLVLAARRVEKLEKIVQIIKANSGEAIFAKTDVTKREDNKKLVELAIERYGKVDAIFLNAGIMPNSPLSALKEDEWEQMIDINIKGVLNGIAAVLPSFIAQKSGHIIATSSVAGLKAYPGGAVYGATKWAVRDLMEVLRMESAQEGTNIRTATIYPAAINTELLETITDKETEQGMTSLYKQYGITPDRIASIVAYAIDQPEDVNVNEFTVGPTSQPW
&gt;YRTP_BACSU HYPOTHETICAL 25.3 KD PROTEIN IN RTP 5'REGION (ORF238)
MQSLQHKTALITGGGRGIGRATALALAKEGVNIGLIGRTSANVEKVAEEVKALGVKAAFAAADVKDADQVNQAVAQVKEQLGDIDILINNAGISKFGGFLDLSADEWENIIQVNLMGVYHVTRAVLPEMIERKAGDIINISSTAGQRGAAVTSAYSASKFAVLGLTESLMQEVRKHNIRVSALTPSTVASDMSIELNLTDGNPEKVMQPEDLAEYMVAQLKLDPRIFIKTAGLWSTNP
&gt;CSGA_MYXXA no comment
MRAFATNVCTGPVDVLINNAGVSGLWCALGDVDYADMARTFTINALGPLR
VTSAMLPGLRQGALRRVAHVTSRMGSLAANTDGGAYAYRMSKAALNMAVR
SMSTDLRPEGFVTVLLHPGWVQTDMGGPDATLPAPDSVRGMLRVIDGLNP


<font color=red>  [Part of this file has been deleted for brevity]</font>

FSIAAMNELELK
&gt;FVT1_HUMAN no comment
MLLLAAAFLVAFVLLLYMVSPLISPKPLALPGAHVVVTGGSSGIGKCIAI
ECYKQGAFITLVARNEDKLLQAKKEIEMHSINDKQVVLCISVDVSQDYNQ
VENVIKQAQEKLGPVDMLVNCAGMAVSGKFEDLEVSTFERLMSINYLGSV
YPSRAVITTMKERRVGRIVFVSSQAGQLGLFGFTAYSASKFAIRGLAEAL
QMEVKPYNVYITVAYPPDTDTPGFAEENRTKPLETRLISETTSVCKPEQV
AKQIVKDAIQGNFNSSLGSDGYMLSALTCGMAPVTSITEGLQQVVTMGLF
RTIALFYLGSFDSIVRRCMMQREKSENADKTA
&gt;HMTR_LEIMA no comment
MTAPTVPVALVTGAAKRLGRSIAEGLHAEGYAVCLHYHRSAAEANALSAT
LNARRPNSAITVQADLSNVATAPVSGADGSAPVTLFTRCAELVAACYTHW
GRCDVLVNNASSFYPTPLLRNDEDGHEPCVGDREAMETATADLFGSNAIA
PYFLIKAFAHRSRHPSQASRTNYSIINMVDAMTNQPLLGYTIYTMAKGAL
EGLTRSAALELAPLQIRVNGVGPGLSVLVDDMPPAVWEGHRSKVPLYQRD
SSAAEVSDVVIFLCSSKAKYITGTCVKVDGGYSLTRA
&gt;MAS1_AGRRA no comment
MHQLWAYDVGTLGCVSYHALPDIKRHSPKSGHLYLNKPSLRSFILQCPSL
ARTLVLPSHQPVSRSSTSSAMVQPISTRKKCTCKVKNIGVCRAPARTSVS
MELANAKRFSPATFSANFLSXSVVCSPLLRAIQTALIANIGFLCFDIDED
LKERDFGKHEGGYGPLKMFEDNYPDCEDTEMFSLRVAKALTHAKNENTLF
VSHGGVLRVIAALLGVDLTKEHTNNGRVLHFRRGFSHWTVEIHQSPVILV
SGSNRGVGKAIAEDLIAHGYRLSLGARKVKDLEVAFGPQDEWLHYARFDA
EDHGTMAAWVTAAVEKFGRIDGLVNNAGYGEPVNLDKHVDYQRFHLQWYI
NCVAPLRMTELCLPHLYETGSGRIVNINSMSGQRVLNPLVGYNMTKHALG
GLTKTTQHVGWDRRCAAIDICLGFVATDMSAWTDLIASKDMIQPEDIAKL
VREAIERPNRAYVPRSEVMCIKEATR
&gt;PCR_PEA no comment
MALQTASMLPASFSIPKEGKIGASLKDSTLFGVSSLSDSLKGDFTSSALR
CKELRQKVGAVRAETAAPATPAVNKSSSEGKKTLRKGNVVITGASSGLGL
ATAKALAESGKWHVIMACRDYLKAARAAKSAGLAKENYTIMHLDLASLDS
VRQFVDNFRRSEMPLDVLINNAAVYFPTAKEPSFTADGFEISVGTNHLGH
FLLSRLLLEDLKKSDYPSKRLIIVGSITGNTNTLAGNVPPKANLGDLRGL
AGGLTGLNSSAMIDGGDFDGAKAYKDSKVCNMLTMQEFHRRYHEETGITF
ASLYPGCIATTGLFREHIPLFRTLFPPFQKYITKGYVSEEESGKRLAQVV
SDPSLTKSGVYWSWNNASASFENQLSQEASDAEKARKVWEVSEKLVGLA
&gt;RFBB_NEIGO no comment
MQTEGKKNILVTGGAGFIGSAVVRHIIQNTRDSVVNLDKLTYAGNLESLT
DIADNPRYAFEQVDICDRAELDRVFAQYRPDAVMHLAAESHVDRAIGSAG
EFIRTNIVGTFDLLEAARAYWQQMPSEKREAFRFHHISTDEVYGDLHGTD
DLFTETTPYAPSSPYSASKAAADHLVRAWQRTYRLPSIVSNCSNNYGPRQ
FPEKLIPLMILNALSGKPLPVYGDGAQIRDWLFVEDHARALYQVVTEGVV
GETYNIGGHNEKTNLEVVKTICALLEELAPEKPAGVARYEDLITFVQDRP
GHDARYAVDAAKIRRDLGWLPLETFESGLRKTVQWYLDNKTRRQNA
&gt;YURA_MYXXA no comment
RQHTGGLHGGDELPDGVGDGCLQRPGTRAGAVARQAGVRVFAAGRRLPQL
QAADEAPGGRRHRGARGVDVTKADATLERIRALDAEAGGLDLVVANAGVG
GTTNAKRLPWERVRGIIDTNVTGAAATLSAVLPQMVERKRGHLVGVSSLA
GFRGLPATRYSASKAFLSTFMESLRVDLRGTGVRVTCIYPGFVKSELTAT
NNFPMPFLMETHDAVELMGKGIVRGDAEVSFPWQLAVPTRMAKVLPNPLF
DAAARRLR
</pre>
</td></tr></table><p>





<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
include file="inc/reportformats.ihtml"
   -->




<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: ex2.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: ex3.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>

<a name="output.3"></a>
<h3>Output files for usage example 3</h3>
<p><h3>File: ex4.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>

<a name="output.4"></a>
<h3>Output files for usage example 4</h3>
<p><h3>File: ex5.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>

<a name="output.5"></a>
<h3>Output files for usage example 5</h3>
<p><h3>File: ex6.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>

<a name="output.6"></a>
<h3>Output files for usage example 6</h3>
<p><h3>File: ex7.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>

<a name="output.7"></a>
<h3>Output files for usage example 7</h3>
<p><h3>File: ex8.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>

<a name="output.8"></a>
<h3>Output files for usage example 8</h3>
<p><h3>File: ex9.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>

<a name="output.9"></a>
<h3>Output files for usage example 9</h3>
<p><h3>File: ex1.html</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
&lt;!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"&gt;
&lt;HTML&gt;
&lt;HEAD&gt;
&lt;meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1"&gt;
&lt;TITLE&gt;MEME&lt;/TITLE&gt;
&lt;STYLE type="text/css"&gt;
  TD.invisible { color: '#D5F0FF'; }
  TD.c0 { background: aqua; color: black; }
  TD.cw0 { background: aqua; color: black; font: 50% sans-serif; }
  TD.c1 { background: blue; color: white; }
  TD.cw1 { background: blue; color: white; font: 50% sans-serif; }
  TD.c2 { background: red; color: white; }
  TD.cw2 { background: red; color: white; font: 50% sans-serif; }
  TD.c3 { background: fuchsia; color: black; }
  TD.cw3 { background: fuchsia; color: black; font: 50% sans-serif; }
  TD.c4 { background: yellow; color: black; }
  TD.cw4 { background: yellow; color: black; font: 50% sans-serif; }
  TD.c5 { background: lime; color: black; }
  TD.cw5 { background: lime; color: black; font: 50% sans-serif; }
  TD.c6 { background: teal; color: white; }
  TD.cw6 { background: teal; color: white; font: 50% sans-serif; }
  TD.c7 { background: #444444; color: white; }
  TD.cw7 { background: #444444; color: white; font: 50% sans-serif; }
  TD.c8 { background: green; color: white; }
  TD.cw8 { background: green; color: white; font: 50% sans-serif; }
  TD.c9 { background: silver; color: black; }
  TD.cw9 { background: silver; color: black; font: 50% sans-serif; }
  TD.c10 { background: purple; color: white; }
  TD.cw10 { background: purple; color: white; font: 50% sans-serif; }
  TD.c11 { background: olive; color: black; }
  TD.cw11 { background: olive; color: black; font: 50% sans-serif; }
  TD.c12 { background: navy; color: white; }
  TD.cw12 { background: navy; color: white; font: 50% sans-serif; }
  TD.c13 { background: maroon; color: white; }
  TD.cw13 { background: maroon; color: white; font: 50% sans-serif; }
  TD.c14 { background: black; color: white; }
  TD.cw14 { background: black; color: white; font: 50% sans-serif; }
  TD.c15 { background: white; color: black; }
  TD.cw15 { background: white; color: black; font: 50% sans-serif; }
  B.red { color: red; }
  TD.red { color: red; }
  TH.red { color: red; }
  B.blue { color: blue; }
  TD.blue { color: blue; }
  TH.blue { color: blue; }
  B.orange { color: orange; }
  TD.orange { color: orange; }
  TH.orange { color: orange; }
  B.green { color: green; }
  TD.green { color: green; }


<font color=red>  [Part of this file has been deleted for brevity]</font>

position one first.
The motif is preceded by a line starting with
"letter-probability matrix:" and containing the length of the alphabet, width
of the motif, number of occurrences of the motif, and the &lt;I&gt;E&lt;/I&gt;-value of the
motif.
&lt;p&gt;
&lt;b&gt;Note:&lt;/b&gt; Earlier versions
of MEME gave the posterior probabilities--the probability after applying
a prior on letter frequencies--rather than the observed frequencies.
These versions of MEME also gave the number of &lt;I&gt;possible&lt;/I&gt;
positions for the motif rather than the actual number of occurrences.
The output from these earlier versions of MEME can be distinguished
by "n=" rather than "nsites=" in the line preceding the matrix.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=regular_expression_doc2 HREF=#regular_expression1&gt;&lt;H4&gt;
Regular Expression&lt;/H4&gt;&lt;/A&gt;
This is the &lt;A HREF=#consensus_doc2&gt;multilevel consensus&lt;/A&gt; expressed as
a regular expression for convenience.  Regular expressions can
be used for searching for against sequences (using, for example,
&lt;A HREF="http://nar.oxfordjournals.org/cgi/content/full/33/suppl_2/W262"&gt;PatMatch&lt;/A&gt;)  
but the search accuracy will usually be better with the PSSM (using,
for example
&lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.)
MEME regular expressions are interpreted as follows: 
single letters match that letter; groups of letters in square brackets 
match any of the letters in the group.

&lt;P&gt;
&lt;LI&gt; &lt;A NAME=motif-summary-doc2 HREF=#motif-summary&gt;&lt;H4&gt;
Motif Summary Tiling&lt;/H4&gt;&lt;/A&gt;
The motif summary tiling is done using the same algorithm as used
by &lt;A HREF=http://meme.nbcr.net/meme/mast-intro.html&gt;MAST&lt;/A&gt;.
The motif occurrences shown in the motif summary
&lt;B&gt;may not be exactly the same as those reported in each motif section&lt;/B&gt;
because only motifs with a position &lt;em&gt;p&lt;/em&gt;-value of 0.0001 that
don't overlap other, more significant motif occurrences are shown.
The format of the machine readable motif-summary is:
&lt;pre&gt;
[sequence_name combined_&lt;em&gt;p&lt;/em&gt;-value number_of_motif_occurrences [motif_number start_of_motif position_&lt;em&gt;p&lt;/em&gt;-value]+]+
&lt;/pre&gt;
See the documentation for 
&lt;A HREF=http://meme.nbcr.net/meme/mast-output.html&gt;MAST output&lt;/A&gt; for the definition of position and
combined &lt;em&gt;p&lt;/em&gt;-values.

&lt;/UL&gt;
&lt;HR&gt;&lt;TABLE SUMMARY='buttons' ALIGN=LEFT CELLSPACING=0&gt;&lt;TR&gt;
  &lt;TD BGCOLOR='#DDDDFF'&gt;&lt;A HREF="#top_buttons"&gt;&lt;B&gt;Go to top&lt;/B&gt;&lt;/A&gt;&lt;/TABLE&gt;&lt;BR&gt;
&lt;/FORM&gt;
&lt;/BODY&gt;
&lt;/HTML&gt;
</td></tr></table><p>





<p>  The MEME results consist of:
<ul>  
<li>         The version of MEME and the date it was released. 
<li>         The reference to cite if you use MEME in your research. 
<li>          A description of the sequences you submitted (the "training set")
         showing the name, "weight" and length of each sequence. 
<li>         The command line summary detailing the parameters with which you
         ran MEME. 
<li>         Information on each of the motifs MEME discovered, including: 
  <ol>
  <li>       1.A summary line showing the width, number of occurrences, log
                likelihood ratio and statistical significance of the motif. 
  <li>       2.A simplified position-specific probability matrix. 
  <li>       3.A diagram showing the degree of conservation at each motif
                position. 
  <li>       4.A multilevel consensus sequence showing the most conserved
                letter(s) at each motif position. 
  <li>       5.The occurrences of the motif sorted by p-value and aligned with
                each other. 
  <li>       6.Block diagrams of the occurrences of the motif within each
                sequence in the training set. 
  <li>       7.The motif in BLOCKS format. 
  <li>       8.A position-specific scoring matrix (PSSM) for use by the
                MAST database search program. 
  <li>       9.The position specific probability matrix (PSPM) describing the
  </ol>              motif. 

<li>     A summary of motifs showing an optimized (non-overlapping) tiling of
         all of the motifs onto each of the sequences in the training set. 
  
<li>         The reason why MEME stopped and the name of the CPU on which it ran. 
<li>         This explanation of how to interpret MEME results.  
</ul>




<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.

	If you wish to include the standard description of what data
	files are and how to use embossdata to inspect and retrieve
	them, use:
include file="inc/localfiles.ihtml"
   -->

None.













<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->
<H3>1. Command-line arguments</H3>
The following original MEME options are not supported:
<br><pre>
-h         : Use -help to get help information.
-dna	   : EMBOSS will specify whether sequences use a DNA alphabet 
             automatically.
-protein   : EMBOSS will specify whether sequences use a protein alphabet 
             automatically.
</pre>

<p>
The following additional options are provided:
<br><pre>
outfile    : Application output that was normally written to stdout.
</pre>

Note: ememe makes a temporary local copy of its input sequence data.  You must ensure there is sufficient disk space for this in the directory that ememe is run.


<H3>2. Installing EMBASSY MEME</H3>
The EMBASSY MEME package contains "wrapper" applications providing an EMBOSS-style interface to the applications in the original MEME package version 3.0.14 developed by Timothy L. Bailey.  Please read the file README in the EMBASSY MEME package distribution for installation instructions.


<H3>3. Installing original MEME</H3>
To use EMBASSY MEME, you will first need to download and install the original MEME package:
<pre>
WWW home:       http://meme.sdsc.edu/meme/
Distribution:   http://meme.nbcr.net/downloads/old_versions/  
</pre>
Please read the file README in the the original MEME package distribution for installation instructions.


<H3>4. Setting up MEME</H3>
For the EMBASSY MEME package to work, the directory containing the original MEME executables *must* be in your path. For example if you executables were installed to "/usr/local/meme/bin", then type:
<pre>
set path=(/usr/local/meme/bin/ $path)
rehash
</pre>


<H3> 5. Getting help </H3>
Once you have installed the original MEME, type
<pre>
meme > meme.txt 
mast > mast.txt 
</pre>
to retrieve the meme and mast documentation into text files. The same documentation is given here and in the ememe documentation.
<p>Please read the 'Notes' section below for a description of the differences between the original and EMBASSY MEME, particularly which application command line options are supported. 



<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->
<p>(MEME) Timothy L. Bailey and Charles Elkan, "Fitting a mixture model by expectation maximization to discover motifs in biopolymers", Proceedings of the Second International Conference on Intelligent Systems for Molecular Biology, pp. 28-36, AAAI Press, Menlo Park, California, 1994.
<p>(MAST) Timothy L. Bailey and Michael Gribskov, "Combining evidence using p-values: application to sequence homology searches", Bioinformatics, Vol. 14, pp. 48-54, 1998. 







<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->
<H3>Input data</H3>


<H3>Sequence input </H3>
Note: ememe makes a temporary local copy of its input sequence data.  You must ensure there is sufficient disk space for this in the directory that ememe is run.
<p>The user must provide the full filename of a sequence database for the sequence input ("seqset" ACD option), not an indirect reference, e.g. a USA is NOT acceptable.  This is because meme (which ememe wraps) does not support USAs, and a full sequence database is too big to write to a temporary file that the original meme would understand.






<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th><th>Description</th></tr>
<tr>
<td><a href="/emboss/apps/antigenic.html">antigenic</a></td>
<td>Finds antigenic sites in proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/digest.html">digest</a></td>
<td>Protein proteolytic enzyme or reagent cleavage digest</td>
</tr>

<tr>
<td><a href="emast.html">emast</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/emboss/apps/epestfind.html">epestfind</a></td>
<td>Finds PEST motifs as potential proteolytic cleavage sites</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzzpro.html">fuzzpro</a></td>
<td>Protein pattern search</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzztran.html">fuzztran</a></td>
<td>Protein pattern search after translation</td>
</tr>

<tr>
<td><a href="/emboss/apps/helixturnhelix.html">helixturnhelix</a></td>
<td>Report nucleic acid binding motifs</td>
</tr>

<tr>
<td><a href="/emboss/apps/oddcomp.html">oddcomp</a></td>
<td>Find protein sequence regions with a biased composition</td>
</tr>

<tr>
<td><a href="/embassy/meme/omeme.html">omeme</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatdb.html">patmatdb</a></td>
<td>Search a protein sequence with a motif</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatmotifs.html">patmatmotifs</a></td>
<td>Search a PROSITE motif database with a protein sequence</td>
</tr>

<tr>
<td><a href="/emboss/apps/pepcoil.html">pepcoil</a></td>
<td>Predicts coiled coil regions</td>
</tr>

<tr>
<td><a href="/emboss/apps/preg.html">preg</a></td>
<td>Regular expression search of a protein sequence</td>
</tr>

<tr>
<td><a href="/emboss/apps/pscan.html">pscan</a></td>
<td>Scans proteins using PRINTS</td>
</tr>

<tr>
<td><a href="/emboss/apps/sigcleave.html">sigcleave</a></td>
<td>Reports protein signal cleavage sites</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

Jon Ison (jison&nbsp;&copy;&nbsp;rfcgr.mrc.ac.uk)
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK



<p>This program is an EMBASSY wrapper to a program written by Timothy L. Bailey as part of his meme package.
<p>Please report any bugs to the EMBOSS bug team in the first instance, not to Timothy L. Bailey.

<H2>
    History
</H2>

None.




<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.











</BODY>
</HTML>

