<HTML>
<HEAD>
  <TITLE>
  EMBASSY: CBSTOOLS: enetphos
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
enetphos
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>



<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Reports ser, thr and tyr phosphorylation sites in eukaryotic proteins
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>


The NetPhos 2.0 server produces neural network predictions for serine,
threonine and tyrosine phosphorylation sites in eukaryotic proteins.


<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with enetphos</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>enetphos </b>
Reports ser, thr and tyr phosphorylation sites in eukaryotic proteins
Input (aligned) sequence set: <b>EFTU_HUMAN.fsa</b>
Output file [eftu_human.enetphos]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Reports ser, thr and tyr phosphorylation sites in eukaryotic proteins
Version: EMBOSS:6.3.0

   Standard (Mandatory) qualifiers:
  [-sequence]          seqset     (Aligned) sequence set filename and optional
                                  format, or reference (input USA)
  [-outfile]           outfile    [*.enetphos] Output file name

   Additional (Optional) qualifiers:
   -cutoff             float      [0.0] Report only scores above this value
                                  (Any numeric value)

   Advanced (Unprompted) qualifiers:
   -plot               boolean    [N] Produce graphics
   -two                boolean    [N] Run generic predictions only
   -kinase             boolean    [N] Run kinase-specific only
   -residue            menu       [all] Residues to predict with (Values: all
                                  (all); serine (serine); threonine
                                  (threonine); tyrosine (tyrosine))
   -gff                boolean    [N] Produce gff output
   -best               boolean    [N] Report only the highest score for each
                                  residue
   -addseq             boolean    [N] Add sequence to gff output

   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>seqset</td>
<td>(Aligned) sequence set filename and optional format, or reference (input USA)</td>
<td>Readable set of sequences</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>outfile</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.enetphos</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-cutoff</td>
<td>float</td>
<td>Report only scores above this value</td>
<td>Any numeric value</td>
<td>0.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-plot</td>
<td>boolean</td>
<td>Produce graphics</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-two</td>
<td>boolean</td>
<td>Run generic predictions only</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-kinase</td>
<td>boolean</td>
<td>Run kinase-specific only</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-residue</td>
<td>list</td>
<td>Residues to predict with</td>
<td><table><tr><td>all</td> <td><i>(all)</i></td></tr><tr><td>serine</td> <td><i>(serine)</i></td></tr><tr><td>threonine</td> <td><i>(threonine)</i></td></tr><tr><td>tyrosine</td> <td><i>(tyrosine)</i></td></tr></table></td>
<td>all</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-gff</td>
<td>boolean</td>
<td>Produce gff output</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-best</td>
<td>boolean</td>
<td>Report only the highest score for each residue</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-addseq</td>
<td>boolean</td>
<td>Add sequence to gff output</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-sequence" associated seqset qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-outfile" associated outfile qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory2<br>-odirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>enetphos</b> reads any normal sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: EFTU_HUMAN.fsa</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;EFTU_HUMAN  P49411 Elongation factor Tu, mitochondrial precursor (EF-Tu) (P43). - Homo sapiens (Human).
MAAATLLRATPHFSGLAAGRTFLLQGLLRLLKAPALPLLCRGLAVEAKKTYVRDKPHVNV
GTIGHVDHGKTTLTAAITKILAEGGGAKFKKYEEIDNAPEERARGITINAAHVEYSTAAR
HYAHTDCPGHADYVKNMITGTAPLDGCILVVAANDGPMPQTREHLLLARQIGVEHVVVYV
NKADAVQDSEMVELVELEIRELLTEFGYKGEETPVIVGSALCALEGRDPELGLKSVQKLL
DAVDTYIPVPARDLEKPFLLPVEAVYSVPGRGTVVTGTLERGILKKGDECELLGHSKNIR
TVVTGIEMFHKSLERAEAGDNLGALVRGLKREDLRRGLVMVKPGSIKPHQKVEAQVYILS
KEEGGRHKPFVSHFMPVMFSLTWDMACRIILPPEKELAMPGEDLKFNLILRQPMILEKGQ
RFTLRDGNRTIGTGLVTNTLAMTEEEKNIKWG
</pre>
</td></tr></table><p>





<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:

        #include file="inc/reportformats.ihtml"
   -->


<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: eftu_human.enetphos</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
&gt;EFTU_HUMAN	452 amino acids
#
# netphos-3.1b prediction results
#
# Sequence		   # x   Context     Score   Kinase    Answer
# -------------------------------------------------------------------
# EFTU_HUMAN               5 T   MAAATLLRA   0.776   PKC        YES
# EFTU_HUMAN               5 T   MAAATLLRA   0.520   cdc2       YES
# EFTU_HUMAN               5 T   MAAATLLRA   0.442   CaM-II      . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.428   GSK3        . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.382   CKI         . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.350   DNAPK       . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.345   p38MAPK     . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.285   CKII        . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.226   ATM         . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.221   PKG         . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.204   RSK         . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.187   cdk5        . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.172   unsp        . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.125   PKA         . 
# EFTU_HUMAN               5 T   MAAATLLRA   0.073   PKB         . 
#
# EFTU_HUMAN              10 T   LLRATPHFS   0.555   cdk5       YES
# EFTU_HUMAN              10 T   LLRATPHFS   0.531   p38MAPK    YES
# EFTU_HUMAN              10 T   LLRATPHFS   0.490   GSK3        . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.440   CaM-II      . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.369   CKI         . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.343   DNAPK       . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.341   PKG         . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.322   cdc2        . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.303   unsp        . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.275   RSK         . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.266   ATM         . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.249   CKII        . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.190   PKA         . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.099   PKC         . 
# EFTU_HUMAN              10 T   LLRATPHFS   0.093   PKB         . 
#
# EFTU_HUMAN              14 S   TPHFSGLAA   0.547   cdc2       YES
# EFTU_HUMAN              14 S   TPHFSGLAA   0.468   CaM-II      . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.444   GSK3        . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.371   CKI         . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.342   DNAPK       . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.332   CKII        . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.306   PKG         . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.304   PKC         . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.290   p38MAPK     . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.270   ATM         . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.248   RSK         . 
# EFTU_HUMAN              14 S   TPHFSGLAA   0.199   cdk5        . 


<font color=red>  [Part of this file has been deleted for brevity]</font>

# EFTU_HUMAN             439 T   LVTNTLAMT   0.446   CKII        . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.441   CaM-II      . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.425   CKI         . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.424   GSK3        . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.420   cdc2        . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.364   PKA         . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.279   p38MAPK     . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.275   ATM         . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.264   PKG         . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.192   RSK         . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.160   cdk5        . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.160   unsp        . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.138   PKC         . 
# EFTU_HUMAN             439 T   LVTNTLAMT   0.078   PKB         . 
#
# EFTU_HUMAN             443 T   TLAMTEEEK   0.678   CKII       YES
# EFTU_HUMAN             443 T   TLAMTEEEK   0.480   CKI         . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.447   unsp        . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.443   GSK3        . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.425   CaM-II      . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.376   cdc2        . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.348   DNAPK       . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.315   PKG         . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.277   ATM         . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.251   p38MAPK     . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.249   RSK         . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.157   cdk5        . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.132   PKC         . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.081   PKB         . 
# EFTU_HUMAN             443 T   TLAMTEEEK   0.066   PKA         . 
#
    MAAATLLRATPHFSGLAAGRTFLLQGLLRLLKAPALPLLCRGLAVEAKKT   #     50
    YVRDKPHVNVGTIGHVDHGKTTLTAAITKILAEGGGAKFKKYEEIDNAPE   #    100
    ERARGITINAAHVEYSTAARHYAHTDCPGHADYVKNMITGTAPLDGCILV   #    150
    VAANDGPMPQTREHLLLARQIGVEHVVVYVNKADAVQDSEMVELVELEIR   #    200
    ELLTEFGYKGEETPVIVGSALCALEGRDPELGLKSVQKLLDAVDTYIPVP   #    250
    ARDLEKPFLLPVEAVYSVPGRGTVVTGTLERGILKKGDECELLGHSKNIR   #    300
    TVVTGIEMFHKSLERAEAGDNLGALVRGLKREDLRRGLVMVKPGSIKPHQ   #    350
    KVEAQVYILSKEEGGRHKPFVSHFMPVMFSLTWDMACRIILPPEKELAMP   #    400
    GEDLKFNLILRQPMILEKGQRFTLRDGNRTIGTGLVTNTLAMTEEEKNIK   #    450
    WG                                                   #    500
%1  ....T....T...S...................................T   #     50
%1  Y....................T.T.................Y........   #    100
%1  ......T........S................Y.......T.........   #    150
%1  ..........T.................Y.........S...........   #    200
%1  .......Y....T...............................TY....   #    250
%1  ...............Y......T....T.................S....   #    300
%1  ...T.......S................................S.....   #    350
%1  .........S........................................   #    400
%1  ......................T.........T.....T...T.......   #    450
%1  ..
</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

None.



<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="/emboss/apps/antigenic.html">antigenic</a></td>
<td>Finds antigenic sites in proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/digest.html">digest</a></td>
<td>Reports on protein proteolytic enzyme or reagent cleavage sites</td>
</tr>

<tr>
<td><a href="echlorop.html">echlorop</a></td>
<td>Reports presence of chloroplast transit peptides</td>
</tr>

<tr>
<td><a href="/embassy/iprscan/eiprscan.html">eiprscan</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="elipop.html">elipop</a></td>
<td>Prediction of lipoproteins</td>
</tr>

<tr>
<td><a href="/embassy/memenew/emast.html">emast</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/embassy/memenew/ememe.html">ememe</a></td>
<td>Multiple EM for Motif Elicitation</td>
</tr>

<tr>
<td><a href="/embassy/memenew/ememetext.html">ememetext</a></td>
<td>Multiple EM for Motif Elicitation. Text file only</td>
</tr>

<tr>
<td><a href="enetnglyc.html">enetnglyc</a></td>
<td>Reports N-glycosylation sites in human proteins</td>
</tr>

<tr>
<td><a href="enetoglyc.html">enetoglyc</a></td>
<td>Reports mucin type GalNAc O-glycosylation sites in mammalian proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/epestfind.html">epestfind</a></td>
<td>Finds PEST motifs as potential proteolytic cleavage sites</td>
</tr>

<tr>
<td><a href="eprop.html">eprop</a></td>
<td>Reports propeptide cleavage sites in proteins</td>
</tr>

<tr>
<td><a href="esignalp.html">esignalp</a></td>
<td>Reports protein signal cleavage sites</td>
</tr>

<tr>
<td><a href="etmhmm.html">etmhmm</a></td>
<td>Reports transmembrane helices</td>
</tr>

<tr>
<td><a href="eyinoyang.html">eyinoyang</a></td>
<td>Reports O-(beta)-GlcNAc attachment sites</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzzpro.html">fuzzpro</a></td>
<td>Search for patterns in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzztran.html">fuzztran</a></td>
<td>Search for patterns in protein sequences (translated)</td>
</tr>

<tr>
<td><a href="/emboss/apps/helixturnhelix.html">helixturnhelix</a></td>
<td>Identify nucleic acid-binding motifs in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/oddcomp.html">oddcomp</a></td>
<td>Identify proteins with specified sequence word composition</td>
</tr>

<tr>
<td><a href="/embassy/meme/omeme.html">omeme</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatdb.html">patmatdb</a></td>
<td>Searches protein sequences with a sequence motif</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatmotifs.html">patmatmotifs</a></td>
<td>Scan a protein sequence with motifs from the PROSITE database</td>
</tr>

<tr>
<td><a href="/emboss/apps/pepcoil.html">pepcoil</a></td>
<td>Predicts coiled coil regions in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/preg.html">preg</a></td>
<td>Regular expression search of protein sequence(s)</td>
</tr>

<tr>
<td><a href="/emboss/apps/pscan.html">pscan</a></td>
<td>Scans protein sequence(s) with fingerprints from the PRINTS database</td>
</tr>

<tr>
<td><a href="/emboss/apps/sigcleave.html">sigcleave</a></td>
<td>Reports on signal cleavage sites in a protein sequence</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

This program is an EMBOSS wrapper for a program written by the CBS group
http://www.cbs.dtu.dk/index.shtml

<p>
The original CBS group application must be licensed and installed to
use this wrapper.

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.

<H2>
    History
</H2>





<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.


<H2>
    Comments
</H2>

None

</BODY>
</HTML>

