<!-- START OF HEADER -->
<HTML><HEAD>
<TITLE> EMBASSY: SEQWORDS documentation. </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<!--#include virtual="/apps/header1.inc" -->
<H2> SEQWORDS documentation
<!--#include virtual="/apps/header2.inc" -->

<!-- END OF HEADER -->




<!-- CONTENTS
     This always includes the sections below.
     Other subsections can be added for individual applications.
-->
<H2>CONTENTS </H2>
<b> <a href="#1.0">1.0     SUMMARY                   </a></b><br>
<b> <a href="#2.0">2.0     INPUTS & OUTPUTS          </a></b><br>
<b> <a href="#3.0">3.0     INPUT FILE FORMAT         </a></b><br>
<b> <a href="#4.0">4.0     OUTPUT FILE FORMAT        </a></b><br>
<b> <a href="#5.0">5.0     DATA FILES                </a></b><br>
<b> <a href="#6.0">6.0     USAGE                     </a></b><br>    
<b> <a href="#7.0">7.0     KNOWN BUGS & WARNINGS     </a></b><br>    
<b> <a href="#8.0">8.0     NOTES                     </a></b><br>
<b> <a href="#9.0">9.0     DESCRIPTION               </a></b><br>
<b> <a href="#10.0">10.0   ALGORITHM                 </a></b><br>
<b> <a href="#11.0">11.0   RELATED APPLICATIONS      </a></b><br>
<b> <a href="#12.0">12.0   DIAGNOSTIC ERROR MESSAGES </a></b><br>
<b> <a href="#13.0">13.0   AUTHORS                   </a></b><br>
<b> <a href="#14.0">14.0   REFERENCES                </a></b><br>


<!-- SUMMARY
     Succint description of the application, particularly its inputs, outputs
     and what it does.  The same text is given at the top of the source (.c)
     file and in the <documentation> attribute of the <application definition>
     of the ACD file.
-->
<a name="1.0"></a>
<br><br><br><H2> 1.0   SUMMARY  </H2>
Generates DHF files (domain hits files) of database hits (sequences) from Swissprot matching keywords from a keywords file.

<!--#include file="inc/seqwords.ione" -->



<!-- INPUTS & OUTPUTS
     Short summary of the application inputs and outputs in its different 
     modes of usage (if appropriate). More detail than the summary.
-->
<a name="2.0"></a>
<br><br><br><H2> 2.0   INPUTS & OUTPUTS          </H2>
SEQWORDS searches a swissprot-format sequence database with keywords taken from a keywords file, and writes a DHF file (domain hits file) of sequences whose swissprot entries contains at least one of the keywords.  If an entry contains a keyword in a domain record of the feature table, then the sequence of the domain is written to the output file, otherwise the entire sequence is written.  The user specifies the name of the swissprot-format sequence database (input), keywords file (input) and DHF file (output).


<!-- INPUT FILE FORMAT 
     Description and example(s) of input file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.
     Use "<b>SEQWORDS</b> reads any normal sequence USAs." if
     appropriate.
-->



<a name="3.0"></a>
<br><br><br><H2> 3.0   INPUT FILE FORMAT         </H2>

The keywords file (below) contains lists of keywords specific to a number of SCOP or CATH nodes, e.g. families and superfamilies.  Each list of keywords is given after a block of SCOP or CATH classification records; for family-specific search terms, the block must contain a CL, FO, SF and an FA record (see below). For superfamily-specific terms, clearly only the CL, FO and SF should be specified. A single keyword must be given per line after the record 'TE'. Each block of SCOP or CATH classification records and search terms must be delimited by the record '//' (the file should also end with this record). 
<br>It is possible to provide search terms above the level of superfamily, for example, fold and class-specific search terms for SCOP by using the CL and FO records only as appropriate.  However, text searches of swissprot for members of scop folds and classes are unlikely to produce specific or meaningful results.

<!--#include file="inc/seqwords.input" -->






<!-- OUTPUT FILE FORMAT 
     Description and example(s) of output file formats.  Should provide enough
     information to write and parse the file.  Should describe the format in 
     unusual cases - null input, etc.  
     If the standard description of the avalable report formats is required, 
     use:   #include file="inc/reportformats.ihtml"
     Use "Outputs a graph to the specified graphics device."
     or  "outputs a report format file. The default format is ..."
     if appropriate.
-->
<a name="4.0"></a>
<br><br><br><H2> 4.0   OUTPUT FILE FORMAT         </H2>

<b>DHF file (domain hits file)</b>
<br>The format of the DHF file (domain hits file) of hit sequences generated by SEQWORDS (Figure 1) is described fully in <a href="seqsearch.html">SEQSEARCH documentation</a> and only summarised here. The file contains two lines per hit, the first is a description of the hit in 16 text tokens delimited by '^'. The second line contains the protein sequence. The first 4 tokens refer to the hit (sequence) itself, the tokens are as follows:
<UL>
<LI>(i) Accession number
<LI>(ii)  Database code, 
<LI>(iii - iv) Start and end positions of the hit relative to the full length sequence. 
</UL>

<br>The next 9 tokens refer to the domain family, superfamily etc for which the terms were defined (in the keywords file) and are as follows: 
<UL>
<LI>(v)  Type of domain (one of 'SCOP' or 'CATH'), 
<LI>(vi)  SCOP or CATH domain identifier.  
<LI>(vii) SCOP or CATH node unique identifier, e.g. SCOP Family Sunid.
<LI>(viii) Domain class.  Textual description of the 'Class' (SCOP and CATH domains). 
<LI>(ix) Domain architecture. Textual description of the 'Architecture' (CATH only). 
<LI>(x) Domain topology. Textual description of the 'Topology' (CATH only).  
<LI>(xi) Domain fold. Textual description of the 'Fold' (SCOP domains only).  
<LI>(xii) Domain superfamily. Textual description of the 'Superfamily' (SCOP and CATH domains).  
<LI>(xiii) Domain family. Textual description of the 'Fold' (SCOP only). 
</UL>

<br>The next 4 tokens refer to the hit, specifically, information about the search result as follows: 
<UL>
<LI>(xiv) Model type.  The type of model that was used to generate the hit. For DHF files generated by using SEQWORDS a value of KEYWORD is given.  Several other values are possible, however, see <a href="seqsearch.html">SEQSEARCH documentation</a>.
<LI>(xv) SC - Score of hit from search algoritm  (not written by SEQWORDS).
<LI>(xvi) P-value of hit (not written by SEQWORDS).
<LI>(xvii) E-value of hit (not written by SEQWORDS).
</UL>


<!--#include file="inc/seqwords.output" -->



<!-- DATA FILES         
     Any data files used (e.g. translation table file, substitution matrix 
     etc.  This includes example data file formats if they are not obvious.
     For a standard description of what data files are and how embossdata can
     be used to inspect and retrieve them, use:
     #include file="inc/localfiles.ihtml" 
-->

<a name="5.0"></a>
<br><br><br><H2> 5.0   DATA FILES                </H2>
SEQWORDS does not use a data file.


<!-- USAGE
     Example usage, as run from the command-line.
     Many examples illustrating different behaviours is good.
-->
<a name="6.0"></a>
<br><br><br><H2> 6.0   USAGE                     </H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" >
<tr><td>
<pre>
<!--#include file="inc/seqwords.ihelp" --></pre>
</pre>
</td></tr>
</table>


<H3> 6.1   COMMAND LINE ARGUMENTS </H3>
 
<!--#include file="inc/seqwords.itable" -->

<H3> 6.2 EXAMPLE SESSION </H3>
An example of interactive use of SEQWORDS is shown below.

<!--#include file="inc/seqwords.usage" -->

<!-- <br><br>
The keywords file test_data/seqwords.terms was read and used to search the swissprot database test_data/seqwords.seq.  The DHF file test_data/seqwords/seqwords.dhf containing the results of the search was written.
<br><br>


<!-- KNOWN BUGS & WARNINGS
     Bugs that have not yet been fixed, easily missued features, problems
     and caveats etc. Potentially stupid things the program will let you do.
-->
<a name="7.0"></a>
<br><br><br><H2> 7.0   KNOWN BUGS & WARNINGS     </H2>
SEQWORDS is <b>slow</b> - swissprot is read multiple times (once for each list of terms).  Changing it to do a single file read would require modifying the program to take an array of hitlist and terms structures. An easy-ish change.








<!-- NOTES
     Important general remarks, including:
       Restrictions.
       Interesting behaviour.
       Useful things you can do with this program.
       Future plans.
       etc.
-->
<a name="8.0"></a>
<br><br><br><H2> 8.0   NOTES                     </H2>



<H3> 8.1   GLOSSARY OF FILE TYPES    </H3>
<table BORDER CELLSPACING=5 CELLPADDING=5 BGCOLOR="#f5f5ff" >
<tr>
<td><b>FILE TYPE</b></td>
<td><b>FORMAT</b></td>
<td><b>DESCRIPTION</b></td>
<td><b>CREATED BY <b></td>
<td><b>SEE ALSO</b></td>
</tr>
<tr>
<td><b>Domain hits file </b></td>
<td> DHF format (FASTA-like). </td>
<td> Database hits (sequences) with domain classification information.  The hits are relatives to a SCOP or CATH family (or other node in the structural hierarchies) and are found from a search of a discriminating element (e.g. a protein signature, hidden Markov model, simple frequency matrix, Gribskov profile or Hennikoff profile) against a sequence database. </td>
<td> <a href="seqsearch.html">SEQSEARCH</a> (hits retrieved by PSIBLAST). <a href="sigscan.html">SIGSCAN</a> (hits retrieved by sparse protein signature). <a href="libscan.html">LIBSCAN</a> (hits retrieved by various types of HMM and profile). </td>
<td> N.A. </td>
</tr>
<tr>
<td><b>Keywords file </b></td>
<td> Text</td>
<td> Contains a list of keywords specific to a number of SCOP families and superfamilies used by SEQWORDS to search a sequence database. </td>
<td> N.A. </td>
<td> N.A. </td>
</tr>
</table>

<!--#include file="inc/seqwords.history" -->
<!--#include file="inc/seqwords.comment" -->

<!-- DESCRIPTION
     A complete, non-technical, user-level description of the application.
-->
<a name="9.0"></a>
<br><br><br><H2> 9.0   DESCRIPTION               </H2>
Retrieval of domain sequence information from swissprot by mutliple keywords can be a time-consuming process.  SEQWORDS parses a file of keywords and the swissprot database and writes a file of sequences whose swissprot entries contains at least one of the keywords.  The domain sequence is taken if the keyword appears in the domain record of the feature table, otherwise the full-length sequence is taken.


<!-- ALGORITHM
     A technical description of algorithmic aspects, describing exactly how
     the key aspects of the application work.
-->
<a name="10.0"></a>
<br><br><br><H2> 10.0  ALGORITHM                 </H2>
None.



<!-- RELATED APPLICATIONS
     Other applications that either generate the input, use the output or 
     are in some other way related to the application are described here. 
-->

<a name="11.0"></a>
<br><br><br><H2> 11.0  RELATED APPLICATIONS      </H2>

<!--#include file="inc/seqwords.isee" -->



<!-- DIAGNOSTIC ERROR MESSAGES
     Description of error messages or log file, if one is written.
-->
<a name="12.0"></a>
<br><br><br><H2> 12.0  DIAGNOSTIC ERROR MESSAGES </H2>
None.



<!-- AUTHORS
     The main author first then all others. 
-->
<a name="13.0"></a>
<br><br><br><H2> 13.0  AUTHORS                   </H2>
Jon Ison <a href="mailto:jison@rfcgr.mrc.ac.uk">(jison@rfcgr.mrc.ac.uk)</a>
<br>
MRC Rosalind Franklin Centre for Genomics Research
Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SB, UK




<!-- REFERENCES
     Quote the paper where the application was first published, described, used etc. 
-->
<a name="14.0"></a>
<br><br><br><H2> 14.0  REFERENCES                </H2>

<!--#include file="inc/embossref.ihtml" -->


<H3>14.1 Other useful references  </H3>


</BODY>
</HTML>



