<!--START OF HEADER - DON'T ALTER -->

<HTML>
<HEAD>
  <TITLE>
  EMBOSS: meme
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">



<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
meme
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>


<!--END OF HEADER-->






<H2>
Wiki
</H2>

The master copies of EMBOSS documentation are available
at <a href="http://emboss.open-bio.org/wiki/Appdocs">
http://emboss.open-bio.org/wiki/Appdocs</a>
on the EMBOSS Wiki.

<p>
Please help by correcting and extending the Wiki pages.

<H2>
    Function
</H2>
Motif detection
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

<!-- 
	This is the main part of the document.
        This is what the user looks at to find out if this program will do what he wants. 
        Long description.
        Why it is useful.
        When it should/shouldn't be used.
  -->

EMBASSY MEME is a port of the original meme v3.0.1 applications written by  Timothy Bailey.


<H2>
    Algorithm
</H2>

<!-- 
        Algorithms
	  Pseudocode/Vague description/Handwaving/web links to useful pages
  -->

Please read the file README distributed with the original MEME. 













<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with omeme</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>omeme -protein </b>
Motif detection
Input sequence(s): <b>memepep.dat</b>
Motif Distribution
      oops : Oops Distribution
     zoops : Zoops Distribution
       tcm : TCM Distribution
Model to use [zoops]: <b></b>
Maximum number of motifs to find [1]: <b>3</b>
MEME program output file [carp_rhich.omeme]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
Motif detection
Version: EMBOSS:6.3.0

   Standard (Mandatory) qualifiers:
  [-sequence]          seqall     Sequence(s) filename and optional format, or
                                  reference (input USA)
   -model              menu       [zoops] Model to use (Values: oops (Oops
                                  Distribution); zoops (Zoops Distribution);
                                  tcm (TCM Distribution))
   -nmotifs            integer    [1] Maximum number of motifs to find (Any
                                  integer value)
  [-outfile]           outfile    [*.omeme] MEME program output file

   Additional (Optional) qualifiers:
   -ntype              menu       [pair] Method to use (Values: pair (Pairwise
                                  method); blend (Blending method))
   -protein            boolean    [N] Assume sequences are proteins
   -nucleic            boolean    [N] Assume sequences are DNA
   -palindromes        boolean    [N] Allow palindromes
   -ponly              boolean    [N] Force palindromes
   -[no]shorten        boolean    [Y] Allow motifs shorter than MINW
   -nsites             float      [0.] Expected number of sites for each motif
                                  (Any numeric value)
   -minsites           float      [0.] Minimum number of sites for each motif
                                  (Any numeric value)
   -maxsites           float      [0.] Maximum number of sites for each motif
                                  (Any numeric value)
   -w                  integer    [0] Starting motif width to try (Any integer
                                  value)
   -minw               integer    [8] Minimum starting motif width to try (Any
                                  integer value)
   -maxw               integer    [57] Maximum starting motif width to try
                                  (Any integer value)
   -prior              menu       [dirichlet] Prior to use (Values: dirichlet
                                  (Dirichlet prior); dmix (dmix prior); mega
                                  (mega prior); megap (megap prior); addone
                                  (addone prior))
   -[no]brief          boolean    [Y] Don't print documentation
   -b                  float      [-1.0] Strength of the prior (Any numeric
                                  value)
   -spmap              menu       [uni] Mapping start (Values: uni (uni map
                                  type); pam (pam map type))
   -spfuzz             float      [-1.0] Fuzziness of sequence to theta
                                  mapping (Any numeric value)
   -maxiter            integer    [50] Maximum EM iterations to run (Any
                                  integer value)
   -distance           float      [1e-3] EM convergence criterion (Any numeric
                                  value)
   -cons               string     Consensus sequence to start EM from (Any
                                  string)
   -chi                float      [1.0] Cutoff for p-value (Any numeric value)
   -adj                menu       [root] Type (Values: none (no LRT); bon (bon
                                  LRT); root (root LRT))
   -maxsize            integer    [100000] Maximum dataset size in characters
                                  (Any integer value)
   -page               integer    [80] Width of page (Any integer value)
   -status             boolean    [N] Print progress reports
   -v                  boolean    [N] Verbose mode
   -cfive              boolean    [N] Use 5' to 3' complementary strand as
                                  well
   -cthree             boolean    [N] Use 3' to 5' complementary strand as
                                  well
   -wthree             boolean    [N] Use 3' to 5' main strand as well
   -prob               float      [1.0] Starting point confidence level (Any
                                  numeric value)
   -seed               integer    [0] Seed for random numbers in sampling (Any
                                  integer value)
   -seqfrac            float      [1.0] Fraction of sequences to use (Any
                                  numeric value)
   -[no]align          boolean    [Y] Print aligned motif occurrences
   -trace              boolean    [N] Trace starting points
   -allprint           boolean    [N] Print all debugging information
   -wprint             boolean    [N] Print erasure matrix
   -zprint             boolean    [N] Print missing information matrix
   -llprint            boolean    [N] Print log likelihood during EM
   -startsprint        boolean    [N] Print starting points
   -fastaprint         boolean    [N] Print sites in FASTA format
   -timer              integer    [0] Timer type (Any integer value)

   Advanced (Unprompted) qualifiers: (none)
   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options and exit. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages
   -version            boolean    Report version number and exit

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left">Qualifier</th>
<th align="left">Type</th>
<th align="left">Description</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Standard (Mandatory) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-sequence]<br>(Parameter 1)</td>
<td>seqall</td>
<td>Sequence(s) filename and optional format, or reference (input USA)</td>
<td>Readable sequence(s)</td>
<td><b>Required</b></td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-model</td>
<td>list</td>
<td>Model to use</td>
<td><table><tr><td>oops</td> <td><i>(Oops Distribution)</i></td></tr><tr><td>zoops</td> <td><i>(Zoops Distribution)</i></td></tr><tr><td>tcm</td> <td><i>(TCM Distribution)</i></td></tr></table></td>
<td>zoops</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-nmotifs</td>
<td>integer</td>
<td>Maximum number of motifs to find</td>
<td>Any integer value</td>
<td>1</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>[-outfile]<br>(Parameter 2)</td>
<td>outfile</td>
<td>MEME program output file</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.omeme</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Additional (Optional) qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td>-ntype</td>
<td>list</td>
<td>Method to use</td>
<td><table><tr><td>pair</td> <td><i>(Pairwise method)</i></td></tr><tr><td>blend</td> <td><i>(Blending method)</i></td></tr></table></td>
<td>pair</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-protein</td>
<td>boolean</td>
<td>Assume sequences are proteins</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-nucleic</td>
<td>boolean</td>
<td>Assume sequences are DNA</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-palindromes</td>
<td>boolean</td>
<td>Allow palindromes</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-ponly</td>
<td>boolean</td>
<td>Force palindromes</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]shorten</td>
<td>boolean</td>
<td>Allow motifs shorter than MINW</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-nsites</td>
<td>float</td>
<td>Expected number of sites for each motif</td>
<td>Any numeric value</td>
<td>0.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-minsites</td>
<td>float</td>
<td>Minimum number of sites for each motif</td>
<td>Any numeric value</td>
<td>0.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maxsites</td>
<td>float</td>
<td>Maximum number of sites for each motif</td>
<td>Any numeric value</td>
<td>0.</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-w</td>
<td>integer</td>
<td>Starting motif width to try</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-minw</td>
<td>integer</td>
<td>Minimum starting motif width to try</td>
<td>Any integer value</td>
<td>8</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maxw</td>
<td>integer</td>
<td>Maximum starting motif width to try</td>
<td>Any integer value</td>
<td>57</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-prior</td>
<td>list</td>
<td>Prior to use</td>
<td><table><tr><td>dirichlet</td> <td><i>(Dirichlet prior)</i></td></tr><tr><td>dmix</td> <td><i>(dmix prior)</i></td></tr><tr><td>mega</td> <td><i>(mega prior)</i></td></tr><tr><td>megap</td> <td><i>(megap prior)</i></td></tr><tr><td>addone</td> <td><i>(addone prior)</i></td></tr></table></td>
<td>dirichlet</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]brief</td>
<td>boolean</td>
<td>Don't print documentation</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-b</td>
<td>float</td>
<td>Strength of the prior</td>
<td>Any numeric value</td>
<td>-1.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-spmap</td>
<td>list</td>
<td>Mapping start</td>
<td><table><tr><td>uni</td> <td><i>(uni map type)</i></td></tr><tr><td>pam</td> <td><i>(pam map type)</i></td></tr></table></td>
<td>uni</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-spfuzz</td>
<td>float</td>
<td>Fuzziness of sequence to theta mapping</td>
<td>Any numeric value</td>
<td>-1.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maxiter</td>
<td>integer</td>
<td>Maximum EM iterations to run</td>
<td>Any integer value</td>
<td>50</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-distance</td>
<td>float</td>
<td>EM convergence criterion</td>
<td>Any numeric value</td>
<td>1e-3</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-cons</td>
<td>string</td>
<td>Consensus sequence to start EM from</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-chi</td>
<td>float</td>
<td>Cutoff for p-value</td>
<td>Any numeric value</td>
<td>1.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-adj</td>
<td>list</td>
<td>Type</td>
<td><table><tr><td>none</td> <td><i>(no LRT)</i></td></tr><tr><td>bon</td> <td><i>(bon LRT)</i></td></tr><tr><td>root</td> <td><i>(root LRT)</i></td></tr></table></td>
<td>root</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-maxsize</td>
<td>integer</td>
<td>Maximum dataset size in characters</td>
<td>Any integer value</td>
<td>100000</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-page</td>
<td>integer</td>
<td>Width of page</td>
<td>Any integer value</td>
<td>80</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-status</td>
<td>boolean</td>
<td>Print progress reports</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-v</td>
<td>boolean</td>
<td>Verbose mode</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-cfive</td>
<td>boolean</td>
<td>Use 5' to 3' complementary strand as well</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-cthree</td>
<td>boolean</td>
<td>Use 3' to 5' complementary strand as well</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-wthree</td>
<td>boolean</td>
<td>Use 3' to 5' main strand as well</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-prob</td>
<td>float</td>
<td>Starting point confidence level</td>
<td>Any numeric value</td>
<td>1.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-seed</td>
<td>integer</td>
<td>Seed for random numbers in sampling</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-seqfrac</td>
<td>float</td>
<td>Fraction of sequences to use</td>
<td>Any numeric value</td>
<td>1.0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-[no]align</td>
<td>boolean</td>
<td>Print aligned motif occurrences</td>
<td>Boolean value Yes/No</td>
<td>Yes</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-trace</td>
<td>boolean</td>
<td>Trace starting points</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-allprint</td>
<td>boolean</td>
<td>Print all debugging information</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-wprint</td>
<td>boolean</td>
<td>Print erasure matrix</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-zprint</td>
<td>boolean</td>
<td>Print missing information matrix</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-llprint</td>
<td>boolean</td>
<td>Print log likelihood during EM</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-startsprint</td>
<td>boolean</td>
<td>Print starting points</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-fastaprint</td>
<td>boolean</td>
<td>Print sites in FASTA format</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr bgcolor="#FFFFCC">
<td>-timer</td>
<td>integer</td>
<td>Timer type</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Advanced (Unprompted) qualifiers</th>
</tr>

<tr>
<td colspan=5>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>Associated qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-sequence" associated seqall qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sbegin1<br>-sbegin_sequence</td>
<td>integer</td>
<td>Start of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -send1<br>-send_sequence</td>
<td>integer</td>
<td>End of each sequence to be used</td>
<td>Any integer value</td>
<td>0</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sreverse1<br>-sreverse_sequence</td>
<td>boolean</td>
<td>Reverse (if DNA)</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sask1<br>-sask_sequence</td>
<td>boolean</td>
<td>Ask for begin/end/reverse</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -snucleotide1<br>-snucleotide_sequence</td>
<td>boolean</td>
<td>Sequence is nucleotide</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sprotein1<br>-sprotein_sequence</td>
<td>boolean</td>
<td>Sequence is protein</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -slower1<br>-slower_sequence</td>
<td>boolean</td>
<td>Make lower case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -supper1<br>-supper_sequence</td>
<td>boolean</td>
<td>Make upper case</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sformat1<br>-sformat_sequence</td>
<td>string</td>
<td>Input sequence format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sdbname1<br>-sdbname_sequence</td>
<td>string</td>
<td>Database name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -sid1<br>-sid_sequence</td>
<td>string</td>
<td>Entryname</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -ufo1<br>-ufo_sequence</td>
<td>string</td>
<td>UFO features</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fformat1<br>-fformat_sequence</td>
<td>string</td>
<td>Features format</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fopenfile1<br>-fopenfile_sequence</td>
<td>string</td>
<td>Features file name</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>"-outfile" associated outfile qualifiers
</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -odirectory2<br>-odirectory_outfile</td>
<td>string</td>
<td>Output directory</td>
<td>Any string</td>
<td>&nbsp;</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=5>General qualifiers</th>
</tr>

<tr bgcolor="#FFFFCC">
<td> -auto</td>
<td>boolean</td>
<td>Turn off prompts</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -stdout</td>
<td>boolean</td>
<td>Write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -filter</td>
<td>boolean</td>
<td>Read first file from standard input, write first file to standard output</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -options</td>
<td>boolean</td>
<td>Prompt for standard and additional values</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -debug</td>
<td>boolean</td>
<td>Write debug output to program.dbg</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -verbose</td>
<td>boolean</td>
<td>Report some/full command line options</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -help</td>
<td>boolean</td>
<td>Report command line options and exit. More information on associated and general qualifiers can be found with -help -verbose</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -warning</td>
<td>boolean</td>
<td>Report warnings</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -error</td>
<td>boolean</td>
<td>Report errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -fatal</td>
<td>boolean</td>
<td>Report fatal errors</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -die</td>
<td>boolean</td>
<td>Report dying program messages</td>
<td>Boolean value Yes/No</td>
<td>Y</td>
</tr>

<tr bgcolor="#FFFFCC">
<td> -version</td>
<td>boolean</td>
<td>Report version number and exit</td>
<td>Boolean value Yes/No</td>
<td>N</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>meme</b> reads any normal sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: memepep.dat</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;CARP_RHICH P06026 RHIZOPUSPEPSIN PRECURSOR (EC 3.4.23.21)
MKFTLISSCIAIAALAVAVDAAPGEKKISIPLAKNPNYKPSAKNAIQKAIAKYNKHKINT
STGGIVPDAGVGTVPMTDYGNDVEYYGQVTIGTPGKKFNLDFDTGSSDLWIASTLCTNCG
SRQTKYDPKQSSTYQADGRTWSISYGDGSSASGILAKDNVNLGGLLIKGQTIELAKREAA
SFANGPNDGLLGLGFDTITTVRGVKTPMDNLISQGLISRPIFGVYLGKASNGGGGEYIFG
GYDSTKFKGSLTTVPIDNSRGWWGITVDRATVGTSTVASSFDGILDTGTTLLILPNNVAA
SVARAYGASDNGDGTYTISCDTSRFKPLVFSINGASFQVSPDSLVFEEYQGQCIAGFGYG
NFDFAIIGDTFLKNNYVVFNQGVPEVQIAPVAQ
&gt;CARP_YEAST P07267 SACCHAROPEPSIN PRECURSOR (EC 3.4.23.25) (ASPARTATE PROTEASE)
MFSLKALLPLALLLVSANQVAAKVHKAKIYKHELSDEMKEVTFEQHLAHLGQKYLTQFEK
ANPEVVFSREHPFFTEGGHDVPLTNYLNAQYYTDITLGTPPQNFKVILDTGSSNLWVPSN
ECGSLACFLHSKYDHEASSSYKANGTEFAIQYGTGSLEGYISQDTLSIGDLTIPKQDFAE
ATSEPGLTFAFGKFDGILGLGYDTISVDKVVPPFYNAIQQDLLDEKRFAFYLGDTSKDTE
NGGEATFGGIDESKFKGDITWLPVRRKAYWEVKFEGIGLGDEYAELESHGAAIDTGTSLI
TLPSGLAEMINAEIGAKKGWTGQYTLDCNTRDNLPDLIFNFNGYNFTIGPYDYTLEVSGS
CISAITPMDFPEPVGPLAIVGDAFLRKYYSIYDLGNNAVGLAKAI
&gt;CATD_HUMAN P07339 CATHEPSIN D PRECURSOR (EC 3.4.23.5)
MQPSSLLPLALCLLAAPASALVRIPLHKFTSIRRTMSEVGGSVEDLIAKGPVSKYSQAVP
AVTEGPIPEVLKNYMDAQYYGEIGIGTPPQCFTVVFDTGSSNLWVPSIHCKLLDIACWIH
HKYNSDKSSTYVKNGTSFDIHYGSGSLSGYLSQDTVSVPCQSASSASALGGVKVERQVFG
EATKQPGITFIAAKFDGILGMAYPRISVNNVLPVFDNLMQQKLVDQNIFSFYLSRDPDAQ
PGGELMLGGTDSKYYKGSLSYLNVTRKAYWQVHLDQVEVASGLTLCKEGCEAIVDTGTSL
MVGPVDEVRELQKAIGAVPLIQGEYMIPCEKVSTLPAITLKLGGKGYKLSPEDYTLKVSQ
AGKTLCLSGFMGMDIPPPSGPLWILGDVFIGRYYTVFDRDNNRVGFAEAARL
&gt;CHYM_BOVIN P00794 PROCHYMOSIN A AND B PRECURSORS (EC 3.4.23.4) (PREPRORENNIN)
MRCLVVLLAVFALSQGTEITRIPLYKGKSLRKALKEHGLLEDFLQKQQYGISSKYSGFGE
VASVPLTNYLDSQYFGKIYLGTPPQEFTVLFDTGSSDFWVPSIYCKSNACKNHQRFDPRK
SSTFQNLGKPLSIHYGTGSMQGILGYDTVTVSNIVDIQQTVGLSTQEPGDVFTYAEFDGI
LGMAYPSLASEYSIPVFDNMMNRHLVAQDLFSVYMDRNGQESMLTLGAIDPSYYTGSLHW
VPVTVQQYWQFTVDSVTISGVVVACEGGCQAILDTGTSKLVGPSSDILNIQQAIGATQNQ
YGEFDIDCDNLSYMPTVVFEINGKMYPLTPSAYTSQDQGFCTSGFQSENHSQKWILGDVF
IREYYSVFDRANNLVGLAKAI
&gt;PEPA_ASPAW P17946 ASPERGILLOPEPSIN A PRECURSOR (EC 3.4.23.18)
MVVFSKTAALVLGLSSAVSAAPAPTRKGFTINQIARPANKTRTINLPGMYARSLAKFGGT
VPQSVKEAASKGSAVTTPQNNDEEYLTPVTVGKSTLHLDFDTGSADLWVFSDELPSSEQT
GHDLYTPSSSATKLSGYTWDISYGDGSSASGDVYRDTVTVGGVTTNKQAVEAASKISSEF
VQNTANDGLLGLAFSSINTVQPKAQTTFFDTVKSQLDSPLFAVQLKHDAPGVYDFGYIDD
SKYTGSITYTDADSSQGYWGFSTDGYSIGDGSSSSSGFSAIADTGTTLILLDDEIVSAYY
EQVSGASGETEAGGYVFSCSTNPPDFTVVIGDYKAVVPGKYINYAPISTGSSTCFGGIQS
NSGLGLSILGDVFLKSQYVVFNSEGPKLGFAAQA
</pre>
</td></tr></table><p>





<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:
include file="inc/reportformats.ihtml"
   -->

<b>meme</b> 
outputs a graph to the specified graphics device. 
outputs a report format file. The default format is ...


<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: carp_rhich.omeme</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>
********************************************************************************
MEME - Motif discovery tool
********************************************************************************
MEME version 2.3.1 (Release date: 2000/11/05 21:47:56)

For further information on how to interpret these results or to get
a copy of the MEME software please access http://www.sdsc.edu/MEME.

This file may be used as input to the MAST algorithm for searching
sequence databases for matches to groups of motifs.  MAST is available
for interactive use and downloading at http://www.sdsc.edu/MEME.
********************************************************************************


********************************************************************************
REFERENCE
********************************************************************************
If you use this program in your research, please cite:

Timothy L. Bailey and Charles Elkan,
"Fitting a mixture model by expectation maximization to discover
motifs in biopolymers", Proceedings of the Second International
Conference on Intelligent Systems for Molecular Biology, pp. 28-36,
AAAI Press, Menlo Park, California, 1994.
********************************************************************************
********************************************************************************
TRAINING SET
********************************************************************************
DATAFILE= ../../data/memepep.dat (deleted by web version of MEME)
ALPHABET= ACDEFGHIKLMNPQRSTVWY
Sequence name           Weight Length  Sequence name           Weight Length  
-------------           ------ ------  -------------           ------ ------  
CARP_RHICH              1.0000    393  CARP_YEAST              1.0000    405  
CATD_HUMAN              1.0000    412  CHYM_BOVIN              1.0000    381  
PEPA_ASPAW              1.0000    394  
********************************************************************************



********************************************************************************
MOTIF  1		width =   8	sites =  5.0
********************************************************************************
Simplified     A  55:2::2:
motif letter-  C  :22:::::
probability    D  ::::::::
matrix         E  ::::::::
               F  ::::::::
               G  ::::::::
               H  ::::332:
               I  2::2::::


<font color=red>  [Part of this file has been deleted for brevity]</font>

letter-probability matrix: alength= 20 w= 16 n= 1910
 0.345527  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.178967  0.009902  0.010727  0.335557  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.842714  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.010727  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.176047  0.009749  0.515285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.177394  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.844888  0.003738  0.009380  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.010727  0.002224  0.005447 
 0.012194  0.003027  0.841956  0.010390  0.006714  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.010727  0.002224  0.005447 
 0.178861  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.176568  0.510727  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.840047  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.010727  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.342714  0.009749  0.515285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.010727  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.178221  0.003738  0.009380  0.343082  0.015285  0.003844  0.007680  0.008446  0.006793  0.341983  0.012300  0.009902  0.010727  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.177057  0.006714  0.011555  0.003738  0.009380  0.176415  0.015285  0.003844  0.174346  0.008446  0.006793  0.175316  0.178967  0.009902  0.010727  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.174346  0.008446  0.173459  0.008650  0.012300  0.009902  0.010727  0.002224  0.505447 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.010727  0.002224  0.838780 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.345634  0.176568  0.344060  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.006714  0.011555  0.003738  0.176047  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.677394  0.002224  0.005447 
 0.012194  0.003027  0.008623  0.010390  0.673381  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.007680  0.008446  0.006793  0.008650  0.012300  0.009902  0.010727  0.002224  0.172114 
 0.012194  0.003027  0.508623  0.010390  0.006714  0.011555  0.003738  0.009380  0.009749  0.015285  0.003844  0.341013  0.008446  0.006793  0.008650  0.012300  0.009902  0.010727  0.002224  0.005447 

Stopped because nmotifs = 3 reached.


********************************************************************************
DEBUG INFORMATION
********************************************************************************
This information can also be useful in the event you wish to report a
problem with the MEME software.

model:  mod=         zoops    nmotifs=         3    chi=             1
width:  minw=            8    maxw=           57    shorten=       yes
lambda: minsites=        0    maxsites=        5
theta:  prob=            1    spmap=         uni    spfuzz=        0.5
em:     prior=   dirichlet    b=               1    maxiter=        50
        distance=    0.001
data:   n=            1985    N=               5
strands:    w53
sample: seed=            0    seqfrac=         1
LRT:    adj=          root
Letter frequencies:
A 0.074 C 0.014 D 0.061 E 0.039 F 0.048 G 0.100 H 0.013 I 0.058 K 0.052 
L 0.084 M 0.014 N 0.040 P 0.048 Q 0.042 R 0.021 S 0.094 T 0.072 V 0.069 
W 0.010 Y 0.049 
Non-redundant database letter frequencies:
A 0.073 C 0.018 D 0.052 E 0.062 F 0.040 G 0.069 H 0.022 I 0.056 K 0.058 
L 0.092 M 0.023 N 0.046 P 0.051 Q 0.041 R 0.052 S 0.074 T 0.059 V 0.064 
W 0.013 Y 0.033 
Effective length of alphabet = 20
Entropy of dataset (bits) =     -4.1

meme -protein 
********************************************************************************

</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

<!-- 
        Any data files used (e.g. translation table file)
        This includes example data file formats if they are
        not obvious.

	If you wish to include the standard description of what data
	files are and how to use embossdata to inspect and retrieve
	them, use:
include file="inc/localfiles.ihtml"
   -->

None


<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="/emboss/apps/antigenic.html">antigenic</a></td>
<td>Finds antigenic sites in proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/digest.html">digest</a></td>
<td>Reports on protein proteolytic enzyme or reagent cleavage sites</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/echlorop.html">echlorop</a></td>
<td>Reports presence of chloroplast transit peptides</td>
</tr>

<tr>
<td><a href="/embassy/iprscan/eiprscan.html">eiprscan</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/elipop.html">elipop</a></td>
<td>Prediction of lipoproteins</td>
</tr>

<tr>
<td><a href="/embassy/memenew/emast.html">emast</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/embassy/memenew/ememe.html">ememe</a></td>
<td>Multiple EM for Motif Elicitation</td>
</tr>

<tr>
<td><a href="/embassy/memenew/ememetext.html">ememetext</a></td>
<td>Multiple EM for Motif Elicitation. Text file only</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/enetnglyc.html">enetnglyc</a></td>
<td>Reports N-glycosylation sites in human proteins</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/enetoglyc.html">enetoglyc</a></td>
<td>Reports mucin type GalNAc O-glycosylation sites in mammalian proteins</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/enetphos.html">enetphos</a></td>
<td>Reports ser, thr and tyr phosphorylation sites in eukaryotic proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/epestfind.html">epestfind</a></td>
<td>Finds PEST motifs as potential proteolytic cleavage sites</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/eprop.html">eprop</a></td>
<td>Reports propeptide cleavage sites in proteins</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/esignalp.html">esignalp</a></td>
<td>Reports protein signal cleavage sites</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/etmhmm.html">etmhmm</a></td>
<td>Reports transmembrane helices</td>
</tr>

<tr>
<td><a href="/embassy/cbstools/eyinoyang.html">eyinoyang</a></td>
<td>Reports O-(beta)-GlcNAc attachment sites</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzzpro.html">fuzzpro</a></td>
<td>Search for patterns in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzztran.html">fuzztran</a></td>
<td>Search for patterns in protein sequences (translated)</td>
</tr>

<tr>
<td><a href="/emboss/apps/helixturnhelix.html">helixturnhelix</a></td>
<td>Identify nucleic acid-binding motifs in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/oddcomp.html">oddcomp</a></td>
<td>Identify proteins with specified sequence word composition</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatdb.html">patmatdb</a></td>
<td>Searches protein sequences with a sequence motif</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatmotifs.html">patmatmotifs</a></td>
<td>Scan a protein sequence with motifs from the PROSITE database</td>
</tr>

<tr>
<td><a href="/emboss/apps/pepcoil.html">pepcoil</a></td>
<td>Predicts coiled coil regions in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/preg.html">preg</a></td>
<td>Regular expression search of protein sequence(s)</td>
</tr>

<tr>
<td><a href="/emboss/apps/pscan.html">pscan</a></td>
<td>Scans protein sequence(s) with fingerprints from the PRINTS database</td>
</tr>

<tr>
<td><a href="/emboss/apps/sigcleave.html">sigcleave</a></td>
<td>Reports on signal cleavage sites in a protein sequence</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

This program is an EMBOSS conversion of a program written by Thomas L.
Bailey as part of the MEME package.

<p>
Please report all bugs to the EMBOSS bug team (emboss-bug&nbsp;&copy;&nbsp;emboss.open-bio.org) not to the original author.

<H2>
    History
</H2>





<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.











</BODY>
</HTML>

