<HTML>
<HEAD>
  <TITLE>
  EMBASSY: CBSTOOLS: echlorop
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
echlorop
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>



<H2>
    Function
</H2>
Reports presence of chloroplast transit peptides
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>


The ChloroP server predicts the presence of chloroplast transit
peptides (cTP) in protein sequences and the location of potential cTP
cleavage sites.







<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with echlorop</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>echlorop </b>
Reports presence of chloroplast transit peptides
Input (aligned) sequence set: <b>test1.fsa</b>
Output file [test1.echlorop]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>
<p>
<b>Example 2</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>echlorop </b>
Reports presence of chloroplast transit peptides
Input (aligned) sequence set: <b>test12.fsa</b>
Output file [test12.echlorop]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.2">Go to the input files for this example</a><br><a href="#output.2">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-sequence]          seqset     (Aligned) sequence set filename and optional
                                  format, or reference (input USA)
  [-outfile]           outfile    [*.echlorop] Output file name

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers:
   -full               boolean    [N] Show full output

   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>(Aligned) sequence set filename and optional format, or reference (input USA)</td>
<td>Readable set of sequences</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.echlorop</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-full</td>
<td>Show full output</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>echlorop</b> reads any normal sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: test1.fsa</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;P48786;       PATHOGENESIS-RELATED HOMEODOMAIN PROTEIN (PRHP).
MEEISDPKPNALEQVLPTVPNGKCTAPVQMESLAVDVQKVSGEAKVRICSCWCEIVRSPEDLTKLVPCNDFAEDIKLFDS
DPMQQEAESSIGIPLIPKQVTMSHNHDHESGSEMVSNEVMQENHVIATENTYQKSDFDRINMGQKETMPEEVIHKSFLES
STSSIDILLNNHNSYQSGLPPENAVTDCKQVQLGHRSDDAIKNSGLVELVIGQKNVAKSPSQLVETGKRGRGRPRKVQTG
LEQLVIGQKTAAKSSSQLGDTGKRSRGRPRKVQNSPTSFLENINMEQKETIPEQVTQNSILESLTIPTDNQSRTYNSDQS
ELPPENAAKNCNHAQFGHQSDDTTKISGFKELVIGQETVAKSPSQLVDAGKRGRGRPRKVQTGLEQLVPVQETAAKSSSQ
LGDTGKRSRGRPRKVQDSPTSLGGNVKVVPEKGKDSQELSVNSSRSLRSRSQEKSIEPDVNNIVADEGADREKPRKKRKK
RMEENRVDEFCRIRTHLRYLLHRIKYEKNFLDAYSGEGWKGQSLDKIKPEKELKRAKAEIFGRKLKIRDLFQRLDLARSE
GRLPEILFDSRGEIDSEDIFCAKCGSKDVTLSNDIILCDGACDRGFHQFCLDPPLLKEYIPPDDEGWLCPGCECKIDCIK
LLNDSQETNILLGDSWEKVFAEEAAAAASGKNLDDNSGLPSDDSEDDDYDPGGPDLDEKVQGDDSSTDESDYQSESDDMQ
VIRQKNSRGLPSDDSEDDEYDPSGLVTDQMYKDSSCSDFTSDSEDFTGVFDDYKDTGKAQGPLASTPDHVRNNEEGCGHP
EQGDTAPLYPRRQVESLDYKKLNDIEFSKMCDILDILSSQLDVIICTGNQEEYGNTSSDSSDEDYMVTSSPDKNNSDKEA
TAMERGRESGDLELDQKARESTHNRRYIKKFAVEGTDSFLSRSCEDSAAPVAGSKSTSKTLHGEHATQRLLQSFKENQYP
QRAVKESLAAELALSVRQVSNWFNNRRWSFRHSSRIGSDVAKFDSNDTPRQKSIDMSGPSLKSVLDSATYSEIEKKEQDT
ASLGLTEGCDRYMTLNMVADEGNVHTPCIAETREEKTEVGIKPQQNPL
</pre>
</td></tr></table><p>

<a name="input.2"></a>
<h3>Input files for usage example 2</h3>
<p><h3>File: test12.fsa</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;P11043_it_is_a_very_very_long_name;       3-PHOSPHOSHIKIMATE 1-CARBOXYVINYLTRANSFERASE PRECURSOR (EC 2
MAQINNMAQGIQTLNPNSNFHKPQVPKSSSFLVFGSKKLKNSANSMLVLKKDSIFMQKFCSFRISASVATAQKPSEIVLQ
PIKEISGTVKLPGSKSLSNRILLLAALSEGTTVVDNLLSSDDIHYMLGALKTLGLHVEEDSANQRAVVEGCGGLFPVGKE
SKEEIQLFLGNAGTAMRPLTAAVTVAGGNSRYVLDGVPRMRERPISDLVDGLKQLGAEVDCFLGTKCPPVRIVSKGGLPG
GKVKLSGSISSQYLTALLMAAPLALGDVEIEIIDKLISVPYVEMTLKLMERFGISVEHSSSWDRFFVRGGQKYKSPGKAF
VEGDASSASYFLAGAAVTGGTITVEGCGTNSLQGDVKFAEVLEKMGAEVTWTENSVTVKGPPRSSSGRKHLRAIDVNMNK
MPDVAMTLAVVALYADGPTAIRDVASWRVKETERMIAICTELRKLGATVEEGPDYCIITPPEKLNVTDIDTYDDHRMAMA
FSLAACADVPVTINDPGCTRKTFPNYFDVLQQYSKH
&gt;P07505;       SUPEROXIDE DISMUTASE, CHLOROPLAST PRECURSOR (CU-ZN) (EC 1.15
MAAHTILASAPSHTTFSLISPFSSTPTNALSSSLQSSSFNGLSFKLSPTTQSLSLSTSAASKPLTIVAATKKAVAVLKGT
SNVEGVVTLTQEDDGPTTVNVRISGLAPGKHGFHLHEFGDTTNGCMSTGPHFNPDKKTHGAPEDEVRHAGDLGNIVANTD
GVAEATIVDNQIPLTGPNSVVGRALVVHELEDDLGKGGHELSPTTGNAGGRLACGVVGLTPV
&gt;P12360;       CHLOROPHYLL A-B BINDING PROTEIN OF LHCI TYPE I PRECURSOR (CA
MASNTLMSCGIPAVCPSFLSSTKSKFAAAMPVYVGATNFMSRFSMSADWMPGQPRPSYLDGSAPGDFGFDSLGLGEVPAN
LERYKESELIHCRWAMLAVPGIIVPEALGLGNWVKAQEWAAIPGGQATYLGQPVPWGTLPTILAIEFLAIAFVEHQRSME
KDSEKKKYPGGAFDPLGYSKDPAKFEELKVKEIKNGRLALLAIVGFCVQQSAYLGTGPLENLATHLADPWHNNIGDVIIP
KGIFPN
&gt;P12352;       PHOTOSYSTEM I REACTION CENTRE SUBUNIT IV PRECURSOR (PHOTOSYS
MQALSSRVNIAAKPQRAQRLVVRAEEVKAAPKKEVGPKRGSLVKILRPESYWFNQVGKVVSVDQSGVRYPVVVRFENQNY
AGVTTNNYALDEVVAAK
&gt;Q01289;       PROTOCHLOROPHYLLIDE REDUCTASE PRECURSOR (EC 1.3.1.33) (PCR) 
MALQTASMLPASFSIPKEGKIGASLKDSTLFGVSSLSDSLKGDFTSSALRCKELRQKVGAVRAETAAPATPAVNKSSSEG
KKTLRKGNVVITGASSGLGLATAKALAESGKWHVIMACRDYLKAARAAKSAGLAKENYTIMHLDLASLDSVRQFVDNFRR
SEMPLDVLINNAAVYFPTAKEPSFTADGFEISVGTNHLGHFLLSRLLLEDLKKSDYPSKRLIIVGSITGNTNTLAGNVPP
KANLGDLRGLAGGLTGLNSSAMIDGGDFDGAKAYKDSKVCNMLTMQEFHRRYHEETGITFASLYPGCIATTGLFREHIPL
FRTLFPPFQKYITKGYVSEEESGKRLAQVVSDPSLTKSGVYWSWNNASASFENQLSQEASDAEKARKVWEVSEKLVGLA 
&gt;P08817;       ACYL CARRIER PROTEIN II PRECURSOR (ACP II).
MASAAASAVSFARPVKAICVNSVSFSALRKDNVSFRLQPVPQRFSVCCAAKKETVEKVCDIVKSQLALSDDTEVSGSSTF
ADLGADSLDTVEIVMGLEEAFGISVEESSAQTIATVEDAANLIDSLVGK
&gt;P07263;       HISTIDYL-TRNA SYNTHETASE, MITOCHONDRIAL PRECURSOR (EC 6.1.1.
MLSRSLNKVVTSIKSSSIIRMSSATAAATSAPTANAANALKASKAPKKGKLQVSLKTPKGTKDWADSDMVIREAIFSTLS
GLFKKHGGVTIDTPVFELREILAGKYGEDSKLIYNLEDQGGELCSLRYDLTVPFARYVAMNNIQSIKRYHIAKVYRRDQP
AMTKGRMREFYQCDFDVAGTFESMVPDSECLSILVEGLTSLGIKDFKIKLNHRKILDGIFQIAGVKDEDVRKISSAVDKL
DKSPWEAVKKEMTEEKGQSEETADKIGEYVKLNGSLKEIHAVLSADANITSNEKAKQGLDDIATLMKYTEAFDIDSFISF
DLSLARGLDYYTGLIYEVVTSASAPPENASELKKKAKSAEDASEFVGVGSIAAGGRYDNLVNMFSEASGKKSTQIPCVGI
SFGVERIFSLIKQRINSSTTIKPTATQVFVMAFGGGKDWTGYLPERMKVTKQLWDAGIEAEYVYKAKANPRKQFDAAEKA
GCHIAVILGKEEYLEGKLRVKRLGQEFADDDGELVSAADIVPIVQEKLSQIHEDGLNEVTRLIKGL
&gt;P07597;       NONSPECIFIC LIPID-TRANSFER PROTEIN 1 PRECURSOR (LTP 1) (PAPI
MARAQVLLMAAALVLMLTAAPRAAVALNCGQVDSKMKPCLTYVQGGPGPSGECCNGVRDLHNQAQSSGDRQTVCNCLKGI
ARGIHNLNLNNAASIPSKCNVNVPYTISPDIDCSRIY
&gt;P48786;       PATHOGENESIS-RELATED HOMEODOMAIN PROTEIN (PRHP).
MEEISDPKPNALEQVLPTVPNGKCTAPVQMESLAVDVQKVSGEAKVRICSCWCEIVRSPEDLTKLVPCNDFAEDIKLFDS
DPMQQEAESSIGIPLIPKQVTMSHNHDHESGSEMVSNEVMQENHVIATENTYQKSDFDRINMGQKETMPEEVIHKSFLES
STSSIDILLNNHNSYQSGLPPENAVTDCKQVQLGHRSDDAIKNSGLVELVIGQKNVAKSPSQLVETGKRGRGRPRKVQTG
LEQLVIGQKTAAKSSSQLGDTGKRSRGRPRKVQNSPTSFLENINMEQKETIPEQVTQNSILESLTIPTDNQSRTYNSDQS
ELPPENAAKNCNHAQFGHQSDDTTKISGFKELVIGQETVAKSPSQLVDAGKRGRGRPRKVQTGLEQLVPVQETAAKSSSQ
LGDTGKRSRGRPRKVQDSPTSLGGNVKVVPEKGKDSQELSVNSSRSLRSRSQEKSIEPDVNNIVADEGADREKPRKKRKK
RMEENRVDEFCRIRTHLRYLLHRIKYEKNFLDAYSGEGWKGQSLDKIKPEKELKRAKAEIFGRKLKIRDLFQRLDLARSE
GRLPEILFDSRGEIDSEDIFCAKCGSKDVTLSNDIILCDGACDRGFHQFCLDPPLLKEYIPPDDEGWLCPGCECKIDCIK
LLNDSQETNILLGDSWEKVFAEEAAAAASGKNLDDNSGLPSDDSEDDDYDPGGPDLDEKVQGDDSSTDESDYQSESDDMQ
VIRQKNSRGLPSDDSEDDEYDPSGLVTDQMYKDSSCSDFTSDSEDFTGVFDDYKDTGKAQGPLASTPDHVRNNEEGCGHP
EQGDTAPLYPRRQVESLDYKKLNDIEFSKMCDILDILSSQLDVIICTGNQEEYGNTSSDSSDEDYMVTSSPDKNNSDKEA
TAMERGRESGDLELDQKARESTHNRRYIKKFAVEGTDSFLSRSCEDSAAPVAGSKSTSKTLHGEHATQRLLQSFKENQYP
QRAVKESLAAELALSVRQVSNWFNNRRWSFRHSSRIGSDVAKFDSNDTPRQKSIDMSGPSLKSVLDSATYSEIEKKEQDT
ASLGLTEGCDRYMTLNMVADEGNVHTPCIAETREEKTEVGIKPQQNPL
&gt;Q01238;       MAJOR BASIC NUCLEAR PROTEIN 2 (P14 PROTEIN ALPHA CHAIN).
MKAMKATKKAMTKTGLAEALAPKPSSARRIAPPSSRAWPPSAQEVKKTGKLIIPGLVMVKTRKKPATKAGKREMFGKVVL
VKAQPAKTVVKAYPVKALKDNF
&gt;P35334;       POLYGALACTURONASE INHIBITOR PRECURSOR (POLYGALACTURONASE-INH
MTQFNIPVTMSSSLSIILVILVSLRTALSELCNPQDKQALLQIKKDLGNPTTLSSWLPTTDCCNRTWLGVLCDTDTQTYR
VNNLDLSGHNLPKPYPIPSSLANLPYLNFLYIGGINNLVGPIPPAIAKLTQLHYLYITHTNVSGAIPDFLSQIKTLVTLD
FSYNALSGTLPPSISSLPNLGGITFDGNRISGAIPDSYGSFSKLFTAMTISRNRLTGKIPPTFANLNLAFVDLSRNMLEG
DASVLFGSDKNTKKIHLAKNSLAFDLGKVGLSKNLNGLDLRNNRIYGTLPQGLTQLKFLQSLNVSFNNLCGEIPQGGNLK
RFDVSSYANNKCLCGSPLPSCT
&gt;P13086;       SUCCINYL-COA LIGASE (GDP-FORMING), ALPHA-CHAIN PRECURSOR (EC
MVSGSSGLAAARLLSRTFLLQQNGIRHGSYTASRKNIYIDKNTKVICQGFTGKQGTFHSQQALEYGTKLVGGTTPGKGGK
KHLGLPVFNTVKEAKEKTGATASVIYVPPPFAAAAINEAIDAEIPLVVCITEGIPQQDMLRVKHKLTRQGKTRLIGPNCP
GIINPGECKIGIMPGHIHKKGRIGIVSRSGTLTYEAVHQTTQVGLGQSLCIGIGGDPFNGTNFIDCLDVFLKDPATEGIV
LIGEIGGHAEENAAEFLKEHNSGPKAKPVVSFIAGITAPPGRRMGHAGAIIAGGKGGAKEKISALQSAGVIVSMSPAQLG
TCMYKEFEKRKML

</pre>
</td></tr></table><p>





<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:

        #include file="inc/reportformats.ihtml"
   -->



<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: test1.echlorop</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>

### chlorop v1.1 prediction results ###########################
Number of query sequences: 1

Name                  Length       Score  cTP       CS-    cTP-
                                                  score  length
---------------------------------------------------------------
P48786;                 1088	   0.434   -	 -0.663	    28
---------------------------------------------------------------
 
</pre>
</td></tr></table><p>

<a name="output.2"></a>
<h3>Output files for usage example 2</h3>
<p><h3>File: test12.echlorop</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>

### chlorop v1.1 prediction results ###########################
Number of query sequences: 12

Name                  Length       Score  cTP       CS-    cTP-
                                                  score  length
---------------------------------------------------------------
P11043_it_is_a_very_     516	   0.580   Y	  6.765	    65
P07505;                  222	   0.596   Y	 11.122	    67
P12360;                  246	   0.566   Y	  1.302	    42
P12352;                   97	   0.470   -	 13.692	    23
Q01289;                  399	   0.532   Y	 10.699	    62
P08817;                  129	   0.559   Y	 12.650	    47
P07263;                  546	   0.534   Y	  4.495	    41
P07597;                  117	   0.465   -	  4.575	    22
P48786;                 1088	   0.434   -	 -0.663	    28
Q01238;                  102	   0.533   Y	  8.666	    41
P35334;                  342	   0.465   -	  4.342	    71
P13086;                  333	   0.460   -	  0.747	    48
---------------------------------------------------------------
 
</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

None.



<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="/emboss/apps/antigenic.html">antigenic</a></td>
<td>Finds antigenic sites in proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/digest.html">digest</a></td>
<td>Reports on protein proteolytic enzyme or reagent cleavage sites</td>
</tr>

<tr>
<td><a href="/embassy/iprscan/eiprscan.html">eiprscan</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="elipop.html">elipop</a></td>
<td>Prediction of lipoproteins</td>
</tr>

<tr>
<td><a href="/embassy/memenew/emast.html">emast</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/embassy/memenew/ememe.html">ememe</a></td>
<td>Multiple EM for Motif Elicitation</td>
</tr>

<tr>
<td><a href="/embassy/memenew/ememetext.html">ememetext</a></td>
<td>Multiple EM for Motif Elicitation. Text file only</td>
</tr>

<tr>
<td><a href="enetnglyc.html">enetnglyc</a></td>
<td>Reports N-glycosylation sites in human proteins</td>
</tr>

<tr>
<td><a href="enetoglyc.html">enetoglyc</a></td>
<td>Reports mucin type GalNAc O-glycosylation sites in mammalian proteins</td>
</tr>

<tr>
<td><a href="enetphos.html">enetphos</a></td>
<td>Reports ser, thr and tyr phosphorylation sites in eukaryotic proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/epestfind.html">epestfind</a></td>
<td>Finds PEST motifs as potential proteolytic cleavage sites</td>
</tr>

<tr>
<td><a href="eprop.html">eprop</a></td>
<td>Reports propeptide cleavage sites in proteins</td>
</tr>

<tr>
<td><a href="esignalp.html">esignalp</a></td>
<td>Reports protein signal cleavage sites</td>
</tr>

<tr>
<td><a href="etmhmm.html">etmhmm</a></td>
<td>Reports transmembrane helices</td>
</tr>

<tr>
<td><a href="eyinoyang.html">eyinoyang</a></td>
<td>Reports O-(beta)-GlcNAc attachment sites</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzzpro.html">fuzzpro</a></td>
<td>Search for patterns in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzztran.html">fuzztran</a></td>
<td>Search for patterns in protein sequences (translated)</td>
</tr>

<tr>
<td><a href="/emboss/apps/helixturnhelix.html">helixturnhelix</a></td>
<td>Identify nucleic acid-binding motifs in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/oddcomp.html">oddcomp</a></td>
<td>Identify proteins with specified sequence word composition</td>
</tr>

<tr>
<td><a href="/embassy/meme/omeme.html">omeme</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatdb.html">patmatdb</a></td>
<td>Searches protein sequences with a sequence motif</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatmotifs.html">patmatmotifs</a></td>
<td>Scan a protein sequence with motifs from the PROSITE database</td>
</tr>

<tr>
<td><a href="/emboss/apps/pepcoil.html">pepcoil</a></td>
<td>Predicts coiled coil regions in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/preg.html">preg</a></td>
<td>Regular expression search of protein sequence(s)</td>
</tr>

<tr>
<td><a href="/emboss/apps/pscan.html">pscan</a></td>
<td>Scans protein sequence(s) with fingerprints from the PRINTS database</td>
</tr>

<tr>
<td><a href="/emboss/apps/sigcleave.html">sigcleave</a></td>
<td>Reports on signal cleavage sites in a protein sequence</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

This program is an EMBOSS wrapper for a program written by the CBS group
http://www.cbs.dtu.dk/index.shtml

<p>
The original CBS group application must be licensed and installed to
use this wrapper.


<p>
Although we take every care to ensure that the results of the EMBOSS
version are identical to those from the original package, we recommend
that you check your inputs give the same results in both versions
before publication.

<p>
Please report all bugs in the EMBOSS version to the EMBOSS bug team,
not to the original author.

<H2>
    History
</H2>





<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.


<H2>
    Comments
</H2>

None

</BODY>
</HTML>

