<HTML>
<HEAD>
  <TITLE>
  EMBASSY: CBSTOOLS: eprop
  </TITLE>
</HEAD>
<BODY BGCOLOR="#FFFFFF" text="#000000">

<table align=center border=0 cellspacing=0 cellpadding=0>
<tr><td valign=top>
<A HREF="/" ONMOUSEOVER="self.status='Go to the EMBOSS home page';return true"><img border=0 src="emboss_icon.jpg" alt="" width=150 height=48></a>
</td>
<td align=left valign=middle>
<b><font size="+6">
eprop
</font></b>
</td></tr>
</table>
<br>&nbsp;
<p>



<H2>
    Function
</H2>
Reports propeptide cleavage sites in proteins
<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->




<H2>
    Description
</H2>

 
ProP 1.0 server predicts arginine and lysine propeptide cleavage sites
in eukaryotic protein sequences using an ensemble of neural
networks. Furin-specific prediction is the default. It is also
possible to perform a general proprotein convertase (PC) prediction.

<H2>
    Usage
</H2>

<!--  
	Example usage, as run from the command-line.
        Many examples illustrating different behaviours is good.
 -->

<b>Here is a sample session with eprop</b>
<p>

<p>
<table width="90%"><tr><td bgcolor="#CCFFFF"><pre>

% <b>eprop </b>
Reports propeptide cleavage sites in proteins
Input (aligned) sequence set: <b>EDA_HUMAN.fsa</b>
Output file [eda_human.eprop]: <b></b>

</pre></td></tr></table><p>
<p>
<a href="#input.1">Go to the input files for this example</a><br><a href="#output.1">Go to the output files for this example</a><p><p>



<H2>
    Command line arguments
</H2>

<table CELLSPACING=0 CELLPADDING=3 BGCOLOR="#f5f5ff" ><tr><td>
<pre>
   Standard (Mandatory) qualifiers:
  [-sequence]          seqset     (Aligned) sequence set filename and optional
                                  format, or reference (input USA)
  [-outfile]           outfile    [*.eprop] Output file name

   Additional (Optional) qualifiers: (none)
   Advanced (Unprompted) qualifiers:
   -plot               boolean    [N] Produce graphics
   -pcprediction       boolean    [N] Perform proprotein convertase prediction
   -signalp            boolean    [N] Include signal peptide cleavage site
                                  prediction

   Associated qualifiers:

   "-sequence" associated qualifiers
   -sbegin1            integer    Start of each sequence to be used
   -send1              integer    End of each sequence to be used
   -sreverse1          boolean    Reverse (if DNA)
   -sask1              boolean    Ask for begin/end/reverse
   -snucleotide1       boolean    Sequence is nucleotide
   -sprotein1          boolean    Sequence is protein
   -slower1            boolean    Make lower case
   -supper1            boolean    Make upper case
   -sformat1           string     Input sequence format
   -sdbname1           string     Database name
   -sid1               string     Entryname
   -ufo1               string     UFO features
   -fformat1           string     Features format
   -fopenfile1         string     Features file name

   "-outfile" associated qualifiers
   -odirectory2        string     Output directory

   General qualifiers:
   -auto               boolean    Turn off prompts
   -stdout             boolean    Write first file to standard output
   -filter             boolean    Read first file from standard input, write
                                  first file to standard output
   -options            boolean    Prompt for standard and additional values
   -debug              boolean    Write debug output to program.dbg
   -verbose            boolean    Report some/full command line options
   -help               boolean    Report command line options. More
                                  information on associated and general
                                  qualifiers can be found with -help -verbose
   -warning            boolean    Report warnings
   -error              boolean    Report errors
   -fatal              boolean    Report fatal errors
   -die                boolean    Report dying program messages

</pre>
</td></tr></table>
<P>
<table border cellspacing=0 cellpadding=3 bgcolor="#ccccff">
<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Standard (Mandatory) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>[-sequence]<br>(Parameter 1)</td>
<td>(Aligned) sequence set filename and optional format, or reference (input USA)</td>
<td>Readable set of sequences</td>
<td><b>Required</b></td>
</tr>

<tr>
<td>[-outfile]<br>(Parameter 2)</td>
<td>Output file name</td>
<td>Output file</td>
<td><i>&lt;*&gt;</i>.eprop</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Additional (Optional) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td colspan=4>(none)</td>
</tr>

<tr bgcolor="#FFFFCC">
<th align="left" colspan=2>Advanced (Unprompted) qualifiers</th>
<th align="left">Allowed values</th>
<th align="left">Default</th>
</tr>

<tr>
<td>-plot</td>
<td>Produce graphics</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-pcprediction</td>
<td>Perform proprotein convertase prediction</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

<tr>
<td>-signalp</td>
<td>Include signal peptide cleavage site prediction</td>
<td>Boolean value Yes/No</td>
<td>No</td>
</tr>

</table>

<!--
DON'T WRITE ANYTHING HERE.
IT IS DONE FOR YOU.
-->








<H2>
    Input file format
</H2>

<!-- 
        This includes example input file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null input, etc. 
   -->

<b>eprop</b> reads any normal sequence USAs.

<p>


<a name="input.1"></a>
<h3>Input files for usage example </h3>
<p><h3>File: EDA_HUMAN.fsa</h3>
<table width="90%"><tr><td bgcolor="#FFCCFF">
<pre>
&gt;EDA_HUMAN   Q92838 Ectodysplasin A (Ectodermal dysplasia protein) (EDA protein). - Homo sapiens (Human).
MGYPEVERRELLPAAAPRERGSQGCGCGGAPARAGEGNSCLLFLGFFGLSLALHLLTLCC
YLELRSELRRERGAESRLGGSGTPGTSGTLSSLGGLDPDSPITSHLGQPSPKQQPLEPGE
AALHSDSQDGHQMALLNFFFPDEKPYSEEESRRVRRNKRSKSNEGADGPVKNKKKGKKAG
PPGPNGPPGPPGPPGPQGPPGIPGIPGIPGTTVMGPPGPPGPPGPQGPPGLQGPSGAADK
AGTRENQPAVVHLQGQGSAIQVKNDLSGGVLNDWSRITMNPKVFKLHPRSGELEVLVDGT
YFIYSQVEVYYINFTDFASYEVVVDEKPFLQCTRSIETGKTNYNTCYTAGVCLLKARQKI
AVKMVHADISINMSKHTTFFGAIRLGEAPAS
</pre>
</td></tr></table><p>





<H2>
    Output file format
</H2>

<!-- 
        This includes example output file formats.
        This should be a detailed description and example - assume
        someone will want to parse this file and will want to know what
        happens in unusual cases - null output, errors etc. 

        If you wish to include the standard description of the avalable
	report formats, use:

        #include file="inc/reportformats.ihtml"
   -->

<p>


<a name="output.1"></a>
<h3>Output files for usage example </h3>
<p><h3>File: eda_human.eprop</h3>
<table width="90%"><tr><td bgcolor="#CCFFCC">
<pre>


         ##### ProP v.1.0b ProPeptide Cleavage Site Prediction #####


         ##### Furin-type cleavage site prediction (Arginine/Lysine residues) #####


  391 EDA_HUMAN   
MGYPEVERRELLPAAAPRERGSQGCGCGGAPARAGEGNSCLLFLGFFGLSLALHLLTLCCYLELRSELRRERGAESRLGG      80
SGTPGTSGTLSSLGGLDPDSPITSHLGQPSPKQQPLEPGEAALHSDSQDGHQMALLNFFFPDEKPYSEEESRRVRRNKRS     160
KSNEGADGPVKNKKKGKKAGPPGPNGPPGPPGPPGPQGPPGIPGIPGIPGTTVMGPPGPPGPPGPQGPPGLQGPSGAADK     240
AGTRENQPAVVHLQGQGSAIQVKNDLSGGVLNDWSRITMNPKVFKLHPRSGELEVLVDGTYFIYSQVEVYYINFTDFASY     320
EVVVDEKPFLQCTRSIETGKTNYNTCYTAGVCLLKARQKIAVKMVHADISINMSKHTTFFGAIRLGEAPAS              400
................................................................................      80
..............................................................................P.     160
................................................................................     240
................................................................................     320
.......................................................................              400
 
Propeptide cleavage sites predicted:	Arg(R)/Lys(K): 1




Name           Pos    Context     Score  Pred
____________________________v_________________
EDA_HUMAN        8    GYPEVER|RE  0.057    .
EDA_HUMAN        9    YPEVERR|EL  0.128    .
EDA_HUMAN       18    LPAAAPR|ER  0.135    .
EDA_HUMAN       20    AAAPRER|GS  0.102    .
EDA_HUMAN       33    CGGAPAR|AG  0.090    .
EDA_HUMAN       65    CCYLELR|SE  0.089    .
EDA_HUMAN       69    ELRSELR|RE  0.081    .
EDA_HUMAN       70    LRSELRR|ER  0.195    .
EDA_HUMAN       72    SELRRER|GA  0.293    .
EDA_HUMAN       77    ERGAESR|LG  0.112    .
EDA_HUMAN      112    LGQPSPK|QQ  0.064    .
EDA_HUMAN      144    FFFPDEK|PY  0.060    .
EDA_HUMAN      152    YSEEESR|RV  0.096    .
EDA_HUMAN      153    SEEESRR|VR  0.114    .
EDA_HUMAN      155    EESRRVR|RN  0.204    .
EDA_HUMAN      156    ESRRVRR|NK  0.482    .
EDA_HUMAN      158    RRVRRNK|RS  0.182    .
EDA_HUMAN      159    RVRRNKR|SK  0.819 *ProP*
EDA_HUMAN      161    RRNKRSK|SN  0.131    .
EDA_HUMAN      171    GADGPVK|NK  0.060    .
EDA_HUMAN      173    DGPVKNK|KK  0.066    .
EDA_HUMAN      174    GPVKNKK|KG  0.119    .
EDA_HUMAN      175    PVKNKKK|GK  0.156    .
EDA_HUMAN      177    KNKKKGK|KA  0.116    .
EDA_HUMAN      178    NKKKGKK|AG  0.120    .
EDA_HUMAN      240    PSGAADK|AG  0.060    .
EDA_HUMAN      244    ADKAGTR|EN  0.082    .
EDA_HUMAN      263    GSAIQVK|ND  0.066    .
EDA_HUMAN      276    VLNDWSR|IT  0.077    .
EDA_HUMAN      282    RITMNPK|VF  0.057    .
EDA_HUMAN      285    MNPKVFK|LH  0.064    .
EDA_HUMAN      289    VFKLHPR|SG  0.116    .
EDA_HUMAN      327    EVVVDEK|PF  0.060    .
EDA_HUMAN      334    PFLQCTR|SI  0.152    .
EDA_HUMAN      340    RSIETGK|TN  0.061    .
EDA_HUMAN      355    AGVCLLK|AR  0.059    .
EDA_HUMAN      357    VCLLKAR|QK  0.099    .
EDA_HUMAN      359    LLKARQK|IA  0.073    .
EDA_HUMAN      363    RQKIAVK|MV  0.078    .
EDA_HUMAN      375    ISINMSK|HT  0.067    .
EDA_HUMAN      384    TFFGAIR|LG  0.076    .
____________________________^_________________




</pre>
</td></tr></table><p>





<H2>
    Data files
</H2>

None.




<H2>
    Notes
</H2>

<!-- 
        Restrictions.
        Interesting behaviour.
        Useful things you can do with this program.
   -->

None.







<H2>
    References
</H2>

<!-- 
        Bibliography for methods used.
   -->

None.








<H2>
    Warnings
</H2>

<!-- 
        Potentially stupid things the program will let you do.
   -->

None.







<H2>
    Diagnostic Error Messages
</H2>

<!-- 
        Error messages specific to this program, eg:
        "FATAL xxx" - means you have not set up the xxx data using program <b>prog</b>.<p>
   -->

None.







<H2>
    Exit status
</H2>

<!-- 
        Description of the exit status for various error conditions
   -->

It always exits with status 0.








<H2>
    Known bugs
</H2>


<!-- 
        Bugs noted but not yet fixed.
   -->

None.








<!--
<H2>
    See also
</H2>
-->
<h2><a name="See also">See also</a></h2>
<table border cellpadding=4 bgcolor="#FFFFF0">
<tr><th>Program name</th>
<th>Description</th></tr>
<tr>
<td><a href="/emboss/apps/antigenic.html">antigenic</a></td>
<td>Finds antigenic sites in proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/digest.html">digest</a></td>
<td>Reports on protein proteolytic enzyme or reagent cleavage sites</td>
</tr>

<tr>
<td><a href="echlorop.html">echlorop</a></td>
<td>Reports presence of chloroplast transit peptides</td>
</tr>

<tr>
<td><a href="/embassy/iprscan/eiprscan.html">eiprscan</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="elipop.html">elipop</a></td>
<td>Prediction of lipoproteins</td>
</tr>

<tr>
<td><a href="/embassy/memenew/emast.html">emast</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/embassy/memenew/ememe.html">ememe</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="enetnglyc.html">enetnglyc</a></td>
<td>Reports N-glycosylation sites in human proteins</td>
</tr>

<tr>
<td><a href="enetoglyc.html">enetoglyc</a></td>
<td>Reports mucin type GalNAc O-glycosylation sites in mammalian proteins</td>
</tr>

<tr>
<td><a href="enetphos.html">enetphos</a></td>
<td>Reports ser, thr and tyr phosphorylation sites in eukaryotic proteins</td>
</tr>

<tr>
<td><a href="/emboss/apps/epestfind.html">epestfind</a></td>
<td>Finds PEST motifs as potential proteolytic cleavage sites</td>
</tr>

<tr>
<td><a href="esignalp.html">esignalp</a></td>
<td>Reports protein signal cleavage sites</td>
</tr>

<tr>
<td><a href="etmhmm.html">etmhmm</a></td>
<td>Reports transmembrane helices</td>
</tr>

<tr>
<td><a href="eyinoyang.html">eyinoyang</a></td>
<td>Reports O-(beta)-GlcNAc attachment sites</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzzpro.html">fuzzpro</a></td>
<td>Search for patterns in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/fuzztran.html">fuzztran</a></td>
<td>Search for patterns in protein sequences (translated)</td>
</tr>

<tr>
<td><a href="/emboss/apps/helixturnhelix.html">helixturnhelix</a></td>
<td>Identify nucleic acid-binding motifs in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/oddcomp.html">oddcomp</a></td>
<td>Identify proteins with specified sequence word composition</td>
</tr>

<tr>
<td><a href="/embassy/meme/omeme.html">omeme</a></td>
<td>Motif detection</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatdb.html">patmatdb</a></td>
<td>Searches protein sequences with a sequence motif</td>
</tr>

<tr>
<td><a href="/emboss/apps/patmatmotifs.html">patmatmotifs</a></td>
<td>Scan a protein sequence with motifs from the PROSITE database</td>
</tr>

<tr>
<td><a href="/emboss/apps/pepcoil.html">pepcoil</a></td>
<td>Predicts coiled coil regions in protein sequences</td>
</tr>

<tr>
<td><a href="/emboss/apps/preg.html">preg</a></td>
<td>Regular expression search of protein sequence(s)</td>
</tr>

<tr>
<td><a href="/emboss/apps/pscan.html">pscan</a></td>
<td>Scans protein sequence(s) with fingerprints from the PRINTS database</td>
</tr>

<tr>
<td><a href="/emboss/apps/sigcleave.html">sigcleave</a></td>
<td>Reports on signal cleavage sites in a protein sequence</td>
</tr>

</table>
<!-- 
        Add any comments about other associated programs (to prepare
        data files?) that seealso doesn't find. 
   -->










<H2>
    Author(s)
</H2>

This program is an EMBOSS wrapper for a program written by the CBS group
http://www.cbs.dtu.dk/index.shtml

<p>
The original CBS group application must be licensed and installed to
use this wrapper.


<p>
Although we take every care to ensure that the results of the EMBOSS
version are identical to those from the original package, we recommend
that you check your inputs give the same results in both versions
before publication.

<p>
Please report all bugs in the EMBOSS version to the EMBOSS bug team,
not to the original author.

<H2>
    History
</H2>





<H2>
    Target users
</H2>

<!--
        For general users, requested by one user, for EMBOSS site
        maintainers, for EMBOSS developers etc.
        eg:
        "This program is intended to be used by everyone and everything,
        from naive users to embedded scripts." 
	Which is easy to include using:
   -->

This program is intended to be used by everyone and everything, from naive users to embedded scripts.


<H2>
    Comments
</H2>

None

</BODY>
</HTML>

